{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "feda9365-8bde-471e-9b80-35032e409922",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Are you sure you want to start benchmarking and delete previous results? (True/False) True\n"
     ]
    },
    {
     "ename": "RuntimeError",
     "evalue": "No active exception to reraise",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mRuntimeError\u001b[0m                              Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[228], line 119\u001b[0m\n\u001b[1;32m    117\u001b[0m results[[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mfinished\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcompleted\u001b[39m\u001b[38;5;124m\"\u001b[39m]] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[1;32m    118\u001b[0m results\u001b[38;5;241m.\u001b[39mto_csv(file_path)\n\u001b[0;32m--> 119\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m\n\u001b[1;32m    121\u001b[0m shutil\u001b[38;5;241m.\u001b[39mrmtree(subresults_path, ignore_errors\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[1;32m    122\u001b[0m os\u001b[38;5;241m.\u001b[39mmkdir(subresults_path)\n",
      "\u001b[0;31mRuntimeError\u001b[0m: No active exception to reraise"
     ]
    }
   ],
   "source": [
    "import itertools\n",
    "import os.path\n",
    "import argparse\n",
    "\n",
    "import numpy as np\n",
    "from pandarallel import pandarallel\n",
    "import pandas as pd\n",
    "from sklearn.pipeline import make_pipeline\n",
    "from sklearn.preprocessing import StandardScaler, MinMaxScaler, FunctionTransformer\n",
    "from sklearn.cluster import KMeans\n",
    "from mvlearn.decomposition import AJIVE, GroupPCA\n",
    "from mvlearn.cluster import MultiviewSpectralClustering, MultiviewCoRegSpectralClustering\n",
    "from imvc.datasets import LoadDataset\n",
    "from imvc.transformers import MultiViewTransformer, ConcatenateViews\n",
    "from imvc.algorithms import NMFC\n",
    "\n",
    "from utils import Utils\n",
    "\n",
    "\n",
    "folder_results = \"results\"\n",
    "folder_subresults = \"subresults\"\n",
    "filelame = \"complete_algorithms_evaluation.csv\"\n",
    "file_path = os.path.join(folder_results, filelame)\n",
    "subresults_path = os.path.join(folder_results, folder_subresults)\n",
    "logs_file = os.path.join(folder_results, 'logs.txt')\n",
    "error_file = os.path.join(folder_results, 'error.txt')\n",
    "\n",
    "random_state = 42\n",
    "\n",
    "# parser = argparse.ArgumentParser()\n",
    "# parser.add_argument('-continue_benchmarking', default= False, action='store_true')\n",
    "# parser.add_argument('-n_jobs', default= 1, type= int)\n",
    "# args = parser.parse_args()\n",
    "\n",
    "continue_benchmarking = False\n",
    "n_jobs = 1\n",
    "if n_jobs > 1:\n",
    "    pandarallel.initialize(nb_workers= args.n_jobs)\n",
    "\n",
    "datasets = [\n",
    "    # \"simulated_InterSIM\",\n",
    "    # \"simulated_netMUG\",\n",
    "    \"nutrimouse_genotype\",\n",
    "    \"nutrimouse_diet\",\n",
    "    \"bbcsport\",\n",
    "    \"buaa\",\n",
    "    \"metabric\",\n",
    "    \"digits\",\n",
    "    \"bdgp\",\n",
    "    \"tcga\",\n",
    "    \"caltech101\",\n",
    "    \"nuswide\",\n",
    "]\n",
    "amputation_mechanisms = [\"EDM\", 'MCAR', 'MAR', 'MNAR']\n",
    "probs = np.arange(100, step= 10)\n",
    "imputation = [True, False]\n",
    "runs_per_alg = np.arange(10)\n",
    "algorithms = {\n",
    "    \"Concat\": {\"alg\": make_pipeline(ConcatenateViews(),\n",
    "                                    StandardScaler().set_output(transform='pandas'),\n",
    "                                    KMeans()), \"params\": {}},\n",
    "    \"NMFC\": {\"alg\": make_pipeline(ConcatenateViews(),\n",
    "                                  MinMaxScaler().set_output(transform='pandas'),\n",
    "                                  NMFC().set_output(transform='pandas')), \"params\": {}},\n",
    "    \"MVSpectralClustering\": {\"alg\": make_pipeline(MultiViewTransformer(StandardScaler().set_output(transform= \"pandas\")),\n",
    "                                                  MultiviewSpectralClustering()),\n",
    "                             \"params\": {}},\n",
    "    \"MVCoRegSpectralClustering\": {\"alg\": make_pipeline(MultiViewTransformer(StandardScaler().set_output(transform= \"pandas\")),\n",
    "                                                       MultiviewCoRegSpectralClustering()),\n",
    "                                  \"params\": {}},\n",
    "    \"GroupPCA\": {\"alg\": make_pipeline(MultiViewTransformer(StandardScaler()), GroupPCA(), StandardScaler(), KMeans()),\n",
    "                 \"params\": {}},\n",
    "    \"AJIVE\": {\"alg\": make_pipeline(MultiViewTransformer(StandardScaler()), AJIVE(), MultiViewTransformer(FunctionTransformer(pd.DataFrame)),\n",
    "                                   ConcatenateViews(), StandardScaler(), KMeans()),\n",
    "              \"params\": {}},\n",
    "    \"SNF\": {},\n",
    "    \"intNMF\": {},\n",
    "    \"jNMF\": {},\n",
    "}\n",
    "indexes_results = {\"dataset\": datasets, \"algorithm\": list(algorithms.keys()), \"missing_percentage\": probs,\n",
    "                   \"amputation_mechanism\": amputation_mechanisms, \"imputation\": imputation, \"run_n\": runs_per_alg}\n",
    "indexes_names = list(indexes_results.keys())\n",
    "\n",
    "if not continue_benchmarking:\n",
    "    if not eval(input(\"Are you sure you want to start benchmarking and delete previous results? (True/False)\")):\n",
    "        raise Exception\n",
    "    results = pd.DataFrame(datasets, columns= [\"dataset\"])\n",
    "    for k,v in {k:v for k,v in indexes_results.items() if k != \"dataset\"}.items():\n",
    "        results = results.merge(pd.Series(v, name= k), how= \"cross\")\n",
    "    results.loc[(results[\"amputation_mechanism\"] == \"EDM\") & (\n",
    "                results[\"missing_percentage\"] == 0), \"amputation_mechanism\"] = \"'None'\"\n",
    "    results = results.set_index(indexes_names)\n",
    "\n",
    "    idx_to_drop = results.xs(0, level=\"missing_percentage\",\n",
    "                             drop_level=False).xs(True, level=\"imputation\", drop_level=False).index\n",
    "    results = results.drop(idx_to_drop)\n",
    "    for amputation_mechanism in amputation_mechanisms[1:]:\n",
    "        idx_to_drop = results.xs(0, level=\"missing_percentage\",\n",
    "                                 drop_level=False).xs(amputation_mechanism, level=\"amputation_mechanism\", drop_level=False).index\n",
    "        results = results.drop(idx_to_drop)\n",
    "    results_amputation_mechanism_none = results.xs(0, level=\"missing_percentage\", drop_level=False)\n",
    "    results_amputation_mechanism_none_tochange = results_amputation_mechanism_none.index.to_frame()\n",
    "    results_amputation_mechanism_none_tochange[\"amputation_mechanism\"] = \"None\"\n",
    "    results.loc[results_amputation_mechanism_none.index].index = pd.MultiIndex.from_frame(results_amputation_mechanism_none_tochange)\n",
    "\n",
    "    for amputation_mechanism, dataset in itertools.product([\"MAR\", \"MNAR\"], [\"nutrimouse_genotype\",\n",
    "                                                                             \"nutrimouse_diet\",\n",
    "                                                                             \"metabric\",\n",
    "                                                                             \"bdgp\",\n",
    "                                                                             \"buaa\",\n",
    "                                                                             # \"simulated_netMUG\",\n",
    "                                                                             ]):\n",
    "        idx_to_drop = results.xs(dataset, level=\"dataset\",\n",
    "                                 drop_level=False).xs(amputation_mechanism, level=\"amputation_mechanism\", drop_level=False).index\n",
    "        results = results.drop(idx_to_drop)\n",
    "\n",
    "    results[[\"finished\", \"completed\"]] = False\n",
    "    results.to_csv(file_path)\n",
    "    raise\n",
    "\n",
    "    shutil.rmtree(subresults_path, ignore_errors=True)\n",
    "    os.mkdir(subresults_path)\n",
    "\n",
    "    os.remove(logs_file) if os.path.exists(logs_file) else None\n",
    "    os.remove(error_file) if os.path.exists(error_file) else None\n",
    "    open(logs_file, 'w').close()\n",
    "    open(error_file, 'w').close()\n",
    "else:\n",
    "    results = pd.read_csv(file_path, index_col= indexes_names)\n",
    "    results = Utils.collect_subresults(results=results, subresults_path=subresults_path, indexes_names=indexes_names)\n",
    "\n",
    "unfinished_results = results.loc[~results[\"finished\"]]\n",
    "\n",
    "for dataset_name in unfinished_results.index.get_level_values(\"dataset\").unique():\n",
    "    names = dataset_name.split(\"_\")\n",
    "    if \"simulated\" in names:\n",
    "        names = [\"_\".join(names)]\n",
    "    x_name,y_name = names if len(names) > 1 else (names[0], \"0\")\n",
    "    Xs, y = LoadDataset.load_dataset(dataset_name=x_name, return_y=True, shuffle= False)\n",
    "    y = y[y_name]\n",
    "    n_clusters = y.nunique()\n",
    "    unfinished_results_dataset = unfinished_results.loc[[dataset_name]]\n",
    "\n",
    "    if n_jobs == 1:\n",
    "        iterator = pd.DataFrame(unfinished_results_dataset.index.to_list(), columns=indexes_names)\n",
    "        iterator.apply(lambda x: Utils.run_iteration(idx= x, results= results, Xs=Xs, y=y, n_clusters=n_clusters,\n",
    "                                                     algorithms=algorithms, random_state=random_state, subresults_path=subresults_path,\n",
    "                                                     logs_file=logs_file, error_file=error_file), axis= 1)\n",
    "    else:\n",
    "        try:\n",
    "            unfinished_results_dataset_idx = unfinished_results_dataset.xs(0, level=\"missing_percentage\", drop_level=False).index\n",
    "            iterator = pd.DataFrame(unfinished_results_dataset_idx.to_list(), columns= indexes_names)\n",
    "            iterator.parallel_apply(lambda x: Utils.run_iteration(idx= x, results= results, Xs=Xs, y=y,\n",
    "                                                                  n_clusters=n_clusters,\n",
    "                                                                  algorithms=algorithms,\n",
    "                                                                  random_state=random_state,\n",
    "                                                                  subresults_path=subresults_path,\n",
    "                                                                  logs_file=logs_file,\n",
    "                                                                  error_file=error_file), axis= 1)\n",
    "            results = Utils.collect_subresults(results=results, subresults_path=subresults_path,\n",
    "                                               indexes_names=indexes_names)\n",
    "            results.to_csv(file_path)\n",
    "\n",
    "            unfinished_results_dataset_idx = unfinished_results_dataset.drop(unfinished_results_dataset_idx).index\n",
    "            iterator = pd.DataFrame(unfinished_results_dataset_idx.to_list(), columns=indexes_names)\n",
    "        except KeyError:\n",
    "            iterator = pd.DataFrame(unfinished_results_dataset.index.to_list(), columns=indexes_names)\n",
    "\n",
    "        iterator.parallel_apply(lambda x: Utils.run_iteration(idx= x, results= results, Xs=Xs, y=y,\n",
    "                                                                            n_clusters=n_clusters,\n",
    "                                                                            algorithms=algorithms,\n",
    "                                                                            random_state=random_state,\n",
    "                                                                            subresults_path=subresults_path,\n",
    "                                                                            logs_file=logs_file,\n",
    "                                                                            error_file=error_file), axis= 1)\n",
    "        results = Utils.collect_subresults(results=results, subresults_path=subresults_path,\n",
    "                                           indexes_names=indexes_names)\n",
    "        results.to_csv(file_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6e349f64-cddc-49d4-be8c-93c7329a56ea",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>38</th>\n",
       "      <th>39</th>\n",
       "      <th>40</th>\n",
       "      <th>41</th>\n",
       "      <th>42</th>\n",
       "      <th>43</th>\n",
       "      <th>44</th>\n",
       "      <th>45</th>\n",
       "      <th>46</th>\n",
       "      <th>47</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15.044314</td>\n",
       "      <td>31.836497</td>\n",
       "      <td>40.997210</td>\n",
       "      <td>34.390896</td>\n",
       "      <td>7.736058</td>\n",
       "      <td>17.370114</td>\n",
       "      <td>25.391800</td>\n",
       "      <td>29.198195</td>\n",
       "      <td>5.377202</td>\n",
       "      <td>14.717360</td>\n",
       "      <td>...</td>\n",
       "      <td>0.037056</td>\n",
       "      <td>0.040868</td>\n",
       "      <td>0.011358</td>\n",
       "      <td>0.020355</td>\n",
       "      <td>0.028393</td>\n",
       "      <td>0.035676</td>\n",
       "      <td>0.013326</td>\n",
       "      <td>0.022278</td>\n",
       "      <td>0.029836</td>\n",
       "      <td>0.033345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>28.504582</td>\n",
       "      <td>53.951531</td>\n",
       "      <td>62.548719</td>\n",
       "      <td>84.996720</td>\n",
       "      <td>22.503153</td>\n",
       "      <td>45.091515</td>\n",
       "      <td>59.365519</td>\n",
       "      <td>90.261557</td>\n",
       "      <td>23.175209</td>\n",
       "      <td>61.951504</td>\n",
       "      <td>...</td>\n",
       "      <td>0.061419</td>\n",
       "      <td>0.057223</td>\n",
       "      <td>0.023957</td>\n",
       "      <td>0.039699</td>\n",
       "      <td>0.045649</td>\n",
       "      <td>0.054297</td>\n",
       "      <td>0.021933</td>\n",
       "      <td>0.031987</td>\n",
       "      <td>0.038050</td>\n",
       "      <td>0.047752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15.947757</td>\n",
       "      <td>19.039989</td>\n",
       "      <td>7.196868</td>\n",
       "      <td>1.775596</td>\n",
       "      <td>8.391891</td>\n",
       "      <td>10.695923</td>\n",
       "      <td>5.011634</td>\n",
       "      <td>2.016594</td>\n",
       "      <td>6.809082</td>\n",
       "      <td>7.125768</td>\n",
       "      <td>...</td>\n",
       "      <td>0.012585</td>\n",
       "      <td>0.016084</td>\n",
       "      <td>0.007273</td>\n",
       "      <td>0.008067</td>\n",
       "      <td>0.008445</td>\n",
       "      <td>0.010746</td>\n",
       "      <td>0.007856</td>\n",
       "      <td>0.009487</td>\n",
       "      <td>0.007954</td>\n",
       "      <td>0.007093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>13.135696</td>\n",
       "      <td>28.928518</td>\n",
       "      <td>38.080037</td>\n",
       "      <td>55.357058</td>\n",
       "      <td>7.541382</td>\n",
       "      <td>15.375125</td>\n",
       "      <td>19.820881</td>\n",
       "      <td>38.400375</td>\n",
       "      <td>7.608873</td>\n",
       "      <td>16.643876</td>\n",
       "      <td>...</td>\n",
       "      <td>0.042706</td>\n",
       "      <td>0.056762</td>\n",
       "      <td>0.014946</td>\n",
       "      <td>0.022783</td>\n",
       "      <td>0.030973</td>\n",
       "      <td>0.039278</td>\n",
       "      <td>0.014053</td>\n",
       "      <td>0.021040</td>\n",
       "      <td>0.027943</td>\n",
       "      <td>0.040719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9.598944</td>\n",
       "      <td>17.575873</td>\n",
       "      <td>23.974306</td>\n",
       "      <td>56.529306</td>\n",
       "      <td>7.932658</td>\n",
       "      <td>15.406384</td>\n",
       "      <td>20.424018</td>\n",
       "      <td>35.888609</td>\n",
       "      <td>8.962675</td>\n",
       "      <td>19.784140</td>\n",
       "      <td>...</td>\n",
       "      <td>0.053178</td>\n",
       "      <td>0.069009</td>\n",
       "      <td>0.019230</td>\n",
       "      <td>0.030706</td>\n",
       "      <td>0.040181</td>\n",
       "      <td>0.052220</td>\n",
       "      <td>0.013286</td>\n",
       "      <td>0.019051</td>\n",
       "      <td>0.024898</td>\n",
       "      <td>0.035943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9139</th>\n",
       "      <td>16.226805</td>\n",
       "      <td>69.214218</td>\n",
       "      <td>90.441376</td>\n",
       "      <td>68.436279</td>\n",
       "      <td>15.313380</td>\n",
       "      <td>61.463430</td>\n",
       "      <td>77.947826</td>\n",
       "      <td>66.263331</td>\n",
       "      <td>14.241716</td>\n",
       "      <td>57.003396</td>\n",
       "      <td>...</td>\n",
       "      <td>0.048697</td>\n",
       "      <td>0.045058</td>\n",
       "      <td>0.017056</td>\n",
       "      <td>0.036708</td>\n",
       "      <td>0.048240</td>\n",
       "      <td>0.038688</td>\n",
       "      <td>0.017844</td>\n",
       "      <td>0.038576</td>\n",
       "      <td>0.051452</td>\n",
       "      <td>0.044039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9140</th>\n",
       "      <td>12.503711</td>\n",
       "      <td>32.980913</td>\n",
       "      <td>68.166927</td>\n",
       "      <td>79.869925</td>\n",
       "      <td>11.415075</td>\n",
       "      <td>29.113732</td>\n",
       "      <td>52.375020</td>\n",
       "      <td>78.490055</td>\n",
       "      <td>12.988374</td>\n",
       "      <td>31.037411</td>\n",
       "      <td>...</td>\n",
       "      <td>0.033886</td>\n",
       "      <td>0.054953</td>\n",
       "      <td>0.010437</td>\n",
       "      <td>0.018520</td>\n",
       "      <td>0.031220</td>\n",
       "      <td>0.050817</td>\n",
       "      <td>0.010087</td>\n",
       "      <td>0.018301</td>\n",
       "      <td>0.032092</td>\n",
       "      <td>0.049505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9141</th>\n",
       "      <td>4.972273</td>\n",
       "      <td>10.890188</td>\n",
       "      <td>31.764150</td>\n",
       "      <td>62.417002</td>\n",
       "      <td>4.454803</td>\n",
       "      <td>8.302686</td>\n",
       "      <td>20.663929</td>\n",
       "      <td>49.419153</td>\n",
       "      <td>2.767894</td>\n",
       "      <td>4.794864</td>\n",
       "      <td>...</td>\n",
       "      <td>0.011959</td>\n",
       "      <td>0.023671</td>\n",
       "      <td>0.006361</td>\n",
       "      <td>0.008392</td>\n",
       "      <td>0.011643</td>\n",
       "      <td>0.019725</td>\n",
       "      <td>0.007316</td>\n",
       "      <td>0.010598</td>\n",
       "      <td>0.019110</td>\n",
       "      <td>0.029684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9142</th>\n",
       "      <td>18.828697</td>\n",
       "      <td>47.957211</td>\n",
       "      <td>75.842385</td>\n",
       "      <td>147.081311</td>\n",
       "      <td>10.364175</td>\n",
       "      <td>38.001792</td>\n",
       "      <td>73.559660</td>\n",
       "      <td>125.925574</td>\n",
       "      <td>9.352314</td>\n",
       "      <td>34.360579</td>\n",
       "      <td>...</td>\n",
       "      <td>0.038313</td>\n",
       "      <td>0.059128</td>\n",
       "      <td>0.011211</td>\n",
       "      <td>0.024217</td>\n",
       "      <td>0.038212</td>\n",
       "      <td>0.055324</td>\n",
       "      <td>0.011959</td>\n",
       "      <td>0.026034</td>\n",
       "      <td>0.040628</td>\n",
       "      <td>0.064625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9143</th>\n",
       "      <td>3.652957</td>\n",
       "      <td>12.171284</td>\n",
       "      <td>28.649336</td>\n",
       "      <td>70.898222</td>\n",
       "      <td>3.409030</td>\n",
       "      <td>11.801292</td>\n",
       "      <td>28.810325</td>\n",
       "      <td>63.334852</td>\n",
       "      <td>3.075232</td>\n",
       "      <td>11.206886</td>\n",
       "      <td>...</td>\n",
       "      <td>0.017938</td>\n",
       "      <td>0.036368</td>\n",
       "      <td>0.004456</td>\n",
       "      <td>0.009116</td>\n",
       "      <td>0.018277</td>\n",
       "      <td>0.036154</td>\n",
       "      <td>0.004874</td>\n",
       "      <td>0.010228</td>\n",
       "      <td>0.020914</td>\n",
       "      <td>0.043835</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9144 rows × 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              0          1          2           3          4          5   \n",
       "0     15.044314  31.836497  40.997210   34.390896   7.736058  17.370114  \\\n",
       "1     28.504582  53.951531  62.548719   84.996720  22.503153  45.091515   \n",
       "2     15.947757  19.039989   7.196868    1.775596   8.391891  10.695923   \n",
       "3     13.135696  28.928518  38.080037   55.357058   7.541382  15.375125   \n",
       "4      9.598944  17.575873  23.974306   56.529306   7.932658  15.406384   \n",
       "...         ...        ...        ...         ...        ...        ...   \n",
       "9139  16.226805  69.214218  90.441376   68.436279  15.313380  61.463430   \n",
       "9140  12.503711  32.980913  68.166927   79.869925  11.415075  29.113732   \n",
       "9141   4.972273  10.890188  31.764150   62.417002   4.454803   8.302686   \n",
       "9142  18.828697  47.957211  75.842385  147.081311  10.364175  38.001792   \n",
       "9143   3.652957  12.171284  28.649336   70.898222   3.409030  11.801292   \n",
       "\n",
       "              6           7          8          9  ...        38        39   \n",
       "0     25.391800   29.198195   5.377202  14.717360  ...  0.037056  0.040868  \\\n",
       "1     59.365519   90.261557  23.175209  61.951504  ...  0.061419  0.057223   \n",
       "2      5.011634    2.016594   6.809082   7.125768  ...  0.012585  0.016084   \n",
       "3     19.820881   38.400375   7.608873  16.643876  ...  0.042706  0.056762   \n",
       "4     20.424018   35.888609   8.962675  19.784140  ...  0.053178  0.069009   \n",
       "...         ...         ...        ...        ...  ...       ...       ...   \n",
       "9139  77.947826   66.263331  14.241716  57.003396  ...  0.048697  0.045058   \n",
       "9140  52.375020   78.490055  12.988374  31.037411  ...  0.033886  0.054953   \n",
       "9141  20.663929   49.419153   2.767894   4.794864  ...  0.011959  0.023671   \n",
       "9142  73.559660  125.925574   9.352314  34.360579  ...  0.038313  0.059128   \n",
       "9143  28.810325   63.334852   3.075232  11.206886  ...  0.017938  0.036368   \n",
       "\n",
       "            40        41        42        43        44        45        46   \n",
       "0     0.011358  0.020355  0.028393  0.035676  0.013326  0.022278  0.029836  \\\n",
       "1     0.023957  0.039699  0.045649  0.054297  0.021933  0.031987  0.038050   \n",
       "2     0.007273  0.008067  0.008445  0.010746  0.007856  0.009487  0.007954   \n",
       "3     0.014946  0.022783  0.030973  0.039278  0.014053  0.021040  0.027943   \n",
       "4     0.019230  0.030706  0.040181  0.052220  0.013286  0.019051  0.024898   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "9139  0.017056  0.036708  0.048240  0.038688  0.017844  0.038576  0.051452   \n",
       "9140  0.010437  0.018520  0.031220  0.050817  0.010087  0.018301  0.032092   \n",
       "9141  0.006361  0.008392  0.011643  0.019725  0.007316  0.010598  0.019110   \n",
       "9142  0.011211  0.024217  0.038212  0.055324  0.011959  0.026034  0.040628   \n",
       "9143  0.004456  0.009116  0.018277  0.036154  0.004874  0.010228  0.020914   \n",
       "\n",
       "            47  \n",
       "0     0.033345  \n",
       "1     0.047752  \n",
       "2     0.007093  \n",
       "3     0.040719  \n",
       "4     0.035943  \n",
       "...        ...  \n",
       "9139  0.044039  \n",
       "9140  0.049505  \n",
       "9141  0.029684  \n",
       "9142  0.064625  \n",
       "9143  0.043835  \n",
       "\n",
       "[9144 rows x 48 columns]"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from imvc.datasets import LoadDataset\n",
    "import numpy as np\n",
    "Xs, y = LoadDataset.load_dataset(dataset_name=\"caltech101\", return_y=True, shuffle= False)\n",
    "Xs[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "23b5cddf-a795-4a2c-baa4-09e126bbeb06",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>1974</th>\n",
       "      <th>1975</th>\n",
       "      <th>1976</th>\n",
       "      <th>1977</th>\n",
       "      <th>1978</th>\n",
       "      <th>1979</th>\n",
       "      <th>1980</th>\n",
       "      <th>1981</th>\n",
       "      <th>1982</th>\n",
       "      <th>1983</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.261661</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.343632</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.329767</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.167035</td>\n",
       "      <td>0.394578</td>\n",
       "      <td>...</td>\n",
       "      <td>0.158582</td>\n",
       "      <td>0.228006</td>\n",
       "      <td>0.187460</td>\n",
       "      <td>0.172032</td>\n",
       "      <td>0.139624</td>\n",
       "      <td>0.293360</td>\n",
       "      <td>0.187875</td>\n",
       "      <td>0.254355</td>\n",
       "      <td>0.223806</td>\n",
       "      <td>0.189381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.014952</td>\n",
       "      <td>0.053008</td>\n",
       "      <td>0.138431</td>\n",
       "      <td>0.391251</td>\n",
       "      <td>0.352965</td>\n",
       "      <td>0.217056</td>\n",
       "      <td>0.023932</td>\n",
       "      <td>0.016740</td>\n",
       "      <td>0.004268</td>\n",
       "      <td>0.194551</td>\n",
       "      <td>...</td>\n",
       "      <td>0.226811</td>\n",
       "      <td>0.209077</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.024079</td>\n",
       "      <td>0.108644</td>\n",
       "      <td>0.199408</td>\n",
       "      <td>0.093917</td>\n",
       "      <td>0.129881</td>\n",
       "      <td>0.140024</td>\n",
       "      <td>0.222356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.102156</td>\n",
       "      <td>0.302656</td>\n",
       "      <td>0.292165</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.134023</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000422</td>\n",
       "      <td>0.297752</td>\n",
       "      <td>0.359863</td>\n",
       "      <td>0.011706</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.168027</td>\n",
       "      <td>0.188040</td>\n",
       "      <td>0.072884</td>\n",
       "      <td>0.163923</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.238917</td>\n",
       "      <td>0.020608</td>\n",
       "      <td>0.006927</td>\n",
       "      <td>0.124900</td>\n",
       "      <td>0.113634</td>\n",
       "      <td>...</td>\n",
       "      <td>0.210533</td>\n",
       "      <td>0.074725</td>\n",
       "      <td>0.250818</td>\n",
       "      <td>0.315082</td>\n",
       "      <td>0.323974</td>\n",
       "      <td>0.102916</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.328875</td>\n",
       "      <td>0.391070</td>\n",
       "      <td>0.275135</td>\n",
       "      <td>0.030786</td>\n",
       "      <td>0.029746</td>\n",
       "      <td>0.040101</td>\n",
       "      <td>0.048513</td>\n",
       "      <td>0.024524</td>\n",
       "      <td>0.032631</td>\n",
       "      <td>0.028272</td>\n",
       "      <td>...</td>\n",
       "      <td>0.282160</td>\n",
       "      <td>0.127711</td>\n",
       "      <td>0.187435</td>\n",
       "      <td>0.278200</td>\n",
       "      <td>0.214201</td>\n",
       "      <td>0.259537</td>\n",
       "      <td>0.265158</td>\n",
       "      <td>0.290671</td>\n",
       "      <td>0.330026</td>\n",
       "      <td>0.244864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9139</th>\n",
       "      <td>0.039068</td>\n",
       "      <td>0.007367</td>\n",
       "      <td>0.084344</td>\n",
       "      <td>0.381240</td>\n",
       "      <td>0.390044</td>\n",
       "      <td>0.120257</td>\n",
       "      <td>0.016993</td>\n",
       "      <td>0.024952</td>\n",
       "      <td>0.014783</td>\n",
       "      <td>0.014419</td>\n",
       "      <td>...</td>\n",
       "      <td>0.245075</td>\n",
       "      <td>0.207252</td>\n",
       "      <td>0.001578</td>\n",
       "      <td>0.144414</td>\n",
       "      <td>0.054743</td>\n",
       "      <td>0.077117</td>\n",
       "      <td>0.084481</td>\n",
       "      <td>0.280723</td>\n",
       "      <td>0.212753</td>\n",
       "      <td>0.262370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9140</th>\n",
       "      <td>0.011986</td>\n",
       "      <td>0.291720</td>\n",
       "      <td>0.319846</td>\n",
       "      <td>0.328607</td>\n",
       "      <td>0.286253</td>\n",
       "      <td>0.143306</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001630</td>\n",
       "      <td>0.086209</td>\n",
       "      <td>...</td>\n",
       "      <td>0.168624</td>\n",
       "      <td>0.024420</td>\n",
       "      <td>0.013202</td>\n",
       "      <td>0.103529</td>\n",
       "      <td>0.146153</td>\n",
       "      <td>0.116770</td>\n",
       "      <td>0.137495</td>\n",
       "      <td>0.121783</td>\n",
       "      <td>0.144657</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9141</th>\n",
       "      <td>0.149710</td>\n",
       "      <td>0.099214</td>\n",
       "      <td>0.203577</td>\n",
       "      <td>0.200623</td>\n",
       "      <td>0.079059</td>\n",
       "      <td>0.098057</td>\n",
       "      <td>0.151877</td>\n",
       "      <td>0.138999</td>\n",
       "      <td>0.136936</td>\n",
       "      <td>0.148251</td>\n",
       "      <td>...</td>\n",
       "      <td>0.293888</td>\n",
       "      <td>0.348856</td>\n",
       "      <td>0.300672</td>\n",
       "      <td>0.323372</td>\n",
       "      <td>0.332723</td>\n",
       "      <td>0.315752</td>\n",
       "      <td>0.340056</td>\n",
       "      <td>0.354810</td>\n",
       "      <td>0.340426</td>\n",
       "      <td>0.344751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9142</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000070</td>\n",
       "      <td>0.000099</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000190</td>\n",
       "      <td>0.000465</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000595</td>\n",
       "      <td>0.033671</td>\n",
       "      <td>...</td>\n",
       "      <td>0.248481</td>\n",
       "      <td>0.222414</td>\n",
       "      <td>0.001434</td>\n",
       "      <td>0.072547</td>\n",
       "      <td>0.103934</td>\n",
       "      <td>0.063300</td>\n",
       "      <td>0.057856</td>\n",
       "      <td>0.129774</td>\n",
       "      <td>0.207614</td>\n",
       "      <td>0.186621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9143</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001285</td>\n",
       "      <td>0.005373</td>\n",
       "      <td>0.006973</td>\n",
       "      <td>0.005226</td>\n",
       "      <td>0.004229</td>\n",
       "      <td>0.003136</td>\n",
       "      <td>0.001720</td>\n",
       "      <td>0.001118</td>\n",
       "      <td>0.064458</td>\n",
       "      <td>...</td>\n",
       "      <td>0.265564</td>\n",
       "      <td>0.188231</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.031089</td>\n",
       "      <td>0.127476</td>\n",
       "      <td>0.165567</td>\n",
       "      <td>0.181716</td>\n",
       "      <td>0.200003</td>\n",
       "      <td>0.194710</td>\n",
       "      <td>0.010137</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9144 rows × 1984 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             0         1         2         3         4         5         6   \n",
       "0     0.261661  0.400000  0.400000  0.400000  0.343632  0.400000  0.329767  \\\n",
       "1     0.014952  0.053008  0.138431  0.391251  0.352965  0.217056  0.023932   \n",
       "2     0.000000  0.000000  0.102156  0.302656  0.292165  0.000000  0.000000   \n",
       "3     0.168027  0.188040  0.072884  0.163923  0.400000  0.238917  0.020608   \n",
       "4     0.328875  0.391070  0.275135  0.030786  0.029746  0.040101  0.048513   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "9139  0.039068  0.007367  0.084344  0.381240  0.390044  0.120257  0.016993   \n",
       "9140  0.011986  0.291720  0.319846  0.328607  0.286253  0.143306  0.000000   \n",
       "9141  0.149710  0.099214  0.203577  0.200623  0.079059  0.098057  0.151877   \n",
       "9142  0.000000  0.000000  0.000070  0.000099  0.000000  0.000190  0.000465   \n",
       "9143  0.000000  0.001285  0.005373  0.006973  0.005226  0.004229  0.003136   \n",
       "\n",
       "             7         8         9  ...      1974      1975      1976   \n",
       "0     0.400000  0.167035  0.394578  ...  0.158582  0.228006  0.187460  \\\n",
       "1     0.016740  0.004268  0.194551  ...  0.226811  0.209077  0.000000   \n",
       "2     0.000000  0.000000  0.134023  ...  0.000000  0.000000  0.000000   \n",
       "3     0.006927  0.124900  0.113634  ...  0.210533  0.074725  0.250818   \n",
       "4     0.024524  0.032631  0.028272  ...  0.282160  0.127711  0.187435   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "9139  0.024952  0.014783  0.014419  ...  0.245075  0.207252  0.001578   \n",
       "9140  0.000000  0.001630  0.086209  ...  0.168624  0.024420  0.013202   \n",
       "9141  0.138999  0.136936  0.148251  ...  0.293888  0.348856  0.300672   \n",
       "9142  0.000000  0.000595  0.033671  ...  0.248481  0.222414  0.001434   \n",
       "9143  0.001720  0.001118  0.064458  ...  0.265564  0.188231  0.000000   \n",
       "\n",
       "          1977      1978      1979      1980      1981      1982      1983  \n",
       "0     0.172032  0.139624  0.293360  0.187875  0.254355  0.223806  0.189381  \n",
       "1     0.024079  0.108644  0.199408  0.093917  0.129881  0.140024  0.222356  \n",
       "2     0.000000  0.000422  0.297752  0.359863  0.011706  0.000000  0.000000  \n",
       "3     0.315082  0.323974  0.102916  0.000000  0.000000  0.000000  0.000000  \n",
       "4     0.278200  0.214201  0.259537  0.265158  0.290671  0.330026  0.244864  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "9139  0.144414  0.054743  0.077117  0.084481  0.280723  0.212753  0.262370  \n",
       "9140  0.103529  0.146153  0.116770  0.137495  0.121783  0.144657  0.000000  \n",
       "9141  0.323372  0.332723  0.315752  0.340056  0.354810  0.340426  0.344751  \n",
       "9142  0.072547  0.103934  0.063300  0.057856  0.129774  0.207614  0.186621  \n",
       "9143  0.031089  0.127476  0.165567  0.181716  0.200003  0.194710  0.010137  \n",
       "\n",
       "[9144 rows x 1984 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Xs[3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3098eb0e-c702-45b7-8887-2d6d2356dfa2",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>1974</th>\n",
       "      <th>1975</th>\n",
       "      <th>1976</th>\n",
       "      <th>1977</th>\n",
       "      <th>1978</th>\n",
       "      <th>1979</th>\n",
       "      <th>1980</th>\n",
       "      <th>1981</th>\n",
       "      <th>1982</th>\n",
       "      <th>1983</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.261661</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.343632</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.329767</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.167035</td>\n",
       "      <td>0.394578</td>\n",
       "      <td>...</td>\n",
       "      <td>0.158582</td>\n",
       "      <td>0.228006</td>\n",
       "      <td>0.187460</td>\n",
       "      <td>0.172032</td>\n",
       "      <td>0.139624</td>\n",
       "      <td>0.293360</td>\n",
       "      <td>0.187875</td>\n",
       "      <td>0.254355</td>\n",
       "      <td>0.223806</td>\n",
       "      <td>0.189381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.014952</td>\n",
       "      <td>0.053008</td>\n",
       "      <td>0.138431</td>\n",
       "      <td>0.391251</td>\n",
       "      <td>0.352965</td>\n",
       "      <td>0.217056</td>\n",
       "      <td>0.023932</td>\n",
       "      <td>0.016740</td>\n",
       "      <td>0.004268</td>\n",
       "      <td>0.194551</td>\n",
       "      <td>...</td>\n",
       "      <td>0.226811</td>\n",
       "      <td>0.209077</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.024079</td>\n",
       "      <td>0.108644</td>\n",
       "      <td>0.199408</td>\n",
       "      <td>0.093917</td>\n",
       "      <td>0.129881</td>\n",
       "      <td>0.140024</td>\n",
       "      <td>0.222356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.102156</td>\n",
       "      <td>0.302656</td>\n",
       "      <td>0.292165</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.134023</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000422</td>\n",
       "      <td>0.297752</td>\n",
       "      <td>0.359863</td>\n",
       "      <td>0.011706</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.168027</td>\n",
       "      <td>0.188040</td>\n",
       "      <td>0.072884</td>\n",
       "      <td>0.163923</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.238917</td>\n",
       "      <td>0.020608</td>\n",
       "      <td>0.006927</td>\n",
       "      <td>0.124900</td>\n",
       "      <td>0.113634</td>\n",
       "      <td>...</td>\n",
       "      <td>0.210533</td>\n",
       "      <td>0.074725</td>\n",
       "      <td>0.250818</td>\n",
       "      <td>0.315082</td>\n",
       "      <td>0.323974</td>\n",
       "      <td>0.102916</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.328875</td>\n",
       "      <td>0.391070</td>\n",
       "      <td>0.275135</td>\n",
       "      <td>0.030786</td>\n",
       "      <td>0.029746</td>\n",
       "      <td>0.040101</td>\n",
       "      <td>0.048513</td>\n",
       "      <td>0.024524</td>\n",
       "      <td>0.032631</td>\n",
       "      <td>0.028272</td>\n",
       "      <td>...</td>\n",
       "      <td>0.282160</td>\n",
       "      <td>0.127711</td>\n",
       "      <td>0.187435</td>\n",
       "      <td>0.278200</td>\n",
       "      <td>0.214201</td>\n",
       "      <td>0.259537</td>\n",
       "      <td>0.265158</td>\n",
       "      <td>0.290671</td>\n",
       "      <td>0.330026</td>\n",
       "      <td>0.244864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9148</th>\n",
       "      <td>0.039068</td>\n",
       "      <td>0.007367</td>\n",
       "      <td>0.084344</td>\n",
       "      <td>0.381240</td>\n",
       "      <td>0.390044</td>\n",
       "      <td>0.120257</td>\n",
       "      <td>0.016993</td>\n",
       "      <td>0.024952</td>\n",
       "      <td>0.014783</td>\n",
       "      <td>0.014419</td>\n",
       "      <td>...</td>\n",
       "      <td>0.245075</td>\n",
       "      <td>0.207252</td>\n",
       "      <td>0.001578</td>\n",
       "      <td>0.144414</td>\n",
       "      <td>0.054743</td>\n",
       "      <td>0.077117</td>\n",
       "      <td>0.084481</td>\n",
       "      <td>0.280723</td>\n",
       "      <td>0.212753</td>\n",
       "      <td>0.262370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9149</th>\n",
       "      <td>0.011986</td>\n",
       "      <td>0.291720</td>\n",
       "      <td>0.319846</td>\n",
       "      <td>0.328607</td>\n",
       "      <td>0.286253</td>\n",
       "      <td>0.143306</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001630</td>\n",
       "      <td>0.086209</td>\n",
       "      <td>...</td>\n",
       "      <td>0.168624</td>\n",
       "      <td>0.024420</td>\n",
       "      <td>0.013202</td>\n",
       "      <td>0.103529</td>\n",
       "      <td>0.146153</td>\n",
       "      <td>0.116770</td>\n",
       "      <td>0.137495</td>\n",
       "      <td>0.121783</td>\n",
       "      <td>0.144657</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9150</th>\n",
       "      <td>0.149710</td>\n",
       "      <td>0.099214</td>\n",
       "      <td>0.203577</td>\n",
       "      <td>0.200623</td>\n",
       "      <td>0.079059</td>\n",
       "      <td>0.098057</td>\n",
       "      <td>0.151877</td>\n",
       "      <td>0.138999</td>\n",
       "      <td>0.136936</td>\n",
       "      <td>0.148251</td>\n",
       "      <td>...</td>\n",
       "      <td>0.293888</td>\n",
       "      <td>0.348856</td>\n",
       "      <td>0.300672</td>\n",
       "      <td>0.323372</td>\n",
       "      <td>0.332723</td>\n",
       "      <td>0.315752</td>\n",
       "      <td>0.340056</td>\n",
       "      <td>0.354810</td>\n",
       "      <td>0.340426</td>\n",
       "      <td>0.344751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9151</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000070</td>\n",
       "      <td>0.000099</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000190</td>\n",
       "      <td>0.000465</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000595</td>\n",
       "      <td>0.033671</td>\n",
       "      <td>...</td>\n",
       "      <td>0.248481</td>\n",
       "      <td>0.222414</td>\n",
       "      <td>0.001434</td>\n",
       "      <td>0.072547</td>\n",
       "      <td>0.103934</td>\n",
       "      <td>0.063300</td>\n",
       "      <td>0.057856</td>\n",
       "      <td>0.129774</td>\n",
       "      <td>0.207614</td>\n",
       "      <td>0.186621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9152</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001285</td>\n",
       "      <td>0.005373</td>\n",
       "      <td>0.006973</td>\n",
       "      <td>0.005226</td>\n",
       "      <td>0.004229</td>\n",
       "      <td>0.003136</td>\n",
       "      <td>0.001720</td>\n",
       "      <td>0.001118</td>\n",
       "      <td>0.064458</td>\n",
       "      <td>...</td>\n",
       "      <td>0.265564</td>\n",
       "      <td>0.188231</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.031089</td>\n",
       "      <td>0.127476</td>\n",
       "      <td>0.165567</td>\n",
       "      <td>0.181716</td>\n",
       "      <td>0.200003</td>\n",
       "      <td>0.194710</td>\n",
       "      <td>0.010137</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9153 rows × 1984 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             0         1         2         3         4         5         6   \n",
       "0     0.261661  0.400000  0.400000  0.400000  0.343632  0.400000  0.329767  \\\n",
       "1     0.014952  0.053008  0.138431  0.391251  0.352965  0.217056  0.023932   \n",
       "2     0.000000  0.000000  0.102156  0.302656  0.292165  0.000000  0.000000   \n",
       "3     0.168027  0.188040  0.072884  0.163923  0.400000  0.238917  0.020608   \n",
       "4     0.328875  0.391070  0.275135  0.030786  0.029746  0.040101  0.048513   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "9148  0.039068  0.007367  0.084344  0.381240  0.390044  0.120257  0.016993   \n",
       "9149  0.011986  0.291720  0.319846  0.328607  0.286253  0.143306  0.000000   \n",
       "9150  0.149710  0.099214  0.203577  0.200623  0.079059  0.098057  0.151877   \n",
       "9151  0.000000  0.000000  0.000070  0.000099  0.000000  0.000190  0.000465   \n",
       "9152  0.000000  0.001285  0.005373  0.006973  0.005226  0.004229  0.003136   \n",
       "\n",
       "             7         8         9  ...      1974      1975      1976   \n",
       "0     0.400000  0.167035  0.394578  ...  0.158582  0.228006  0.187460  \\\n",
       "1     0.016740  0.004268  0.194551  ...  0.226811  0.209077  0.000000   \n",
       "2     0.000000  0.000000  0.134023  ...  0.000000  0.000000  0.000000   \n",
       "3     0.006927  0.124900  0.113634  ...  0.210533  0.074725  0.250818   \n",
       "4     0.024524  0.032631  0.028272  ...  0.282160  0.127711  0.187435   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "9148  0.024952  0.014783  0.014419  ...  0.245075  0.207252  0.001578   \n",
       "9149  0.000000  0.001630  0.086209  ...  0.168624  0.024420  0.013202   \n",
       "9150  0.138999  0.136936  0.148251  ...  0.293888  0.348856  0.300672   \n",
       "9151  0.000000  0.000595  0.033671  ...  0.248481  0.222414  0.001434   \n",
       "9152  0.001720  0.001118  0.064458  ...  0.265564  0.188231  0.000000   \n",
       "\n",
       "          1977      1978      1979      1980      1981      1982      1983  \n",
       "0     0.172032  0.139624  0.293360  0.187875  0.254355  0.223806  0.189381  \n",
       "1     0.024079  0.108644  0.199408  0.093917  0.129881  0.140024  0.222356  \n",
       "2     0.000000  0.000422  0.297752  0.359863  0.011706  0.000000  0.000000  \n",
       "3     0.315082  0.323974  0.102916  0.000000  0.000000  0.000000  0.000000  \n",
       "4     0.278200  0.214201  0.259537  0.265158  0.290671  0.330026  0.244864  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "9148  0.144414  0.054743  0.077117  0.084481  0.280723  0.212753  0.262370  \n",
       "9149  0.103529  0.146153  0.116770  0.137495  0.121783  0.144657  0.000000  \n",
       "9150  0.323372  0.332723  0.315752  0.340056  0.354810  0.340426  0.344751  \n",
       "9151  0.072547  0.103934  0.063300  0.057856  0.129774  0.207614  0.186621  \n",
       "9152  0.031089  0.127476  0.165567  0.181716  0.200003  0.194710  0.010137  \n",
       "\n",
       "[9153 rows x 1984 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"imvc/datasets/data/caltech101/upload/caltech101_3.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "id": "67c5d271-09f7-4737-b7f2-be99e681910f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "Xs[3].iloc[:500].to_csv(f\"imvc/datasets/data/caltech101/caltech101_3_{i}.csv\", index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e1cbd491-a603-4239-9613-128ded8ef096",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for i, mini_X in enumerate(np.array_split(Xs[3], 10)):\n",
    "    mini_X.to_csv(f\"imvc/datasets/data/caltech101/upload/caltech101_3_{i}.csv\", index= False)\n",
    "for i, mini_X in enumerate(np.array_split(Xs[5], 10)):\n",
    "    mini_X.to_csv(f\"imvc/datasets/data/caltech101/upload/caltech101_5_{i}.csv\", index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "id": "3c49d08c-fb28-4305-a4aa-9ee76c3456f7",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>finished</th>\n",
       "      <th>completed</th>\n",
       "      <th>n_samples</th>\n",
       "      <th>n_incomplete_samples</th>\n",
       "      <th>n_complete_samples</th>\n",
       "      <th>time</th>\n",
       "      <th>n_clustered_samples</th>\n",
       "      <th>percentage_clustered_samples</th>\n",
       "      <th>comments</th>\n",
       "      <th>stratified</th>\n",
       "      <th>...</th>\n",
       "      <th>bal_acc</th>\n",
       "      <th>ami</th>\n",
       "      <th>ari</th>\n",
       "      <th>completeness</th>\n",
       "      <th>random_acc</th>\n",
       "      <th>random_f1</th>\n",
       "      <th>silhouette</th>\n",
       "      <th>vrc</th>\n",
       "      <th>db</th>\n",
       "      <th>dunn</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"50\" valign=\"top\">simulated_InterSIM</th>\n",
       "      <th rowspan=\"10\" valign=\"top\">Concat</th>\n",
       "      <th rowspan=\"10\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"10\" valign=\"top\">True</th>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"10\" valign=\"top\">NMFC</th>\n",
       "      <th rowspan=\"10\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"10\" valign=\"top\">True</th>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"10\" valign=\"top\">MVSpectralClustering</th>\n",
       "      <th rowspan=\"10\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"10\" valign=\"top\">True</th>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"10\" valign=\"top\">MVCoRegSpectralClustering</th>\n",
       "      <th rowspan=\"10\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"10\" valign=\"top\">True</th>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"10\" valign=\"top\">GroupPCA</th>\n",
       "      <th rowspan=\"10\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"10\" valign=\"top\">True</th>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>50 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                  finished   \n",
       "dataset            algorithm                 missing_percentage imputation run_n             \n",
       "simulated_InterSIM Concat                    0                  True       0         False  \\\n",
       "                                                                           1         False   \n",
       "                                                                           2         False   \n",
       "                                                                           3         False   \n",
       "                                                                           4         False   \n",
       "                                                                           5         False   \n",
       "                                                                           6         False   \n",
       "                                                                           7         False   \n",
       "                                                                           8         False   \n",
       "                                                                           9         False   \n",
       "                   NMFC                      0                  True       0         False   \n",
       "                                                                           1         False   \n",
       "                                                                           2         False   \n",
       "                                                                           3         False   \n",
       "                                                                           4         False   \n",
       "                                                                           5         False   \n",
       "                                                                           6         False   \n",
       "                                                                           7         False   \n",
       "                                                                           8         False   \n",
       "                                                                           9         False   \n",
       "                   MVSpectralClustering      0                  True       0         False   \n",
       "                                                                           1         False   \n",
       "                                                                           2         False   \n",
       "                                                                           3         False   \n",
       "                                                                           4         False   \n",
       "                                                                           5         False   \n",
       "                                                                           6         False   \n",
       "                                                                           7         False   \n",
       "                                                                           8         False   \n",
       "                                                                           9         False   \n",
       "                   MVCoRegSpectralClustering 0                  True       0         False   \n",
       "                                                                           1         False   \n",
       "                                                                           2         False   \n",
       "                                                                           3         False   \n",
       "                                                                           4         False   \n",
       "                                                                           5         False   \n",
       "                                                                           6         False   \n",
       "                                                                           7         False   \n",
       "                                                                           8         False   \n",
       "                                                                           9         False   \n",
       "                   GroupPCA                  0                  True       0         False   \n",
       "                                                                           1         False   \n",
       "                                                                           2         False   \n",
       "                                                                           3         False   \n",
       "                                                                           4         False   \n",
       "                                                                           5         False   \n",
       "                                                                           6         False   \n",
       "                                                                           7         False   \n",
       "                                                                           8         False   \n",
       "                                                                           9         False   \n",
       "\n",
       "                                                                                  completed   \n",
       "dataset            algorithm                 missing_percentage imputation run_n              \n",
       "simulated_InterSIM Concat                    0                  True       0          False  \\\n",
       "                                                                           1          False   \n",
       "                                                                           2          False   \n",
       "                                                                           3          False   \n",
       "                                                                           4          False   \n",
       "                                                                           5          False   \n",
       "                                                                           6          False   \n",
       "                                                                           7          False   \n",
       "                                                                           8          False   \n",
       "                                                                           9          False   \n",
       "                   NMFC                      0                  True       0          False   \n",
       "                                                                           1          False   \n",
       "                                                                           2          False   \n",
       "                                                                           3          False   \n",
       "                                                                           4          False   \n",
       "                                                                           5          False   \n",
       "                                                                           6          False   \n",
       "                                                                           7          False   \n",
       "                                                                           8          False   \n",
       "                                                                           9          False   \n",
       "                   MVSpectralClustering      0                  True       0          False   \n",
       "                                                                           1          False   \n",
       "                                                                           2          False   \n",
       "                                                                           3          False   \n",
       "                                                                           4          False   \n",
       "                                                                           5          False   \n",
       "                                                                           6          False   \n",
       "                                                                           7          False   \n",
       "                                                                           8          False   \n",
       "                                                                           9          False   \n",
       "                   MVCoRegSpectralClustering 0                  True       0          False   \n",
       "                                                                           1          False   \n",
       "                                                                           2          False   \n",
       "                                                                           3          False   \n",
       "                                                                           4          False   \n",
       "                                                                           5          False   \n",
       "                                                                           6          False   \n",
       "                                                                           7          False   \n",
       "                                                                           8          False   \n",
       "                                                                           9          False   \n",
       "                   GroupPCA                  0                  True       0          False   \n",
       "                                                                           1          False   \n",
       "                                                                           2          False   \n",
       "                                                                           3          False   \n",
       "                                                                           4          False   \n",
       "                                                                           5          False   \n",
       "                                                                           6          False   \n",
       "                                                                           7          False   \n",
       "                                                                           8          False   \n",
       "                                                                           9          False   \n",
       "\n",
       "                                                                                  n_samples   \n",
       "dataset            algorithm                 missing_percentage imputation run_n              \n",
       "simulated_InterSIM Concat                    0                  True       0            NaN  \\\n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   NMFC                      0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   MVSpectralClustering      0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   GroupPCA                  0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "\n",
       "                                                                                  n_incomplete_samples   \n",
       "dataset            algorithm                 missing_percentage imputation run_n                         \n",
       "simulated_InterSIM Concat                    0                  True       0                       NaN  \\\n",
       "                                                                           1                       NaN   \n",
       "                                                                           2                       NaN   \n",
       "                                                                           3                       NaN   \n",
       "                                                                           4                       NaN   \n",
       "                                                                           5                       NaN   \n",
       "                                                                           6                       NaN   \n",
       "                                                                           7                       NaN   \n",
       "                                                                           8                       NaN   \n",
       "                                                                           9                       NaN   \n",
       "                   NMFC                      0                  True       0                       NaN   \n",
       "                                                                           1                       NaN   \n",
       "                                                                           2                       NaN   \n",
       "                                                                           3                       NaN   \n",
       "                                                                           4                       NaN   \n",
       "                                                                           5                       NaN   \n",
       "                                                                           6                       NaN   \n",
       "                                                                           7                       NaN   \n",
       "                                                                           8                       NaN   \n",
       "                                                                           9                       NaN   \n",
       "                   MVSpectralClustering      0                  True       0                       NaN   \n",
       "                                                                           1                       NaN   \n",
       "                                                                           2                       NaN   \n",
       "                                                                           3                       NaN   \n",
       "                                                                           4                       NaN   \n",
       "                                                                           5                       NaN   \n",
       "                                                                           6                       NaN   \n",
       "                                                                           7                       NaN   \n",
       "                                                                           8                       NaN   \n",
       "                                                                           9                       NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0                       NaN   \n",
       "                                                                           1                       NaN   \n",
       "                                                                           2                       NaN   \n",
       "                                                                           3                       NaN   \n",
       "                                                                           4                       NaN   \n",
       "                                                                           5                       NaN   \n",
       "                                                                           6                       NaN   \n",
       "                                                                           7                       NaN   \n",
       "                                                                           8                       NaN   \n",
       "                                                                           9                       NaN   \n",
       "                   GroupPCA                  0                  True       0                       NaN   \n",
       "                                                                           1                       NaN   \n",
       "                                                                           2                       NaN   \n",
       "                                                                           3                       NaN   \n",
       "                                                                           4                       NaN   \n",
       "                                                                           5                       NaN   \n",
       "                                                                           6                       NaN   \n",
       "                                                                           7                       NaN   \n",
       "                                                                           8                       NaN   \n",
       "                                                                           9                       NaN   \n",
       "\n",
       "                                                                                  n_complete_samples   \n",
       "dataset            algorithm                 missing_percentage imputation run_n                       \n",
       "simulated_InterSIM Concat                    0                  True       0                     NaN  \\\n",
       "                                                                           1                     NaN   \n",
       "                                                                           2                     NaN   \n",
       "                                                                           3                     NaN   \n",
       "                                                                           4                     NaN   \n",
       "                                                                           5                     NaN   \n",
       "                                                                           6                     NaN   \n",
       "                                                                           7                     NaN   \n",
       "                                                                           8                     NaN   \n",
       "                                                                           9                     NaN   \n",
       "                   NMFC                      0                  True       0                     NaN   \n",
       "                                                                           1                     NaN   \n",
       "                                                                           2                     NaN   \n",
       "                                                                           3                     NaN   \n",
       "                                                                           4                     NaN   \n",
       "                                                                           5                     NaN   \n",
       "                                                                           6                     NaN   \n",
       "                                                                           7                     NaN   \n",
       "                                                                           8                     NaN   \n",
       "                                                                           9                     NaN   \n",
       "                   MVSpectralClustering      0                  True       0                     NaN   \n",
       "                                                                           1                     NaN   \n",
       "                                                                           2                     NaN   \n",
       "                                                                           3                     NaN   \n",
       "                                                                           4                     NaN   \n",
       "                                                                           5                     NaN   \n",
       "                                                                           6                     NaN   \n",
       "                                                                           7                     NaN   \n",
       "                                                                           8                     NaN   \n",
       "                                                                           9                     NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0                     NaN   \n",
       "                                                                           1                     NaN   \n",
       "                                                                           2                     NaN   \n",
       "                                                                           3                     NaN   \n",
       "                                                                           4                     NaN   \n",
       "                                                                           5                     NaN   \n",
       "                                                                           6                     NaN   \n",
       "                                                                           7                     NaN   \n",
       "                                                                           8                     NaN   \n",
       "                                                                           9                     NaN   \n",
       "                   GroupPCA                  0                  True       0                     NaN   \n",
       "                                                                           1                     NaN   \n",
       "                                                                           2                     NaN   \n",
       "                                                                           3                     NaN   \n",
       "                                                                           4                     NaN   \n",
       "                                                                           5                     NaN   \n",
       "                                                                           6                     NaN   \n",
       "                                                                           7                     NaN   \n",
       "                                                                           8                     NaN   \n",
       "                                                                           9                     NaN   \n",
       "\n",
       "                                                                                  time   \n",
       "dataset            algorithm                 missing_percentage imputation run_n         \n",
       "simulated_InterSIM Concat                    0                  True       0       NaN  \\\n",
       "                                                                           1       NaN   \n",
       "                                                                           2       NaN   \n",
       "                                                                           3       NaN   \n",
       "                                                                           4       NaN   \n",
       "                                                                           5       NaN   \n",
       "                                                                           6       NaN   \n",
       "                                                                           7       NaN   \n",
       "                                                                           8       NaN   \n",
       "                                                                           9       NaN   \n",
       "                   NMFC                      0                  True       0       NaN   \n",
       "                                                                           1       NaN   \n",
       "                                                                           2       NaN   \n",
       "                                                                           3       NaN   \n",
       "                                                                           4       NaN   \n",
       "                                                                           5       NaN   \n",
       "                                                                           6       NaN   \n",
       "                                                                           7       NaN   \n",
       "                                                                           8       NaN   \n",
       "                                                                           9       NaN   \n",
       "                   MVSpectralClustering      0                  True       0       NaN   \n",
       "                                                                           1       NaN   \n",
       "                                                                           2       NaN   \n",
       "                                                                           3       NaN   \n",
       "                                                                           4       NaN   \n",
       "                                                                           5       NaN   \n",
       "                                                                           6       NaN   \n",
       "                                                                           7       NaN   \n",
       "                                                                           8       NaN   \n",
       "                                                                           9       NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0       NaN   \n",
       "                                                                           1       NaN   \n",
       "                                                                           2       NaN   \n",
       "                                                                           3       NaN   \n",
       "                                                                           4       NaN   \n",
       "                                                                           5       NaN   \n",
       "                                                                           6       NaN   \n",
       "                                                                           7       NaN   \n",
       "                                                                           8       NaN   \n",
       "                                                                           9       NaN   \n",
       "                   GroupPCA                  0                  True       0       NaN   \n",
       "                                                                           1       NaN   \n",
       "                                                                           2       NaN   \n",
       "                                                                           3       NaN   \n",
       "                                                                           4       NaN   \n",
       "                                                                           5       NaN   \n",
       "                                                                           6       NaN   \n",
       "                                                                           7       NaN   \n",
       "                                                                           8       NaN   \n",
       "                                                                           9       NaN   \n",
       "\n",
       "                                                                                  n_clustered_samples   \n",
       "dataset            algorithm                 missing_percentage imputation run_n                        \n",
       "simulated_InterSIM Concat                    0                  True       0                      NaN  \\\n",
       "                                                                           1                      NaN   \n",
       "                                                                           2                      NaN   \n",
       "                                                                           3                      NaN   \n",
       "                                                                           4                      NaN   \n",
       "                                                                           5                      NaN   \n",
       "                                                                           6                      NaN   \n",
       "                                                                           7                      NaN   \n",
       "                                                                           8                      NaN   \n",
       "                                                                           9                      NaN   \n",
       "                   NMFC                      0                  True       0                      NaN   \n",
       "                                                                           1                      NaN   \n",
       "                                                                           2                      NaN   \n",
       "                                                                           3                      NaN   \n",
       "                                                                           4                      NaN   \n",
       "                                                                           5                      NaN   \n",
       "                                                                           6                      NaN   \n",
       "                                                                           7                      NaN   \n",
       "                                                                           8                      NaN   \n",
       "                                                                           9                      NaN   \n",
       "                   MVSpectralClustering      0                  True       0                      NaN   \n",
       "                                                                           1                      NaN   \n",
       "                                                                           2                      NaN   \n",
       "                                                                           3                      NaN   \n",
       "                                                                           4                      NaN   \n",
       "                                                                           5                      NaN   \n",
       "                                                                           6                      NaN   \n",
       "                                                                           7                      NaN   \n",
       "                                                                           8                      NaN   \n",
       "                                                                           9                      NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0                      NaN   \n",
       "                                                                           1                      NaN   \n",
       "                                                                           2                      NaN   \n",
       "                                                                           3                      NaN   \n",
       "                                                                           4                      NaN   \n",
       "                                                                           5                      NaN   \n",
       "                                                                           6                      NaN   \n",
       "                                                                           7                      NaN   \n",
       "                                                                           8                      NaN   \n",
       "                                                                           9                      NaN   \n",
       "                   GroupPCA                  0                  True       0                      NaN   \n",
       "                                                                           1                      NaN   \n",
       "                                                                           2                      NaN   \n",
       "                                                                           3                      NaN   \n",
       "                                                                           4                      NaN   \n",
       "                                                                           5                      NaN   \n",
       "                                                                           6                      NaN   \n",
       "                                                                           7                      NaN   \n",
       "                                                                           8                      NaN   \n",
       "                                                                           9                      NaN   \n",
       "\n",
       "                                                                                  percentage_clustered_samples   \n",
       "dataset            algorithm                 missing_percentage imputation run_n                                 \n",
       "simulated_InterSIM Concat                    0                  True       0                               NaN  \\\n",
       "                                                                           1                               NaN   \n",
       "                                                                           2                               NaN   \n",
       "                                                                           3                               NaN   \n",
       "                                                                           4                               NaN   \n",
       "                                                                           5                               NaN   \n",
       "                                                                           6                               NaN   \n",
       "                                                                           7                               NaN   \n",
       "                                                                           8                               NaN   \n",
       "                                                                           9                               NaN   \n",
       "                   NMFC                      0                  True       0                               NaN   \n",
       "                                                                           1                               NaN   \n",
       "                                                                           2                               NaN   \n",
       "                                                                           3                               NaN   \n",
       "                                                                           4                               NaN   \n",
       "                                                                           5                               NaN   \n",
       "                                                                           6                               NaN   \n",
       "                                                                           7                               NaN   \n",
       "                                                                           8                               NaN   \n",
       "                                                                           9                               NaN   \n",
       "                   MVSpectralClustering      0                  True       0                               NaN   \n",
       "                                                                           1                               NaN   \n",
       "                                                                           2                               NaN   \n",
       "                                                                           3                               NaN   \n",
       "                                                                           4                               NaN   \n",
       "                                                                           5                               NaN   \n",
       "                                                                           6                               NaN   \n",
       "                                                                           7                               NaN   \n",
       "                                                                           8                               NaN   \n",
       "                                                                           9                               NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0                               NaN   \n",
       "                                                                           1                               NaN   \n",
       "                                                                           2                               NaN   \n",
       "                                                                           3                               NaN   \n",
       "                                                                           4                               NaN   \n",
       "                                                                           5                               NaN   \n",
       "                                                                           6                               NaN   \n",
       "                                                                           7                               NaN   \n",
       "                                                                           8                               NaN   \n",
       "                                                                           9                               NaN   \n",
       "                   GroupPCA                  0                  True       0                               NaN   \n",
       "                                                                           1                               NaN   \n",
       "                                                                           2                               NaN   \n",
       "                                                                           3                               NaN   \n",
       "                                                                           4                               NaN   \n",
       "                                                                           5                               NaN   \n",
       "                                                                           6                               NaN   \n",
       "                                                                           7                               NaN   \n",
       "                                                                           8                               NaN   \n",
       "                                                                           9                               NaN   \n",
       "\n",
       "                                                                                 comments   \n",
       "dataset            algorithm                 missing_percentage imputation run_n            \n",
       "simulated_InterSIM Concat                    0                  True       0          NaN  \\\n",
       "                                                                           1          NaN   \n",
       "                                                                           2          NaN   \n",
       "                                                                           3          NaN   \n",
       "                                                                           4          NaN   \n",
       "                                                                           5          NaN   \n",
       "                                                                           6          NaN   \n",
       "                                                                           7          NaN   \n",
       "                                                                           8          NaN   \n",
       "                                                                           9          NaN   \n",
       "                   NMFC                      0                  True       0          NaN   \n",
       "                                                                           1          NaN   \n",
       "                                                                           2          NaN   \n",
       "                                                                           3          NaN   \n",
       "                                                                           4          NaN   \n",
       "                                                                           5          NaN   \n",
       "                                                                           6          NaN   \n",
       "                                                                           7          NaN   \n",
       "                                                                           8          NaN   \n",
       "                                                                           9          NaN   \n",
       "                   MVSpectralClustering      0                  True       0          NaN   \n",
       "                                                                           1          NaN   \n",
       "                                                                           2          NaN   \n",
       "                                                                           3          NaN   \n",
       "                                                                           4          NaN   \n",
       "                                                                           5          NaN   \n",
       "                                                                           6          NaN   \n",
       "                                                                           7          NaN   \n",
       "                                                                           8          NaN   \n",
       "                                                                           9          NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0          NaN   \n",
       "                                                                           1          NaN   \n",
       "                                                                           2          NaN   \n",
       "                                                                           3          NaN   \n",
       "                                                                           4          NaN   \n",
       "                                                                           5          NaN   \n",
       "                                                                           6          NaN   \n",
       "                                                                           7          NaN   \n",
       "                                                                           8          NaN   \n",
       "                                                                           9          NaN   \n",
       "                   GroupPCA                  0                  True       0          NaN   \n",
       "                                                                           1          NaN   \n",
       "                                                                           2          NaN   \n",
       "                                                                           3          NaN   \n",
       "                                                                           4          NaN   \n",
       "                                                                           5          NaN   \n",
       "                                                                           6          NaN   \n",
       "                                                                           7          NaN   \n",
       "                                                                           8          NaN   \n",
       "                                                                           9          NaN   \n",
       "\n",
       "                                                                                 stratified   \n",
       "dataset            algorithm                 missing_percentage imputation run_n              \n",
       "simulated_InterSIM Concat                    0                  True       0            NaN  \\\n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   NMFC                      0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   MVSpectralClustering      0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   GroupPCA                  0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "\n",
       "                                                                                  ...   \n",
       "dataset            algorithm                 missing_percentage imputation run_n  ...   \n",
       "simulated_InterSIM Concat                    0                  True       0      ...  \\\n",
       "                                                                           1      ...   \n",
       "                                                                           2      ...   \n",
       "                                                                           3      ...   \n",
       "                                                                           4      ...   \n",
       "                                                                           5      ...   \n",
       "                                                                           6      ...   \n",
       "                                                                           7      ...   \n",
       "                                                                           8      ...   \n",
       "                                                                           9      ...   \n",
       "                   NMFC                      0                  True       0      ...   \n",
       "                                                                           1      ...   \n",
       "                                                                           2      ...   \n",
       "                                                                           3      ...   \n",
       "                                                                           4      ...   \n",
       "                                                                           5      ...   \n",
       "                                                                           6      ...   \n",
       "                                                                           7      ...   \n",
       "                                                                           8      ...   \n",
       "                                                                           9      ...   \n",
       "                   MVSpectralClustering      0                  True       0      ...   \n",
       "                                                                           1      ...   \n",
       "                                                                           2      ...   \n",
       "                                                                           3      ...   \n",
       "                                                                           4      ...   \n",
       "                                                                           5      ...   \n",
       "                                                                           6      ...   \n",
       "                                                                           7      ...   \n",
       "                                                                           8      ...   \n",
       "                                                                           9      ...   \n",
       "                   MVCoRegSpectralClustering 0                  True       0      ...   \n",
       "                                                                           1      ...   \n",
       "                                                                           2      ...   \n",
       "                                                                           3      ...   \n",
       "                                                                           4      ...   \n",
       "                                                                           5      ...   \n",
       "                                                                           6      ...   \n",
       "                                                                           7      ...   \n",
       "                                                                           8      ...   \n",
       "                                                                           9      ...   \n",
       "                   GroupPCA                  0                  True       0      ...   \n",
       "                                                                           1      ...   \n",
       "                                                                           2      ...   \n",
       "                                                                           3      ...   \n",
       "                                                                           4      ...   \n",
       "                                                                           5      ...   \n",
       "                                                                           6      ...   \n",
       "                                                                           7      ...   \n",
       "                                                                           8      ...   \n",
       "                                                                           9      ...   \n",
       "\n",
       "                                                                                 bal_acc   \n",
       "dataset            algorithm                 missing_percentage imputation run_n           \n",
       "simulated_InterSIM Concat                    0                  True       0         NaN  \\\n",
       "                                                                           1         NaN   \n",
       "                                                                           2         NaN   \n",
       "                                                                           3         NaN   \n",
       "                                                                           4         NaN   \n",
       "                                                                           5         NaN   \n",
       "                                                                           6         NaN   \n",
       "                                                                           7         NaN   \n",
       "                                                                           8         NaN   \n",
       "                                                                           9         NaN   \n",
       "                   NMFC                      0                  True       0         NaN   \n",
       "                                                                           1         NaN   \n",
       "                                                                           2         NaN   \n",
       "                                                                           3         NaN   \n",
       "                                                                           4         NaN   \n",
       "                                                                           5         NaN   \n",
       "                                                                           6         NaN   \n",
       "                                                                           7         NaN   \n",
       "                                                                           8         NaN   \n",
       "                                                                           9         NaN   \n",
       "                   MVSpectralClustering      0                  True       0         NaN   \n",
       "                                                                           1         NaN   \n",
       "                                                                           2         NaN   \n",
       "                                                                           3         NaN   \n",
       "                                                                           4         NaN   \n",
       "                                                                           5         NaN   \n",
       "                                                                           6         NaN   \n",
       "                                                                           7         NaN   \n",
       "                                                                           8         NaN   \n",
       "                                                                           9         NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0         NaN   \n",
       "                                                                           1         NaN   \n",
       "                                                                           2         NaN   \n",
       "                                                                           3         NaN   \n",
       "                                                                           4         NaN   \n",
       "                                                                           5         NaN   \n",
       "                                                                           6         NaN   \n",
       "                                                                           7         NaN   \n",
       "                                                                           8         NaN   \n",
       "                                                                           9         NaN   \n",
       "                   GroupPCA                  0                  True       0         NaN   \n",
       "                                                                           1         NaN   \n",
       "                                                                           2         NaN   \n",
       "                                                                           3         NaN   \n",
       "                                                                           4         NaN   \n",
       "                                                                           5         NaN   \n",
       "                                                                           6         NaN   \n",
       "                                                                           7         NaN   \n",
       "                                                                           8         NaN   \n",
       "                                                                           9         NaN   \n",
       "\n",
       "                                                                                 ami   \n",
       "dataset            algorithm                 missing_percentage imputation run_n       \n",
       "simulated_InterSIM Concat                    0                  True       0     NaN  \\\n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   NMFC                      0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   MVSpectralClustering      0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   GroupPCA                  0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "\n",
       "                                                                                 ari   \n",
       "dataset            algorithm                 missing_percentage imputation run_n       \n",
       "simulated_InterSIM Concat                    0                  True       0     NaN  \\\n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   NMFC                      0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   MVSpectralClustering      0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   GroupPCA                  0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "\n",
       "                                                                                 completeness   \n",
       "dataset            algorithm                 missing_percentage imputation run_n                \n",
       "simulated_InterSIM Concat                    0                  True       0              NaN  \\\n",
       "                                                                           1              NaN   \n",
       "                                                                           2              NaN   \n",
       "                                                                           3              NaN   \n",
       "                                                                           4              NaN   \n",
       "                                                                           5              NaN   \n",
       "                                                                           6              NaN   \n",
       "                                                                           7              NaN   \n",
       "                                                                           8              NaN   \n",
       "                                                                           9              NaN   \n",
       "                   NMFC                      0                  True       0              NaN   \n",
       "                                                                           1              NaN   \n",
       "                                                                           2              NaN   \n",
       "                                                                           3              NaN   \n",
       "                                                                           4              NaN   \n",
       "                                                                           5              NaN   \n",
       "                                                                           6              NaN   \n",
       "                                                                           7              NaN   \n",
       "                                                                           8              NaN   \n",
       "                                                                           9              NaN   \n",
       "                   MVSpectralClustering      0                  True       0              NaN   \n",
       "                                                                           1              NaN   \n",
       "                                                                           2              NaN   \n",
       "                                                                           3              NaN   \n",
       "                                                                           4              NaN   \n",
       "                                                                           5              NaN   \n",
       "                                                                           6              NaN   \n",
       "                                                                           7              NaN   \n",
       "                                                                           8              NaN   \n",
       "                                                                           9              NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0              NaN   \n",
       "                                                                           1              NaN   \n",
       "                                                                           2              NaN   \n",
       "                                                                           3              NaN   \n",
       "                                                                           4              NaN   \n",
       "                                                                           5              NaN   \n",
       "                                                                           6              NaN   \n",
       "                                                                           7              NaN   \n",
       "                                                                           8              NaN   \n",
       "                                                                           9              NaN   \n",
       "                   GroupPCA                  0                  True       0              NaN   \n",
       "                                                                           1              NaN   \n",
       "                                                                           2              NaN   \n",
       "                                                                           3              NaN   \n",
       "                                                                           4              NaN   \n",
       "                                                                           5              NaN   \n",
       "                                                                           6              NaN   \n",
       "                                                                           7              NaN   \n",
       "                                                                           8              NaN   \n",
       "                                                                           9              NaN   \n",
       "\n",
       "                                                                                 random_acc   \n",
       "dataset            algorithm                 missing_percentage imputation run_n              \n",
       "simulated_InterSIM Concat                    0                  True       0            NaN  \\\n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   NMFC                      0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   MVSpectralClustering      0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   GroupPCA                  0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "\n",
       "                                                                                 random_f1   \n",
       "dataset            algorithm                 missing_percentage imputation run_n             \n",
       "simulated_InterSIM Concat                    0                  True       0           NaN  \\\n",
       "                                                                           1           NaN   \n",
       "                                                                           2           NaN   \n",
       "                                                                           3           NaN   \n",
       "                                                                           4           NaN   \n",
       "                                                                           5           NaN   \n",
       "                                                                           6           NaN   \n",
       "                                                                           7           NaN   \n",
       "                                                                           8           NaN   \n",
       "                                                                           9           NaN   \n",
       "                   NMFC                      0                  True       0           NaN   \n",
       "                                                                           1           NaN   \n",
       "                                                                           2           NaN   \n",
       "                                                                           3           NaN   \n",
       "                                                                           4           NaN   \n",
       "                                                                           5           NaN   \n",
       "                                                                           6           NaN   \n",
       "                                                                           7           NaN   \n",
       "                                                                           8           NaN   \n",
       "                                                                           9           NaN   \n",
       "                   MVSpectralClustering      0                  True       0           NaN   \n",
       "                                                                           1           NaN   \n",
       "                                                                           2           NaN   \n",
       "                                                                           3           NaN   \n",
       "                                                                           4           NaN   \n",
       "                                                                           5           NaN   \n",
       "                                                                           6           NaN   \n",
       "                                                                           7           NaN   \n",
       "                                                                           8           NaN   \n",
       "                                                                           9           NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0           NaN   \n",
       "                                                                           1           NaN   \n",
       "                                                                           2           NaN   \n",
       "                                                                           3           NaN   \n",
       "                                                                           4           NaN   \n",
       "                                                                           5           NaN   \n",
       "                                                                           6           NaN   \n",
       "                                                                           7           NaN   \n",
       "                                                                           8           NaN   \n",
       "                                                                           9           NaN   \n",
       "                   GroupPCA                  0                  True       0           NaN   \n",
       "                                                                           1           NaN   \n",
       "                                                                           2           NaN   \n",
       "                                                                           3           NaN   \n",
       "                                                                           4           NaN   \n",
       "                                                                           5           NaN   \n",
       "                                                                           6           NaN   \n",
       "                                                                           7           NaN   \n",
       "                                                                           8           NaN   \n",
       "                                                                           9           NaN   \n",
       "\n",
       "                                                                                 silhouette   \n",
       "dataset            algorithm                 missing_percentage imputation run_n              \n",
       "simulated_InterSIM Concat                    0                  True       0            NaN  \\\n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   NMFC                      0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   MVSpectralClustering      0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "                   GroupPCA                  0                  True       0            NaN   \n",
       "                                                                           1            NaN   \n",
       "                                                                           2            NaN   \n",
       "                                                                           3            NaN   \n",
       "                                                                           4            NaN   \n",
       "                                                                           5            NaN   \n",
       "                                                                           6            NaN   \n",
       "                                                                           7            NaN   \n",
       "                                                                           8            NaN   \n",
       "                                                                           9            NaN   \n",
       "\n",
       "                                                                                 vrc   \n",
       "dataset            algorithm                 missing_percentage imputation run_n       \n",
       "simulated_InterSIM Concat                    0                  True       0     NaN  \\\n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   NMFC                      0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   MVSpectralClustering      0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   GroupPCA                  0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "\n",
       "                                                                                  db   \n",
       "dataset            algorithm                 missing_percentage imputation run_n       \n",
       "simulated_InterSIM Concat                    0                  True       0     NaN  \\\n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   NMFC                      0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   MVSpectralClustering      0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   MVCoRegSpectralClustering 0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "                   GroupPCA                  0                  True       0     NaN   \n",
       "                                                                           1     NaN   \n",
       "                                                                           2     NaN   \n",
       "                                                                           3     NaN   \n",
       "                                                                           4     NaN   \n",
       "                                                                           5     NaN   \n",
       "                                                                           6     NaN   \n",
       "                                                                           7     NaN   \n",
       "                                                                           8     NaN   \n",
       "                                                                           9     NaN   \n",
       "\n",
       "                                                                                  dunn  \n",
       "dataset            algorithm                 missing_percentage imputation run_n        \n",
       "simulated_InterSIM Concat                    0                  True       0       NaN  \n",
       "                                                                           1       NaN  \n",
       "                                                                           2       NaN  \n",
       "                                                                           3       NaN  \n",
       "                                                                           4       NaN  \n",
       "                                                                           5       NaN  \n",
       "                                                                           6       NaN  \n",
       "                                                                           7       NaN  \n",
       "                                                                           8       NaN  \n",
       "                                                                           9       NaN  \n",
       "                   NMFC                      0                  True       0       NaN  \n",
       "                                                                           1       NaN  \n",
       "                                                                           2       NaN  \n",
       "                                                                           3       NaN  \n",
       "                                                                           4       NaN  \n",
       "                                                                           5       NaN  \n",
       "                                                                           6       NaN  \n",
       "                                                                           7       NaN  \n",
       "                                                                           8       NaN  \n",
       "                                                                           9       NaN  \n",
       "                   MVSpectralClustering      0                  True       0       NaN  \n",
       "                                                                           1       NaN  \n",
       "                                                                           2       NaN  \n",
       "                                                                           3       NaN  \n",
       "                                                                           4       NaN  \n",
       "                                                                           5       NaN  \n",
       "                                                                           6       NaN  \n",
       "                                                                           7       NaN  \n",
       "                                                                           8       NaN  \n",
       "                                                                           9       NaN  \n",
       "                   MVCoRegSpectralClustering 0                  True       0       NaN  \n",
       "                                                                           1       NaN  \n",
       "                                                                           2       NaN  \n",
       "                                                                           3       NaN  \n",
       "                                                                           4       NaN  \n",
       "                                                                           5       NaN  \n",
       "                                                                           6       NaN  \n",
       "                                                                           7       NaN  \n",
       "                                                                           8       NaN  \n",
       "                                                                           9       NaN  \n",
       "                   GroupPCA                  0                  True       0       NaN  \n",
       "                                                                           1       NaN  \n",
       "                                                                           2       NaN  \n",
       "                                                                           3       NaN  \n",
       "                                                                           4       NaN  \n",
       "                                                                           5       NaN  \n",
       "                                                                           6       NaN  \n",
       "                                                                           7       NaN  \n",
       "                                                                           8       NaN  \n",
       "                                                                           9       NaN  \n",
       "\n",
       "[50 rows x 33 columns]"
      ]
     },
     "execution_count": 205,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results.xs(0, level= \"missing_percentage\", drop_level=False).xs(True, level= \"imputation\", drop_level=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "de24f30f-8146-4a45-9410-0d255038ebdd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>finished</th>\n",
       "      <th>completed</th>\n",
       "      <th>n_samples</th>\n",
       "      <th>n_incomplete_samples</th>\n",
       "      <th>n_complete_samples</th>\n",
       "      <th>time</th>\n",
       "      <th>n_clustered_samples</th>\n",
       "      <th>percentage_clustered_samples</th>\n",
       "      <th>comments</th>\n",
       "      <th>stratified</th>\n",
       "      <th>...</th>\n",
       "      <th>MCC_artificial_performance</th>\n",
       "      <th>F1_artificial_performance</th>\n",
       "      <th>precision_artificial_performance</th>\n",
       "      <th>recall_artificial_performance</th>\n",
       "      <th>bal_acc_artificial_performance</th>\n",
       "      <th>ami_artificial_performance</th>\n",
       "      <th>ari_artificial_performance</th>\n",
       "      <th>completeness_artificial_performance</th>\n",
       "      <th>random_acc_artificial_performance</th>\n",
       "      <th>random_f1_artificial_performance</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nutrimouse_diet</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">AJIVE</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">30</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">False</th>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">40</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">True</th>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">metabric</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">jNMF</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">90</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14942 rows × 77 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                               finished   \n",
       "dataset         algorithm missing_percentage imputation run_n             \n",
       "nutrimouse_diet AJIVE     30                 False      8         False  \\\n",
       "                                                        9         False   \n",
       "                          40                 True       0         False   \n",
       "                                                        1         False   \n",
       "                                                        2         False   \n",
       "...                                                                 ...   \n",
       "metabric        jNMF      90                 False      5         False   \n",
       "                                                        6         False   \n",
       "                                                        7         False   \n",
       "                                                        8         False   \n",
       "                                                        9         False   \n",
       "\n",
       "                                                               completed   \n",
       "dataset         algorithm missing_percentage imputation run_n              \n",
       "nutrimouse_diet AJIVE     30                 False      8          False  \\\n",
       "                                                        9          False   \n",
       "                          40                 True       0          False   \n",
       "                                                        1          False   \n",
       "                                                        2          False   \n",
       "...                                                                  ...   \n",
       "metabric        jNMF      90                 False      5          False   \n",
       "                                                        6          False   \n",
       "                                                        7          False   \n",
       "                                                        8          False   \n",
       "                                                        9          False   \n",
       "\n",
       "                                                               n_samples   \n",
       "dataset         algorithm missing_percentage imputation run_n              \n",
       "nutrimouse_diet AJIVE     30                 False      8            NaN  \\\n",
       "                                                        9            NaN   \n",
       "                          40                 True       0            NaN   \n",
       "                                                        1            NaN   \n",
       "                                                        2            NaN   \n",
       "...                                                                  ...   \n",
       "metabric        jNMF      90                 False      5            NaN   \n",
       "                                                        6            NaN   \n",
       "                                                        7            NaN   \n",
       "                                                        8            NaN   \n",
       "                                                        9            NaN   \n",
       "\n",
       "                                                               n_incomplete_samples   \n",
       "dataset         algorithm missing_percentage imputation run_n                         \n",
       "nutrimouse_diet AJIVE     30                 False      8                       NaN  \\\n",
       "                                                        9                       NaN   \n",
       "                          40                 True       0                       NaN   \n",
       "                                                        1                       NaN   \n",
       "                                                        2                       NaN   \n",
       "...                                                                             ...   \n",
       "metabric        jNMF      90                 False      5                       NaN   \n",
       "                                                        6                       NaN   \n",
       "                                                        7                       NaN   \n",
       "                                                        8                       NaN   \n",
       "                                                        9                       NaN   \n",
       "\n",
       "                                                               n_complete_samples   \n",
       "dataset         algorithm missing_percentage imputation run_n                       \n",
       "nutrimouse_diet AJIVE     30                 False      8                     NaN  \\\n",
       "                                                        9                     NaN   \n",
       "                          40                 True       0                     NaN   \n",
       "                                                        1                     NaN   \n",
       "                                                        2                     NaN   \n",
       "...                                                                           ...   \n",
       "metabric        jNMF      90                 False      5                     NaN   \n",
       "                                                        6                     NaN   \n",
       "                                                        7                     NaN   \n",
       "                                                        8                     NaN   \n",
       "                                                        9                     NaN   \n",
       "\n",
       "                                                               time   \n",
       "dataset         algorithm missing_percentage imputation run_n         \n",
       "nutrimouse_diet AJIVE     30                 False      8       NaN  \\\n",
       "                                                        9       NaN   \n",
       "                          40                 True       0       NaN   \n",
       "                                                        1       NaN   \n",
       "                                                        2       NaN   \n",
       "...                                                             ...   \n",
       "metabric        jNMF      90                 False      5       NaN   \n",
       "                                                        6       NaN   \n",
       "                                                        7       NaN   \n",
       "                                                        8       NaN   \n",
       "                                                        9       NaN   \n",
       "\n",
       "                                                               n_clustered_samples   \n",
       "dataset         algorithm missing_percentage imputation run_n                        \n",
       "nutrimouse_diet AJIVE     30                 False      8                      NaN  \\\n",
       "                                                        9                      NaN   \n",
       "                          40                 True       0                      NaN   \n",
       "                                                        1                      NaN   \n",
       "                                                        2                      NaN   \n",
       "...                                                                            ...   \n",
       "metabric        jNMF      90                 False      5                      NaN   \n",
       "                                                        6                      NaN   \n",
       "                                                        7                      NaN   \n",
       "                                                        8                      NaN   \n",
       "                                                        9                      NaN   \n",
       "\n",
       "                                                               percentage_clustered_samples   \n",
       "dataset         algorithm missing_percentage imputation run_n                                 \n",
       "nutrimouse_diet AJIVE     30                 False      8                               NaN  \\\n",
       "                                                        9                               NaN   \n",
       "                          40                 True       0                               NaN   \n",
       "                                                        1                               NaN   \n",
       "                                                        2                               NaN   \n",
       "...                                                                                     ...   \n",
       "metabric        jNMF      90                 False      5                               NaN   \n",
       "                                                        6                               NaN   \n",
       "                                                        7                               NaN   \n",
       "                                                        8                               NaN   \n",
       "                                                        9                               NaN   \n",
       "\n",
       "                                                              comments   \n",
       "dataset         algorithm missing_percentage imputation run_n            \n",
       "nutrimouse_diet AJIVE     30                 False      8          NaN  \\\n",
       "                                                        9          NaN   \n",
       "                          40                 True       0          NaN   \n",
       "                                                        1          NaN   \n",
       "                                                        2          NaN   \n",
       "...                                                                ...   \n",
       "metabric        jNMF      90                 False      5          NaN   \n",
       "                                                        6          NaN   \n",
       "                                                        7          NaN   \n",
       "                                                        8          NaN   \n",
       "                                                        9          NaN   \n",
       "\n",
       "                                                              stratified  ...   \n",
       "dataset         algorithm missing_percentage imputation run_n             ...   \n",
       "nutrimouse_diet AJIVE     30                 False      8            NaN  ...  \\\n",
       "                                                        9            NaN  ...   \n",
       "                          40                 True       0            NaN  ...   \n",
       "                                                        1            NaN  ...   \n",
       "                                                        2            NaN  ...   \n",
       "...                                                                  ...  ...   \n",
       "metabric        jNMF      90                 False      5            NaN  ...   \n",
       "                                                        6            NaN  ...   \n",
       "                                                        7            NaN  ...   \n",
       "                                                        8            NaN  ...   \n",
       "                                                        9            NaN  ...   \n",
       "\n",
       "                                                              MCC_artificial_performance   \n",
       "dataset         algorithm missing_percentage imputation run_n                              \n",
       "nutrimouse_diet AJIVE     30                 False      8                            NaN  \\\n",
       "                                                        9                            NaN   \n",
       "                          40                 True       0                            NaN   \n",
       "                                                        1                            NaN   \n",
       "                                                        2                            NaN   \n",
       "...                                                                                  ...   \n",
       "metabric        jNMF      90                 False      5                            NaN   \n",
       "                                                        6                            NaN   \n",
       "                                                        7                            NaN   \n",
       "                                                        8                            NaN   \n",
       "                                                        9                            NaN   \n",
       "\n",
       "                                                              F1_artificial_performance   \n",
       "dataset         algorithm missing_percentage imputation run_n                             \n",
       "nutrimouse_diet AJIVE     30                 False      8                           NaN  \\\n",
       "                                                        9                           NaN   \n",
       "                          40                 True       0                           NaN   \n",
       "                                                        1                           NaN   \n",
       "                                                        2                           NaN   \n",
       "...                                                                                 ...   \n",
       "metabric        jNMF      90                 False      5                           NaN   \n",
       "                                                        6                           NaN   \n",
       "                                                        7                           NaN   \n",
       "                                                        8                           NaN   \n",
       "                                                        9                           NaN   \n",
       "\n",
       "                                                              precision_artificial_performance   \n",
       "dataset         algorithm missing_percentage imputation run_n                                    \n",
       "nutrimouse_diet AJIVE     30                 False      8                                  NaN  \\\n",
       "                                                        9                                  NaN   \n",
       "                          40                 True       0                                  NaN   \n",
       "                                                        1                                  NaN   \n",
       "                                                        2                                  NaN   \n",
       "...                                                                                        ...   \n",
       "metabric        jNMF      90                 False      5                                  NaN   \n",
       "                                                        6                                  NaN   \n",
       "                                                        7                                  NaN   \n",
       "                                                        8                                  NaN   \n",
       "                                                        9                                  NaN   \n",
       "\n",
       "                                                              recall_artificial_performance   \n",
       "dataset         algorithm missing_percentage imputation run_n                                 \n",
       "nutrimouse_diet AJIVE     30                 False      8                               NaN  \\\n",
       "                                                        9                               NaN   \n",
       "                          40                 True       0                               NaN   \n",
       "                                                        1                               NaN   \n",
       "                                                        2                               NaN   \n",
       "...                                                                                     ...   \n",
       "metabric        jNMF      90                 False      5                               NaN   \n",
       "                                                        6                               NaN   \n",
       "                                                        7                               NaN   \n",
       "                                                        8                               NaN   \n",
       "                                                        9                               NaN   \n",
       "\n",
       "                                                              bal_acc_artificial_performance   \n",
       "dataset         algorithm missing_percentage imputation run_n                                  \n",
       "nutrimouse_diet AJIVE     30                 False      8                                NaN  \\\n",
       "                                                        9                                NaN   \n",
       "                          40                 True       0                                NaN   \n",
       "                                                        1                                NaN   \n",
       "                                                        2                                NaN   \n",
       "...                                                                                      ...   \n",
       "metabric        jNMF      90                 False      5                                NaN   \n",
       "                                                        6                                NaN   \n",
       "                                                        7                                NaN   \n",
       "                                                        8                                NaN   \n",
       "                                                        9                                NaN   \n",
       "\n",
       "                                                              ami_artificial_performance   \n",
       "dataset         algorithm missing_percentage imputation run_n                              \n",
       "nutrimouse_diet AJIVE     30                 False      8                            NaN  \\\n",
       "                                                        9                            NaN   \n",
       "                          40                 True       0                            NaN   \n",
       "                                                        1                            NaN   \n",
       "                                                        2                            NaN   \n",
       "...                                                                                  ...   \n",
       "metabric        jNMF      90                 False      5                            NaN   \n",
       "                                                        6                            NaN   \n",
       "                                                        7                            NaN   \n",
       "                                                        8                            NaN   \n",
       "                                                        9                            NaN   \n",
       "\n",
       "                                                              ari_artificial_performance   \n",
       "dataset         algorithm missing_percentage imputation run_n                              \n",
       "nutrimouse_diet AJIVE     30                 False      8                            NaN  \\\n",
       "                                                        9                            NaN   \n",
       "                          40                 True       0                            NaN   \n",
       "                                                        1                            NaN   \n",
       "                                                        2                            NaN   \n",
       "...                                                                                  ...   \n",
       "metabric        jNMF      90                 False      5                            NaN   \n",
       "                                                        6                            NaN   \n",
       "                                                        7                            NaN   \n",
       "                                                        8                            NaN   \n",
       "                                                        9                            NaN   \n",
       "\n",
       "                                                              completeness_artificial_performance   \n",
       "dataset         algorithm missing_percentage imputation run_n                                       \n",
       "nutrimouse_diet AJIVE     30                 False      8                                     NaN  \\\n",
       "                                                        9                                     NaN   \n",
       "                          40                 True       0                                     NaN   \n",
       "                                                        1                                     NaN   \n",
       "                                                        2                                     NaN   \n",
       "...                                                                                           ...   \n",
       "metabric        jNMF      90                 False      5                                     NaN   \n",
       "                                                        6                                     NaN   \n",
       "                                                        7                                     NaN   \n",
       "                                                        8                                     NaN   \n",
       "                                                        9                                     NaN   \n",
       "\n",
       "                                                               random_acc_artificial_performance   \n",
       "dataset         algorithm missing_percentage imputation run_n                                      \n",
       "nutrimouse_diet AJIVE     30                 False      8                                    NaN  \\\n",
       "                                                        9                                    NaN   \n",
       "                          40                 True       0                                    NaN   \n",
       "                                                        1                                    NaN   \n",
       "                                                        2                                    NaN   \n",
       "...                                                                                          ...   \n",
       "metabric        jNMF      90                 False      5                                    NaN   \n",
       "                                                        6                                    NaN   \n",
       "                                                        7                                    NaN   \n",
       "                                                        8                                    NaN   \n",
       "                                                        9                                    NaN   \n",
       "\n",
       "                                                               random_f1_artificial_performance  \n",
       "dataset         algorithm missing_percentage imputation run_n                                    \n",
       "nutrimouse_diet AJIVE     30                 False      8                                   NaN  \n",
       "                                                        9                                   NaN  \n",
       "                          40                 True       0                                   NaN  \n",
       "                                                        1                                   NaN  \n",
       "                                                        2                                   NaN  \n",
       "...                                                                                         ...  \n",
       "metabric        jNMF      90                 False      5                                   NaN  \n",
       "                                                        6                                   NaN  \n",
       "                                                        7                                   NaN  \n",
       "                                                        8                                   NaN  \n",
       "                                                        9                                   NaN  \n",
       "\n",
       "[14942 rows x 77 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unfinished_results = results.loc[~results[\"finished\"]]\n",
    "unfinished_results.drop(unfinished_results.loc[[\"bbcsport\"]].xs(0, level=2, drop_level=False).index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "a5c2e495-33df-4f77-8297-4bd25144a069",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for i in os.listdir(\"imvc/datasets/data/buaa\"):\n",
    "    if \"y.csv\" in i:\n",
    "        continue\n",
    "    if \".csv\" in i:\n",
    "        pd.read_csv(os.path.join(\"imvc/datasets/data/buaa\", i), index_col= 0).iloc[:90].to_csv(os.path.join(\"imvc/datasets/data/buaa\", i), index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "1644e25a-8bdd-4592-b661-f578a1527387",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "simulated_netMUG\n",
      "simulated_netMUG_1.csv (1000, 2000)\n",
      "simulated_netMUG_0.csv (1000, 2000)\n",
      "simulated_netMUG_y.csv (1000, 1)\n",
      "\n",
      ".ipynb_checkpoints\n",
      "\n",
      "simulated_InterSIM\n",
      "simulated_InterSIM_1.csv (500, 131)\n",
      "simulated_InterSIM_y.csv (500, 1)\n",
      "simulated_InterSIM_0.csv (500, 367)\n",
      "simulated_InterSIM_2.csv (500, 160)\n",
      "\n",
      "bdgp\n",
      "bdgp_1.csv (2500, 79)\n",
      "bdgp_y.csv (2500, 2)\n",
      "metadata.json (0, 2)\n",
      "bdgp_0.csv (2500, 1750)\n",
      "\n",
      "tcga\n",
      "tcga_2.csv (2437, 132)\n",
      "tcga_3.csv (2437, 2044)\n",
      "tcga_y.csv (2437, 2)\n",
      "tcga_0.csv (2437, 216)\n",
      "metadata.json (0, 14)\n",
      "tcga_1.csv (2437, 16117)\n",
      "\n",
      "nuswide\n",
      "nuswide_4.csv (30000, 129)\n",
      "nuswide_3.csv (30000, 74)\n",
      "nuswide_y.csv (30000, 2)\n",
      "nuswide_1.csv (30000, 226)\n",
      "nuswide_0.csv (30000, 65)\n",
      "nuswide_2.csv (30000, 145)\n",
      "\n",
      "nutrimouse\n",
      "nutrimouse_y.csv (40, 3)\n",
      "nutrimouse_0.csv (40, 120)\n",
      "nutrimouse_1.csv (40, 21)\n",
      "metadata.json (0, 9)\n",
      "\n",
      "digits\n",
      "digits_5.csv (2000, 240)\n",
      "digits_3.csv (2000, 47)\n",
      "digits_4.csv (2000, 76)\n",
      "digits_y.csv (2000, 2)\n",
      "digits_0.csv (2000, 6)\n",
      "digits_2.csv (2000, 216)\n",
      "digits_1.csv (2000, 64)\n",
      "metadata.json (0, 6)\n",
      "\n",
      "caltech101\n",
      "caltech101_y.csv (9144, 2)\n",
      "caltech101_2.csv (9144, 254)\n",
      "caltech101_1.csv (9144, 40)\n",
      "caltech101_0.csv (9144, 48)\n",
      "caltech101_5.csv (9144, 928)\n",
      "caltech101_3.csv (9144, 1984)\n",
      "caltech101_4.csv (9144, 512)\n",
      "metadata.json (0, 108)\n",
      "\n",
      "bbcsport\n",
      "bbcsport_3.csv (116, 2158)\n",
      "bbcsport_1.csv (116, 2063)\n",
      "bbcsport_0.csv (116, 1991)\n",
      "bbcsport_2.csv (116, 2113)\n",
      "bbcsport_y.csv (116, 2)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "for i in os.listdir(\"imvc/datasets/data/\"):\n",
    "    if i == \"metabric\":\n",
    "        continue\n",
    "    print(i)\n",
    "    for j in os.listdir(os.path.join(\"imvc/datasets/data/\", i)):\n",
    "        if os.path.isfile(os.path.join(\"imvc/datasets/data/\", i, j)):\n",
    "            a = pd.read_csv(os.path.join(\"imvc/datasets/data/\", i, j))\n",
    "            print(j, a.shape)\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "9b7e341c-cf30-4a4f-b32c-d2ee9a52cd03",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>finished</th>\n",
       "      <th>completed</th>\n",
       "      <th>n_samples</th>\n",
       "      <th>n_incomplete_samples</th>\n",
       "      <th>n_complete_samples</th>\n",
       "      <th>time</th>\n",
       "      <th>n_clustered_samples</th>\n",
       "      <th>percentage_clustered_samples</th>\n",
       "      <th>comments</th>\n",
       "      <th>stratified</th>\n",
       "      <th>...</th>\n",
       "      <th>MCC_artificial_performance</th>\n",
       "      <th>F1_artificial_performance</th>\n",
       "      <th>precision_artificial_performance</th>\n",
       "      <th>recall_artificial_performance</th>\n",
       "      <th>bal_acc_artificial_performance</th>\n",
       "      <th>ami_artificial_performance</th>\n",
       "      <th>ari_artificial_performance</th>\n",
       "      <th>completeness_artificial_performance</th>\n",
       "      <th>random_acc_artificial_performance</th>\n",
       "      <th>random_f1_artificial_performance</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"11\" valign=\"top\">nutrimouse_genotype</th>\n",
       "      <th rowspan=\"11\" valign=\"top\">SNF</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>0</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.038531</td>\n",
       "      <td>40.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>{}</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.034154</td>\n",
       "      <td>40.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>{}</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.42229</td>\n",
       "      <td>40.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>{}</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.659519</td>\n",
       "      <td>40.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>{}</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.591319</td>\n",
       "      <td>40.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>{}</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">90</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>5</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{'ValueError: Percentiles must be in the range...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{'ValueError: Percentiles must be in the range...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{'ValueError: Percentiles must be in the range...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{'ValueError: Percentiles must be in the range...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>{'ValueError: Percentiles must be in the range...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows × 77 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                   finished   \n",
       "dataset             algorithm missing_percentage imputation run_n             \n",
       "nutrimouse_genotype SNF       0                  False      0          True  \\\n",
       "                                                            1          True   \n",
       "                                                            2          True   \n",
       "                                                            3          True   \n",
       "                                                            4          True   \n",
       "...                                                                     ...   \n",
       "                              90                 False      5          True   \n",
       "                                                            6          True   \n",
       "                                                            7          True   \n",
       "                                                            8          True   \n",
       "                                                            9          True   \n",
       "\n",
       "                                                                   completed   \n",
       "dataset             algorithm missing_percentage imputation run_n              \n",
       "nutrimouse_genotype SNF       0                  False      0           True  \\\n",
       "                                                            1           True   \n",
       "                                                            2           True   \n",
       "                                                            3           True   \n",
       "                                                            4           True   \n",
       "...                                                                      ...   \n",
       "                              90                 False      5          False   \n",
       "                                                            6          False   \n",
       "                                                            7          False   \n",
       "                                                            8          False   \n",
       "                                                            9          False   \n",
       "\n",
       "                                                                  n_samples   \n",
       "dataset             algorithm missing_percentage imputation run_n             \n",
       "nutrimouse_genotype SNF       0                  False      0          40.0  \\\n",
       "                                                            1          40.0   \n",
       "                                                            2          40.0   \n",
       "                                                            3          40.0   \n",
       "                                                            4          40.0   \n",
       "...                                                                     ...   \n",
       "                              90                 False      5           NaN   \n",
       "                                                            6           NaN   \n",
       "                                                            7           NaN   \n",
       "                                                            8           NaN   \n",
       "                                                            9           NaN   \n",
       "\n",
       "                                                                  n_incomplete_samples   \n",
       "dataset             algorithm missing_percentage imputation run_n                        \n",
       "nutrimouse_genotype SNF       0                  False      0                      0.0  \\\n",
       "                                                            1                      0.0   \n",
       "                                                            2                      0.0   \n",
       "                                                            3                      0.0   \n",
       "                                                            4                      0.0   \n",
       "...                                                                                ...   \n",
       "                              90                 False      5                      NaN   \n",
       "                                                            6                      NaN   \n",
       "                                                            7                      NaN   \n",
       "                                                            8                      NaN   \n",
       "                                                            9                      NaN   \n",
       "\n",
       "                                                                  n_complete_samples   \n",
       "dataset             algorithm missing_percentage imputation run_n                      \n",
       "nutrimouse_genotype SNF       0                  False      0                   40.0  \\\n",
       "                                                            1                   40.0   \n",
       "                                                            2                   40.0   \n",
       "                                                            3                   40.0   \n",
       "                                                            4                   40.0   \n",
       "...                                                                              ...   \n",
       "                              90                 False      5                    NaN   \n",
       "                                                            6                    NaN   \n",
       "                                                            7                    NaN   \n",
       "                                                            8                    NaN   \n",
       "                                                            9                    NaN   \n",
       "\n",
       "                                                                       time   \n",
       "dataset             algorithm missing_percentage imputation run_n             \n",
       "nutrimouse_genotype SNF       0                  False      0      0.038531  \\\n",
       "                                                            1      0.034154   \n",
       "                                                            2       0.42229   \n",
       "                                                            3      0.659519   \n",
       "                                                            4      0.591319   \n",
       "...                                                                     ...   \n",
       "                              90                 False      5           NaN   \n",
       "                                                            6           NaN   \n",
       "                                                            7           NaN   \n",
       "                                                            8           NaN   \n",
       "                                                            9           NaN   \n",
       "\n",
       "                                                                  n_clustered_samples   \n",
       "dataset             algorithm missing_percentage imputation run_n                       \n",
       "nutrimouse_genotype SNF       0                  False      0                    40.0  \\\n",
       "                                                            1                    40.0   \n",
       "                                                            2                    40.0   \n",
       "                                                            3                    40.0   \n",
       "                                                            4                    40.0   \n",
       "...                                                                               ...   \n",
       "                              90                 False      5                     NaN   \n",
       "                                                            6                     NaN   \n",
       "                                                            7                     NaN   \n",
       "                                                            8                     NaN   \n",
       "                                                            9                     NaN   \n",
       "\n",
       "                                                                  percentage_clustered_samples   \n",
       "dataset             algorithm missing_percentage imputation run_n                                \n",
       "nutrimouse_genotype SNF       0                  False      0                            100.0  \\\n",
       "                                                            1                            100.0   \n",
       "                                                            2                            100.0   \n",
       "                                                            3                            100.0   \n",
       "                                                            4                            100.0   \n",
       "...                                                                                        ...   \n",
       "                              90                 False      5                              NaN   \n",
       "                                                            6                              NaN   \n",
       "                                                            7                              NaN   \n",
       "                                                            8                              NaN   \n",
       "                                                            9                              NaN   \n",
       "\n",
       "                                                                                                            comments   \n",
       "dataset             algorithm missing_percentage imputation run_n                                                      \n",
       "nutrimouse_genotype SNF       0                  False      0                                                     {}  \\\n",
       "                                                            1                                                     {}   \n",
       "                                                            2                                                     {}   \n",
       "                                                            3                                                     {}   \n",
       "                                                            4                                                     {}   \n",
       "...                                                                                                              ...   \n",
       "                              90                 False      5      {'ValueError: Percentiles must be in the range...   \n",
       "                                                            6      {'ValueError: Percentiles must be in the range...   \n",
       "                                                            7      {'ValueError: Percentiles must be in the range...   \n",
       "                                                            8      {'ValueError: Percentiles must be in the range...   \n",
       "                                                            9      {'ValueError: Percentiles must be in the range...   \n",
       "\n",
       "                                                                  stratified   \n",
       "dataset             algorithm missing_percentage imputation run_n              \n",
       "nutrimouse_genotype SNF       0                  False      0          False  \\\n",
       "                                                            1          False   \n",
       "                                                            2          False   \n",
       "                                                            3          False   \n",
       "                                                            4          False   \n",
       "...                                                                      ...   \n",
       "                              90                 False      5            NaN   \n",
       "                                                            6            NaN   \n",
       "                                                            7            NaN   \n",
       "                                                            8            NaN   \n",
       "                                                            9            NaN   \n",
       "\n",
       "                                                                   ...   \n",
       "dataset             algorithm missing_percentage imputation run_n  ...   \n",
       "nutrimouse_genotype SNF       0                  False      0      ...  \\\n",
       "                                                            1      ...   \n",
       "                                                            2      ...   \n",
       "                                                            3      ...   \n",
       "                                                            4      ...   \n",
       "...                                                                ...   \n",
       "                              90                 False      5      ...   \n",
       "                                                            6      ...   \n",
       "                                                            7      ...   \n",
       "                                                            8      ...   \n",
       "                                                            9      ...   \n",
       "\n",
       "                                                                  MCC_artificial_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                              \n",
       "nutrimouse_genotype SNF       0                  False      0                            NaN  \\\n",
       "                                                            1                            NaN   \n",
       "                                                            2                            NaN   \n",
       "                                                            3                            NaN   \n",
       "                                                            4                            NaN   \n",
       "...                                                                                      ...   \n",
       "                              90                 False      5                            NaN   \n",
       "                                                            6                            NaN   \n",
       "                                                            7                            NaN   \n",
       "                                                            8                            NaN   \n",
       "                                                            9                            NaN   \n",
       "\n",
       "                                                                  F1_artificial_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                             \n",
       "nutrimouse_genotype SNF       0                  False      0                           NaN  \\\n",
       "                                                            1                           NaN   \n",
       "                                                            2                           NaN   \n",
       "                                                            3                           NaN   \n",
       "                                                            4                           NaN   \n",
       "...                                                                                     ...   \n",
       "                              90                 False      5                           NaN   \n",
       "                                                            6                           NaN   \n",
       "                                                            7                           NaN   \n",
       "                                                            8                           NaN   \n",
       "                                                            9                           NaN   \n",
       "\n",
       "                                                                  precision_artificial_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                                    \n",
       "nutrimouse_genotype SNF       0                  False      0                                  NaN  \\\n",
       "                                                            1                                  NaN   \n",
       "                                                            2                                  NaN   \n",
       "                                                            3                                  NaN   \n",
       "                                                            4                                  NaN   \n",
       "...                                                                                            ...   \n",
       "                              90                 False      5                                  NaN   \n",
       "                                                            6                                  NaN   \n",
       "                                                            7                                  NaN   \n",
       "                                                            8                                  NaN   \n",
       "                                                            9                                  NaN   \n",
       "\n",
       "                                                                  recall_artificial_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                                 \n",
       "nutrimouse_genotype SNF       0                  False      0                               NaN  \\\n",
       "                                                            1                               NaN   \n",
       "                                                            2                               NaN   \n",
       "                                                            3                               NaN   \n",
       "                                                            4                               NaN   \n",
       "...                                                                                         ...   \n",
       "                              90                 False      5                               NaN   \n",
       "                                                            6                               NaN   \n",
       "                                                            7                               NaN   \n",
       "                                                            8                               NaN   \n",
       "                                                            9                               NaN   \n",
       "\n",
       "                                                                  bal_acc_artificial_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                                  \n",
       "nutrimouse_genotype SNF       0                  False      0                                NaN  \\\n",
       "                                                            1                                NaN   \n",
       "                                                            2                                NaN   \n",
       "                                                            3                                NaN   \n",
       "                                                            4                                NaN   \n",
       "...                                                                                          ...   \n",
       "                              90                 False      5                                NaN   \n",
       "                                                            6                                NaN   \n",
       "                                                            7                                NaN   \n",
       "                                                            8                                NaN   \n",
       "                                                            9                                NaN   \n",
       "\n",
       "                                                                  ami_artificial_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                              \n",
       "nutrimouse_genotype SNF       0                  False      0                            NaN  \\\n",
       "                                                            1                            NaN   \n",
       "                                                            2                            NaN   \n",
       "                                                            3                            NaN   \n",
       "                                                            4                            NaN   \n",
       "...                                                                                      ...   \n",
       "                              90                 False      5                            NaN   \n",
       "                                                            6                            NaN   \n",
       "                                                            7                            NaN   \n",
       "                                                            8                            NaN   \n",
       "                                                            9                            NaN   \n",
       "\n",
       "                                                                  ari_artificial_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                              \n",
       "nutrimouse_genotype SNF       0                  False      0                            NaN  \\\n",
       "                                                            1                            NaN   \n",
       "                                                            2                            NaN   \n",
       "                                                            3                            NaN   \n",
       "                                                            4                            NaN   \n",
       "...                                                                                      ...   \n",
       "                              90                 False      5                            NaN   \n",
       "                                                            6                            NaN   \n",
       "                                                            7                            NaN   \n",
       "                                                            8                            NaN   \n",
       "                                                            9                            NaN   \n",
       "\n",
       "                                                                  completeness_artificial_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                                       \n",
       "nutrimouse_genotype SNF       0                  False      0                                     NaN  \\\n",
       "                                                            1                                     NaN   \n",
       "                                                            2                                     NaN   \n",
       "                                                            3                                     NaN   \n",
       "                                                            4                                     NaN   \n",
       "...                                                                                               ...   \n",
       "                              90                 False      5                                     NaN   \n",
       "                                                            6                                     NaN   \n",
       "                                                            7                                     NaN   \n",
       "                                                            8                                     NaN   \n",
       "                                                            9                                     NaN   \n",
       "\n",
       "                                                                  random_acc_artificial_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                                     \n",
       "nutrimouse_genotype SNF       0                  False      0                                   NaN  \\\n",
       "                                                            1                                   NaN   \n",
       "                                                            2                                   NaN   \n",
       "                                                            3                                   NaN   \n",
       "                                                            4                                   NaN   \n",
       "...                                                                                             ...   \n",
       "                              90                 False      5                                   NaN   \n",
       "                                                            6                                   NaN   \n",
       "                                                            7                                   NaN   \n",
       "                                                            8                                   NaN   \n",
       "                                                            9                                   NaN   \n",
       "\n",
       "                                                                  random_f1_artificial_performance  \n",
       "dataset             algorithm missing_percentage imputation run_n                                   \n",
       "nutrimouse_genotype SNF       0                  False      0                                  NaN  \n",
       "                                                            1                                  NaN  \n",
       "                                                            2                                  NaN  \n",
       "                                                            3                                  NaN  \n",
       "                                                            4                                  NaN  \n",
       "...                                                                                            ...  \n",
       "                              90                 False      5                                  NaN  \n",
       "                                                            6                                  NaN  \n",
       "                                                            7                                  NaN  \n",
       "                                                            8                                  NaN  \n",
       "                                                            9                                  NaN  \n",
       "\n",
       "[100 rows x 77 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "finished_results = results[results[\"finished\"]]\n",
    "finished_results.loc[[\"nutrimouse_genotype\"]].xs(\"SNF\", level=1, drop_level=False).xs(False, level=3, drop_level=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c7ba1a4-9005-4882-acec-edd62f153b64",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "%tb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "d5268465-57d3-4d0a-8bdd-476e2865b5c0",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "iterator = finished_results.loc[[\"nutrimouse_genotype\"]].xs(\"SNF\", level=1, drop_level=False).xs(False, level=3, drop_level=False).index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "bf134aca-e6b8-459e-b8ab-9168082cb934",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 0, 'imputation': False, 'run_n': 0} \t 2023-12-05 14:40:29.658451\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 0, 'imputation': False, 'run_n': 1} \t 2023-12-05 14:40:29.841562\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 0, 'imputation': False, 'run_n': 2} \t 2023-12-05 14:40:30.179149\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 0, 'imputation': False, 'run_n': 3} \t 2023-12-05 14:40:30.319688\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 0, 'imputation': False, 'run_n': 4} \t 2023-12-05 14:40:30.516572\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 0, 'imputation': False, 'run_n': 5} \t 2023-12-05 14:40:30.680208\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 0, 'imputation': False, 'run_n': 6} \t 2023-12-05 14:40:30.904797\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 0, 'imputation': False, 'run_n': 7} \t 2023-12-05 14:40:31.068916\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 0, 'imputation': False, 'run_n': 8} \t 2023-12-05 14:40:31.299065\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 0, 'imputation': False, 'run_n': 9} \t 2023-12-05 14:40:31.448579\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 10, 'imputation': False, 'run_n': 0} \t 2023-12-05 14:40:31.608767\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 10, 'imputation': False, 'run_n': 1} \t 2023-12-05 14:40:31.827336\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 10, 'imputation': False, 'run_n': 2} \t 2023-12-05 14:40:32.114008\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 10, 'imputation': False, 'run_n': 3} \t 2023-12-05 14:40:32.320446\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 10, 'imputation': False, 'run_n': 4} \t 2023-12-05 14:40:32.509891\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 10, 'imputation': False, 'run_n': 5} \t 2023-12-05 14:40:32.701993\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 10, 'imputation': False, 'run_n': 6} \t 2023-12-05 14:40:32.924645\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 10, 'imputation': False, 'run_n': 7} \t 2023-12-05 14:40:33.460809\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 10, 'imputation': False, 'run_n': 8} \t 2023-12-05 14:40:33.674239\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 10, 'imputation': False, 'run_n': 9} \t 2023-12-05 14:40:33.890632\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 20, 'imputation': False, 'run_n': 0} \t 2023-12-05 14:40:34.079844\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 20, 'imputation': False, 'run_n': 1} \t 2023-12-05 14:40:34.339925\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 20, 'imputation': False, 'run_n': 2} \t 2023-12-05 14:40:34.527401\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 20, 'imputation': False, 'run_n': 3} \t 2023-12-05 14:40:34.775914\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 20, 'imputation': False, 'run_n': 4} \t 2023-12-05 14:40:34.989057\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 20, 'imputation': False, 'run_n': 5} \t 2023-12-05 14:40:35.203665\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 20, 'imputation': False, 'run_n': 6} \t 2023-12-05 14:40:35.390642\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 20, 'imputation': False, 'run_n': 7} \t 2023-12-05 14:40:35.665829\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 20, 'imputation': False, 'run_n': 8} \t 2023-12-05 14:40:35.888297\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 20, 'imputation': False, 'run_n': 9} \t 2023-12-05 14:40:36.093979\n",
      "4\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 30, 'imputation': False, 'run_n': 0} \t 2023-12-05 14:40:36.283672\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 30, 'imputation': False, 'run_n': 1} \t 2023-12-05 14:40:36.461008\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 30, 'imputation': False, 'run_n': 2} \t 2023-12-05 14:40:36.640611\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 30, 'imputation': False, 'run_n': 3} \t 2023-12-05 14:40:36.918936\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 30, 'imputation': False, 'run_n': 4} \t 2023-12-05 14:40:37.127759\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 30, 'imputation': False, 'run_n': 5} \t 2023-12-05 14:40:37.327321\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 30, 'imputation': False, 'run_n': 6} \t 2023-12-05 14:40:37.525329\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 30, 'imputation': False, 'run_n': 7} \t 2023-12-05 14:40:37.785164\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 30, 'imputation': False, 'run_n': 8} \t 2023-12-05 14:40:38.000151\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 30, 'imputation': False, 'run_n': 9} \t 2023-12-05 14:40:38.185044\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 40, 'imputation': False, 'run_n': 0} \t 2023-12-05 14:40:38.398892\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 40, 'imputation': False, 'run_n': 1} \t 2023-12-05 14:40:38.594874\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 40, 'imputation': False, 'run_n': 2} \t 2023-12-05 14:40:38.831721\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 40, 'imputation': False, 'run_n': 3} \t 2023-12-05 14:40:39.044194\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 40, 'imputation': False, 'run_n': 4} \t 2023-12-05 14:40:39.230951\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 40, 'imputation': False, 'run_n': 5} \t 2023-12-05 14:40:39.471791\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 40, 'imputation': False, 'run_n': 6} \t 2023-12-05 14:40:39.682977\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 40, 'imputation': False, 'run_n': 7} \t 2023-12-05 14:40:39.883358\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 40, 'imputation': False, 'run_n': 8} \t 2023-12-05 14:40:40.123783\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 40, 'imputation': False, 'run_n': 9} \t 2023-12-05 14:40:40.311616\n",
      "3\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 50, 'imputation': False, 'run_n': 0} \t 2023-12-05 14:40:40.575162\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 50, 'imputation': False, 'run_n': 1} \t 2023-12-05 14:40:40.775625\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 50, 'imputation': False, 'run_n': 2} \t 2023-12-05 14:40:41.002142\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 50, 'imputation': False, 'run_n': 3} \t 2023-12-05 14:40:41.209638\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 50, 'imputation': False, 'run_n': 4} \t 2023-12-05 14:40:41.496847\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 50, 'imputation': False, 'run_n': 5} \t 2023-12-05 14:40:41.693684\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 50, 'imputation': False, 'run_n': 6} \t 2023-12-05 14:40:41.904653\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 50, 'imputation': False, 'run_n': 7} \t 2023-12-05 14:40:42.114528\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 50, 'imputation': False, 'run_n': 8} \t 2023-12-05 14:40:42.406497\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 50, 'imputation': False, 'run_n': 9} \t 2023-12-05 14:40:42.639821\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 60, 'imputation': False, 'run_n': 0} \t 2023-12-05 14:40:42.857476\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 60, 'imputation': False, 'run_n': 1} \t 2023-12-05 14:40:43.066815\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 60, 'imputation': False, 'run_n': 2} \t 2023-12-05 14:40:43.264688\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 60, 'imputation': False, 'run_n': 3} \t 2023-12-05 14:40:44.168822\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 60, 'imputation': False, 'run_n': 4} \t 2023-12-05 14:40:44.393261\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 60, 'imputation': False, 'run_n': 5} \t 2023-12-05 14:40:44.580406\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 60, 'imputation': False, 'run_n': 6} \t 2023-12-05 14:40:44.868281\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 60, 'imputation': False, 'run_n': 7} \t 2023-12-05 14:40:45.058120\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 60, 'imputation': False, 'run_n': 8} \t 2023-12-05 14:40:45.321415\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 60, 'imputation': False, 'run_n': 9} \t 2023-12-05 14:40:45.501123\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 70, 'imputation': False, 'run_n': 0} \t 2023-12-05 14:40:45.805354\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 70, 'imputation': False, 'run_n': 1} \t 2023-12-05 14:40:46.013014\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 70, 'imputation': False, 'run_n': 2} \t 2023-12-05 14:40:46.215296\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 70, 'imputation': False, 'run_n': 3} \t 2023-12-05 14:40:46.423970\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 70, 'imputation': False, 'run_n': 4} \t 2023-12-05 14:40:46.689149\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 70, 'imputation': False, 'run_n': 5} \t 2023-12-05 14:40:46.984258\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 70, 'imputation': False, 'run_n': 6} \t 2023-12-05 14:40:47.154724\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 70, 'imputation': False, 'run_n': 7} \t 2023-12-05 14:40:47.455375\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 70, 'imputation': False, 'run_n': 8} \t 2023-12-05 14:40:47.622600\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 70, 'imputation': False, 'run_n': 9} \t 2023-12-05 14:40:47.847263\n",
      "2\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 80, 'imputation': False, 'run_n': 0} \t 2023-12-05 14:40:48.021566\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 80, 'imputation': False, 'run_n': 1} \t 2023-12-05 14:40:48.312319\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 80, 'imputation': False, 'run_n': 2} \t 2023-12-05 14:40:48.496043\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 80, 'imputation': False, 'run_n': 3} \t 2023-12-05 14:40:49.018913\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 80, 'imputation': False, 'run_n': 4} \t 2023-12-05 14:40:49.211212\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 80, 'imputation': False, 'run_n': 5} \t 2023-12-05 14:40:49.415516\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 80, 'imputation': False, 'run_n': 6} \t 2023-12-05 14:40:49.646280\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 80, 'imputation': False, 'run_n': 7} \t 2023-12-05 14:40:49.836631\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 80, 'imputation': False, 'run_n': 8} \t 2023-12-05 14:40:50.256608\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 80, 'imputation': False, 'run_n': 9} \t 2023-12-05 14:40:50.425323\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 90, 'imputation': False, 'run_n': 0} \t 2023-12-05 14:40:50.602741\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 90, 'imputation': False, 'run_n': 1} \t 2023-12-05 14:40:50.797474\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 90, 'imputation': False, 'run_n': 2} \t 2023-12-05 14:40:50.994954\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 90, 'imputation': False, 'run_n': 3} \t 2023-12-05 14:40:51.336386\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 90, 'imputation': False, 'run_n': 4} \t 2023-12-05 14:40:51.530559\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 90, 'imputation': False, 'run_n': 5} \t 2023-12-05 14:40:51.802946\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 90, 'imputation': False, 'run_n': 6} \t 2023-12-05 14:40:52.012751\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 90, 'imputation': False, 'run_n': 7} \t 2023-12-05 14:40:52.212445\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 90, 'imputation': False, 'run_n': 8} \t 2023-12-05 14:40:52.411659\n",
      "1\n",
      "{'dataset': 'nutrimouse_genotype', 'algorithm': 'SNF', 'missing_percentage': 90, 'imputation': False, 'run_n': 9} \t 2023-12-05 14:40:52.631805\n",
      "1\n"
     ]
    }
   ],
   "source": [
    "import time\n",
    "from collections import defaultdict\n",
    "from datetime import datetime\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from bignmf.models.jnmf.integrative import IntegrativeJnmf\n",
    "from bignmf.models.jnmf.standard import StandardJnmf\n",
    "from reval.utils import kuhn_munkres_algorithm\n",
    "from sklearn import metrics\n",
    "from sklearn.cluster import spectral_clustering\n",
    "from sklearn.impute import SimpleImputer\n",
    "from sklearn.pipeline import make_pipeline\n",
    "from sklearn.preprocessing import StandardScaler, MinMaxScaler\n",
    "from snf import compute\n",
    "from validclust import dunn\n",
    "\n",
    "from imvc.transformers import MultiViewTransformer, ConcatenateViews\n",
    "from imvc.utils import DatasetUtils\n",
    "\n",
    "for idx in iterator:\n",
    "    row = results.loc[[idx]]\n",
    "    row_index = row.index\n",
    "    alg_name, impute, p, run_n = (\n",
    "        row_index.get_level_values(\"algorithm\")[0],\n",
    "        row_index.get_level_values(\"imputation\")[0],\n",
    "        row_index.get_level_values(\"missing_percentage\")[0] / 100,\n",
    "        row_index.get_level_values(\"run_n\")[0])\n",
    "\n",
    "    print(row.drop(columns=row.columns).reset_index().to_dict(orient=\"records\")[0], \"\\t\", datetime.now())\n",
    "    alg = algorithms[alg_name]\n",
    "    train_Xs = DatasetUtils.shuffle_imvd(Xs=Xs, random_state=random_state + run_n)\n",
    "    y_train = y.loc[train_Xs[0].index]\n",
    "    errors_dict = defaultdict(int)\n",
    "    if p != 0:\n",
    "        try:\n",
    "            assert n_clusters < len(train_Xs[0]) * (1-p)\n",
    "        except AssertionError as exception:\n",
    "            errors_dict[f\"{type(exception).__name__}: {exception}; n_clusters < len(train_Xs[0]) * (1-p)\"] += 1\n",
    "            # print(errors_dict, row.drop(columns=row.columns).reset_index().to_dict(orient=\"records\")[0])\n",
    "            results.loc[idx, [\"finished\", \"comments\"]] = True, errors_dict\n",
    "            results.to_csv(file_path)\n",
    "            # return results.loc[idx]\n",
    "        try:\n",
    "            train_Xs = DatasetUtils.add_random_noise_to_views(Xs=train_Xs, p=round(p, 2),\n",
    "                                                              random_state=random_state + run_n,\n",
    "                                                              assess_percentage=True, stratify=y_train)\n",
    "            strat = True\n",
    "        except ValueError:\n",
    "            try:\n",
    "                train_Xs = DatasetUtils.add_random_noise_to_views(Xs=train_Xs, p=round(p, 2),\n",
    "                                                                  random_state=random_state + run_n,\n",
    "                                                                  assess_percentage=True)\n",
    "                strat = False\n",
    "            except Exception as exception:\n",
    "                errors_dict[f\"{type(exception).__name__}: {exception}\"] += 1\n",
    "                # print(errors_dict, row.drop(columns=row.columns).reset_index().to_dict(orient=\"records\")[0])\n",
    "                results.loc[idx, [\"finished\", \"comments\"]] = True, errors_dict\n",
    "                results.to_csv(file_path)\n",
    "                # return results.loc[idx]\n",
    "    else:\n",
    "        strat = False\n",
    "\n",
    "    if impute:\n",
    "        train_Xs = MultiViewTransformer(SimpleImputer(strategy=\"mean\").set_output(transform=\"pandas\")).fit_transform(\n",
    "            train_Xs)\n",
    "    else:\n",
    "        train_Xs = DatasetUtils.select_complete_samples(Xs=train_Xs)\n",
    "        y_train = y_train.loc[train_Xs[0].index]\n",
    "\n",
    "    try:\n",
    "        start_time = time.perf_counter()\n",
    "        if alg_name == \"SNF\":\n",
    "            preprocessing_step = MultiViewTransformer(StandardScaler().set_output(transform=\"pandas\"))\n",
    "            train_Xs = preprocessing_step.fit_transform(train_Xs)\n",
    "            k_snf = np.ceil(len(y_train)/10).astype(int)\n",
    "            print(k_snf)\n",
    "            affinities = compute.make_affinity(train_Xs, normalize=False, K= k_snf)\n",
    "            fused = compute.snf(affinities, K= k_snf)\n",
    "            clusters = spectral_clustering(fused, n_clusters=n_clusters, random_state=random_state + run_n)\n",
    "        elif alg_name == \"intNMF\":\n",
    "            preprocessing_step = MultiViewTransformer(MinMaxScaler().set_output(transform=\"pandas\"))\n",
    "            train_Xs = preprocessing_step.fit_transform(train_Xs)\n",
    "            model = IntegrativeJnmf({k: v for k, v in enumerate(train_Xs)}, k=n_clusters, lamb=0.1)\n",
    "            model.run(trials=50, iterations=100, verbose=False)\n",
    "            model.cluster_data()\n",
    "            clusters = np.argmax(model.w_cluster, axis=1)\n",
    "        elif alg_name == \"jNMF\":\n",
    "            preprocessing_step = make_pipeline(MultiViewTransformer(MinMaxScaler().set_output(transform=\"pandas\")))\n",
    "            train_Xs = preprocessing_step.fit_transform(train_Xs)\n",
    "            model = StandardJnmf({k: v for k, v in enumerate(train_Xs)}, k=n_clusters)\n",
    "            model.run(trials=50, iterations=100, verbose=False)\n",
    "            model.cluster_data()\n",
    "            clusters = np.argmax(model.w_cluster, axis=1)\n",
    "        else:\n",
    "            model, params = alg[\"alg\"], alg[\"params\"]\n",
    "            if alg_name == \"GroupPCA\":\n",
    "                model[1].set_params(n_components=n_clusters, random_state=random_state + run_n, multiview_output=False)\n",
    "            elif alg_name == \"AJIVE\":\n",
    "                model[1].set_params(random_state=random_state + run_n)\n",
    "            if alg_name == \"NMFC\":\n",
    "                model[-1].set_params(n_components=n_clusters, random_state=random_state + run_n)\n",
    "            else:\n",
    "                model[-1].set_params(n_clusters=n_clusters, random_state=random_state + run_n)\n",
    "            clusters = model.fit_predict(train_Xs)\n",
    "    except ValueError as exception:\n",
    "        if alg_name == \"AJIVE\" and len(y_train) < 5:\n",
    "            errors_dict[f\"{type(exception).__name__}: {exception}\"] += 1\n",
    "            # print(errors_dict, row.drop(columns=row.columns).reset_index().to_dict(orient=\"records\")[0])\n",
    "            results.loc[idx, [\"finished\", \"comments\"]] = True, errors_dict\n",
    "            results.to_csv(file_path)\n",
    "            # return results.loc[idx]\n",
    "        if alg_name == \"SNF\" and len(y_train) < 17:\n",
    "            errors_dict[f\"{type(exception).__name__}: {exception}\"] += 1\n",
    "            # print(errors_dict, row.drop(columns=row.columns).reset_index().to_dict(orient=\"records\")[0])\n",
    "            results.loc[idx, [\"finished\", \"comments\"]] = True, errors_dict\n",
    "            results.to_csv(file_path)\n",
    "            # return results.loc[idx]\n",
    "        if alg_name == \"intNMF\" and len(y_train) < 5:\n",
    "            errors_dict[f\"{type(exception).__name__}: {exception}\"] += 1\n",
    "            # print(errors_dict, row.drop(columns=row.columns).reset_index().to_dict(orient=\"records\")[0])\n",
    "            results.loc[idx, [\"finished\", \"comments\"]] = True, errors_dict\n",
    "            results.to_csv(file_path)\n",
    "            # return results.loc[idx]\n",
    "        else:\n",
    "            raise\n",
    "\n",
    "    clusters = pd.Series(clusters, index=y_train.index)\n",
    "\n",
    "    elapsed_time = time.perf_counter() - start_time\n",
    "\n",
    "    if alg_name in [\"NMFC\"]:\n",
    "        train_X = model.transform(train_Xs)\n",
    "    elif alg_name in [\"SNF\"]:\n",
    "        train_X = preprocessing_step.transform(train_Xs)\n",
    "    elif alg_name in [\"intNMF\", \"jNMF\"]:\n",
    "        train_X = model.w\n",
    "    else:\n",
    "        train_X = model[:-1].transform(train_Xs)\n",
    "    if isinstance(train_X, list):\n",
    "        train_X = ConcatenateViews().fit_transform(train_X)\n",
    "    if not isinstance(train_X, pd.DataFrame):\n",
    "        train_X = pd.DataFrame(train_X, index=y_train.index)\n",
    "\n",
    "    assert train_X.index.equals(y_train.index)\n",
    "    assert train_Xs[0].index.equals(y_train.index)\n",
    "\n",
    "    if p > 0:\n",
    "        best_solution = pd.MultiIndex.from_arrays(\n",
    "            [[row_index.get_level_values(level=level)[0]] if level != \"missing_percentage\" else [0]\n",
    "             for level in row_index.names], names=row_index.names)\n",
    "        best_solution = results.loc[best_solution].iloc[0]\n",
    "        y_train_total = pd.Series(best_solution[\"y_true\"], index=best_solution[\"y_pred_idx\"])\n",
    "        best_solution = pd.Series(best_solution[\"y_pred\"], index=best_solution[\"y_pred_idx\"])\n",
    "    else:\n",
    "        best_solution = None\n",
    "        y_train_total = None\n",
    "\n",
    "    dict_results = Utils.save_record(train_Xs=train_Xs, train_X=train_X, clusters=clusters, y=y_train, p=p, y_true_total=y_train_total,\n",
    "                               best_solution=best_solution, elapsed_time=elapsed_time, strat=strat,\n",
    "                               random_state=random_state, errors_dict=errors_dict)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "e8c5bc5c-0f63-4a42-b480-8f9629d81cfd",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "train_Xs = DatasetUtils.shuffle_imvd(Xs=Xs, random_state=random_state + run_n)\n",
    "y_train = y.loc[train_Xs[0].index]\n",
    "train_Xs = DatasetUtils.add_random_noise_to_views(Xs=train_Xs, p=round(p, 2),\n",
    "                                                              random_state=random_state + run_n,\n",
    "                                                              assess_percentage=True, stratify=y_train)\n",
    "train_Xs = DatasetUtils.select_complete_samples(Xs=train_Xs)\n",
    "y_train = y_train.loc[train_Xs[0].index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "a399f5b0-0a87-43b5-a3e5-ed889a11b75a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(16, 16)"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "preprocessing_step = MultiViewTransformer(StandardScaler().set_output(transform=\"pandas\"))\n",
    "train_Xs = preprocessing_step.fit_transform(train_Xs)\n",
    "k_snf = np.ceil(len(y_train)/10).astype(int)\n",
    "aff = compute.make_affinity(train_Xs, normalize=False, K= k_snf)\n",
    "aff[1].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "16aca288-0e9b-4b54-8a20-fd2771d0b47e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[8.17888478e-01, 3.65788169e-02, 2.43160808e-01, 2.38397328e-02,\n",
       "        2.74530611e-03, 7.32549852e-03, 1.36156155e-01, 3.64192253e-04,\n",
       "        3.67052593e-02, 2.17831412e-03, 1.53018811e-01, 2.99521108e-03,\n",
       "        2.46207380e-03, 3.64147880e-04, 2.14004989e-02, 2.48877638e-03],\n",
       "       [3.65788169e-02, 8.43450897e-01, 4.89686247e-02, 4.40409230e-03,\n",
       "        1.55403572e-02, 4.40224789e-02, 7.10256111e-03, 7.45412247e-03,\n",
       "        2.59595045e-01, 1.08554513e-02, 9.42230365e-03, 4.62753688e-02,\n",
       "        1.35503984e-02, 7.43400467e-03, 1.31759921e-01, 1.29612998e-02],\n",
       "       [2.43160808e-01, 4.89686247e-02, 8.21065943e-01, 1.96412592e-02,\n",
       "        3.40814059e-03, 9.38110542e-03, 1.19007067e-01, 5.48592142e-04,\n",
       "        5.08643961e-02, 2.32211758e-03, 1.31357059e-01, 4.20535775e-03,\n",
       "        2.88066982e-03, 5.48419839e-04, 2.86873768e-02, 2.75469464e-03],\n",
       "       [2.38397328e-02, 4.40409230e-03, 1.96412592e-02, 9.71374665e-01,\n",
       "        4.24767808e-02, 2.32350094e-02, 7.81886899e-02, 4.34495469e-05,\n",
       "        4.27228363e-03, 8.52091252e-02, 6.42787813e-02, 2.09064533e-02,\n",
       "        4.73583020e-02, 4.34491268e-05, 8.81397402e-03, 8.13328779e-02],\n",
       "       [2.74530611e-03, 1.55403572e-02, 3.40814059e-03, 4.24767808e-02,\n",
       "        7.74832111e-01, 1.38195325e-01, 4.11137015e-03, 7.83234772e-05,\n",
       "        1.91904820e-02, 1.16944628e-01, 3.18464532e-03, 2.02063668e-02,\n",
       "        2.10997132e-01, 7.83219385e-05, 5.24059953e-02, 9.97519028e-02],\n",
       "       [7.32549852e-03, 4.40224789e-02, 9.38110542e-03, 2.32350094e-02,\n",
       "        1.38195325e-01, 8.17726590e-01, 3.39640610e-03, 1.57534277e-04,\n",
       "        5.34300776e-02, 7.06750341e-02, 3.20671936e-03, 1.39231397e-02,\n",
       "        1.21688061e-01, 1.57521839e-04, 1.30645507e-01, 7.41945078e-02],\n",
       "       [1.36156155e-01, 7.10256111e-03, 1.19007067e-01, 7.81886899e-02,\n",
       "        4.11137015e-03, 3.39640610e-03, 8.49003367e-01, 1.06355004e-04,\n",
       "        7.01416610e-03, 6.88499149e-03, 2.66051240e-01, 2.49718807e-03,\n",
       "        4.44715118e-03, 1.06353138e-04, 4.85263094e-03, 6.98488845e-03],\n",
       "       [3.64192253e-04, 7.45412247e-03, 5.48592142e-04, 4.34495469e-05,\n",
       "        7.83234772e-05, 1.57534277e-04, 1.06355004e-04, 9.74254389e-01,\n",
       "        4.11682304e-03, 5.65118383e-05, 1.26672339e-04, 1.05153247e-01,\n",
       "        7.17552913e-05, 4.16386874e-01, 8.67641735e-04, 5.74287938e-05],\n",
       "       [3.67052593e-02, 2.59595045e-01, 5.08643961e-02, 4.27228363e-03,\n",
       "        1.91904820e-02, 5.34300776e-02, 7.01416610e-03, 4.11682304e-03,\n",
       "        8.29834247e-01, 1.15111590e-02, 9.34188039e-03, 2.90497974e-02,\n",
       "        1.62800916e-02, 4.11322350e-03, 1.56130856e-01, 1.37577103e-02],\n",
       "       [2.17831412e-03, 1.08554513e-02, 2.32211758e-03, 8.52091252e-02,\n",
       "        1.16944628e-01, 7.06750341e-02, 6.88499149e-03, 5.65118383e-05,\n",
       "        1.15111590e-02, 7.88493331e-01, 4.84430197e-03, 4.42795781e-02,\n",
       "        1.32161469e-01, 5.65111812e-05, 2.81017529e-02, 2.14756896e-01],\n",
       "       [1.53018811e-01, 9.42230365e-03, 1.31357059e-01, 6.42787813e-02,\n",
       "        3.18464532e-03, 3.20671936e-03, 2.66051240e-01, 1.26672339e-04,\n",
       "        9.34188039e-03, 4.84430197e-03, 8.35086990e-01, 2.12357566e-03,\n",
       "        3.26318052e-03, 1.26667780e-04, 6.12845356e-03, 4.98766293e-03],\n",
       "       [2.99521108e-03, 4.62753688e-02, 4.20535775e-03, 2.09064533e-02,\n",
       "        2.02063668e-02, 1.39231397e-02, 2.49718807e-03, 1.05153247e-01,\n",
       "        2.90497974e-02, 4.42795781e-02, 2.12357566e-03, 9.85389619e-01,\n",
       "        2.22730779e-02, 1.03262347e-01, 1.20041162e-02, 5.09982539e-02],\n",
       "       [2.46207380e-03, 1.35503984e-02, 2.88066982e-03, 4.73583020e-02,\n",
       "        2.10997132e-01, 1.21688061e-01, 4.44715118e-03, 7.17552913e-05,\n",
       "        1.62800916e-02, 1.32161469e-01, 3.26318052e-03, 2.22730779e-02,\n",
       "        7.72106731e-01, 7.17543562e-05, 4.59194790e-02, 1.09122074e-01],\n",
       "       [3.64147880e-04, 7.43400467e-03, 5.48419839e-04, 4.34491268e-05,\n",
       "        7.83219385e-05, 1.57521839e-04, 1.06353138e-04, 4.16386874e-01,\n",
       "        4.11322350e-03, 5.65111812e-05, 1.26667780e-04, 1.03262347e-01,\n",
       "        7.17543562e-05, 9.76012438e-01, 8.67373724e-04, 5.74278262e-05],\n",
       "       [2.14004989e-02, 1.31759921e-01, 2.86873768e-02, 8.81397402e-03,\n",
       "        5.24059953e-02, 1.30645507e-01, 4.85263094e-03, 8.67641735e-04,\n",
       "        1.56130856e-01, 2.81017529e-02, 6.12845356e-03, 1.20041162e-02,\n",
       "        4.59194790e-02, 8.67373724e-04, 8.45323864e-01, 3.21016480e-02],\n",
       "       [2.48877638e-03, 1.29612998e-02, 2.75469464e-03, 8.13328779e-02,\n",
       "        9.97519028e-02, 7.41945078e-02, 6.98488845e-03, 5.74287938e-05,\n",
       "        1.37577103e-02, 2.14756896e-01, 4.98766293e-03, 5.09982539e-02,\n",
       "        1.09122074e-01, 5.74278262e-05, 3.21016480e-02, 8.12179733e-01]])"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "compute.snf(aff, K= k_snf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "e0aa768e-afb9-4bce-ab54-4153b3f0442d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "1\n"
     ]
    }
   ],
   "source": [
    "Wk = [0] * len(aff)\n",
    "Wsum = np.zeros(aff[0].shape)\n",
    "\n",
    "# get number of modalities informing each subject x subject affinity\n",
    "n_aff = len(aff) - np.sum([np.isnan(a) for a in aff], axis=0)\n",
    "\n",
    "for n, mat in enumerate(aff):\n",
    "    print(1)\n",
    "    # normalize affinity matrix based on strength of edges\n",
    "    mat = mat / np.nansum(mat, axis=1, keepdims=True)\n",
    "    aff[n] = check_symmetric(mat, raise_warning=False)\n",
    "    # apply KNN threshold to normalized affinity matrix\n",
    "    Wk[n] = _find_dominate_set(aff[n], int(K))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "024c424f-bacf-4cc1-9b57-bce1304905f5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "68.75"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Wk = aff[n].copy()\n",
    "cutoff = 100 - (100 * (K / len(Wk)))\n",
    "cutoff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "aae93027-3115-4d1c-b6fb-46808f79f082",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'survivalDFS':         PatientID     Survival  Death\n",
       " MB-0002   MB-0002  2539.000000    0.0\n",
       " MB-0005   MB-0005  4893.000000    1.0\n",
       " MB-0006   MB-0006  4947.999999    0.0\n",
       " MB-0010   MB-0010   234.000000    1.0\n",
       " MB-0014   MB-0014  4929.999999    0.0\n",
       " ...           ...          ...    ...\n",
       " MB-0291   MB-0291  1176.000000    1.0\n",
       " MB-0444   MB-0444  1664.000000    1.0\n",
       " MB-0482   MB-0482   746.000000    1.0\n",
       " MB-0519   MB-0519  3329.000001    0.0\n",
       " MB-0593   MB-0593  1040.000000    1.0\n",
       " \n",
       " [1980 rows x 3 columns],\n",
       " 'survival':         PatientID     Survival  Death\n",
       " MB-0002   MB-0002  2539.000000    0.0\n",
       " MB-0005   MB-0005  4893.000000    1.0\n",
       " MB-0006   MB-0006  4947.999999    0.0\n",
       " MB-0010   MB-0010   234.000000    1.0\n",
       " MB-0014   MB-0014  4929.999999    0.0\n",
       " ...           ...          ...    ...\n",
       " MB-0291   MB-0291  1176.000000    1.0\n",
       " MB-0444   MB-0444  1664.000000    1.0\n",
       " MB-0482   MB-0482   746.000000    1.0\n",
       " MB-0519   MB-0519  3329.000001    0.0\n",
       " MB-0593   MB-0593  1040.000000    1.0\n",
       " \n",
       " [1980 rows x 3 columns],\n",
       " 'clinical':                         SAMPLE_ID   OS_MONTHS OS_STATUS  DFS_MONTHS   \n",
       " MB-0002  BRCA-METABRIC-S1-MB-0002   84.633333    LIVING   84.633333  \\\n",
       " MB-0005  BRCA-METABRIC-S1-MB-0005  163.100000  DECEASED  163.100000   \n",
       " MB-0006  BRCA-METABRIC-S1-MB-0006  164.933333    LIVING  164.933333   \n",
       " MB-0010  BRCA-METABRIC-S1-MB-0010    7.800000  DECEASED    7.800000   \n",
       " MB-0014  BRCA-METABRIC-S1-MB-0014  164.333333    LIVING  164.333333   \n",
       " ...                           ...         ...       ...         ...   \n",
       " MB-0291  BRCA-METABRIC-S1-MB-0291   39.200000  DECEASED   39.200000   \n",
       " MB-0444  BRCA-METABRIC-S1-MB-0444   55.466667  DECEASED   55.466667   \n",
       " MB-0482  BRCA-METABRIC-S1-MB-0482   24.866667  DECEASED   24.866667   \n",
       " MB-0519  BRCA-METABRIC-S1-MB-0519  110.966667    LIVING  110.966667   \n",
       " MB-0593  BRCA-METABRIC-S1-MB-0593   34.666667  DECEASED   34.666667   \n",
       " \n",
       "                   DFS_STATUS INTCLUST  COHORT  AGE_AT_DIAGNOSIS LATERALITY   \n",
       " MB-0002          DiseaseFree     4ER+       1             43.19          r  \\\n",
       " MB-0005  Recurred/Progressed        3       1             48.87          r   \n",
       " MB-0006          DiseaseFree        9       1             47.68          r   \n",
       " MB-0010  Recurred/Progressed        7       1             78.77          l   \n",
       " MB-0014          DiseaseFree        3       1             56.45          r   \n",
       " ...                      ...      ...     ...               ...        ...   \n",
       " MB-0291  Recurred/Progressed        5       1             51.19          r   \n",
       " MB-0444  Recurred/Progressed        8       1             49.87          l   \n",
       " MB-0482  Recurred/Progressed        5       1             39.53          r   \n",
       " MB-0519          DiseaseFree     4ER+       1             61.99          r   \n",
       " MB-0593  Recurred/Progressed        5       1             62.34          l   \n",
       " \n",
       "            NPI  ... ER_STATUS HER2_STATUS  PR_STATUS CLAUDIN_SUBTYPE   \n",
       " MB-0002  4.020  ...         +           -          +            LumA  \\\n",
       " MB-0005  4.030  ...         +           -          +            LumB   \n",
       " MB-0006  4.050  ...         +           -          +            LumB   \n",
       " MB-0010  4.062  ...         +           -          +            LumB   \n",
       " MB-0014  4.020  ...         +           -          +            LumB   \n",
       " ...        ...  ...       ...         ...        ...             ...   \n",
       " MB-0291  5.052  ...         +           +          +            LumB   \n",
       " MB-0444  5.066  ...         +           -          +            LumB   \n",
       " MB-0482  5.032  ...         -           +          -          Normal   \n",
       " MB-0519  6.130  ...         +           -          -     claudin-low   \n",
       " MB-0593  5.180  ...         -           +          -            Her2   \n",
       " \n",
       "         CHEMOTHERAPY HORMONE_THERAPY RADIO_THERAPY SAMPLE_TYPE    CANCER_TYPE   \n",
       " MB-0002           NO             YES           YES     Primary  Breast Cancer  \\\n",
       " MB-0005          YES             YES            NO     Primary  Breast Cancer   \n",
       " MB-0006          YES             YES           YES     Primary  Breast Cancer   \n",
       " MB-0010           NO             YES           YES     Primary  Breast Cancer   \n",
       " MB-0014          YES             YES           YES     Primary  Breast Cancer   \n",
       " ...              ...             ...           ...         ...            ...   \n",
       " MB-0291          YES             YES            NO     Primary  Breast Cancer   \n",
       " MB-0444           NO             YES           YES     Primary  Breast Cancer   \n",
       " MB-0482          YES              NO           YES     Primary  Breast Cancer   \n",
       " MB-0519          YES             YES           YES     Primary  Breast Cancer   \n",
       " MB-0593          YES              NO           YES     Primary  Breast Cancer   \n",
       " \n",
       "                      CANCER_TYPE_DETAILED  \n",
       " MB-0002  Breast Invasive Ductal Carcinoma  \n",
       " MB-0005  Breast Invasive Ductal Carcinoma  \n",
       " MB-0006  Breast Invasive Ductal Carcinoma  \n",
       " MB-0010  Breast Invasive Ductal Carcinoma  \n",
       " MB-0014  Breast Invasive Ductal Carcinoma  \n",
       " ...                                   ...  \n",
       " MB-0291  Breast Invasive Ductal Carcinoma  \n",
       " MB-0444  Breast Invasive Ductal Carcinoma  \n",
       " MB-0482  Breast Invasive Ductal Carcinoma  \n",
       " MB-0519  Breast Invasive Ductal Carcinoma  \n",
       " MB-0593  Breast Invasive Ductal Carcinoma  \n",
       " \n",
       " [1980 rows x 31 columns],\n",
       " 'mydatGE': <xarray.DataArray (dim_0: 995, dim_1: 48787)>\n",
       " array([[7.96349322, 5.57949414, 5.21390845, ..., 5.37350273, 7.39767234,\n",
       "         5.37213751],\n",
       "        [8.45269326, 5.4568271 , 5.20648335, ..., 5.46485272, 5.65141389,\n",
       "         5.35855981],\n",
       "        [7.58801812, 5.35056424, 5.41857562, ..., 5.39656165, 6.8094217 ,\n",
       "         5.51390586],\n",
       "        ...,\n",
       "        [8.85862168, 5.49357852, 5.1606083 , ..., 5.25477262, 6.62060542,\n",
       "         5.63166168],\n",
       "        [8.41586735, 5.5844999 , 5.38666009, ..., 5.40462609, 7.43711579,\n",
       "         5.59051551],\n",
       "        [9.9027757 , 5.09404444, 5.40253774, ..., 5.09218386, 7.65651733,\n",
       "         5.26636799]])\n",
       " Coordinates:\n",
       "   * dim_0    (dim_0) <U7 'MB-0005' 'MB-0048' 'MB-0025' ... 'MB-4313' 'MB-4823'\n",
       "   * dim_1    (dim_1) <U12 'ILMN_1802380' 'ILMN_1893287' ... 'ILMN_1846115',\n",
       " 'mydatCNV':        chrom     start       end  geneid   genename  MB-0000  MB-0005   \n",
       " 1          1      1743      6658  643635  LOC643635      0.0      0.0  \\\n",
       " 8          1      3411      6658  643643  LOC643643      0.0      0.0   \n",
       " 11         1      3757     20510  653635  LOC653635      0.0      0.0   \n",
       " 24         1     23390     25938  645520  LOC645520      0.0      0.0   \n",
       " 40         1     42315     43258   79504     OR4G4P      0.0      0.0   \n",
       " ...      ...       ...       ...     ...        ...      ...      ...   \n",
       " 867315     Y  27104950  27105882  360009  LOC360009      0.0      0.0   \n",
       " 867317     Y  27150218  27190190  347613     PARP4P      0.0      0.0   \n",
       " 867319     Y  57410778  57411705  352905  LOC352905      0.0      0.0   \n",
       " 867327     Y  57617003  57617949  644939  LOC644939      0.0      0.0   \n",
       " 867341     Y  57727405  57728306  644970  LOC644970      0.0      0.0   \n",
       " \n",
       "         MB-0006  MB-0014  MB-0020  ...  MB-7289  MB-7291  MB-7292  MB-7293   \n",
       " 1           0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0  \\\n",
       " 8           0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " 11          0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " 24          0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " 40          0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " ...         ...      ...      ...  ...      ...      ...      ...      ...   \n",
       " 867315      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " 867317      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " 867319      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " 867327      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " 867341      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " \n",
       "         MB-7294  MB-7295  MB-7296  MB-7297  MB-7298  MB-7299  \n",
       " 1           0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 8           0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 11          0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 24          0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 40          0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " ...         ...      ...      ...      ...      ...      ...  \n",
       " 867315      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 867317      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 867319      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 867327      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 867341      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " \n",
       " [28918 rows x 1000 columns]}"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "parsed2 = rdata.parser.parse_file(\"imvc/datasets/data/metabric/METABRIC_validation.RData\")\n",
    "converted2 = rdata.conversion.convert(parsed2)\n",
    "converted2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "a490b8f0-ffc7-486e-a093-0d2409c616e5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ILMN_1802380</th>\n",
       "      <th>ILMN_1893287</th>\n",
       "      <th>ILMN_1736104</th>\n",
       "      <th>ILMN_1792389</th>\n",
       "      <th>ILMN_1854015</th>\n",
       "      <th>ILMN_1904757</th>\n",
       "      <th>ILMN_1740305</th>\n",
       "      <th>ILMN_1665168</th>\n",
       "      <th>ILMN_2375156</th>\n",
       "      <th>ILMN_1705423</th>\n",
       "      <th>...</th>\n",
       "      <th>ILMN_1659781</th>\n",
       "      <th>ILMN_1908807</th>\n",
       "      <th>ILMN_1701127</th>\n",
       "      <th>ILMN_1751164</th>\n",
       "      <th>ILMN_1843643</th>\n",
       "      <th>ILMN_1852347</th>\n",
       "      <th>ILMN_1909210</th>\n",
       "      <th>ILMN_1693941</th>\n",
       "      <th>ILMN_1846115</th>\n",
       "      <th>ILMN_1709472</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MB-0362</th>\n",
       "      <td>8.676978</td>\n",
       "      <td>5.298711</td>\n",
       "      <td>5.430877</td>\n",
       "      <td>6.075331</td>\n",
       "      <td>5.595625</td>\n",
       "      <td>5.453928</td>\n",
       "      <td>5.490240</td>\n",
       "      <td>4.994525</td>\n",
       "      <td>5.838270</td>\n",
       "      <td>5.498484</td>\n",
       "      <td>...</td>\n",
       "      <td>5.161796</td>\n",
       "      <td>5.504426</td>\n",
       "      <td>6.353215</td>\n",
       "      <td>5.472163</td>\n",
       "      <td>4.836483</td>\n",
       "      <td>5.319774</td>\n",
       "      <td>5.075572</td>\n",
       "      <td>7.304643</td>\n",
       "      <td>5.251843</td>\n",
       "      <td>5.049591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0346</th>\n",
       "      <td>9.653589</td>\n",
       "      <td>5.378801</td>\n",
       "      <td>5.199253</td>\n",
       "      <td>6.687887</td>\n",
       "      <td>6.010127</td>\n",
       "      <td>5.454185</td>\n",
       "      <td>5.150210</td>\n",
       "      <td>5.346010</td>\n",
       "      <td>5.600876</td>\n",
       "      <td>5.454868</td>\n",
       "      <td>...</td>\n",
       "      <td>5.197392</td>\n",
       "      <td>5.276526</td>\n",
       "      <td>6.132355</td>\n",
       "      <td>5.289956</td>\n",
       "      <td>5.316819</td>\n",
       "      <td>5.141559</td>\n",
       "      <td>5.420024</td>\n",
       "      <td>7.933324</td>\n",
       "      <td>5.450611</td>\n",
       "      <td>5.316790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0386</th>\n",
       "      <td>9.033589</td>\n",
       "      <td>5.606122</td>\n",
       "      <td>5.449121</td>\n",
       "      <td>5.910885</td>\n",
       "      <td>5.683969</td>\n",
       "      <td>5.501577</td>\n",
       "      <td>5.385091</td>\n",
       "      <td>5.247467</td>\n",
       "      <td>6.030718</td>\n",
       "      <td>5.296832</td>\n",
       "      <td>...</td>\n",
       "      <td>8.087722</td>\n",
       "      <td>5.322883</td>\n",
       "      <td>6.366335</td>\n",
       "      <td>5.672556</td>\n",
       "      <td>5.466419</td>\n",
       "      <td>5.508364</td>\n",
       "      <td>5.349676</td>\n",
       "      <td>7.580336</td>\n",
       "      <td>5.235394</td>\n",
       "      <td>5.461617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0574</th>\n",
       "      <td>8.814855</td>\n",
       "      <td>5.316155</td>\n",
       "      <td>5.309371</td>\n",
       "      <td>5.628740</td>\n",
       "      <td>5.479983</td>\n",
       "      <td>5.471941</td>\n",
       "      <td>5.416758</td>\n",
       "      <td>5.316523</td>\n",
       "      <td>5.849428</td>\n",
       "      <td>5.373278</td>\n",
       "      <td>...</td>\n",
       "      <td>5.780062</td>\n",
       "      <td>5.466534</td>\n",
       "      <td>6.424048</td>\n",
       "      <td>5.492811</td>\n",
       "      <td>5.193150</td>\n",
       "      <td>5.419906</td>\n",
       "      <td>5.124697</td>\n",
       "      <td>6.903654</td>\n",
       "      <td>5.091927</td>\n",
       "      <td>5.227130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0185</th>\n",
       "      <td>8.736406</td>\n",
       "      <td>5.303613</td>\n",
       "      <td>5.438538</td>\n",
       "      <td>6.392422</td>\n",
       "      <td>6.013500</td>\n",
       "      <td>5.525027</td>\n",
       "      <td>5.769779</td>\n",
       "      <td>5.032408</td>\n",
       "      <td>5.542133</td>\n",
       "      <td>5.552146</td>\n",
       "      <td>...</td>\n",
       "      <td>5.327687</td>\n",
       "      <td>5.495221</td>\n",
       "      <td>6.252966</td>\n",
       "      <td>5.441239</td>\n",
       "      <td>5.299070</td>\n",
       "      <td>5.489531</td>\n",
       "      <td>5.299731</td>\n",
       "      <td>6.848395</td>\n",
       "      <td>5.238651</td>\n",
       "      <td>5.057761</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 48803 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ILMN_1802380  ILMN_1893287  ILMN_1736104  ILMN_1792389  ILMN_1854015   \n",
       "MB-0362      8.676978      5.298711      5.430877      6.075331      5.595625  \\\n",
       "MB-0346      9.653589      5.378801      5.199253      6.687887      6.010127   \n",
       "MB-0386      9.033589      5.606122      5.449121      5.910885      5.683969   \n",
       "MB-0574      8.814855      5.316155      5.309371      5.628740      5.479983   \n",
       "MB-0185      8.736406      5.303613      5.438538      6.392422      6.013500   \n",
       "\n",
       "         ILMN_1904757  ILMN_1740305  ILMN_1665168  ILMN_2375156  ILMN_1705423   \n",
       "MB-0362      5.453928      5.490240      4.994525      5.838270      5.498484  \\\n",
       "MB-0346      5.454185      5.150210      5.346010      5.600876      5.454868   \n",
       "MB-0386      5.501577      5.385091      5.247467      6.030718      5.296832   \n",
       "MB-0574      5.471941      5.416758      5.316523      5.849428      5.373278   \n",
       "MB-0185      5.525027      5.769779      5.032408      5.542133      5.552146   \n",
       "\n",
       "         ...  ILMN_1659781  ILMN_1908807  ILMN_1701127  ILMN_1751164   \n",
       "MB-0362  ...      5.161796      5.504426      6.353215      5.472163  \\\n",
       "MB-0346  ...      5.197392      5.276526      6.132355      5.289956   \n",
       "MB-0386  ...      8.087722      5.322883      6.366335      5.672556   \n",
       "MB-0574  ...      5.780062      5.466534      6.424048      5.492811   \n",
       "MB-0185  ...      5.327687      5.495221      6.252966      5.441239   \n",
       "\n",
       "         ILMN_1843643  ILMN_1852347  ILMN_1909210  ILMN_1693941  ILMN_1846115   \n",
       "MB-0362      4.836483      5.319774      5.075572      7.304643      5.251843  \\\n",
       "MB-0346      5.316819      5.141559      5.420024      7.933324      5.450611   \n",
       "MB-0386      5.466419      5.508364      5.349676      7.580336      5.235394   \n",
       "MB-0574      5.193150      5.419906      5.124697      6.903654      5.091927   \n",
       "MB-0185      5.299070      5.489531      5.299731      6.848395      5.238651   \n",
       "\n",
       "         ILMN_1709472  \n",
       "MB-0362      5.049591  \n",
       "MB-0346      5.316790  \n",
       "MB-0386      5.461617  \n",
       "MB-0574      5.227130  \n",
       "MB-0185      5.057761  \n",
       "\n",
       "[5 rows x 48803 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mydatGE_dis = pd.DataFrame(converted[\"mydatGE\"], index = converted[\"mydatGE\"].dim_0, columns = converted[\"mydatGE\"].dim_1)\n",
    "mydatGE_dis.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "38da1787-4bf6-4612-94aa-fefe82c9ba54",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ILMN_1802380</th>\n",
       "      <th>ILMN_1893287</th>\n",
       "      <th>ILMN_1736104</th>\n",
       "      <th>ILMN_1792389</th>\n",
       "      <th>ILMN_1854015</th>\n",
       "      <th>ILMN_1904757</th>\n",
       "      <th>ILMN_1740305</th>\n",
       "      <th>ILMN_1665168</th>\n",
       "      <th>ILMN_2375156</th>\n",
       "      <th>ILMN_1705423</th>\n",
       "      <th>...</th>\n",
       "      <th>ILMN_1868532</th>\n",
       "      <th>ILMN_1659781</th>\n",
       "      <th>ILMN_1908807</th>\n",
       "      <th>ILMN_1701127</th>\n",
       "      <th>ILMN_1751164</th>\n",
       "      <th>ILMN_1843643</th>\n",
       "      <th>ILMN_1852347</th>\n",
       "      <th>ILMN_1909210</th>\n",
       "      <th>ILMN_1693941</th>\n",
       "      <th>ILMN_1846115</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MB-0005</th>\n",
       "      <td>7.963493</td>\n",
       "      <td>5.579494</td>\n",
       "      <td>5.213908</td>\n",
       "      <td>5.553056</td>\n",
       "      <td>5.786583</td>\n",
       "      <td>5.472386</td>\n",
       "      <td>5.607341</td>\n",
       "      <td>5.265556</td>\n",
       "      <td>5.793398</td>\n",
       "      <td>5.296342</td>\n",
       "      <td>...</td>\n",
       "      <td>5.453946</td>\n",
       "      <td>5.306294</td>\n",
       "      <td>5.569000</td>\n",
       "      <td>6.306927</td>\n",
       "      <td>5.419689</td>\n",
       "      <td>5.362207</td>\n",
       "      <td>5.440643</td>\n",
       "      <td>5.373503</td>\n",
       "      <td>7.397672</td>\n",
       "      <td>5.372138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0048</th>\n",
       "      <td>8.452693</td>\n",
       "      <td>5.456827</td>\n",
       "      <td>5.206483</td>\n",
       "      <td>6.219375</td>\n",
       "      <td>5.747841</td>\n",
       "      <td>5.405220</td>\n",
       "      <td>5.565285</td>\n",
       "      <td>5.327176</td>\n",
       "      <td>5.689668</td>\n",
       "      <td>5.387294</td>\n",
       "      <td>...</td>\n",
       "      <td>5.561369</td>\n",
       "      <td>5.176560</td>\n",
       "      <td>5.547286</td>\n",
       "      <td>6.065291</td>\n",
       "      <td>5.563191</td>\n",
       "      <td>5.180065</td>\n",
       "      <td>5.547610</td>\n",
       "      <td>5.464853</td>\n",
       "      <td>5.651414</td>\n",
       "      <td>5.358560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0025</th>\n",
       "      <td>7.588018</td>\n",
       "      <td>5.350564</td>\n",
       "      <td>5.418576</td>\n",
       "      <td>5.755084</td>\n",
       "      <td>5.733067</td>\n",
       "      <td>5.429913</td>\n",
       "      <td>5.547107</td>\n",
       "      <td>5.204948</td>\n",
       "      <td>5.679817</td>\n",
       "      <td>5.472628</td>\n",
       "      <td>...</td>\n",
       "      <td>5.615135</td>\n",
       "      <td>5.390555</td>\n",
       "      <td>5.400563</td>\n",
       "      <td>6.019489</td>\n",
       "      <td>5.600126</td>\n",
       "      <td>5.206286</td>\n",
       "      <td>5.555487</td>\n",
       "      <td>5.396562</td>\n",
       "      <td>6.809422</td>\n",
       "      <td>5.513906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0083</th>\n",
       "      <td>8.248024</td>\n",
       "      <td>5.284935</td>\n",
       "      <td>5.326922</td>\n",
       "      <td>5.838722</td>\n",
       "      <td>5.594793</td>\n",
       "      <td>5.377686</td>\n",
       "      <td>5.669123</td>\n",
       "      <td>5.073629</td>\n",
       "      <td>5.684246</td>\n",
       "      <td>5.471562</td>\n",
       "      <td>...</td>\n",
       "      <td>5.715747</td>\n",
       "      <td>5.629181</td>\n",
       "      <td>5.636301</td>\n",
       "      <td>6.115126</td>\n",
       "      <td>5.529249</td>\n",
       "      <td>5.052481</td>\n",
       "      <td>5.658572</td>\n",
       "      <td>5.537335</td>\n",
       "      <td>6.611526</td>\n",
       "      <td>5.552590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0053</th>\n",
       "      <td>7.789060</td>\n",
       "      <td>5.364712</td>\n",
       "      <td>5.322772</td>\n",
       "      <td>6.122633</td>\n",
       "      <td>5.683303</td>\n",
       "      <td>5.326070</td>\n",
       "      <td>5.473555</td>\n",
       "      <td>5.233173</td>\n",
       "      <td>5.693977</td>\n",
       "      <td>5.756432</td>\n",
       "      <td>...</td>\n",
       "      <td>5.610843</td>\n",
       "      <td>5.238755</td>\n",
       "      <td>5.611570</td>\n",
       "      <td>5.878613</td>\n",
       "      <td>5.561742</td>\n",
       "      <td>5.317208</td>\n",
       "      <td>5.350093</td>\n",
       "      <td>5.383360</td>\n",
       "      <td>5.646037</td>\n",
       "      <td>5.352785</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 48787 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ILMN_1802380  ILMN_1893287  ILMN_1736104  ILMN_1792389  ILMN_1854015   \n",
       "MB-0005      7.963493      5.579494      5.213908      5.553056      5.786583  \\\n",
       "MB-0048      8.452693      5.456827      5.206483      6.219375      5.747841   \n",
       "MB-0025      7.588018      5.350564      5.418576      5.755084      5.733067   \n",
       "MB-0083      8.248024      5.284935      5.326922      5.838722      5.594793   \n",
       "MB-0053      7.789060      5.364712      5.322772      6.122633      5.683303   \n",
       "\n",
       "         ILMN_1904757  ILMN_1740305  ILMN_1665168  ILMN_2375156  ILMN_1705423   \n",
       "MB-0005      5.472386      5.607341      5.265556      5.793398      5.296342  \\\n",
       "MB-0048      5.405220      5.565285      5.327176      5.689668      5.387294   \n",
       "MB-0025      5.429913      5.547107      5.204948      5.679817      5.472628   \n",
       "MB-0083      5.377686      5.669123      5.073629      5.684246      5.471562   \n",
       "MB-0053      5.326070      5.473555      5.233173      5.693977      5.756432   \n",
       "\n",
       "         ...  ILMN_1868532  ILMN_1659781  ILMN_1908807  ILMN_1701127   \n",
       "MB-0005  ...      5.453946      5.306294      5.569000      6.306927  \\\n",
       "MB-0048  ...      5.561369      5.176560      5.547286      6.065291   \n",
       "MB-0025  ...      5.615135      5.390555      5.400563      6.019489   \n",
       "MB-0083  ...      5.715747      5.629181      5.636301      6.115126   \n",
       "MB-0053  ...      5.610843      5.238755      5.611570      5.878613   \n",
       "\n",
       "         ILMN_1751164  ILMN_1843643  ILMN_1852347  ILMN_1909210  ILMN_1693941   \n",
       "MB-0005      5.419689      5.362207      5.440643      5.373503      7.397672  \\\n",
       "MB-0048      5.563191      5.180065      5.547610      5.464853      5.651414   \n",
       "MB-0025      5.600126      5.206286      5.555487      5.396562      6.809422   \n",
       "MB-0083      5.529249      5.052481      5.658572      5.537335      6.611526   \n",
       "MB-0053      5.561742      5.317208      5.350093      5.383360      5.646037   \n",
       "\n",
       "         ILMN_1846115  \n",
       "MB-0005      5.372138  \n",
       "MB-0048      5.358560  \n",
       "MB-0025      5.513906  \n",
       "MB-0083      5.552590  \n",
       "MB-0053      5.352785  \n",
       "\n",
       "[5 rows x 48787 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mydatGE_val = pd.DataFrame(converted2[\"mydatGE\"], index = converted2[\"mydatGE\"].dim_0, columns = converted2[\"mydatGE\"].dim_1)\n",
    "mydatGE_val.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "b8e0cacf-671a-499c-8aa5-de71c08d53c5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['MB-0005', 'MB-0048', 'MB-0025', 'MB-0083', 'MB-0053', 'MB-0056',\n",
       "       'MB-0068', 'MB-0093', 'MB-0079', 'MB-0108',\n",
       "       ...\n",
       "       'MB-6192', 'MB-4820', 'MB-5527', 'MB-5167', 'MB-5465', 'MB-5453',\n",
       "       'MB-5471', 'MB-5127', 'MB-4313', 'MB-4823'],\n",
       "      dtype='object', length=995)"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mydatGE_val.index.intersection(converted2[\"mydatCNV\"].T.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e36cd69-bdc8-41fc-810b-8b4e98eff65b",
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.DataFrame(converted2[\"mydatGE\"].dim_0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "b2f4dace-a98f-4f59-af3d-c776f1cfb25b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'survivalDFS':         PatientID     Survival  Death\n",
       " MB-0002   MB-0002  2539.000000    0.0\n",
       " MB-0005   MB-0005  4893.000000    1.0\n",
       " MB-0006   MB-0006  4947.999999    0.0\n",
       " MB-0010   MB-0010   234.000000    1.0\n",
       " MB-0014   MB-0014  4929.999999    0.0\n",
       " ...           ...          ...    ...\n",
       " MB-0291   MB-0291  1176.000000    1.0\n",
       " MB-0444   MB-0444  1664.000000    1.0\n",
       " MB-0482   MB-0482   746.000000    1.0\n",
       " MB-0519   MB-0519  3329.000001    0.0\n",
       " MB-0593   MB-0593  1040.000000    1.0\n",
       " \n",
       " [1980 rows x 3 columns],\n",
       " 'survival':         PatientID     Survival  Death\n",
       " MB-0002   MB-0002  2539.000000    0.0\n",
       " MB-0005   MB-0005  4893.000000    1.0\n",
       " MB-0006   MB-0006  4947.999999    0.0\n",
       " MB-0010   MB-0010   234.000000    1.0\n",
       " MB-0014   MB-0014  4929.999999    0.0\n",
       " ...           ...          ...    ...\n",
       " MB-0291   MB-0291  1176.000000    1.0\n",
       " MB-0444   MB-0444  1664.000000    1.0\n",
       " MB-0482   MB-0482   746.000000    1.0\n",
       " MB-0519   MB-0519  3329.000001    0.0\n",
       " MB-0593   MB-0593  1040.000000    1.0\n",
       " \n",
       " [1980 rows x 3 columns],\n",
       " 'clinical':                         SAMPLE_ID   OS_MONTHS OS_STATUS  DFS_MONTHS   \n",
       " MB-0002  BRCA-METABRIC-S1-MB-0002   84.633333    LIVING   84.633333  \\\n",
       " MB-0005  BRCA-METABRIC-S1-MB-0005  163.100000  DECEASED  163.100000   \n",
       " MB-0006  BRCA-METABRIC-S1-MB-0006  164.933333    LIVING  164.933333   \n",
       " MB-0010  BRCA-METABRIC-S1-MB-0010    7.800000  DECEASED    7.800000   \n",
       " MB-0014  BRCA-METABRIC-S1-MB-0014  164.333333    LIVING  164.333333   \n",
       " ...                           ...         ...       ...         ...   \n",
       " MB-0291  BRCA-METABRIC-S1-MB-0291   39.200000  DECEASED   39.200000   \n",
       " MB-0444  BRCA-METABRIC-S1-MB-0444   55.466667  DECEASED   55.466667   \n",
       " MB-0482  BRCA-METABRIC-S1-MB-0482   24.866667  DECEASED   24.866667   \n",
       " MB-0519  BRCA-METABRIC-S1-MB-0519  110.966667    LIVING  110.966667   \n",
       " MB-0593  BRCA-METABRIC-S1-MB-0593   34.666667  DECEASED   34.666667   \n",
       " \n",
       "                   DFS_STATUS INTCLUST  COHORT  AGE_AT_DIAGNOSIS LATERALITY   \n",
       " MB-0002          DiseaseFree     4ER+       1             43.19          r  \\\n",
       " MB-0005  Recurred/Progressed        3       1             48.87          r   \n",
       " MB-0006          DiseaseFree        9       1             47.68          r   \n",
       " MB-0010  Recurred/Progressed        7       1             78.77          l   \n",
       " MB-0014          DiseaseFree        3       1             56.45          r   \n",
       " ...                      ...      ...     ...               ...        ...   \n",
       " MB-0291  Recurred/Progressed        5       1             51.19          r   \n",
       " MB-0444  Recurred/Progressed        8       1             49.87          l   \n",
       " MB-0482  Recurred/Progressed        5       1             39.53          r   \n",
       " MB-0519          DiseaseFree     4ER+       1             61.99          r   \n",
       " MB-0593  Recurred/Progressed        5       1             62.34          l   \n",
       " \n",
       "            NPI  ... ER_STATUS HER2_STATUS  PR_STATUS CLAUDIN_SUBTYPE   \n",
       " MB-0002  4.020  ...         +           -          +            LumA  \\\n",
       " MB-0005  4.030  ...         +           -          +            LumB   \n",
       " MB-0006  4.050  ...         +           -          +            LumB   \n",
       " MB-0010  4.062  ...         +           -          +            LumB   \n",
       " MB-0014  4.020  ...         +           -          +            LumB   \n",
       " ...        ...  ...       ...         ...        ...             ...   \n",
       " MB-0291  5.052  ...         +           +          +            LumB   \n",
       " MB-0444  5.066  ...         +           -          +            LumB   \n",
       " MB-0482  5.032  ...         -           +          -          Normal   \n",
       " MB-0519  6.130  ...         +           -          -     claudin-low   \n",
       " MB-0593  5.180  ...         -           +          -            Her2   \n",
       " \n",
       "         CHEMOTHERAPY HORMONE_THERAPY RADIO_THERAPY SAMPLE_TYPE    CANCER_TYPE   \n",
       " MB-0002           NO             YES           YES     Primary  Breast Cancer  \\\n",
       " MB-0005          YES             YES            NO     Primary  Breast Cancer   \n",
       " MB-0006          YES             YES           YES     Primary  Breast Cancer   \n",
       " MB-0010           NO             YES           YES     Primary  Breast Cancer   \n",
       " MB-0014          YES             YES           YES     Primary  Breast Cancer   \n",
       " ...              ...             ...           ...         ...            ...   \n",
       " MB-0291          YES             YES            NO     Primary  Breast Cancer   \n",
       " MB-0444           NO             YES           YES     Primary  Breast Cancer   \n",
       " MB-0482          YES              NO           YES     Primary  Breast Cancer   \n",
       " MB-0519          YES             YES           YES     Primary  Breast Cancer   \n",
       " MB-0593          YES              NO           YES     Primary  Breast Cancer   \n",
       " \n",
       "                      CANCER_TYPE_DETAILED  \n",
       " MB-0002  Breast Invasive Ductal Carcinoma  \n",
       " MB-0005  Breast Invasive Ductal Carcinoma  \n",
       " MB-0006  Breast Invasive Ductal Carcinoma  \n",
       " MB-0010  Breast Invasive Ductal Carcinoma  \n",
       " MB-0014  Breast Invasive Ductal Carcinoma  \n",
       " ...                                   ...  \n",
       " MB-0291  Breast Invasive Ductal Carcinoma  \n",
       " MB-0444  Breast Invasive Ductal Carcinoma  \n",
       " MB-0482  Breast Invasive Ductal Carcinoma  \n",
       " MB-0519  Breast Invasive Ductal Carcinoma  \n",
       " MB-0593  Breast Invasive Ductal Carcinoma  \n",
       " \n",
       " [1980 rows x 31 columns],\n",
       " 'mydatGE': <xarray.DataArray (dim_0: 997, dim_1: 48803)>\n",
       " array([[8.67697843, 5.2987108 , 5.43087726, ..., 7.30464265, 5.25184266,\n",
       "         5.04959138],\n",
       "        [9.65358947, 5.3788011 , 5.19925326, ..., 7.93332437, 5.45061128,\n",
       "         5.31679043],\n",
       "        [9.03358867, 5.60612206, 5.44912067, ..., 7.58033627, 5.2353943 ,\n",
       "         5.46161666],\n",
       "        ...,\n",
       "        [8.46397248, 5.4275019 , 5.4179514 , ..., 5.53831813, 5.38044942,\n",
       "         5.18999156],\n",
       "        [9.43114084, 5.24035987, 5.25867442, ..., 8.37468289, 5.37471755,\n",
       "         5.24250766],\n",
       "        [9.52282737, 5.41068723, 5.19946056, ..., 8.84363172, 5.5057748 ,\n",
       "         5.47957053]])\n",
       " Coordinates:\n",
       "   * dim_0    (dim_0) <U7 'MB-0362' 'MB-0346' 'MB-0386' ... 'MB-0451' 'MB-0133'\n",
       "   * dim_1    (dim_1) <U12 'ILMN_1802380' 'ILMN_1893287' ... 'ILMN_1709472',\n",
       " 'mydatCNV':        chrom     start       end  geneid   genename  MB-0002  MB-0008   \n",
       " 1          1      1743      6658  643635  LOC643635      0.0      0.0  \\\n",
       " 8          1      3411      6658  643643  LOC643643      0.0      0.0   \n",
       " 11         1      3757     20510  653635  LOC653635      0.0      0.0   \n",
       " 24         1     23390     25938  645520  LOC645520      0.0      0.0   \n",
       " 40         1     42315     43258   79504     OR4G4P      0.0      0.0   \n",
       " ...      ...       ...       ...     ...        ...      ...      ...   \n",
       " 867315     Y  27104950  27105882  360009  LOC360009      0.0      0.0   \n",
       " 867317     Y  27150218  27190190  347613     PARP4P      0.0      0.0   \n",
       " 867319     Y  57410778  57411705  352905  LOC352905      0.0      0.0   \n",
       " 867327     Y  57617003  57617949  644939  LOC644939      0.0      0.0   \n",
       " 867341     Y  57727405  57728306  644970  LOC644970      0.0      0.0   \n",
       " \n",
       "         MB-0010  MB-0035  MB-0036  ...  MB-5634  MB-5635  MB-5638  MB-5642   \n",
       " 1           0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0  \\\n",
       " 8           0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " 11          0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " 24          0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " 40          0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " ...         ...      ...      ...  ...      ...      ...      ...      ...   \n",
       " 867315      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " 867317      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " 867319      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " 867327      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " 867341      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       " \n",
       "         MB-5645  MB-5646  MB-5647  MB-5651  MB-5653  MB-5654  \n",
       " 1           0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 8           0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 11          0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 24          0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 40          0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " ...         ...      ...      ...      ...      ...      ...  \n",
       " 867315      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 867317      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 867319      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 867327      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " 867341      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       " \n",
       " [28918 rows x 1002 columns]}"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "parsed = rdata.parser.parse_file(\"imvc/datasets/data/metabric/METABRIC_discovery.RData\")\n",
    "converted = rdata.conversion.convert(parsed)\n",
    "converted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "18ad66ff-83fd-447b-a541-b7b7e89920d2",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>48793</th>\n",
       "      <th>48794</th>\n",
       "      <th>48795</th>\n",
       "      <th>48796</th>\n",
       "      <th>48797</th>\n",
       "      <th>48798</th>\n",
       "      <th>48799</th>\n",
       "      <th>48800</th>\n",
       "      <th>48801</th>\n",
       "      <th>48802</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8.676978</td>\n",
       "      <td>5.298711</td>\n",
       "      <td>5.430877</td>\n",
       "      <td>6.075331</td>\n",
       "      <td>5.595625</td>\n",
       "      <td>5.453928</td>\n",
       "      <td>5.490240</td>\n",
       "      <td>4.994525</td>\n",
       "      <td>5.838270</td>\n",
       "      <td>5.498484</td>\n",
       "      <td>...</td>\n",
       "      <td>5.161796</td>\n",
       "      <td>5.504426</td>\n",
       "      <td>6.353215</td>\n",
       "      <td>5.472163</td>\n",
       "      <td>4.836483</td>\n",
       "      <td>5.319774</td>\n",
       "      <td>5.075572</td>\n",
       "      <td>7.304643</td>\n",
       "      <td>5.251843</td>\n",
       "      <td>5.049591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9.653589</td>\n",
       "      <td>5.378801</td>\n",
       "      <td>5.199253</td>\n",
       "      <td>6.687887</td>\n",
       "      <td>6.010127</td>\n",
       "      <td>5.454185</td>\n",
       "      <td>5.150210</td>\n",
       "      <td>5.346010</td>\n",
       "      <td>5.600876</td>\n",
       "      <td>5.454868</td>\n",
       "      <td>...</td>\n",
       "      <td>5.197392</td>\n",
       "      <td>5.276526</td>\n",
       "      <td>6.132355</td>\n",
       "      <td>5.289956</td>\n",
       "      <td>5.316819</td>\n",
       "      <td>5.141559</td>\n",
       "      <td>5.420024</td>\n",
       "      <td>7.933324</td>\n",
       "      <td>5.450611</td>\n",
       "      <td>5.316790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9.033589</td>\n",
       "      <td>5.606122</td>\n",
       "      <td>5.449121</td>\n",
       "      <td>5.910885</td>\n",
       "      <td>5.683969</td>\n",
       "      <td>5.501577</td>\n",
       "      <td>5.385091</td>\n",
       "      <td>5.247467</td>\n",
       "      <td>6.030718</td>\n",
       "      <td>5.296832</td>\n",
       "      <td>...</td>\n",
       "      <td>8.087722</td>\n",
       "      <td>5.322883</td>\n",
       "      <td>6.366335</td>\n",
       "      <td>5.672556</td>\n",
       "      <td>5.466419</td>\n",
       "      <td>5.508364</td>\n",
       "      <td>5.349676</td>\n",
       "      <td>7.580336</td>\n",
       "      <td>5.235394</td>\n",
       "      <td>5.461617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8.814855</td>\n",
       "      <td>5.316155</td>\n",
       "      <td>5.309371</td>\n",
       "      <td>5.628740</td>\n",
       "      <td>5.479983</td>\n",
       "      <td>5.471941</td>\n",
       "      <td>5.416758</td>\n",
       "      <td>5.316523</td>\n",
       "      <td>5.849428</td>\n",
       "      <td>5.373278</td>\n",
       "      <td>...</td>\n",
       "      <td>5.780062</td>\n",
       "      <td>5.466534</td>\n",
       "      <td>6.424048</td>\n",
       "      <td>5.492811</td>\n",
       "      <td>5.193150</td>\n",
       "      <td>5.419906</td>\n",
       "      <td>5.124697</td>\n",
       "      <td>6.903654</td>\n",
       "      <td>5.091927</td>\n",
       "      <td>5.227130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8.736406</td>\n",
       "      <td>5.303613</td>\n",
       "      <td>5.438538</td>\n",
       "      <td>6.392422</td>\n",
       "      <td>6.013500</td>\n",
       "      <td>5.525027</td>\n",
       "      <td>5.769779</td>\n",
       "      <td>5.032408</td>\n",
       "      <td>5.542133</td>\n",
       "      <td>5.552146</td>\n",
       "      <td>...</td>\n",
       "      <td>5.327687</td>\n",
       "      <td>5.495221</td>\n",
       "      <td>6.252966</td>\n",
       "      <td>5.441239</td>\n",
       "      <td>5.299070</td>\n",
       "      <td>5.489531</td>\n",
       "      <td>5.299731</td>\n",
       "      <td>6.848395</td>\n",
       "      <td>5.238651</td>\n",
       "      <td>5.057761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>992</th>\n",
       "      <td>8.850144</td>\n",
       "      <td>5.333344</td>\n",
       "      <td>5.331756</td>\n",
       "      <td>6.440723</td>\n",
       "      <td>5.814739</td>\n",
       "      <td>5.210231</td>\n",
       "      <td>5.321674</td>\n",
       "      <td>5.419619</td>\n",
       "      <td>5.828949</td>\n",
       "      <td>5.724069</td>\n",
       "      <td>...</td>\n",
       "      <td>5.218539</td>\n",
       "      <td>5.736583</td>\n",
       "      <td>6.166678</td>\n",
       "      <td>5.284162</td>\n",
       "      <td>5.325168</td>\n",
       "      <td>5.337517</td>\n",
       "      <td>5.569117</td>\n",
       "      <td>6.972185</td>\n",
       "      <td>5.157459</td>\n",
       "      <td>5.539762</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>993</th>\n",
       "      <td>9.132276</td>\n",
       "      <td>5.119812</td>\n",
       "      <td>5.760381</td>\n",
       "      <td>6.315227</td>\n",
       "      <td>6.113891</td>\n",
       "      <td>5.449235</td>\n",
       "      <td>5.066106</td>\n",
       "      <td>5.347050</td>\n",
       "      <td>5.936932</td>\n",
       "      <td>5.603672</td>\n",
       "      <td>...</td>\n",
       "      <td>5.157233</td>\n",
       "      <td>5.578294</td>\n",
       "      <td>5.926179</td>\n",
       "      <td>6.104279</td>\n",
       "      <td>5.399878</td>\n",
       "      <td>5.406930</td>\n",
       "      <td>5.106386</td>\n",
       "      <td>5.499776</td>\n",
       "      <td>5.392051</td>\n",
       "      <td>5.370991</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>994</th>\n",
       "      <td>8.463972</td>\n",
       "      <td>5.427502</td>\n",
       "      <td>5.417951</td>\n",
       "      <td>5.928433</td>\n",
       "      <td>6.192337</td>\n",
       "      <td>5.413672</td>\n",
       "      <td>5.223702</td>\n",
       "      <td>5.211153</td>\n",
       "      <td>5.814744</td>\n",
       "      <td>5.414783</td>\n",
       "      <td>...</td>\n",
       "      <td>5.316751</td>\n",
       "      <td>5.435352</td>\n",
       "      <td>6.207164</td>\n",
       "      <td>6.017801</td>\n",
       "      <td>5.190806</td>\n",
       "      <td>5.273789</td>\n",
       "      <td>5.180971</td>\n",
       "      <td>5.538318</td>\n",
       "      <td>5.380449</td>\n",
       "      <td>5.189992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>9.431141</td>\n",
       "      <td>5.240360</td>\n",
       "      <td>5.258674</td>\n",
       "      <td>8.670881</td>\n",
       "      <td>5.937419</td>\n",
       "      <td>5.537914</td>\n",
       "      <td>5.328730</td>\n",
       "      <td>5.233245</td>\n",
       "      <td>5.734515</td>\n",
       "      <td>5.651650</td>\n",
       "      <td>...</td>\n",
       "      <td>5.184897</td>\n",
       "      <td>5.369399</td>\n",
       "      <td>6.290866</td>\n",
       "      <td>5.603239</td>\n",
       "      <td>5.295244</td>\n",
       "      <td>5.543862</td>\n",
       "      <td>5.513684</td>\n",
       "      <td>8.374683</td>\n",
       "      <td>5.374718</td>\n",
       "      <td>5.242508</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>9.522827</td>\n",
       "      <td>5.410687</td>\n",
       "      <td>5.199461</td>\n",
       "      <td>6.785574</td>\n",
       "      <td>5.793706</td>\n",
       "      <td>5.900175</td>\n",
       "      <td>5.148857</td>\n",
       "      <td>5.133368</td>\n",
       "      <td>6.421485</td>\n",
       "      <td>5.254797</td>\n",
       "      <td>...</td>\n",
       "      <td>5.500127</td>\n",
       "      <td>5.701733</td>\n",
       "      <td>6.675644</td>\n",
       "      <td>5.361947</td>\n",
       "      <td>5.152201</td>\n",
       "      <td>5.353351</td>\n",
       "      <td>5.139631</td>\n",
       "      <td>8.843632</td>\n",
       "      <td>5.505775</td>\n",
       "      <td>5.479571</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>997 rows × 48803 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        0         1         2         3         4         5         6       \n",
       "0    8.676978  5.298711  5.430877  6.075331  5.595625  5.453928  5.490240  \\\n",
       "1    9.653589  5.378801  5.199253  6.687887  6.010127  5.454185  5.150210   \n",
       "2    9.033589  5.606122  5.449121  5.910885  5.683969  5.501577  5.385091   \n",
       "3    8.814855  5.316155  5.309371  5.628740  5.479983  5.471941  5.416758   \n",
       "4    8.736406  5.303613  5.438538  6.392422  6.013500  5.525027  5.769779   \n",
       "..        ...       ...       ...       ...       ...       ...       ...   \n",
       "992  8.850144  5.333344  5.331756  6.440723  5.814739  5.210231  5.321674   \n",
       "993  9.132276  5.119812  5.760381  6.315227  6.113891  5.449235  5.066106   \n",
       "994  8.463972  5.427502  5.417951  5.928433  6.192337  5.413672  5.223702   \n",
       "995  9.431141  5.240360  5.258674  8.670881  5.937419  5.537914  5.328730   \n",
       "996  9.522827  5.410687  5.199461  6.785574  5.793706  5.900175  5.148857   \n",
       "\n",
       "        7         8         9      ...     48793     48794     48795   \n",
       "0    4.994525  5.838270  5.498484  ...  5.161796  5.504426  6.353215  \\\n",
       "1    5.346010  5.600876  5.454868  ...  5.197392  5.276526  6.132355   \n",
       "2    5.247467  6.030718  5.296832  ...  8.087722  5.322883  6.366335   \n",
       "3    5.316523  5.849428  5.373278  ...  5.780062  5.466534  6.424048   \n",
       "4    5.032408  5.542133  5.552146  ...  5.327687  5.495221  6.252966   \n",
       "..        ...       ...       ...  ...       ...       ...       ...   \n",
       "992  5.419619  5.828949  5.724069  ...  5.218539  5.736583  6.166678   \n",
       "993  5.347050  5.936932  5.603672  ...  5.157233  5.578294  5.926179   \n",
       "994  5.211153  5.814744  5.414783  ...  5.316751  5.435352  6.207164   \n",
       "995  5.233245  5.734515  5.651650  ...  5.184897  5.369399  6.290866   \n",
       "996  5.133368  6.421485  5.254797  ...  5.500127  5.701733  6.675644   \n",
       "\n",
       "        48796     48797     48798     48799     48800     48801     48802  \n",
       "0    5.472163  4.836483  5.319774  5.075572  7.304643  5.251843  5.049591  \n",
       "1    5.289956  5.316819  5.141559  5.420024  7.933324  5.450611  5.316790  \n",
       "2    5.672556  5.466419  5.508364  5.349676  7.580336  5.235394  5.461617  \n",
       "3    5.492811  5.193150  5.419906  5.124697  6.903654  5.091927  5.227130  \n",
       "4    5.441239  5.299070  5.489531  5.299731  6.848395  5.238651  5.057761  \n",
       "..        ...       ...       ...       ...       ...       ...       ...  \n",
       "992  5.284162  5.325168  5.337517  5.569117  6.972185  5.157459  5.539762  \n",
       "993  6.104279  5.399878  5.406930  5.106386  5.499776  5.392051  5.370991  \n",
       "994  6.017801  5.190806  5.273789  5.180971  5.538318  5.380449  5.189992  \n",
       "995  5.603239  5.295244  5.543862  5.513684  8.374683  5.374718  5.242508  \n",
       "996  5.361947  5.152201  5.353351  5.139631  8.843632  5.505775  5.479571  \n",
       "\n",
       "[997 rows x 48803 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame(converted[\"mydatGE\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "4b29e560-9c6b-4437-b7fa-cf38d43e639f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>geneid</th>\n",
       "      <th>genename</th>\n",
       "      <th>MB-0002</th>\n",
       "      <th>MB-0008</th>\n",
       "      <th>MB-0010</th>\n",
       "      <th>MB-0035</th>\n",
       "      <th>MB-0036</th>\n",
       "      <th>...</th>\n",
       "      <th>MB-5634</th>\n",
       "      <th>MB-5635</th>\n",
       "      <th>MB-5638</th>\n",
       "      <th>MB-5642</th>\n",
       "      <th>MB-5645</th>\n",
       "      <th>MB-5646</th>\n",
       "      <th>MB-5647</th>\n",
       "      <th>MB-5651</th>\n",
       "      <th>MB-5653</th>\n",
       "      <th>MB-5654</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1743</td>\n",
       "      <td>6658</td>\n",
       "      <td>643635</td>\n",
       "      <td>LOC643635</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>3411</td>\n",
       "      <td>6658</td>\n",
       "      <td>643643</td>\n",
       "      <td>LOC643643</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1</td>\n",
       "      <td>3757</td>\n",
       "      <td>20510</td>\n",
       "      <td>653635</td>\n",
       "      <td>LOC653635</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>1</td>\n",
       "      <td>23390</td>\n",
       "      <td>25938</td>\n",
       "      <td>645520</td>\n",
       "      <td>LOC645520</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>1</td>\n",
       "      <td>42315</td>\n",
       "      <td>43258</td>\n",
       "      <td>79504</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>867315</th>\n",
       "      <td>Y</td>\n",
       "      <td>27104950</td>\n",
       "      <td>27105882</td>\n",
       "      <td>360009</td>\n",
       "      <td>LOC360009</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>867317</th>\n",
       "      <td>Y</td>\n",
       "      <td>27150218</td>\n",
       "      <td>27190190</td>\n",
       "      <td>347613</td>\n",
       "      <td>PARP4P</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>867319</th>\n",
       "      <td>Y</td>\n",
       "      <td>57410778</td>\n",
       "      <td>57411705</td>\n",
       "      <td>352905</td>\n",
       "      <td>LOC352905</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>867327</th>\n",
       "      <td>Y</td>\n",
       "      <td>57617003</td>\n",
       "      <td>57617949</td>\n",
       "      <td>644939</td>\n",
       "      <td>LOC644939</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>867341</th>\n",
       "      <td>Y</td>\n",
       "      <td>57727405</td>\n",
       "      <td>57728306</td>\n",
       "      <td>644970</td>\n",
       "      <td>LOC644970</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28918 rows × 1002 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chrom     start       end  geneid   genename  MB-0002  MB-0008   \n",
       "1          1      1743      6658  643635  LOC643635      0.0      0.0  \\\n",
       "8          1      3411      6658  643643  LOC643643      0.0      0.0   \n",
       "11         1      3757     20510  653635  LOC653635      0.0      0.0   \n",
       "24         1     23390     25938  645520  LOC645520      0.0      0.0   \n",
       "40         1     42315     43258   79504     OR4G4P      0.0      0.0   \n",
       "...      ...       ...       ...     ...        ...      ...      ...   \n",
       "867315     Y  27104950  27105882  360009  LOC360009      0.0      0.0   \n",
       "867317     Y  27150218  27190190  347613     PARP4P      0.0      0.0   \n",
       "867319     Y  57410778  57411705  352905  LOC352905      0.0      0.0   \n",
       "867327     Y  57617003  57617949  644939  LOC644939      0.0      0.0   \n",
       "867341     Y  57727405  57728306  644970  LOC644970      0.0      0.0   \n",
       "\n",
       "        MB-0010  MB-0035  MB-0036  ...  MB-5634  MB-5635  MB-5638  MB-5642   \n",
       "1           0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0  \\\n",
       "8           0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "11          0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "24          0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "40          0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "...         ...      ...      ...  ...      ...      ...      ...      ...   \n",
       "867315      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "867317      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "867319      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "867327      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "867341      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "\n",
       "        MB-5645  MB-5646  MB-5647  MB-5651  MB-5653  MB-5654  \n",
       "1           0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "8           0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "11          0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "24          0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "40          0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "...         ...      ...      ...      ...      ...      ...  \n",
       "867315      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "867317      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "867319      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "867327      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "867341      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "\n",
       "[28918 rows x 1002 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "converted[\"mydatCNV\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "2cec1737-ab61-4b30-904b-9de0c888a592",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MB-0002           LumA\n",
       "MB-0005           LumB\n",
       "MB-0006           LumB\n",
       "MB-0010           LumB\n",
       "MB-0014           LumB\n",
       "              ...     \n",
       "MB-0291           LumB\n",
       "MB-0444           LumB\n",
       "MB-0482         Normal\n",
       "MB-0519    claudin-low\n",
       "MB-0593           Her2\n",
       "Name: CLAUDIN_SUBTYPE, Length: 1980, dtype: string"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "converted[\"clinical\"][\"CLAUDIN_SUBTYPE\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "e02afad5-4ad9-437c-9ead-d04e05c04a20",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "pd.Series(converted[\"data\"][\"cluster\"]).to_csv(\"imvc/datasets/data/simulated_netMUG/simulated_netMUG_y.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "f9df0ed8-5289-46ff-b845-33605bc9d9be",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from sklearn.utils.validation import (check_array, check_symmetric,\n",
    "                                      check_consistent_length)\n",
    "from snf.compute import _find_dominate_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3e683a00-0184-497c-9644-060011be3345",
   "metadata": {},
   "outputs": [],
   "source": [
    "import rpy2.robjects as robjects\n",
    "robjects.r['load'](\"imvc/datasets/data/simulated_netMUG/original/SimulationData.Rdata\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "f80c03dd-04aa-4034-ad22-08d3f7ff537c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "pd.read_csv(\"imvc/datasets/data/simulated_netMUG/simulated_netMUG_y.csv\").to_csv(\"imvc/datasets/data/simulated_netMUG/simulated_netMUG_y.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fc87ca60-bfc6-4c0d-9cf6-e1b42baf1111",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "model.initialize_wh()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "de54d908-f93f-479a-9379-83b8ce314ef8",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "model.update_weights()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ded7339d-456b-425b-a479-ff15c0ba0ab8",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "model.w"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "09e2baf9-1a51-4754-947e-151652f2ae87",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "dict(my_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f8ce9066-04a3-4918-9ada-114769c85f66",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "results.select_dtypes(object).drop(columns= [\"comments\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "324fda0f-66ad-4779-9f6f-c039aebdd1ae",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from datetime import datetime\n",
    "\n",
    "# datetime object containing current date and time\n",
    "now = datetime.now()\n",
    "print(now)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66d4916b-eafd-45ec-ad10-00c528d3dafa",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "def fun(x: int):\n",
    "    \"4\" / 3\n",
    "    return "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bdc1334c-de0a-42c6-a419-fb17337deccb",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "try:\n",
    "    fun()\n",
    "except Exception as exception:\n",
    "    print(type(exception).__name__, \":\", exception)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "546920d8-3b0c-4475-b03a-6da27f6b601a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "defaultdict(defaultdict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "315124ea-e781-4315-9e11-549bbcc56b2f",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "import os.path\n",
    "import time\n",
    "from collections import defaultdict\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from sklearn.impute import SimpleImputer\n",
    "from sklearn.pipeline import make_pipeline\n",
    "from sklearn.preprocessing import StandardScaler, MinMaxScaler, FunctionTransformer\n",
    "from sklearn.cluster import KMeans, spectral_clustering\n",
    "from mvlearn.decomposition import AJIVE, GroupPCA\n",
    "from mvlearn.cluster import MultiviewSpectralClustering, MultiviewCoRegSpectralClustering\n",
    "from snf import compute\n",
    "from bignmf.models.jnmf.integrative import IntegrativeJnmf\n",
    "from bignmf.models.jnmf.standard import StandardJnmf\n",
    "from imvc.datasets import LoadDataset\n",
    "from imvc.utils import DatasetUtils\n",
    "from imvc.transformers import MultiViewTransformer, ConcatenateViews\n",
    "from imvc.algorithms import NMFC\n",
    "\n",
    "from utils import save_record\n",
    "# from utils import GroupPCA\n",
    "\n",
    "folder_name = \"results\"\n",
    "filelame = \"complete_algorithms_evaluation.csv\"\n",
    "file_path = os.path.join(folder_name, filelame)\n",
    "\n",
    "random_state = 42\n",
    "START_BENCHMARKING = False\n",
    "\n",
    "datasets = [\"nutrimouse_genotype\", \"nutrimouse_diet\", \"bbcsport\", \"bdgp\", \"caltech101\", \"digits\", \"tcga_tissue\", \"tcga_survival\", \"nuswide\", \"metabric\"]\n",
    "probs = np.arange(100, step= 10)\n",
    "imputation = [True, False]\n",
    "runs_per_alg = np.arange(10)\n",
    "algorithms = {\n",
    "    \"Concat\": {\"alg\": make_pipeline(ConcatenateViews(),\n",
    "                                    StandardScaler().set_output(transform='pandas'),\n",
    "                                    KMeans()), \"params\": {}},\n",
    "    \"NMFC\": {\"alg\": make_pipeline(ConcatenateViews(),\n",
    "                                  MinMaxScaler().set_output(transform='pandas'),\n",
    "                                  NMFC().set_output(transform='pandas')), \"params\": {}},\n",
    "    \"MVSpectralClustering\": {\"alg\": make_pipeline(MultiViewTransformer(StandardScaler().set_output(transform= \"pandas\")),\n",
    "                                                  MultiviewSpectralClustering()),\n",
    "                             \"params\": {}},\n",
    "    \"MVCoRegSpectralClustering\": {\"alg\": make_pipeline(MultiViewTransformer(StandardScaler().set_output(transform= \"pandas\")),\n",
    "                                                       MultiviewCoRegSpectralClustering()),\n",
    "                                  \"params\": {}},\n",
    "    \"GroupPCA\": {\"alg\": make_pipeline(MultiViewTransformer(StandardScaler()), GroupPCA(), StandardScaler(), KMeans()),\n",
    "                 \"params\": {}},\n",
    "    \"AJIVE\": {\"alg\": make_pipeline(MultiViewTransformer(StandardScaler()), AJIVE(), MultiViewTransformer(FunctionTransformer(pd.DataFrame)),\n",
    "                                   ConcatenateViews(), StandardScaler(), KMeans()),\n",
    "              \"params\": {}},\n",
    "    \"SNF\": {},\n",
    "    \"intNMF\": {},\n",
    "    \"jNMF\": {},\n",
    "}\n",
    "indexes_results = {\"dataset\": datasets, \"algorithm\": list(algorithms.keys()),\n",
    "                   \"missing_percentage\": probs, \"imputation\": imputation, \"run_n\": runs_per_alg}\n",
    "\n",
    "\n",
    "if START_BENCHMARKING:\n",
    "    results = pd.DataFrame(datasets, columns= [\"dataset\"])\n",
    "    for k,v in {k:v for k,v in indexes_results.items() if k != \"dataset\"}.items():\n",
    "        results = results.merge(pd.Series(v, name= k), how= \"cross\")\n",
    "    results = results.set_index(list(indexes_results.keys()))\n",
    "    results[[\"finished\", \"completed\"]] = False\n",
    "else:\n",
    "    results = pd.read_csv(file_path, index_col= list(indexes_results.keys()))\n",
    "    results_ = results.select_dtypes(object).drop(columns= \"comments\").replace(np.nan, \"np.nan\")\n",
    "    for col in results_.columns:\n",
    "        results[col] = results_[col].apply(eval)\n",
    "\n",
    "unfinished_results = results.loc[~results[\"finished\"]]\n",
    "\n",
    "for dataset_name in unfinished_results.index.get_level_values(\"dataset\").unique():\n",
    "    Xs, y = LoadDataset.load_dataset(dataset_name=dataset_name.split(\"_\")[0], return_y=True, shuffle= False)\n",
    "    y = pd.DataFrame(y)\n",
    "    for target in y.columns:\n",
    "        y_series = y[target].squeeze()\n",
    "        n_clusters = y_series.nunique()\n",
    "\n",
    "        for idx_iterator in unfinished_results.loc[unfinished_results.index.get_level_values(\"dataset\") == dataset_name].itertuples():\n",
    "            idx = idx_iterator[0]\n",
    "            row = results.loc[[idx]]\n",
    "            row_index = row.index\n",
    "            print(row.drop(columns= row.columns).reset_index().to_dict(orient=\"records\")[0])\n",
    "            alg_name, impute, p, run_n = (\n",
    "                row_index.get_level_values(\"algorithm\")[0],\n",
    "                row_index.get_level_values(\"imputation\")[0],\n",
    "                row_index.get_level_values(\"missing_percentage\")[0]/100,\n",
    "                row_index.get_level_values(\"run_n\")[0])\n",
    "\n",
    "            alg = algorithms[alg_name]\n",
    "            train_Xs = DatasetUtils.shuffle_imvd(Xs=Xs, random_state= random_state + run_n)\n",
    "            y_train = y_series.loc[train_Xs[0].index]\n",
    "            errors_dict = defaultdict(int)\n",
    "            if p != 0:\n",
    "                # if n_clusters > len(train_Xs[0])*p:\n",
    "                #     continue\n",
    "                try:\n",
    "                    train_Xs = DatasetUtils.add_random_noise_to_views(Xs=train_Xs, p= round(p, 2),\n",
    "                                                                      random_state =random_state + run_n, \n",
    "                                                                      assess_percentage = True, stratify = y_train)\n",
    "                except Exception as exception:\n",
    "                    errors_dict[type(exception).__name__] += 1\n",
    "                    print(errors_dict)\n",
    "                    results.loc[idx, \"finished\"] = True\n",
    "                    results.to_csv(file_path)\n",
    "                    continue\n",
    "\n",
    "            if impute:\n",
    "                train_Xs = MultiViewTransformer(SimpleImputer(strategy=\"mean\").set_output(transform= \"pandas\")).fit_transform(train_Xs)\n",
    "            else:\n",
    "                train_Xs = DatasetUtils.select_complete_samples(Xs = train_Xs)\n",
    "                y_train = y_train.loc[train_Xs[0].index]\n",
    "\n",
    "            try:\n",
    "                start_time = time.perf_counter()\n",
    "                if alg_name == \"SNF\":\n",
    "                    preprocessing_step = MultiViewTransformer(StandardScaler().set_output(transform= \"pandas\"))\n",
    "                    train_Xs = preprocessing_step.fit_transform(train_Xs)\n",
    "                    affinities = compute.make_affinity(train_Xs, normalize= False)\n",
    "                    fused = compute.snf(affinities)\n",
    "                    clusters = spectral_clustering(fused, n_clusters=n_clusters, random_state=random_state + run_n)\n",
    "                elif alg_name == \"intNMF\":\n",
    "                    preprocessing_step = MultiViewTransformer(MinMaxScaler().set_output(transform= \"pandas\"))\n",
    "                    model = IntegrativeJnmf({k:v for k,v in enumerate(train_Xs)}, k= n_clusters, lamb = 0.1)\n",
    "                    raise\n",
    "                    model.run(trials = 50, iterations = 100, verbose=False)\n",
    "                    model.cluster_data()\n",
    "                    clusters = np.argmax(model.w_cluster, axis= 1)\n",
    "                elif alg_name == \"jNMF\":\n",
    "                    pipeline = make_pipeline(MultiViewTransformer(MinMaxScaler().set_output(transform= \"pandas\")))\n",
    "                    model = StandardJnmf({k:v for k,v in enumerate(train_Xs)}, k= n_clusters)\n",
    "                    model.run(trials = 50, iterations = 100, verbose=False)\n",
    "                    model.cluster_data()\n",
    "                    clusters = np.argmax(model.w_cluster, axis= 1)\n",
    "                else:\n",
    "                    model, params = alg[\"alg\"], alg[\"params\"]\n",
    "                    if alg_name == \"GroupPCA\":\n",
    "                        model[1].set_params(n_components=n_clusters, random_state=random_state + run_n, multiview_output=False)\n",
    "                    elif alg_name == \"AJIVE\":\n",
    "                        model[1].set_params(random_state=random_state + run_n)\n",
    "                    if alg_name == \"NMFC\":\n",
    "                        model[-1].set_params(n_components=n_clusters, random_state=random_state + run_n)\n",
    "                    else:\n",
    "                        model[-1].set_params(n_clusters=n_clusters, random_state=random_state + run_n)\n",
    "                    clusters = model.fit_predict(train_Xs)\n",
    "            except ValueError as exception:\n",
    "                if alg_name == \"AJIVE\" and len(y_train) < 5:\n",
    "                    errors_dict[type(exception).__name__] += 1\n",
    "                    print(errors_dict)\n",
    "                    results.loc[idx, \"finished\"] = True\n",
    "                    results.to_csv(file_path)\n",
    "                    continue\n",
    "                if alg_name == \"SNF\" and len(y_train) < 17:\n",
    "                    errors_dict[type(exception).__name__] += 1\n",
    "                    print(errors_dict)\n",
    "                    results.loc[idx, \"finished\"] = True\n",
    "                    results.to_csv(file_path)\n",
    "                    continue\n",
    "                if alg_name == \"intNMF\" and len(y_train) < 5:\n",
    "                    errors_dict[type(exception).__name__] += 1\n",
    "                    print(errors_dict)\n",
    "                    results.loc[idx, \"finished\"] = True\n",
    "                    results.to_csv(file_path)\n",
    "                    continue\n",
    "                else:\n",
    "                    raise\n",
    "\n",
    "            clusters = pd.Series(clusters, index= y_train.index)\n",
    "\n",
    "            elapsed_time = time.perf_counter() - start_time\n",
    "\n",
    "            if alg_name in [\"NMFC\"]:\n",
    "                train_X = model.transform(train_Xs)\n",
    "            elif alg_name in [\"SNF\"]:\n",
    "                train_X = preprocessing_step.transform(train_Xs)\n",
    "            elif alg_name in [\"intNMF\", \"jNMF\"]:\n",
    "                train_X = model.w\n",
    "            else:\n",
    "                train_X = model[:-1].transform(train_Xs)\n",
    "            if isinstance(train_X, list):\n",
    "                train_X = ConcatenateViews().fit_transform(train_X)\n",
    "            if not isinstance(train_X, pd.DataFrame):\n",
    "                train_X = pd.DataFrame(train_X, index= y_train.index)\n",
    "                \n",
    "            assert train_X.index.equals(y_train.index)\n",
    "            assert train_Xs[0].index.equals(y_train.index)\n",
    "\n",
    "            if p > 0:\n",
    "                best_solution = pd.MultiIndex.from_arrays([[row_index.get_level_values(level= level)[0]] if level != \"missing_percentage\" else [0]\n",
    "                                                           for level in row_index.names], names= row_index.names)\n",
    "                best_solution = results.loc[best_solution].iloc[0]\n",
    "                best_solution = pd.Series(best_solution[\"y_pred\"], index= best_solution[\"y_pred_idx\"])\n",
    "                best_solution = best_solution.loc[train_X.index]\n",
    "            else:\n",
    "                best_solution = None\n",
    "\n",
    "            dict_results = save_record(train_Xs=train_Xs, train_X=train_X, clusters=clusters, y=y_train, p= p,\n",
    "                              best_solution = best_solution, elapsed_time=elapsed_time,\n",
    "                              random_state=random_state, errors_dict=errors_dict)\n",
    "            dict_results = pd.DataFrame(pd.Series(dict_results), columns= row_index).T\n",
    "            results.loc[[idx], dict_results.columns] = dict_results\n",
    "            results.loc[idx, \"finished\"] = True\n",
    "            results.to_csv(file_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "id": "a5171fc9-68c3-4822-9cb8-19f2361ee471",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>finished</th>\n",
       "      <th>completed</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">simulated_InterSIM</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">Concat</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">True</th>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nuswide</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">jNMF</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">90</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21600 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                  finished   \n",
       "dataset            algorithm missing_percentage imputation run_n             \n",
       "simulated_InterSIM Concat    0                  True       0         False  \\\n",
       "                                                           1         False   \n",
       "                                                           2         False   \n",
       "                                                           3         False   \n",
       "                                                           4         False   \n",
       "...                                                                    ...   \n",
       "nuswide            jNMF      90                 False      5         False   \n",
       "                                                           6         False   \n",
       "                                                           7         False   \n",
       "                                                           8         False   \n",
       "                                                           9         False   \n",
       "\n",
       "                                                                  completed  \n",
       "dataset            algorithm missing_percentage imputation run_n             \n",
       "simulated_InterSIM Concat    0                  True       0          False  \n",
       "                                                           1          False  \n",
       "                                                           2          False  \n",
       "                                                           3          False  \n",
       "                                                           4          False  \n",
       "...                                                                     ...  \n",
       "nuswide            jNMF      90                 False      5          False  \n",
       "                                                           6          False  \n",
       "                                                           7          False  \n",
       "                                                           8          False  \n",
       "                                                           9          False  \n",
       "\n",
       "[21600 rows x 2 columns]"
      ]
     },
     "execution_count": 276,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "    results = pd.DataFrame(datasets, columns= [\"dataset\"])\n",
    "    for k,v in {k:v for k,v in indexes_results.items() if k != \"dataset\"}.items():\n",
    "        results = results.merge(pd.Series(v, name= k), how= \"cross\")\n",
    "    results = results.set_index(list(indexes_results.keys()))\n",
    "    results[[\"finished\", \"completed\"]] = False\n",
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc5aad2d-963c-42f8-87e5-32262db2cc91",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f06d8c3-bcdc-466e-b6b0-5f8994392c78",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "np.random.rand(list(model.x.values())[0].shape[0], model.k).shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f5dbed8-c780-49cb-9db9-1b0118b7203a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "self.v = {}\n",
    "self.h = {}\n",
    "for key in self.x:\n",
    "    self.h[key] = np.random.rand(self.k, self.x[key].shape[1])\n",
    "    self.v[key] = np.random.rand(number_of_samples, self.k)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90f87c27-4496-47b0-9c23-c8787ff7bff8",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from sklearn import metrics\n",
    "metrics.silhouette_score(train_X, clusters, random_state=random_state)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "281b0bf1-9e90-4382-b083-b97727a04aa7",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for i in range(1000):\n",
    "    try:\n",
    "        model.run(trials = 50, iterations = 100, verbose=0)\n",
    "        # clusters = model.fit_predict(train_Xs)\n",
    "        print(i)\n",
    "    except Exception as ex:\n",
    "        print(i, ex)\n",
    "        pass\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "99616f4f-b844-4f11-80be-7848cf692894",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "results[results[\"finished\"] == True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f07b0ac-5137-4100-ba85-59ab9c7d1b09",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "best_solution.loc[train_X.index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e11fce7a-d10d-4a29-aa9a-ccb1a906d7ab",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "a = pd.read_csv(file_path, index_col= list(indexes_results.keys()))\n",
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5ba40a8d-6eb3-4b61-87bc-c42399cf6ea4",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "a[\"label_sizes\"].apply(str).str.replace(\"nan\", \"np.nan\").apply(eval)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ff9065d1-e624-413e-a468-ae30311d9d56",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "b = a.select_dtypes(object).replace(np.nan, \"np.nan\").drop(columns= \"comments\")\n",
    "b"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f329524-f844-4980-b49a-c8c0916627b7",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for i in b.columns:\n",
    "    print(i)\n",
    "    b[i].apply(eval)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5a007f6c-58dd-4eec-804e-8c43fa0dbb65",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "b.apply(eval, axis= 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c02da898-ce78-45cc-b096-945a0c7f270e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "a = pd.DataFrame(datasets, columns= [\"dataset\"])\n",
    "for k,v in {k:v for k,v in indexes_results.items() if k != \"dataset\"}.items():\n",
    "    a = a.merge(pd.Series(v, name= k), how= \"cross\")\n",
    "a = a.set_index(list(indexes_results.keys()))\n",
    "a[\"finished\"] = False\n",
    "for i in a.itertuples():\n",
    "    print(len(i))\n",
    "    print(i)\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ae3f7de4-c34c-4dda-8965-d6d2b025782c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for i in unfinished_results.iterrows():\n",
    "    print(len(i))\n",
    "    print(i)\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bcaf794a-e432-4563-a6a7-516983310f7e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "len(best_solution.loc[train_X.index])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9a282ec7-3caf-458e-9611-ef70839e349c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "i[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f6bb522-ad3a-4225-8130-61915f8a81c2",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import scipy.io\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "from imvc.datasets import LoadDataset\n",
    "from imvc.utils import DatasetUtils\n",
    "import copy\n",
    "from sklearn.model_selection import train_test_split\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "27e4a119-be24-4784-af50-3ad95e265dfa",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "Xs = DatasetUtils.add_random_noise_to_views(Xs=Xs, p= 0.95, assess_percentage= True, random_state= 42, stratify=y.iloc[:, 1])\n",
    "Xs[0].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6a341a3b-04a7-432e-a26c-eb9b4b1750fc",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "datasets = [\"nutrimouse\", \"bbcsport\", \"bdgp\", \"caltech101\", \"digits\", \"tcga\", \"nuswide\", \"metabric\"]\n",
    "probs = np.arange(0., 1., step= 0.1).round(1) * 100\n",
    "imputation = [True, False]\n",
    "runs_per_alg = np.arange(10)\n",
    "\n",
    "results = pd.DataFrame(datasets, columns= [\"dataset\"]).merge(\n",
    "    pd.Series(probs, name= \"missing_percentage\"), how= \"cross\").merge(\n",
    "    pd.Series(imputation, name= \"imputation\"), how= \"cross\").merge(\n",
    "    pd.Series(runs_per_alg, name= \"run_n\"), how= \"cross\")\n",
    "results = results.set_index(['dataset', 'missing_percentage',\"imputation\", 'run_n'])\n",
    "results[\"finished\"] = False\n",
    "results.to_csv(\"pr.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a77b3af9-d507-4a19-a7f8-ffe5de6062e6",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "pd.read_csv(\"pr.csv\", index_col= ['dataset', 'missing_percentage',\"imputation\", 'run_n'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7aba373c-9a81-4c16-89ac-9a668865f85c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "Xs[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb8c644b-4520-441c-86e8-d8753e22cb0f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "Xs[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8c2dcf3-8b7c-41ea-b85d-55270b34c233",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "DatasetUtils.convert_mvd_into_imvd(Xs, p= 0.1, assess_percentage= True, random_state= random_state)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "72887a96-8cfa-4d8d-a03a-87fa747cf713",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "def split_into_groups(num_elements, num_groups):\n",
    "    # Calculate the base number of elements per group\n",
    "    base_count = num_elements // num_groups\n",
    "    remaining = num_elements % num_groups\n",
    "    \n",
    "    # Initialize the list to store the number of elements in each group\n",
    "    groups_count = [base_count] * num_groups\n",
    "    \n",
    "    # Distribute the remaining elements equally among the groups\n",
    "    for i in range(remaining):\n",
    "        groups_count[i] += 1\n",
    "    \n",
    "    return groups_count\n",
    "\n",
    "# Total number of elements\n",
    "total_elements = 103\n",
    "# Number of groups to split into\n",
    "num_of_groups = 4\n",
    "\n",
    "# Splitting elements into groups\n",
    "result = split_into_groups(total_elements, num_of_groups)\n",
    "\n",
    "# Displaying the number of elements in each group\n",
    "for i, count in enumerate(result):\n",
    "    print(f\"Group {i + 1}: {count} elements\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2edb9198-3e8c-487c-808d-018b84f56754",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "103 % 4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b0357a0b-14ea-4794-b9a0-1eae31b046ad",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "base_count = 103 // 4\n",
    "remaining = 103 % 4\n",
    "\n",
    "# Initialize the list to store the number of elements in each group\n",
    "groups_count = [base_count] * num_groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61df5032-fcc0-4db9-beb9-7735f147f5ef",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for dataset_name in [\n",
    "    # \"nutrimouse\",\n",
    "    \"tcga\"\n",
    "]:\n",
    "    Xs, y = LoadDataset.load_dataset(dataset_name = dataset_name, return_y = True, p= 0.8, assess_percentage=True)\n",
    "    # Xs = GetCompleteSamples().fit_transform(Xs)\n",
    "    # aaaaaaaaaaaaaa\n",
    "    # y.to_csv(os.path.join(\"imvc/datasets/data/\", dataset_name, f\"{dataset_name}_y.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "696c59f3-100e-4c50-a06b-654377a74eab",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "survs = []\n",
    "for dataset_name in os.listdir(\"imvc/datasets/data/tcga/original/\"):\n",
    "    if \"survival\" in dataset_name:\n",
    "        print(dataset_name)\n",
    "        surv = pd.read_csv(os.path.join(\"imvc/datasets/data/tcga/original/\", dataset_name), sep= '\"\"', index_col= 0, engine='python')\n",
    "        if surv.shape[1] <2:\n",
    "            surv = pd.read_csv(os.path.join(\"imvc/datasets/data/tcga/original/\", \"survival_lung.csv\"), sep= \"\\t\", index_col= 0)\n",
    "        surv.index = surv.index.str.replace(\".\", \"-\").str.upper()\n",
    "        surv.index = surv.index.str.extract(pat=r\"(TCGA.{8})\", expand=False)\n",
    "        surv.index.name = None\n",
    "        survs.append(surv[[\"Survival\", \"Death\"]])\n",
    "survs = pd.concat(survs).drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5fc0ece1-0d40-4116-8e7a-8ddbb44e21bc",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "y.index.intersection(survs.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66cce785-b1a5-4444-8d83-83facc6084c3",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "_,met = LoadDataset.load_dataset(dataset_name = \"tcga\", return_metadata= True, p= 0.8, assess_percentage=True)\n",
    "met"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "538ac9f3-9e7e-4511-bc12-66f62561e5d9",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "y[y == 0].dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "743cde78-20a9-435e-959d-738ea8160013",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "pattern = r\"(TCGA.{8})\"\n",
    "matches = surv.index.str.extract(pat=pattern, expand=False)\n",
    "matches"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "87dad0f6-6944-44c7-b6f1-49f8e1d76586",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for dataset_name in [i for i in sorted(os.listdir(\"imvc/datasets/data/\")[1:]) if i != \"metabric\"]:\n",
    "    Xs = LoadDataset.load_dataset(dataset_name = dataset_name, return_y = False)\n",
    "    for p in np.arange(0., 1., step= 0.1).round(1):\n",
    "        imvd = DatasetUtils.convert_mvd_into_imvd(Xs, p= p, random_state = 42, assess_percentage = True)\n",
    "        complete_Xs = GetCompleteSamples().fit_transform(imvd)\n",
    "        print(dataset_name, \"\\t\", p, \"\\t\", len(Xs), \"\\t\", len(DatasetUtils.get_sample_names(imvd)), \"\\t\", len(DatasetUtils.get_sample_names(complete_Xs)))\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ac67995-e7d0-46e6-984a-e685e57dbbf9",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "probs = np.arange(0., 1., step= 0.1).round(1)\n",
    "algorithms = [\"Concat\", \"MOFA\", \"NMFC\", \"MONET\", \"MSNE\", \"SUMO\", \"NEMO\"]\n",
    "runs_per_alg = np.arange(10).tolist()\n",
    "results = pd.DataFrame(os.listdir(\"imvc/datasets/data/\"), columns= [\"dataset\"]).merge(\n",
    "    pd.Series(algorithms, name= \"algorithm\"), how= \"cross\").merge(\n",
    "    pd.Series(probs, name= \"missing_percentage\"), how= \"cross\").merge(\n",
    "    pd.Series(runs_per_alg, name= \"run_n\"), how= \"cross\")\n",
    "results = results.set_index(['dataset', 'algorithm', 'missing_percentage', 'run_n'])\n",
    "results[\"finished\"] = False\n",
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 281,
   "id": "4e940555-6c0d-4be0-8ecb-22ae43daabbb",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "path = \"results/subresults\"\n",
    "files = pd.Series(os.listdir(path)).apply(lambda x: os.path.join(path, x))\n",
    "files = files[files.apply(os.path.isfile)]\n",
    "files = pd.concat(files.apply(pd.read_csv).to_list())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "id": "4169f107-665c-42a4-92b5-416ff8578398",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "      <th>finished</th>\n",
       "      <th>completed</th>\n",
       "      <th>comments</th>\n",
       "      <th>n_samples</th>\n",
       "      <th>n_incomplete_samples</th>\n",
       "      <th>...</th>\n",
       "      <th>bal_acc</th>\n",
       "      <th>ami</th>\n",
       "      <th>ari</th>\n",
       "      <th>completeness</th>\n",
       "      <th>random_acc</th>\n",
       "      <th>random_f1</th>\n",
       "      <th>silhouette</th>\n",
       "      <th>vrc</th>\n",
       "      <th>db</th>\n",
       "      <th>dunn</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>simulated_InterSIM</td>\n",
       "      <td>GroupPCA</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>5</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>simulated_InterSIM</td>\n",
       "      <td>Concat</td>\n",
       "      <td>90</td>\n",
       "      <td>True</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>simulated_InterSIM</td>\n",
       "      <td>Concat</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>8</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>simulated_InterSIM</td>\n",
       "      <td>Concat</td>\n",
       "      <td>60</td>\n",
       "      <td>True</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>simulated_InterSIM</td>\n",
       "      <td>Concat</td>\n",
       "      <td>80</td>\n",
       "      <td>False</td>\n",
       "      <td>6</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>simulated_InterSIM</td>\n",
       "      <td>Concat</td>\n",
       "      <td>50</td>\n",
       "      <td>False</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>simulated_InterSIM</td>\n",
       "      <td>jNMF</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>7</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>{}</td>\n",
       "      <td>500.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.197778</td>\n",
       "      <td>-0.001763</td>\n",
       "      <td>-0.002331</td>\n",
       "      <td>0.001938</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.190476</td>\n",
       "      <td>-0.02149</td>\n",
       "      <td>0.561153</td>\n",
       "      <td>39.720961</td>\n",
       "      <td>0.000909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>simulated_InterSIM</td>\n",
       "      <td>AJIVE</td>\n",
       "      <td>40</td>\n",
       "      <td>True</td>\n",
       "      <td>8</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>simulated_InterSIM</td>\n",
       "      <td>AJIVE</td>\n",
       "      <td>10</td>\n",
       "      <td>True</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>simulated_InterSIM</td>\n",
       "      <td>AJIVE</td>\n",
       "      <td>30</td>\n",
       "      <td>True</td>\n",
       "      <td>8</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>662 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               dataset algorithm  missing_percentage  imputation  run_n   \n",
       "0   simulated_InterSIM  GroupPCA                  60       False      5  \\\n",
       "0   simulated_InterSIM    Concat                  90        True      2   \n",
       "0   simulated_InterSIM    Concat                  60       False      8   \n",
       "0   simulated_InterSIM    Concat                  60        True      2   \n",
       "0   simulated_InterSIM    Concat                  80       False      6   \n",
       "..                 ...       ...                 ...         ...    ...   \n",
       "0   simulated_InterSIM    Concat                  50       False      4   \n",
       "0   simulated_InterSIM      jNMF                   0       False      7   \n",
       "0   simulated_InterSIM     AJIVE                  40        True      8   \n",
       "0   simulated_InterSIM     AJIVE                  10        True      4   \n",
       "0   simulated_InterSIM     AJIVE                  30        True      8   \n",
       "\n",
       "    finished  completed comments  n_samples  n_incomplete_samples  ...   \n",
       "0       True      False      NaN        NaN                   NaN  ...  \\\n",
       "0       True      False      NaN        NaN                   NaN  ...   \n",
       "0       True      False      NaN        NaN                   NaN  ...   \n",
       "0       True      False      NaN        NaN                   NaN  ...   \n",
       "0       True      False      NaN        NaN                   NaN  ...   \n",
       "..       ...        ...      ...        ...                   ...  ...   \n",
       "0       True      False      NaN        NaN                   NaN  ...   \n",
       "0       True       True       {}      500.0                   0.0  ...   \n",
       "0       True      False      NaN        NaN                   NaN  ...   \n",
       "0       True      False      NaN        NaN                   NaN  ...   \n",
       "0       True      False      NaN        NaN                   NaN  ...   \n",
       "\n",
       "     bal_acc       ami       ari  completeness random_acc random_f1   \n",
       "0        NaN       NaN       NaN           NaN        NaN       NaN  \\\n",
       "0        NaN       NaN       NaN           NaN        NaN       NaN   \n",
       "0        NaN       NaN       NaN           NaN        NaN       NaN   \n",
       "0        NaN       NaN       NaN           NaN        NaN       NaN   \n",
       "0        NaN       NaN       NaN           NaN        NaN       NaN   \n",
       "..       ...       ...       ...           ...        ...       ...   \n",
       "0        NaN       NaN       NaN           NaN        NaN       NaN   \n",
       "0   0.197778 -0.001763 -0.002331      0.001938        0.4  0.190476   \n",
       "0        NaN       NaN       NaN           NaN        NaN       NaN   \n",
       "0        NaN       NaN       NaN           NaN        NaN       NaN   \n",
       "0        NaN       NaN       NaN           NaN        NaN       NaN   \n",
       "\n",
       "   silhouette       vrc         db      dunn  \n",
       "0         NaN       NaN        NaN       NaN  \n",
       "0         NaN       NaN        NaN       NaN  \n",
       "0         NaN       NaN        NaN       NaN  \n",
       "0         NaN       NaN        NaN       NaN  \n",
       "0         NaN       NaN        NaN       NaN  \n",
       "..        ...       ...        ...       ...  \n",
       "0         NaN       NaN        NaN       NaN  \n",
       "0    -0.02149  0.561153  39.720961  0.000909  \n",
       "0         NaN       NaN        NaN       NaN  \n",
       "0         NaN       NaN        NaN       NaN  \n",
       "0         NaN       NaN        NaN       NaN  \n",
       "\n",
       "[662 rows x 38 columns]"
      ]
     },
     "execution_count": 282,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "id": "9a3e3715-7aef-43a3-bf61-54cc5c73d61a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>finished</th>\n",
       "      <th>completed</th>\n",
       "      <th>comments</th>\n",
       "      <th>n_samples</th>\n",
       "      <th>n_incomplete_samples</th>\n",
       "      <th>n_complete_samples</th>\n",
       "      <th>time</th>\n",
       "      <th>n_clustered_samples</th>\n",
       "      <th>percentage_clustered_samples</th>\n",
       "      <th>stratified</th>\n",
       "      <th>...</th>\n",
       "      <th>bal_acc</th>\n",
       "      <th>ami</th>\n",
       "      <th>ari</th>\n",
       "      <th>completeness</th>\n",
       "      <th>random_acc</th>\n",
       "      <th>random_f1</th>\n",
       "      <th>silhouette</th>\n",
       "      <th>vrc</th>\n",
       "      <th>db</th>\n",
       "      <th>dunn</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">simulated_InterSIM</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">Concat</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">True</th>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nuswide</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">jNMF</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">90</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21600 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                  finished   \n",
       "dataset            algorithm missing_percentage imputation run_n             \n",
       "simulated_InterSIM Concat    0                  True       0         False  \\\n",
       "                                                           1         False   \n",
       "                                                           2         False   \n",
       "                                                           3         False   \n",
       "                                                           4         False   \n",
       "...                                                                    ...   \n",
       "nuswide            jNMF      90                 False      5         False   \n",
       "                                                           6         False   \n",
       "                                                           7         False   \n",
       "                                                           8         False   \n",
       "                                                           9         False   \n",
       "\n",
       "                                                                  completed   \n",
       "dataset            algorithm missing_percentage imputation run_n              \n",
       "simulated_InterSIM Concat    0                  True       0          False  \\\n",
       "                                                           1          False   \n",
       "                                                           2          False   \n",
       "                                                           3          False   \n",
       "                                                           4          False   \n",
       "...                                                                     ...   \n",
       "nuswide            jNMF      90                 False      5          False   \n",
       "                                                           6          False   \n",
       "                                                           7          False   \n",
       "                                                           8          False   \n",
       "                                                           9          False   \n",
       "\n",
       "                                                                 comments   \n",
       "dataset            algorithm missing_percentage imputation run_n            \n",
       "simulated_InterSIM Concat    0                  True       0          NaN  \\\n",
       "                                                           1          NaN   \n",
       "                                                           2          NaN   \n",
       "                                                           3          NaN   \n",
       "                                                           4          NaN   \n",
       "...                                                                   ...   \n",
       "nuswide            jNMF      90                 False      5          NaN   \n",
       "                                                           6          NaN   \n",
       "                                                           7          NaN   \n",
       "                                                           8          NaN   \n",
       "                                                           9          NaN   \n",
       "\n",
       "                                                                  n_samples   \n",
       "dataset            algorithm missing_percentage imputation run_n              \n",
       "simulated_InterSIM Concat    0                  True       0            NaN  \\\n",
       "                                                           1            NaN   \n",
       "                                                           2            NaN   \n",
       "                                                           3            NaN   \n",
       "                                                           4            NaN   \n",
       "...                                                                     ...   \n",
       "nuswide            jNMF      90                 False      5            NaN   \n",
       "                                                           6            NaN   \n",
       "                                                           7            NaN   \n",
       "                                                           8            NaN   \n",
       "                                                           9            NaN   \n",
       "\n",
       "                                                                  n_incomplete_samples   \n",
       "dataset            algorithm missing_percentage imputation run_n                         \n",
       "simulated_InterSIM Concat    0                  True       0                       NaN  \\\n",
       "                                                           1                       NaN   \n",
       "                                                           2                       NaN   \n",
       "                                                           3                       NaN   \n",
       "                                                           4                       NaN   \n",
       "...                                                                                ...   \n",
       "nuswide            jNMF      90                 False      5                       NaN   \n",
       "                                                           6                       NaN   \n",
       "                                                           7                       NaN   \n",
       "                                                           8                       NaN   \n",
       "                                                           9                       NaN   \n",
       "\n",
       "                                                                  n_complete_samples   \n",
       "dataset            algorithm missing_percentage imputation run_n                       \n",
       "simulated_InterSIM Concat    0                  True       0                     NaN  \\\n",
       "                                                           1                     NaN   \n",
       "                                                           2                     NaN   \n",
       "                                                           3                     NaN   \n",
       "                                                           4                     NaN   \n",
       "...                                                                              ...   \n",
       "nuswide            jNMF      90                 False      5                     NaN   \n",
       "                                                           6                     NaN   \n",
       "                                                           7                     NaN   \n",
       "                                                           8                     NaN   \n",
       "                                                           9                     NaN   \n",
       "\n",
       "                                                                  time   \n",
       "dataset            algorithm missing_percentage imputation run_n         \n",
       "simulated_InterSIM Concat    0                  True       0       NaN  \\\n",
       "                                                           1       NaN   \n",
       "                                                           2       NaN   \n",
       "                                                           3       NaN   \n",
       "                                                           4       NaN   \n",
       "...                                                                ...   \n",
       "nuswide            jNMF      90                 False      5       NaN   \n",
       "                                                           6       NaN   \n",
       "                                                           7       NaN   \n",
       "                                                           8       NaN   \n",
       "                                                           9       NaN   \n",
       "\n",
       "                                                                  n_clustered_samples   \n",
       "dataset            algorithm missing_percentage imputation run_n                        \n",
       "simulated_InterSIM Concat    0                  True       0                      NaN  \\\n",
       "                                                           1                      NaN   \n",
       "                                                           2                      NaN   \n",
       "                                                           3                      NaN   \n",
       "                                                           4                      NaN   \n",
       "...                                                                               ...   \n",
       "nuswide            jNMF      90                 False      5                      NaN   \n",
       "                                                           6                      NaN   \n",
       "                                                           7                      NaN   \n",
       "                                                           8                      NaN   \n",
       "                                                           9                      NaN   \n",
       "\n",
       "                                                                  percentage_clustered_samples   \n",
       "dataset            algorithm missing_percentage imputation run_n                                 \n",
       "simulated_InterSIM Concat    0                  True       0                               NaN  \\\n",
       "                                                           1                               NaN   \n",
       "                                                           2                               NaN   \n",
       "                                                           3                               NaN   \n",
       "                                                           4                               NaN   \n",
       "...                                                                                        ...   \n",
       "nuswide            jNMF      90                 False      5                               NaN   \n",
       "                                                           6                               NaN   \n",
       "                                                           7                               NaN   \n",
       "                                                           8                               NaN   \n",
       "                                                           9                               NaN   \n",
       "\n",
       "                                                                 stratified   \n",
       "dataset            algorithm missing_percentage imputation run_n              \n",
       "simulated_InterSIM Concat    0                  True       0            NaN  \\\n",
       "                                                           1            NaN   \n",
       "                                                           2            NaN   \n",
       "                                                           3            NaN   \n",
       "                                                           4            NaN   \n",
       "...                                                                     ...   \n",
       "nuswide            jNMF      90                 False      5            NaN   \n",
       "                                                           6            NaN   \n",
       "                                                           7            NaN   \n",
       "                                                           8            NaN   \n",
       "                                                           9            NaN   \n",
       "\n",
       "                                                                  ... bal_acc   \n",
       "dataset            algorithm missing_percentage imputation run_n  ...           \n",
       "simulated_InterSIM Concat    0                  True       0      ...     NaN  \\\n",
       "                                                           1      ...     NaN   \n",
       "                                                           2      ...     NaN   \n",
       "                                                           3      ...     NaN   \n",
       "                                                           4      ...     NaN   \n",
       "...                                                               ...     ...   \n",
       "nuswide            jNMF      90                 False      5      ...     NaN   \n",
       "                                                           6      ...     NaN   \n",
       "                                                           7      ...     NaN   \n",
       "                                                           8      ...     NaN   \n",
       "                                                           9      ...     NaN   \n",
       "\n",
       "                                                                 ami ari   \n",
       "dataset            algorithm missing_percentage imputation run_n           \n",
       "simulated_InterSIM Concat    0                  True       0     NaN NaN  \\\n",
       "                                                           1     NaN NaN   \n",
       "                                                           2     NaN NaN   \n",
       "                                                           3     NaN NaN   \n",
       "                                                           4     NaN NaN   \n",
       "...                                                               ..  ..   \n",
       "nuswide            jNMF      90                 False      5     NaN NaN   \n",
       "                                                           6     NaN NaN   \n",
       "                                                           7     NaN NaN   \n",
       "                                                           8     NaN NaN   \n",
       "                                                           9     NaN NaN   \n",
       "\n",
       "                                                                 completeness   \n",
       "dataset            algorithm missing_percentage imputation run_n                \n",
       "simulated_InterSIM Concat    0                  True       0              NaN  \\\n",
       "                                                           1              NaN   \n",
       "                                                           2              NaN   \n",
       "                                                           3              NaN   \n",
       "                                                           4              NaN   \n",
       "...                                                                       ...   \n",
       "nuswide            jNMF      90                 False      5              NaN   \n",
       "                                                           6              NaN   \n",
       "                                                           7              NaN   \n",
       "                                                           8              NaN   \n",
       "                                                           9              NaN   \n",
       "\n",
       "                                                                 random_acc   \n",
       "dataset            algorithm missing_percentage imputation run_n              \n",
       "simulated_InterSIM Concat    0                  True       0            NaN  \\\n",
       "                                                           1            NaN   \n",
       "                                                           2            NaN   \n",
       "                                                           3            NaN   \n",
       "                                                           4            NaN   \n",
       "...                                                                     ...   \n",
       "nuswide            jNMF      90                 False      5            NaN   \n",
       "                                                           6            NaN   \n",
       "                                                           7            NaN   \n",
       "                                                           8            NaN   \n",
       "                                                           9            NaN   \n",
       "\n",
       "                                                                 random_f1   \n",
       "dataset            algorithm missing_percentage imputation run_n             \n",
       "simulated_InterSIM Concat    0                  True       0           NaN  \\\n",
       "                                                           1           NaN   \n",
       "                                                           2           NaN   \n",
       "                                                           3           NaN   \n",
       "                                                           4           NaN   \n",
       "...                                                                    ...   \n",
       "nuswide            jNMF      90                 False      5           NaN   \n",
       "                                                           6           NaN   \n",
       "                                                           7           NaN   \n",
       "                                                           8           NaN   \n",
       "                                                           9           NaN   \n",
       "\n",
       "                                                                 silhouette   \n",
       "dataset            algorithm missing_percentage imputation run_n              \n",
       "simulated_InterSIM Concat    0                  True       0            NaN  \\\n",
       "                                                           1            NaN   \n",
       "                                                           2            NaN   \n",
       "                                                           3            NaN   \n",
       "                                                           4            NaN   \n",
       "...                                                                     ...   \n",
       "nuswide            jNMF      90                 False      5            NaN   \n",
       "                                                           6            NaN   \n",
       "                                                           7            NaN   \n",
       "                                                           8            NaN   \n",
       "                                                           9            NaN   \n",
       "\n",
       "                                                                 vrc  db  dunn  \n",
       "dataset            algorithm missing_percentage imputation run_n                \n",
       "simulated_InterSIM Concat    0                  True       0     NaN NaN   NaN  \n",
       "                                                           1     NaN NaN   NaN  \n",
       "                                                           2     NaN NaN   NaN  \n",
       "                                                           3     NaN NaN   NaN  \n",
       "                                                           4     NaN NaN   NaN  \n",
       "...                                                               ..  ..   ...  \n",
       "nuswide            jNMF      90                 False      5     NaN NaN   NaN  \n",
       "                                                           6     NaN NaN   NaN  \n",
       "                                                           7     NaN NaN   NaN  \n",
       "                                                           8     NaN NaN   NaN  \n",
       "                                                           9     NaN NaN   NaN  \n",
       "\n",
       "[21600 rows x 33 columns]"
      ]
     },
     "execution_count": 283,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files = files.set_index(list(indexes_results.keys()))\n",
    "results.loc[files.index, files.columns] = files\n",
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d8fc5692-a912-4e91-85d6-f07c2537f780",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>#sk-container-id-2 {color: black;background-color: white;}#sk-container-id-2 pre{padding: 0;}#sk-container-id-2 div.sk-toggleable {background-color: white;}#sk-container-id-2 label.sk-toggleable__label {cursor: pointer;display: block;width: 100%;margin-bottom: 0;padding: 0.3em;box-sizing: border-box;text-align: center;}#sk-container-id-2 label.sk-toggleable__label-arrow:before {content: \"▸\";float: left;margin-right: 0.25em;color: #696969;}#sk-container-id-2 label.sk-toggleable__label-arrow:hover:before {color: black;}#sk-container-id-2 div.sk-estimator:hover label.sk-toggleable__label-arrow:before {color: black;}#sk-container-id-2 div.sk-toggleable__content {max-height: 0;max-width: 0;overflow: hidden;text-align: left;background-color: #f0f8ff;}#sk-container-id-2 div.sk-toggleable__content pre {margin: 0.2em;color: black;border-radius: 0.25em;background-color: #f0f8ff;}#sk-container-id-2 input.sk-toggleable__control:checked~div.sk-toggleable__content {max-height: 200px;max-width: 100%;overflow: auto;}#sk-container-id-2 input.sk-toggleable__control:checked~label.sk-toggleable__label-arrow:before {content: \"▾\";}#sk-container-id-2 div.sk-estimator input.sk-toggleable__control:checked~label.sk-toggleable__label {background-color: #d4ebff;}#sk-container-id-2 div.sk-label input.sk-toggleable__control:checked~label.sk-toggleable__label {background-color: #d4ebff;}#sk-container-id-2 input.sk-hidden--visually {border: 0;clip: rect(1px 1px 1px 1px);clip: rect(1px, 1px, 1px, 1px);height: 1px;margin: -1px;overflow: hidden;padding: 0;position: absolute;width: 1px;}#sk-container-id-2 div.sk-estimator {font-family: monospace;background-color: #f0f8ff;border: 1px dotted black;border-radius: 0.25em;box-sizing: border-box;margin-bottom: 0.5em;}#sk-container-id-2 div.sk-estimator:hover {background-color: #d4ebff;}#sk-container-id-2 div.sk-parallel-item::after {content: \"\";width: 100%;border-bottom: 1px solid gray;flex-grow: 1;}#sk-container-id-2 div.sk-label:hover label.sk-toggleable__label {background-color: #d4ebff;}#sk-container-id-2 div.sk-serial::before {content: \"\";position: absolute;border-left: 1px solid gray;box-sizing: border-box;top: 0;bottom: 0;left: 50%;z-index: 0;}#sk-container-id-2 div.sk-serial {display: flex;flex-direction: column;align-items: center;background-color: white;padding-right: 0.2em;padding-left: 0.2em;position: relative;}#sk-container-id-2 div.sk-item {position: relative;z-index: 1;}#sk-container-id-2 div.sk-parallel {display: flex;align-items: stretch;justify-content: center;background-color: white;position: relative;}#sk-container-id-2 div.sk-item::before, #sk-container-id-2 div.sk-parallel-item::before {content: \"\";position: absolute;border-left: 1px solid gray;box-sizing: border-box;top: 0;bottom: 0;left: 50%;z-index: -1;}#sk-container-id-2 div.sk-parallel-item {display: flex;flex-direction: column;z-index: 1;position: relative;background-color: white;}#sk-container-id-2 div.sk-parallel-item:first-child::after {align-self: flex-end;width: 50%;}#sk-container-id-2 div.sk-parallel-item:last-child::after {align-self: flex-start;width: 50%;}#sk-container-id-2 div.sk-parallel-item:only-child::after {width: 0;}#sk-container-id-2 div.sk-dashed-wrapped {border: 1px dashed gray;margin: 0 0.4em 0.5em 0.4em;box-sizing: border-box;padding-bottom: 0.4em;background-color: white;}#sk-container-id-2 div.sk-label label {font-family: monospace;font-weight: bold;display: inline-block;line-height: 1.2em;}#sk-container-id-2 div.sk-label-container {text-align: center;}#sk-container-id-2 div.sk-container {/* jupyter's `normalize.less` sets `[hidden] { display: none; }` but bootstrap.min.css set `[hidden] { display: none !important; }` so we also need the `!important` here to be able to override the default hidden behavior on the sphinx rendered scikit-learn.org. See: https://github.com/scikit-learn/scikit-learn/issues/21755 */display: inline-block !important;position: relative;}#sk-container-id-2 div.sk-text-repr-fallback {display: none;}</style><div id=\"sk-container-id-2\" class=\"sk-top-container\"><div class=\"sk-text-repr-fallback\"><pre>MultivariateAmputation(patterns=[{&#x27;incomplete_vars&#x27;: range(1, 4),\n",
       "                                  &#x27;mechanism&#x27;: &#x27;MAR&#x27;}])</pre><b>In a Jupyter environment, please rerun this cell to show the HTML representation or trust the notebook. <br />On GitHub, the HTML representation is unable to render, please try loading this page with nbviewer.org.</b></div><div class=\"sk-container\" hidden><div class=\"sk-item\"><div class=\"sk-estimator sk-toggleable\"><input class=\"sk-toggleable__control sk-hidden--visually\" id=\"sk-estimator-id-2\" type=\"checkbox\" checked><label for=\"sk-estimator-id-2\" class=\"sk-toggleable__label sk-toggleable__label-arrow\">MultivariateAmputation</label><div class=\"sk-toggleable__content\"><pre>MultivariateAmputation(patterns=[{&#x27;incomplete_vars&#x27;: range(1, 4),\n",
       "                                  &#x27;mechanism&#x27;: &#x27;MAR&#x27;}])</pre></div></div></div></div></div>"
      ],
      "text/plain": [
       "MultivariateAmputation(patterns=[{'incomplete_vars': range(1, 4),\n",
       "                                  'mechanism': 'MAR'}])"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from imvc.datasets import LoadDataset\n",
    "from imvc.transformers import Ampute\n",
    "from imvc.utils import DatasetUtils\n",
    "from pyampute.ampute import MultivariateAmputation\n",
    "import warnings\n",
    "\n",
    "Xs = LoadDataset.load_dataset(dataset_name = \"bbcsport\")\n",
    "\n",
    "missing_view_panel = DatasetUtils.get_missing_view_panel(Xs)\n",
    "Xs = Xs.float()\n",
    "# amp = Ampute(p = 0.5, mechanism= \"MAR\")\n",
    "amp = MultivariateAmputation(patterns= [{\"incomplete_vars\": range(1, missing_view_panel.shape[1]), \"mechanism\": \"MAR\"}])\n",
    "amp.fit(missing_view_panel)\n",
    "X = amp.fit_transform(missing_view_panel)\n",
    "# DatasetUtils.get_n_complete_samples(iXs), DatasetUtils.get_n_incomplete_samples(iXs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "f084e153-0498-4e57-a250-d2edc1272bcb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>116 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     0  1  2  3\n",
       "0    1  1  1  1\n",
       "1    1  1  1  1\n",
       "2    1  1  1  1\n",
       "3    1  0  0  0\n",
       "4    1  0  0  0\n",
       "..  .. .. .. ..\n",
       "111  1  0  0  0\n",
       "112  1  0  0  0\n",
       "113  1  1  1  1\n",
       "114  1  0  0  0\n",
       "115  1  0  0  0\n",
       "\n",
       "[116 rows x 4 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "rng = np.random.default_rng(seed = 42)\n",
    "len(Xs[0])\n",
    "m = 1000\n",
    "n = 10\n",
    "X_compl = rng.standard_normal((len(Xs[0]), len(Xs)))\n",
    "X_compl = pd.DataFrame(X_compl)\n",
    "amp = MultivariateAmputation(patterns= [{\"incomplete_vars\": range(1, X_compl.shape[1]), \"mechanism\": \"MAR\"}])\n",
    "X = amp.fit_transform(X_compl)\n",
    "X[X.notnull()] = 1\n",
    "X = X.fillna(0).astype(int)\n",
    "X"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "id": "6614e83e-1f48-4a0b-8ed1-93e37ec8595c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MultiIndex([(5000,)],\n",
       "           )"
      ]
     },
     "execution_count": 212,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "row = results.iloc[[5000]]\n",
    "row_index = row.index\n",
    "multiidx = []\n",
    "for level in row_index.names:\n",
    "    if level == \"missing_percentage\":\n",
    "        multiidx_value = 0\n",
    "    elif level == \"amputation_mechanism\":\n",
    "        multiidx_value = \"None\"\n",
    "    elif level == \"imputation\":\n",
    "        multiidx_value = False\n",
    "    else:\n",
    "        multiidx_value = row_index.get_level_values(level=level)[0]\n",
    "    multiidx.append([multiidx_value])\n",
    "best_solution = pd.MultiIndex.from_arrays(multiidx, names=row_index.names)\n",
    "best_solution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "407cca22-c8e4-42cc-b538-8639845207ba",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>amputation_mechanism</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "      <th>finished</th>\n",
       "      <th>completed</th>\n",
       "      <th>n_samples</th>\n",
       "      <th>n_incomplete_samples</th>\n",
       "      <th>...</th>\n",
       "      <th>bal_acc</th>\n",
       "      <th>ami</th>\n",
       "      <th>ari</th>\n",
       "      <th>completeness</th>\n",
       "      <th>random_acc</th>\n",
       "      <th>random_f1</th>\n",
       "      <th>silhouette</th>\n",
       "      <th>vrc</th>\n",
       "      <th>db</th>\n",
       "      <th>dunn</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5000</th>\n",
       "      <td>nutrimouse_diet</td>\n",
       "      <td>GroupPCA</td>\n",
       "      <td>50</td>\n",
       "      <td>MCAR</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              dataset algorithm  missing_percentage amputation_mechanism   \n",
       "5000  nutrimouse_diet  GroupPCA                  50                 MCAR  \\\n",
       "\n",
       "      imputation  run_n  finished  completed  n_samples  n_incomplete_samples   \n",
       "5000        True      0     False      False        NaN                   NaN  \\\n",
       "\n",
       "      ...  bal_acc  ami  ari  completeness random_acc random_f1 silhouette   \n",
       "5000  ...      NaN  NaN  NaN           NaN        NaN       NaN        NaN  \\\n",
       "\n",
       "     vrc  db dunn  \n",
       "5000 NaN NaN  NaN  \n",
       "\n",
       "[1 rows x 40 columns]"
      ]
     },
     "execution_count": 211,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "id": "8b12c586-4dce-4a7e-98e0-64cbff2adfd4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0         True\n",
       "1         True\n",
       "2         True\n",
       "3         True\n",
       "4         True\n",
       "         ...  \n",
       "49495    False\n",
       "49496    False\n",
       "49497    False\n",
       "49498    False\n",
       "49499    False\n",
       "Length: 49500, dtype: bool"
      ]
     },
     "execution_count": 205,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(results[\"amputation_mechanism\"] == \"EDM\") & (results[\"missing_percentage\"] == 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "751a6a65-59c5-42ba-8bfc-f7ef0ebf5003",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "amputation_mechanism\n",
       "EDM     16200\n",
       "MCAR    16200\n",
       "MAR      8100\n",
       "MNAR     8100\n",
       "None      900\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results[\"amputation_mechanism\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "2ae9bb18-4528-4712-91e8-cb544b2163b7",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>finished</th>\n",
       "      <th>completed</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>amputation_mechanism</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nutrimouse_genotype</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">Concat</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">NaN</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nuswide</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">jNMF</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">90</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">MNAR</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>49500 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                        finished   \n",
       "dataset             algorithm missing_percentage amputation_mechanism imputation run_n             \n",
       "nutrimouse_genotype Concat    0                  NaN                  False      0         False  \\\n",
       "                                                                                 1         False   \n",
       "                                                                                 2         False   \n",
       "                                                                                 3         False   \n",
       "                                                                                 4         False   \n",
       "...                                                                                          ...   \n",
       "nuswide             jNMF      90                 MNAR                 False      5         False   \n",
       "                                                                                 6         False   \n",
       "                                                                                 7         False   \n",
       "                                                                                 8         False   \n",
       "                                                                                 9         False   \n",
       "\n",
       "                                                                                        completed  \n",
       "dataset             algorithm missing_percentage amputation_mechanism imputation run_n             \n",
       "nutrimouse_genotype Concat    0                  NaN                  False      0          False  \n",
       "                                                                                 1          False  \n",
       "                                                                                 2          False  \n",
       "                                                                                 3          False  \n",
       "                                                                                 4          False  \n",
       "...                                                                                           ...  \n",
       "nuswide             jNMF      90                 MNAR                 False      5          False  \n",
       "                                                                                 6          False  \n",
       "                                                                                 7          False  \n",
       "                                                                                 8          False  \n",
       "                                                                                 9          False  \n",
       "\n",
       "[49500 rows x 2 columns]"
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results = pd.read_csv(\"results/complete_algorithms_evaluation.csv\", index_col = [0,1,2,3,4,5])\n",
    "# results = results.reset_index()\n",
    "# results.loc[results.xs(0, level=\"missing_percentage\", drop_level=False).index].index = results.xs(0, level=\"missing_percentage\", drop_level=False).index.to_frame(index= False).replace(\"EDM\", \"None\").set_index(results.index.names).index\n",
    "results\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "41becfb6-c4e4-40ab-96e7-c5884258da73",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>amputation_mechanism</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nutrimouse_genotype</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">Concat</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">None</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nuswide</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">jNMF</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">None</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>900 rows × 0 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: [(nutrimouse_genotype, Concat, None, 0, False, 0), (nutrimouse_genotype, Concat, None, 0, False, 1), (nutrimouse_genotype, Concat, None, 0, False, 2), (nutrimouse_genotype, Concat, None, 0, False, 3), (nutrimouse_genotype, Concat, None, 0, False, 4), (nutrimouse_genotype, Concat, None, 0, False, 5), (nutrimouse_genotype, Concat, None, 0, False, 6), (nutrimouse_genotype, Concat, None, 0, False, 7), (nutrimouse_genotype, Concat, None, 0, False, 8), (nutrimouse_genotype, Concat, None, 0, False, 9), (nutrimouse_genotype, NMFC, None, 0, False, 0), (nutrimouse_genotype, NMFC, None, 0, False, 1), (nutrimouse_genotype, NMFC, None, 0, False, 2), (nutrimouse_genotype, NMFC, None, 0, False, 3), (nutrimouse_genotype, NMFC, None, 0, False, 4), (nutrimouse_genotype, NMFC, None, 0, False, 5), (nutrimouse_genotype, NMFC, None, 0, False, 6), (nutrimouse_genotype, NMFC, None, 0, False, 7), (nutrimouse_genotype, NMFC, None, 0, False, 8), (nutrimouse_genotype, NMFC, None, 0, False, 9), (nutrimouse_genotype, MVSpectralClustering, None, 0, False, 0), (nutrimouse_genotype, MVSpectralClustering, None, 0, False, 1), (nutrimouse_genotype, MVSpectralClustering, None, 0, False, 2), (nutrimouse_genotype, MVSpectralClustering, None, 0, False, 3), (nutrimouse_genotype, MVSpectralClustering, None, 0, False, 4), (nutrimouse_genotype, MVSpectralClustering, None, 0, False, 5), (nutrimouse_genotype, MVSpectralClustering, None, 0, False, 6), (nutrimouse_genotype, MVSpectralClustering, None, 0, False, 7), (nutrimouse_genotype, MVSpectralClustering, None, 0, False, 8), (nutrimouse_genotype, MVSpectralClustering, None, 0, False, 9), (nutrimouse_genotype, MVCoRegSpectralClustering, None, 0, False, 0), (nutrimouse_genotype, MVCoRegSpectralClustering, None, 0, False, 1), (nutrimouse_genotype, MVCoRegSpectralClustering, None, 0, False, 2), (nutrimouse_genotype, MVCoRegSpectralClustering, None, 0, False, 3), (nutrimouse_genotype, MVCoRegSpectralClustering, None, 0, False, 4), (nutrimouse_genotype, MVCoRegSpectralClustering, None, 0, False, 5), (nutrimouse_genotype, MVCoRegSpectralClustering, None, 0, False, 6), (nutrimouse_genotype, MVCoRegSpectralClustering, None, 0, False, 7), (nutrimouse_genotype, MVCoRegSpectralClustering, None, 0, False, 8), (nutrimouse_genotype, MVCoRegSpectralClustering, None, 0, False, 9), (nutrimouse_genotype, GroupPCA, None, 0, False, 0), (nutrimouse_genotype, GroupPCA, None, 0, False, 1), (nutrimouse_genotype, GroupPCA, None, 0, False, 2), (nutrimouse_genotype, GroupPCA, None, 0, False, 3), (nutrimouse_genotype, GroupPCA, None, 0, False, 4), (nutrimouse_genotype, GroupPCA, None, 0, False, 5), (nutrimouse_genotype, GroupPCA, None, 0, False, 6), (nutrimouse_genotype, GroupPCA, None, 0, False, 7), (nutrimouse_genotype, GroupPCA, None, 0, False, 8), (nutrimouse_genotype, GroupPCA, None, 0, False, 9), (nutrimouse_genotype, AJIVE, None, 0, False, 0), (nutrimouse_genotype, AJIVE, None, 0, False, 1), (nutrimouse_genotype, AJIVE, None, 0, False, 2), (nutrimouse_genotype, AJIVE, None, 0, False, 3), (nutrimouse_genotype, AJIVE, None, 0, False, 4), (nutrimouse_genotype, AJIVE, None, 0, False, 5), (nutrimouse_genotype, AJIVE, None, 0, False, 6), (nutrimouse_genotype, AJIVE, None, 0, False, 7), (nutrimouse_genotype, AJIVE, None, 0, False, 8), (nutrimouse_genotype, AJIVE, None, 0, False, 9), (nutrimouse_genotype, SNF, None, 0, False, 0), (nutrimouse_genotype, SNF, None, 0, False, 1), (nutrimouse_genotype, SNF, None, 0, False, 2), (nutrimouse_genotype, SNF, None, 0, False, 3), (nutrimouse_genotype, SNF, None, 0, False, 4), (nutrimouse_genotype, SNF, None, 0, False, 5), (nutrimouse_genotype, SNF, None, 0, False, 6), (nutrimouse_genotype, SNF, None, 0, False, 7), (nutrimouse_genotype, SNF, None, 0, False, 8), (nutrimouse_genotype, SNF, None, 0, False, 9), (nutrimouse_genotype, intNMF, None, 0, False, 0), (nutrimouse_genotype, intNMF, None, 0, False, 1), (nutrimouse_genotype, intNMF, None, 0, False, 2), (nutrimouse_genotype, intNMF, None, 0, False, 3), (nutrimouse_genotype, intNMF, None, 0, False, 4), (nutrimouse_genotype, intNMF, None, 0, False, 5), (nutrimouse_genotype, intNMF, None, 0, False, 6), (nutrimouse_genotype, intNMF, None, 0, False, 7), (nutrimouse_genotype, intNMF, None, 0, False, 8), (nutrimouse_genotype, intNMF, None, 0, False, 9), (nutrimouse_genotype, jNMF, None, 0, False, 0), (nutrimouse_genotype, jNMF, None, 0, False, 1), (nutrimouse_genotype, jNMF, None, 0, False, 2), (nutrimouse_genotype, jNMF, None, 0, False, 3), (nutrimouse_genotype, jNMF, None, 0, False, 4), (nutrimouse_genotype, jNMF, None, 0, False, 5), (nutrimouse_genotype, jNMF, None, 0, False, 6), (nutrimouse_genotype, jNMF, None, 0, False, 7), (nutrimouse_genotype, jNMF, None, 0, False, 8), (nutrimouse_genotype, jNMF, None, 0, False, 9), (nutrimouse_diet, Concat, None, 0, False, 0), (nutrimouse_diet, Concat, None, 0, False, 1), (nutrimouse_diet, Concat, None, 0, False, 2), (nutrimouse_diet, Concat, None, 0, False, 3), (nutrimouse_diet, Concat, None, 0, False, 4), (nutrimouse_diet, Concat, None, 0, False, 5), (nutrimouse_diet, Concat, None, 0, False, 6), (nutrimouse_diet, Concat, None, 0, False, 7), (nutrimouse_diet, Concat, None, 0, False, 8), (nutrimouse_diet, Concat, None, 0, False, 9), ...]\n",
       "\n",
       "[900 rows x 0 columns]"
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results = pd.read_csv(\"results/complete_algorithms_evaluation.csv\", index_col = [0,1,2,3,4,5])\n",
    "results_amputation_mechanism_complete = results.xs(0, level=\"missing_percentage\", drop_level=False)\n",
    "results_amputation_mechanism_none = results_amputation_mechanism_complete.index.to_frame(index= False).replace(\"EDM\", \"None\")\n",
    "results_amputation_mechanism_none = results_amputation_mechanism_none.set_index(results_amputation_mechanism_none.columns.to_list())\n",
    "results_amputation_mechanism_none_2.index = pd.MultiIndex.from_frame(results_amputation_mechanism_none_tochange)\n",
    "\n",
    "# results.loc[results_amputation_mechanism_none.index].index = pd.MultiIndex.from_frame(results_amputation_mechanism_none_tochange)\n",
    "results_amputation_mechanism_none"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "6035c989-f3ce-452e-a745-98058fbbfe72",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "MultiIndex([('nutrimouse_genotype', 'Concat', 'EDM', 0, False, 0),\n",
       "            ('nutrimouse_genotype', 'Concat', 'EDM', 0, False, 1),\n",
       "            ('nutrimouse_genotype', 'Concat', 'EDM', 0, False, 2),\n",
       "            ('nutrimouse_genotype', 'Concat', 'EDM', 0, False, 3),\n",
       "            ('nutrimouse_genotype', 'Concat', 'EDM', 0, False, 4),\n",
       "            ('nutrimouse_genotype', 'Concat', 'EDM', 0, False, 5),\n",
       "            ('nutrimouse_genotype', 'Concat', 'EDM', 0, False, 6),\n",
       "            ('nutrimouse_genotype', 'Concat', 'EDM', 0, False, 7),\n",
       "            ('nutrimouse_genotype', 'Concat', 'EDM', 0, False, 8),\n",
       "            ('nutrimouse_genotype', 'Concat', 'EDM', 0, False, 9),\n",
       "            ...\n",
       "            (            'nuswide',   'jNMF', 'EDM', 0, False, 0),\n",
       "            (            'nuswide',   'jNMF', 'EDM', 0, False, 1),\n",
       "            (            'nuswide',   'jNMF', 'EDM', 0, False, 2),\n",
       "            (            'nuswide',   'jNMF', 'EDM', 0, False, 3),\n",
       "            (            'nuswide',   'jNMF', 'EDM', 0, False, 4),\n",
       "            (            'nuswide',   'jNMF', 'EDM', 0, False, 5),\n",
       "            (            'nuswide',   'jNMF', 'EDM', 0, False, 6),\n",
       "            (            'nuswide',   'jNMF', 'EDM', 0, False, 7),\n",
       "            (            'nuswide',   'jNMF', 'EDM', 0, False, 8),\n",
       "            (            'nuswide',   'jNMF', 'EDM', 0, False, 9)],\n",
       "           names=['dataset', 'algorithm', 'amputation_mechanism', 'missing_percentage', 'imputation', 'run_n'], length=900)"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results.loc[results_amputation_mechanism_complete.index].index = results_amputation_mechanism_complete.index\n",
    "results.loc[results_amputation_mechanism_complete.index].index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "id": "400877b1-bbaa-4922-a5f8-810f1bb777dc",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>amputation_mechanism</th>\n",
       "      <th>finished</th>\n",
       "      <th>completed</th>\n",
       "      <th>n_samples</th>\n",
       "      <th>n_incomplete_samples</th>\n",
       "      <th>n_complete_samples</th>\n",
       "      <th>time</th>\n",
       "      <th>n_clustered_samples</th>\n",
       "      <th>percentage_clustered_samples</th>\n",
       "      <th>comments</th>\n",
       "      <th>...</th>\n",
       "      <th>MCC_performance</th>\n",
       "      <th>F1_performance</th>\n",
       "      <th>precision_performance</th>\n",
       "      <th>recall_performance</th>\n",
       "      <th>bal_acc_performance</th>\n",
       "      <th>ami_performance</th>\n",
       "      <th>ari_performance</th>\n",
       "      <th>completeness_performance</th>\n",
       "      <th>random_acc_performance</th>\n",
       "      <th>random_f1_performance</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nutrimouse_genotype</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">Concat</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>0</th>\n",
       "      <td>EDM</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>1.418447</td>\n",
       "      <td>40.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>{}</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EDM</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>1.707200</td>\n",
       "      <td>40.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>{}</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EDM</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>1.532106</td>\n",
       "      <td>40.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>{}</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EDM</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>1.413640</td>\n",
       "      <td>40.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>{}</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EDM</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>1.645764</td>\n",
       "      <td>40.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>{}</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nuswide</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">jNMF</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>5</th>\n",
       "      <td>EDM</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>EDM</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>EDM</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>EDM</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>EDM</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>900 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                  amputation_mechanism   \n",
       "dataset             algorithm missing_percentage imputation run_n                        \n",
       "nutrimouse_genotype Concat    0                  False      0                      EDM  \\\n",
       "                                                            1                      EDM   \n",
       "                                                            2                      EDM   \n",
       "                                                            3                      EDM   \n",
       "                                                            4                      EDM   \n",
       "...                                                                                ...   \n",
       "nuswide             jNMF      0                  False      5                      EDM   \n",
       "                                                            6                      EDM   \n",
       "                                                            7                      EDM   \n",
       "                                                            8                      EDM   \n",
       "                                                            9                      EDM   \n",
       "\n",
       "                                                                   finished   \n",
       "dataset             algorithm missing_percentage imputation run_n             \n",
       "nutrimouse_genotype Concat    0                  False      0          True  \\\n",
       "                                                            1          True   \n",
       "                                                            2          True   \n",
       "                                                            3          True   \n",
       "                                                            4          True   \n",
       "...                                                                     ...   \n",
       "nuswide             jNMF      0                  False      5         False   \n",
       "                                                            6         False   \n",
       "                                                            7         False   \n",
       "                                                            8         False   \n",
       "                                                            9         False   \n",
       "\n",
       "                                                                   completed   \n",
       "dataset             algorithm missing_percentage imputation run_n              \n",
       "nutrimouse_genotype Concat    0                  False      0           True  \\\n",
       "                                                            1           True   \n",
       "                                                            2           True   \n",
       "                                                            3           True   \n",
       "                                                            4           True   \n",
       "...                                                                      ...   \n",
       "nuswide             jNMF      0                  False      5          False   \n",
       "                                                            6          False   \n",
       "                                                            7          False   \n",
       "                                                            8          False   \n",
       "                                                            9          False   \n",
       "\n",
       "                                                                   n_samples   \n",
       "dataset             algorithm missing_percentage imputation run_n              \n",
       "nutrimouse_genotype Concat    0                  False      0           40.0  \\\n",
       "                                                            1           40.0   \n",
       "                                                            2           40.0   \n",
       "                                                            3           40.0   \n",
       "                                                            4           40.0   \n",
       "...                                                                      ...   \n",
       "nuswide             jNMF      0                  False      5            NaN   \n",
       "                                                            6            NaN   \n",
       "                                                            7            NaN   \n",
       "                                                            8            NaN   \n",
       "                                                            9            NaN   \n",
       "\n",
       "                                                                   n_incomplete_samples   \n",
       "dataset             algorithm missing_percentage imputation run_n                         \n",
       "nutrimouse_genotype Concat    0                  False      0                       0.0  \\\n",
       "                                                            1                       0.0   \n",
       "                                                            2                       0.0   \n",
       "                                                            3                       0.0   \n",
       "                                                            4                       0.0   \n",
       "...                                                                                 ...   \n",
       "nuswide             jNMF      0                  False      5                       NaN   \n",
       "                                                            6                       NaN   \n",
       "                                                            7                       NaN   \n",
       "                                                            8                       NaN   \n",
       "                                                            9                       NaN   \n",
       "\n",
       "                                                                   n_complete_samples   \n",
       "dataset             algorithm missing_percentage imputation run_n                       \n",
       "nutrimouse_genotype Concat    0                  False      0                    40.0  \\\n",
       "                                                            1                    40.0   \n",
       "                                                            2                    40.0   \n",
       "                                                            3                    40.0   \n",
       "                                                            4                    40.0   \n",
       "...                                                                               ...   \n",
       "nuswide             jNMF      0                  False      5                     NaN   \n",
       "                                                            6                     NaN   \n",
       "                                                            7                     NaN   \n",
       "                                                            8                     NaN   \n",
       "                                                            9                     NaN   \n",
       "\n",
       "                                                                       time   \n",
       "dataset             algorithm missing_percentage imputation run_n             \n",
       "nutrimouse_genotype Concat    0                  False      0      1.418447  \\\n",
       "                                                            1      1.707200   \n",
       "                                                            2      1.532106   \n",
       "                                                            3      1.413640   \n",
       "                                                            4      1.645764   \n",
       "...                                                                     ...   \n",
       "nuswide             jNMF      0                  False      5           NaN   \n",
       "                                                            6           NaN   \n",
       "                                                            7           NaN   \n",
       "                                                            8           NaN   \n",
       "                                                            9           NaN   \n",
       "\n",
       "                                                                   n_clustered_samples   \n",
       "dataset             algorithm missing_percentage imputation run_n                        \n",
       "nutrimouse_genotype Concat    0                  False      0                     40.0  \\\n",
       "                                                            1                     40.0   \n",
       "                                                            2                     40.0   \n",
       "                                                            3                     40.0   \n",
       "                                                            4                     40.0   \n",
       "...                                                                                ...   \n",
       "nuswide             jNMF      0                  False      5                      NaN   \n",
       "                                                            6                      NaN   \n",
       "                                                            7                      NaN   \n",
       "                                                            8                      NaN   \n",
       "                                                            9                      NaN   \n",
       "\n",
       "                                                                   percentage_clustered_samples   \n",
       "dataset             algorithm missing_percentage imputation run_n                                 \n",
       "nutrimouse_genotype Concat    0                  False      0                             100.0  \\\n",
       "                                                            1                             100.0   \n",
       "                                                            2                             100.0   \n",
       "                                                            3                             100.0   \n",
       "                                                            4                             100.0   \n",
       "...                                                                                         ...   \n",
       "nuswide             jNMF      0                  False      5                               NaN   \n",
       "                                                            6                               NaN   \n",
       "                                                            7                               NaN   \n",
       "                                                            8                               NaN   \n",
       "                                                            9                               NaN   \n",
       "\n",
       "                                                                  comments   \n",
       "dataset             algorithm missing_percentage imputation run_n            \n",
       "nutrimouse_genotype Concat    0                  False      0           {}  \\\n",
       "                                                            1           {}   \n",
       "                                                            2           {}   \n",
       "                                                            3           {}   \n",
       "                                                            4           {}   \n",
       "...                                                                    ...   \n",
       "nuswide             jNMF      0                  False      5          NaN   \n",
       "                                                            6          NaN   \n",
       "                                                            7          NaN   \n",
       "                                                            8          NaN   \n",
       "                                                            9          NaN   \n",
       "\n",
       "                                                                   ...   \n",
       "dataset             algorithm missing_percentage imputation run_n  ...   \n",
       "nutrimouse_genotype Concat    0                  False      0      ...  \\\n",
       "                                                            1      ...   \n",
       "                                                            2      ...   \n",
       "                                                            3      ...   \n",
       "                                                            4      ...   \n",
       "...                                                                ...   \n",
       "nuswide             jNMF      0                  False      5      ...   \n",
       "                                                            6      ...   \n",
       "                                                            7      ...   \n",
       "                                                            8      ...   \n",
       "                                                            9      ...   \n",
       "\n",
       "                                                                   MCC_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                    \n",
       "nutrimouse_genotype Concat    0                  False      0                  NaN  \\\n",
       "                                                            1                  NaN   \n",
       "                                                            2                  NaN   \n",
       "                                                            3                  NaN   \n",
       "                                                            4                  NaN   \n",
       "...                                                                            ...   \n",
       "nuswide             jNMF      0                  False      5                  NaN   \n",
       "                                                            6                  NaN   \n",
       "                                                            7                  NaN   \n",
       "                                                            8                  NaN   \n",
       "                                                            9                  NaN   \n",
       "\n",
       "                                                                  F1_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                  \n",
       "nutrimouse_genotype Concat    0                  False      0                NaN  \\\n",
       "                                                            1                NaN   \n",
       "                                                            2                NaN   \n",
       "                                                            3                NaN   \n",
       "                                                            4                NaN   \n",
       "...                                                                          ...   \n",
       "nuswide             jNMF      0                  False      5                NaN   \n",
       "                                                            6                NaN   \n",
       "                                                            7                NaN   \n",
       "                                                            8                NaN   \n",
       "                                                            9                NaN   \n",
       "\n",
       "                                                                  precision_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                         \n",
       "nutrimouse_genotype Concat    0                  False      0                       NaN  \\\n",
       "                                                            1                       NaN   \n",
       "                                                            2                       NaN   \n",
       "                                                            3                       NaN   \n",
       "                                                            4                       NaN   \n",
       "...                                                                                 ...   \n",
       "nuswide             jNMF      0                  False      5                       NaN   \n",
       "                                                            6                       NaN   \n",
       "                                                            7                       NaN   \n",
       "                                                            8                       NaN   \n",
       "                                                            9                       NaN   \n",
       "\n",
       "                                                                  recall_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                      \n",
       "nutrimouse_genotype Concat    0                  False      0                    NaN  \\\n",
       "                                                            1                    NaN   \n",
       "                                                            2                    NaN   \n",
       "                                                            3                    NaN   \n",
       "                                                            4                    NaN   \n",
       "...                                                                              ...   \n",
       "nuswide             jNMF      0                  False      5                    NaN   \n",
       "                                                            6                    NaN   \n",
       "                                                            7                    NaN   \n",
       "                                                            8                    NaN   \n",
       "                                                            9                    NaN   \n",
       "\n",
       "                                                                  bal_acc_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                       \n",
       "nutrimouse_genotype Concat    0                  False      0                     NaN  \\\n",
       "                                                            1                     NaN   \n",
       "                                                            2                     NaN   \n",
       "                                                            3                     NaN   \n",
       "                                                            4                     NaN   \n",
       "...                                                                               ...   \n",
       "nuswide             jNMF      0                  False      5                     NaN   \n",
       "                                                            6                     NaN   \n",
       "                                                            7                     NaN   \n",
       "                                                            8                     NaN   \n",
       "                                                            9                     NaN   \n",
       "\n",
       "                                                                  ami_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                   \n",
       "nutrimouse_genotype Concat    0                  False      0                 NaN  \\\n",
       "                                                            1                 NaN   \n",
       "                                                            2                 NaN   \n",
       "                                                            3                 NaN   \n",
       "                                                            4                 NaN   \n",
       "...                                                                           ...   \n",
       "nuswide             jNMF      0                  False      5                 NaN   \n",
       "                                                            6                 NaN   \n",
       "                                                            7                 NaN   \n",
       "                                                            8                 NaN   \n",
       "                                                            9                 NaN   \n",
       "\n",
       "                                                                  ari_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                   \n",
       "nutrimouse_genotype Concat    0                  False      0                 NaN  \\\n",
       "                                                            1                 NaN   \n",
       "                                                            2                 NaN   \n",
       "                                                            3                 NaN   \n",
       "                                                            4                 NaN   \n",
       "...                                                                           ...   \n",
       "nuswide             jNMF      0                  False      5                 NaN   \n",
       "                                                            6                 NaN   \n",
       "                                                            7                 NaN   \n",
       "                                                            8                 NaN   \n",
       "                                                            9                 NaN   \n",
       "\n",
       "                                                                  completeness_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                            \n",
       "nutrimouse_genotype Concat    0                  False      0                          NaN  \\\n",
       "                                                            1                          NaN   \n",
       "                                                            2                          NaN   \n",
       "                                                            3                          NaN   \n",
       "                                                            4                          NaN   \n",
       "...                                                                                    ...   \n",
       "nuswide             jNMF      0                  False      5                          NaN   \n",
       "                                                            6                          NaN   \n",
       "                                                            7                          NaN   \n",
       "                                                            8                          NaN   \n",
       "                                                            9                          NaN   \n",
       "\n",
       "                                                                  random_acc_performance   \n",
       "dataset             algorithm missing_percentage imputation run_n                          \n",
       "nutrimouse_genotype Concat    0                  False      0                        NaN  \\\n",
       "                                                            1                        NaN   \n",
       "                                                            2                        NaN   \n",
       "                                                            3                        NaN   \n",
       "                                                            4                        NaN   \n",
       "...                                                                                  ...   \n",
       "nuswide             jNMF      0                  False      5                        NaN   \n",
       "                                                            6                        NaN   \n",
       "                                                            7                        NaN   \n",
       "                                                            8                        NaN   \n",
       "                                                            9                        NaN   \n",
       "\n",
       "                                                                  random_f1_performance  \n",
       "dataset             algorithm missing_percentage imputation run_n                        \n",
       "nutrimouse_genotype Concat    0                  False      0                       NaN  \n",
       "                                                            1                       NaN  \n",
       "                                                            2                       NaN  \n",
       "                                                            3                       NaN  \n",
       "                                                            4                       NaN  \n",
       "...                                                                                 ...  \n",
       "nuswide             jNMF      0                  False      5                       NaN  \n",
       "                                                            6                       NaN  \n",
       "                                                            7                       NaN  \n",
       "                                                            8                       NaN  \n",
       "                                                            9                       NaN  \n",
       "\n",
       "[900 rows x 46 columns]"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = results.reset_index(\"amputation_mechanism\")\n",
    "a[\"amputation_mechanism\"] = "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "802c7c7c-e6f3-4be7-a087-0588215b373a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>finished</th>\n",
       "      <th>completed</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>amputation_mechanism</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nutrimouse_genotype</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">Concat</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">EDM</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nuswide</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">jNMF</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">MNAR</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">90</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>49500 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                        finished   \n",
       "dataset             algorithm amputation_mechanism missing_percentage imputation run_n             \n",
       "nutrimouse_genotype Concat    EDM                  0                  False      0         False  \\\n",
       "                                                                                 1         False   \n",
       "                                                                                 2         False   \n",
       "                                                                                 3         False   \n",
       "                                                                                 4         False   \n",
       "...                                                                                          ...   \n",
       "nuswide             jNMF      MNAR                 90                 False      5         False   \n",
       "                                                                                 6         False   \n",
       "                                                                                 7         False   \n",
       "                                                                                 8         False   \n",
       "                                                                                 9         False   \n",
       "\n",
       "                                                                                        completed  \n",
       "dataset             algorithm amputation_mechanism missing_percentage imputation run_n             \n",
       "nutrimouse_genotype Concat    EDM                  0                  False      0          False  \n",
       "                                                                                 1          False  \n",
       "                                                                                 2          False  \n",
       "                                                                                 3          False  \n",
       "                                                                                 4          False  \n",
       "...                                                                                           ...  \n",
       "nuswide             jNMF      MNAR                 90                 False      5          False  \n",
       "                                                                                 6          False  \n",
       "                                                                                 7          False  \n",
       "                                                                                 8          False  \n",
       "                                                                                 9          False  \n",
       "\n",
       "[49500 rows x 2 columns]"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results.loc[results_amputation_mechanism_none_tochange.index].index = results_amputation_mechanism_complete.index\n",
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "9e7d4a9c-4d14-4ca4-b586-ea328456c76b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>finished</th>\n",
       "      <th>completed</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>amputation_mechanism</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nutrimouse_genotype</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">Concat</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">EDM</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nuswide</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">jNMF</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">EDM</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>900 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                       finished   \n",
       "dataset             algorithm amputation_mechanism missing_percentage imputation run_n            \n",
       "nutrimouse_genotype Concat    EDM                  0                  False      0          NaN  \\\n",
       "                                                                                 1          NaN   \n",
       "                                                                                 2          NaN   \n",
       "                                                                                 3          NaN   \n",
       "                                                                                 4          NaN   \n",
       "...                                                                                         ...   \n",
       "nuswide             jNMF      EDM                  0                  False      5          NaN   \n",
       "                                                                                 6          NaN   \n",
       "                                                                                 7          NaN   \n",
       "                                                                                 8          NaN   \n",
       "                                                                                 9          NaN   \n",
       "\n",
       "                                                                                       completed  \n",
       "dataset             algorithm amputation_mechanism missing_percentage imputation run_n            \n",
       "nutrimouse_genotype Concat    EDM                  0                  False      0           NaN  \n",
       "                                                                                 1           NaN  \n",
       "                                                                                 2           NaN  \n",
       "                                                                                 3           NaN  \n",
       "                                                                                 4           NaN  \n",
       "...                                                                                          ...  \n",
       "nuswide             jNMF      EDM                  0                  False      5           NaN  \n",
       "                                                                                 6           NaN  \n",
       "                                                                                 7           NaN  \n",
       "                                                                                 8           NaN  \n",
       "                                                                                 9           NaN  \n",
       "\n",
       "[900 rows x 2 columns]"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results.loc[results_amputation_mechanism_none.index, :] = results_amputation_mechanism_none_2\n",
    "results.loc[results_amputation_mechanism_none.index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "62aac555-2332-40d1-8c07-70a1f674c26a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>finished</th>\n",
       "      <th>completed</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>amputation_mechanism</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nutrimouse_genotype</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">Concat</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">None</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nuswide</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">jNMF</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">None</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>900 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                        finished   \n",
       "dataset             algorithm amputation_mechanism missing_percentage imputation run_n             \n",
       "nutrimouse_genotype Concat    None                 0                  False      0         False  \\\n",
       "                                                                                 1         False   \n",
       "                                                                                 2         False   \n",
       "                                                                                 3         False   \n",
       "                                                                                 4         False   \n",
       "...                                                                                          ...   \n",
       "nuswide             jNMF      None                 0                  False      5         False   \n",
       "                                                                                 6         False   \n",
       "                                                                                 7         False   \n",
       "                                                                                 8         False   \n",
       "                                                                                 9         False   \n",
       "\n",
       "                                                                                        completed  \n",
       "dataset             algorithm amputation_mechanism missing_percentage imputation run_n             \n",
       "nutrimouse_genotype Concat    None                 0                  False      0          False  \n",
       "                                                                                 1          False  \n",
       "                                                                                 2          False  \n",
       "                                                                                 3          False  \n",
       "                                                                                 4          False  \n",
       "...                                                                                           ...  \n",
       "nuswide             jNMF      None                 0                  False      5          False  \n",
       "                                                                                 6          False  \n",
       "                                                                                 7          False  \n",
       "                                                                                 8          False  \n",
       "                                                                                 9          False  \n",
       "\n",
       "[900 rows x 2 columns]"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results_amputation_mechanism_none_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "d334f2b0-acd1-4ed6-bb42-08c8cf059f7b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>finished</th>\n",
       "      <th>completed</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>amputation_mechanism</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nutrimouse_genotype</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">Concat</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">EDM</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nuswide</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">jNMF</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">MNAR</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">90</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>49500 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                        finished   \n",
       "dataset             algorithm amputation_mechanism missing_percentage imputation run_n             \n",
       "nutrimouse_genotype Concat    EDM                  0                  False      0         False  \\\n",
       "                                                                                 1         False   \n",
       "                                                                                 2         False   \n",
       "                                                                                 3         False   \n",
       "                                                                                 4         False   \n",
       "...                                                                                          ...   \n",
       "nuswide             jNMF      MNAR                 90                 False      5         False   \n",
       "                                                                                 6         False   \n",
       "                                                                                 7         False   \n",
       "                                                                                 8         False   \n",
       "                                                                                 9         False   \n",
       "\n",
       "                                                                                        completed  \n",
       "dataset             algorithm amputation_mechanism missing_percentage imputation run_n             \n",
       "nutrimouse_genotype Concat    EDM                  0                  False      0          False  \n",
       "                                                                                 1          False  \n",
       "                                                                                 2          False  \n",
       "                                                                                 3          False  \n",
       "                                                                                 4          False  \n",
       "...                                                                                           ...  \n",
       "nuswide             jNMF      MNAR                 90                 False      5          False  \n",
       "                                                                                 6          False  \n",
       "                                                                                 7          False  \n",
       "                                                                                 8          False  \n",
       "                                                                                 9          False  \n",
       "\n",
       "[49500 rows x 2 columns]"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results.loc[results_amputation_mechanism_none.index].index = pd.MultiIndex.from_frame(results_amputation_mechanism_none_tochange)\n",
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "5750674e-417a-4fc9-bb48-e3a459e7df08",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([('nutrimouse_genotype', 'Concat', 'None', 0, False, 0),\n",
       "       ('nutrimouse_genotype', 'Concat', 'None', 0, False, 1),\n",
       "       ('nutrimouse_genotype', 'Concat', 'None', 0, False, 2),\n",
       "       ('nutrimouse_genotype', 'Concat', 'None', 0, False, 3),\n",
       "       ('nutrimouse_genotype', 'Concat', 'None', 0, False, 4),\n",
       "       ('nutrimouse_genotype', 'Concat', 'None', 0, False, 5),\n",
       "       ('nutrimouse_genotype', 'Concat', 'None', 0, False, 6),\n",
       "       ('nutrimouse_genotype', 'Concat', 'None', 0, False, 7),\n",
       "       ('nutrimouse_genotype', 'Concat', 'None', 0, False, 8),\n",
       "       ('nutrimouse_genotype', 'Concat', 'None', 0, False, 9),\n",
       "       ('nutrimouse_genotype', 'NMFC', 'None', 0, False, 0),\n",
       "       ('nutrimouse_genotype', 'NMFC', 'None', 0, False, 1),\n",
       "       ('nutrimouse_genotype', 'NMFC', 'None', 0, False, 2),\n",
       "       ('nutrimouse_genotype', 'NMFC', 'None', 0, False, 3),\n",
       "       ('nutrimouse_genotype', 'NMFC', 'None', 0, False, 4),\n",
       "       ('nutrimouse_genotype', 'NMFC', 'None', 0, False, 5),\n",
       "       ('nutrimouse_genotype', 'NMFC', 'None', 0, False, 6),\n",
       "       ('nutrimouse_genotype', 'NMFC', 'None', 0, False, 7),\n",
       "       ('nutrimouse_genotype', 'NMFC', 'None', 0, False, 8),\n",
       "       ('nutrimouse_genotype', 'NMFC', 'None', 0, False, 9),\n",
       "       ('nutrimouse_genotype', 'MVSpectralClustering', 'None', 0, False, 0),\n",
       "       ('nutrimouse_genotype', 'MVSpectralClustering', 'None', 0, False, 1),\n",
       "       ('nutrimouse_genotype', 'MVSpectralClustering', 'None', 0, False, 2),\n",
       "       ('nutrimouse_genotype', 'MVSpectralClustering', 'None', 0, False, 3),\n",
       "       ('nutrimouse_genotype', 'MVSpectralClustering', 'None', 0, False, 4),\n",
       "       ('nutrimouse_genotype', 'MVSpectralClustering', 'None', 0, False, 5),\n",
       "       ('nutrimouse_genotype', 'MVSpectralClustering', 'None', 0, False, 6),\n",
       "       ('nutrimouse_genotype', 'MVSpectralClustering', 'None', 0, False, 7),\n",
       "       ('nutrimouse_genotype', 'MVSpectralClustering', 'None', 0, False, 8),\n",
       "       ('nutrimouse_genotype', 'MVSpectralClustering', 'None', 0, False, 9),\n",
       "       ('nutrimouse_genotype', 'MVCoRegSpectralClustering', 'None', 0, False, 0),\n",
       "       ('nutrimouse_genotype', 'MVCoRegSpectralClustering', 'None', 0, False, 1),\n",
       "       ('nutrimouse_genotype', 'MVCoRegSpectralClustering', 'None', 0, False, 2),\n",
       "       ('nutrimouse_genotype', 'MVCoRegSpectralClustering', 'None', 0, False, 3),\n",
       "       ('nutrimouse_genotype', 'MVCoRegSpectralClustering', 'None', 0, False, 4),\n",
       "       ('nutrimouse_genotype', 'MVCoRegSpectralClustering', 'None', 0, False, 5),\n",
       "       ('nutrimouse_genotype', 'MVCoRegSpectralClustering', 'None', 0, False, 6),\n",
       "       ('nutrimouse_genotype', 'MVCoRegSpectralClustering', 'None', 0, False, 7),\n",
       "       ('nutrimouse_genotype', 'MVCoRegSpectralClustering', 'None', 0, False, 8),\n",
       "       ('nutrimouse_genotype', 'MVCoRegSpectralClustering', 'None', 0, False, 9),\n",
       "       ('nutrimouse_genotype', 'GroupPCA', 'None', 0, False, 0),\n",
       "       ('nutrimouse_genotype', 'GroupPCA', 'None', 0, False, 1),\n",
       "       ('nutrimouse_genotype', 'GroupPCA', 'None', 0, False, 2),\n",
       "       ('nutrimouse_genotype', 'GroupPCA', 'None', 0, False, 3),\n",
       "       ('nutrimouse_genotype', 'GroupPCA', 'None', 0, False, 4),\n",
       "       ('nutrimouse_genotype', 'GroupPCA', 'None', 0, False, 5),\n",
       "       ('nutrimouse_genotype', 'GroupPCA', 'None', 0, False, 6),\n",
       "       ('nutrimouse_genotype', 'GroupPCA', 'None', 0, False, 7),\n",
       "       ('nutrimouse_genotype', 'GroupPCA', 'None', 0, False, 8),\n",
       "       ('nutrimouse_genotype', 'GroupPCA', 'None', 0, False, 9),\n",
       "       ('nutrimouse_genotype', 'AJIVE', 'None', 0, False, 0),\n",
       "       ('nutrimouse_genotype', 'AJIVE', 'None', 0, False, 1),\n",
       "       ('nutrimouse_genotype', 'AJIVE', 'None', 0, False, 2),\n",
       "       ('nutrimouse_genotype', 'AJIVE', 'None', 0, False, 3),\n",
       "       ('nutrimouse_genotype', 'AJIVE', 'None', 0, False, 4),\n",
       "       ('nutrimouse_genotype', 'AJIVE', 'None', 0, False, 5),\n",
       "       ('nutrimouse_genotype', 'AJIVE', 'None', 0, False, 6),\n",
       "       ('nutrimouse_genotype', 'AJIVE', 'None', 0, False, 7),\n",
       "       ('nutrimouse_genotype', 'AJIVE', 'None', 0, False, 8),\n",
       "       ('nutrimouse_genotype', 'AJIVE', 'None', 0, False, 9),\n",
       "       ('nutrimouse_genotype', 'SNF', 'None', 0, False, 0),\n",
       "       ('nutrimouse_genotype', 'SNF', 'None', 0, False, 1),\n",
       "       ('nutrimouse_genotype', 'SNF', 'None', 0, False, 2),\n",
       "       ('nutrimouse_genotype', 'SNF', 'None', 0, False, 3),\n",
       "       ('nutrimouse_genotype', 'SNF', 'None', 0, False, 4),\n",
       "       ('nutrimouse_genotype', 'SNF', 'None', 0, False, 5),\n",
       "       ('nutrimouse_genotype', 'SNF', 'None', 0, False, 6),\n",
       "       ('nutrimouse_genotype', 'SNF', 'None', 0, False, 7),\n",
       "       ('nutrimouse_genotype', 'SNF', 'None', 0, False, 8),\n",
       "       ('nutrimouse_genotype', 'SNF', 'None', 0, False, 9),\n",
       "       ('nutrimouse_genotype', 'intNMF', 'None', 0, False, 0),\n",
       "       ('nutrimouse_genotype', 'intNMF', 'None', 0, False, 1),\n",
       "       ('nutrimouse_genotype', 'intNMF', 'None', 0, False, 2),\n",
       "       ('nutrimouse_genotype', 'intNMF', 'None', 0, False, 3),\n",
       "       ('nutrimouse_genotype', 'intNMF', 'None', 0, False, 4),\n",
       "       ('nutrimouse_genotype', 'intNMF', 'None', 0, False, 5),\n",
       "       ('nutrimouse_genotype', 'intNMF', 'None', 0, False, 6),\n",
       "       ('nutrimouse_genotype', 'intNMF', 'None', 0, False, 7),\n",
       "       ('nutrimouse_genotype', 'intNMF', 'None', 0, False, 8),\n",
       "       ('nutrimouse_genotype', 'intNMF', 'None', 0, False, 9),\n",
       "       ('nutrimouse_genotype', 'jNMF', 'None', 0, False, 0),\n",
       "       ('nutrimouse_genotype', 'jNMF', 'None', 0, False, 1),\n",
       "       ('nutrimouse_genotype', 'jNMF', 'None', 0, False, 2),\n",
       "       ('nutrimouse_genotype', 'jNMF', 'None', 0, False, 3),\n",
       "       ('nutrimouse_genotype', 'jNMF', 'None', 0, False, 4),\n",
       "       ('nutrimouse_genotype', 'jNMF', 'None', 0, False, 5),\n",
       "       ('nutrimouse_genotype', 'jNMF', 'None', 0, False, 6),\n",
       "       ('nutrimouse_genotype', 'jNMF', 'None', 0, False, 7),\n",
       "       ('nutrimouse_genotype', 'jNMF', 'None', 0, False, 8),\n",
       "       ('nutrimouse_genotype', 'jNMF', 'None', 0, False, 9),\n",
       "       ('nutrimouse_diet', 'Concat', 'None', 0, False, 0),\n",
       "       ('nutrimouse_diet', 'Concat', 'None', 0, False, 1),\n",
       "       ('nutrimouse_diet', 'Concat', 'None', 0, False, 2),\n",
       "       ('nutrimouse_diet', 'Concat', 'None', 0, False, 3),\n",
       "       ('nutrimouse_diet', 'Concat', 'None', 0, False, 4),\n",
       "       ('nutrimouse_diet', 'Concat', 'None', 0, False, 5),\n",
       "       ('nutrimouse_diet', 'Concat', 'None', 0, False, 6),\n",
       "       ('nutrimouse_diet', 'Concat', 'None', 0, False, 7),\n",
       "       ('nutrimouse_diet', 'Concat', 'None', 0, False, 8),\n",
       "       ('nutrimouse_diet', 'Concat', 'None', 0, False, 9),\n",
       "       ('nutrimouse_diet', 'NMFC', 'None', 0, False, 0),\n",
       "       ('nutrimouse_diet', 'NMFC', 'None', 0, False, 1),\n",
       "       ('nutrimouse_diet', 'NMFC', 'None', 0, False, 2),\n",
       "       ('nutrimouse_diet', 'NMFC', 'None', 0, False, 3),\n",
       "       ('nutrimouse_diet', 'NMFC', 'None', 0, False, 4),\n",
       "       ('nutrimouse_diet', 'NMFC', 'None', 0, False, 5),\n",
       "       ('nutrimouse_diet', 'NMFC', 'None', 0, False, 6),\n",
       "       ('nutrimouse_diet', 'NMFC', 'None', 0, False, 7),\n",
       "       ('nutrimouse_diet', 'NMFC', 'None', 0, False, 8),\n",
       "       ('nutrimouse_diet', 'NMFC', 'None', 0, False, 9),\n",
       "       ('nutrimouse_diet', 'MVSpectralClustering', 'None', 0, False, 0),\n",
       "       ('nutrimouse_diet', 'MVSpectralClustering', 'None', 0, False, 1),\n",
       "       ('nutrimouse_diet', 'MVSpectralClustering', 'None', 0, False, 2),\n",
       "       ('nutrimouse_diet', 'MVSpectralClustering', 'None', 0, False, 3),\n",
       "       ('nutrimouse_diet', 'MVSpectralClustering', 'None', 0, False, 4),\n",
       "       ('nutrimouse_diet', 'MVSpectralClustering', 'None', 0, False, 5),\n",
       "       ('nutrimouse_diet', 'MVSpectralClustering', 'None', 0, False, 6),\n",
       "       ('nutrimouse_diet', 'MVSpectralClustering', 'None', 0, False, 7),\n",
       "       ('nutrimouse_diet', 'MVSpectralClustering', 'None', 0, False, 8),\n",
       "       ('nutrimouse_diet', 'MVSpectralClustering', 'None', 0, False, 9),\n",
       "       ('nutrimouse_diet', 'MVCoRegSpectralClustering', 'None', 0, False, 0),\n",
       "       ('nutrimouse_diet', 'MVCoRegSpectralClustering', 'None', 0, False, 1),\n",
       "       ('nutrimouse_diet', 'MVCoRegSpectralClustering', 'None', 0, False, 2),\n",
       "       ('nutrimouse_diet', 'MVCoRegSpectralClustering', 'None', 0, False, 3),\n",
       "       ('nutrimouse_diet', 'MVCoRegSpectralClustering', 'None', 0, False, 4),\n",
       "       ('nutrimouse_diet', 'MVCoRegSpectralClustering', 'None', 0, False, 5),\n",
       "       ('nutrimouse_diet', 'MVCoRegSpectralClustering', 'None', 0, False, 6),\n",
       "       ('nutrimouse_diet', 'MVCoRegSpectralClustering', 'None', 0, False, 7),\n",
       "       ('nutrimouse_diet', 'MVCoRegSpectralClustering', 'None', 0, False, 8),\n",
       "       ('nutrimouse_diet', 'MVCoRegSpectralClustering', 'None', 0, False, 9),\n",
       "       ('nutrimouse_diet', 'GroupPCA', 'None', 0, False, 0),\n",
       "       ('nutrimouse_diet', 'GroupPCA', 'None', 0, False, 1),\n",
       "       ('nutrimouse_diet', 'GroupPCA', 'None', 0, False, 2),\n",
       "       ('nutrimouse_diet', 'GroupPCA', 'None', 0, False, 3),\n",
       "       ('nutrimouse_diet', 'GroupPCA', 'None', 0, False, 4),\n",
       "       ('nutrimouse_diet', 'GroupPCA', 'None', 0, False, 5),\n",
       "       ('nutrimouse_diet', 'GroupPCA', 'None', 0, False, 6),\n",
       "       ('nutrimouse_diet', 'GroupPCA', 'None', 0, False, 7),\n",
       "       ('nutrimouse_diet', 'GroupPCA', 'None', 0, False, 8),\n",
       "       ('nutrimouse_diet', 'GroupPCA', 'None', 0, False, 9),\n",
       "       ('nutrimouse_diet', 'AJIVE', 'None', 0, False, 0),\n",
       "       ('nutrimouse_diet', 'AJIVE', 'None', 0, False, 1),\n",
       "       ('nutrimouse_diet', 'AJIVE', 'None', 0, False, 2),\n",
       "       ('nutrimouse_diet', 'AJIVE', 'None', 0, False, 3),\n",
       "       ('nutrimouse_diet', 'AJIVE', 'None', 0, False, 4),\n",
       "       ('nutrimouse_diet', 'AJIVE', 'None', 0, False, 5),\n",
       "       ('nutrimouse_diet', 'AJIVE', 'None', 0, False, 6),\n",
       "       ('nutrimouse_diet', 'AJIVE', 'None', 0, False, 7),\n",
       "       ('nutrimouse_diet', 'AJIVE', 'None', 0, False, 8),\n",
       "       ('nutrimouse_diet', 'AJIVE', 'None', 0, False, 9),\n",
       "       ('nutrimouse_diet', 'SNF', 'None', 0, False, 0),\n",
       "       ('nutrimouse_diet', 'SNF', 'None', 0, False, 1),\n",
       "       ('nutrimouse_diet', 'SNF', 'None', 0, False, 2),\n",
       "       ('nutrimouse_diet', 'SNF', 'None', 0, False, 3),\n",
       "       ('nutrimouse_diet', 'SNF', 'None', 0, False, 4),\n",
       "       ('nutrimouse_diet', 'SNF', 'None', 0, False, 5),\n",
       "       ('nutrimouse_diet', 'SNF', 'None', 0, False, 6),\n",
       "       ('nutrimouse_diet', 'SNF', 'None', 0, False, 7),\n",
       "       ('nutrimouse_diet', 'SNF', 'None', 0, False, 8),\n",
       "       ('nutrimouse_diet', 'SNF', 'None', 0, False, 9),\n",
       "       ('nutrimouse_diet', 'intNMF', 'None', 0, False, 0),\n",
       "       ('nutrimouse_diet', 'intNMF', 'None', 0, False, 1),\n",
       "       ('nutrimouse_diet', 'intNMF', 'None', 0, False, 2),\n",
       "       ('nutrimouse_diet', 'intNMF', 'None', 0, False, 3),\n",
       "       ('nutrimouse_diet', 'intNMF', 'None', 0, False, 4),\n",
       "       ('nutrimouse_diet', 'intNMF', 'None', 0, False, 5),\n",
       "       ('nutrimouse_diet', 'intNMF', 'None', 0, False, 6),\n",
       "       ('nutrimouse_diet', 'intNMF', 'None', 0, False, 7),\n",
       "       ('nutrimouse_diet', 'intNMF', 'None', 0, False, 8),\n",
       "       ('nutrimouse_diet', 'intNMF', 'None', 0, False, 9),\n",
       "       ('nutrimouse_diet', 'jNMF', 'None', 0, False, 0),\n",
       "       ('nutrimouse_diet', 'jNMF', 'None', 0, False, 1),\n",
       "       ('nutrimouse_diet', 'jNMF', 'None', 0, False, 2),\n",
       "       ('nutrimouse_diet', 'jNMF', 'None', 0, False, 3),\n",
       "       ('nutrimouse_diet', 'jNMF', 'None', 0, False, 4),\n",
       "       ('nutrimouse_diet', 'jNMF', 'None', 0, False, 5),\n",
       "       ('nutrimouse_diet', 'jNMF', 'None', 0, False, 6),\n",
       "       ('nutrimouse_diet', 'jNMF', 'None', 0, False, 7),\n",
       "       ('nutrimouse_diet', 'jNMF', 'None', 0, False, 8),\n",
       "       ('nutrimouse_diet', 'jNMF', 'None', 0, False, 9),\n",
       "       ('bbcsport', 'Concat', 'None', 0, False, 0),\n",
       "       ('bbcsport', 'Concat', 'None', 0, False, 1),\n",
       "       ('bbcsport', 'Concat', 'None', 0, False, 2),\n",
       "       ('bbcsport', 'Concat', 'None', 0, False, 3),\n",
       "       ('bbcsport', 'Concat', 'None', 0, False, 4),\n",
       "       ('bbcsport', 'Concat', 'None', 0, False, 5),\n",
       "       ('bbcsport', 'Concat', 'None', 0, False, 6),\n",
       "       ('bbcsport', 'Concat', 'None', 0, False, 7),\n",
       "       ('bbcsport', 'Concat', 'None', 0, False, 8),\n",
       "       ('bbcsport', 'Concat', 'None', 0, False, 9),\n",
       "       ('bbcsport', 'NMFC', 'None', 0, False, 0),\n",
       "       ('bbcsport', 'NMFC', 'None', 0, False, 1),\n",
       "       ('bbcsport', 'NMFC', 'None', 0, False, 2),\n",
       "       ('bbcsport', 'NMFC', 'None', 0, False, 3),\n",
       "       ('bbcsport', 'NMFC', 'None', 0, False, 4),\n",
       "       ('bbcsport', 'NMFC', 'None', 0, False, 5),\n",
       "       ('bbcsport', 'NMFC', 'None', 0, False, 6),\n",
       "       ('bbcsport', 'NMFC', 'None', 0, False, 7),\n",
       "       ('bbcsport', 'NMFC', 'None', 0, False, 8),\n",
       "       ('bbcsport', 'NMFC', 'None', 0, False, 9),\n",
       "       ('bbcsport', 'MVSpectralClustering', 'None', 0, False, 0),\n",
       "       ('bbcsport', 'MVSpectralClustering', 'None', 0, False, 1),\n",
       "       ('bbcsport', 'MVSpectralClustering', 'None', 0, False, 2),\n",
       "       ('bbcsport', 'MVSpectralClustering', 'None', 0, False, 3),\n",
       "       ('bbcsport', 'MVSpectralClustering', 'None', 0, False, 4),\n",
       "       ('bbcsport', 'MVSpectralClustering', 'None', 0, False, 5),\n",
       "       ('bbcsport', 'MVSpectralClustering', 'None', 0, False, 6),\n",
       "       ('bbcsport', 'MVSpectralClustering', 'None', 0, False, 7),\n",
       "       ('bbcsport', 'MVSpectralClustering', 'None', 0, False, 8),\n",
       "       ('bbcsport', 'MVSpectralClustering', 'None', 0, False, 9),\n",
       "       ('bbcsport', 'MVCoRegSpectralClustering', 'None', 0, False, 0),\n",
       "       ('bbcsport', 'MVCoRegSpectralClustering', 'None', 0, False, 1),\n",
       "       ('bbcsport', 'MVCoRegSpectralClustering', 'None', 0, False, 2),\n",
       "       ('bbcsport', 'MVCoRegSpectralClustering', 'None', 0, False, 3),\n",
       "       ('bbcsport', 'MVCoRegSpectralClustering', 'None', 0, False, 4),\n",
       "       ('bbcsport', 'MVCoRegSpectralClustering', 'None', 0, False, 5),\n",
       "       ('bbcsport', 'MVCoRegSpectralClustering', 'None', 0, False, 6),\n",
       "       ('bbcsport', 'MVCoRegSpectralClustering', 'None', 0, False, 7),\n",
       "       ('bbcsport', 'MVCoRegSpectralClustering', 'None', 0, False, 8),\n",
       "       ('bbcsport', 'MVCoRegSpectralClustering', 'None', 0, False, 9),\n",
       "       ('bbcsport', 'GroupPCA', 'None', 0, False, 0),\n",
       "       ('bbcsport', 'GroupPCA', 'None', 0, False, 1),\n",
       "       ('bbcsport', 'GroupPCA', 'None', 0, False, 2),\n",
       "       ('bbcsport', 'GroupPCA', 'None', 0, False, 3),\n",
       "       ('bbcsport', 'GroupPCA', 'None', 0, False, 4),\n",
       "       ('bbcsport', 'GroupPCA', 'None', 0, False, 5),\n",
       "       ('bbcsport', 'GroupPCA', 'None', 0, False, 6),\n",
       "       ('bbcsport', 'GroupPCA', 'None', 0, False, 7),\n",
       "       ('bbcsport', 'GroupPCA', 'None', 0, False, 8),\n",
       "       ('bbcsport', 'GroupPCA', 'None', 0, False, 9),\n",
       "       ('bbcsport', 'AJIVE', 'None', 0, False, 0),\n",
       "       ('bbcsport', 'AJIVE', 'None', 0, False, 1),\n",
       "       ('bbcsport', 'AJIVE', 'None', 0, False, 2),\n",
       "       ('bbcsport', 'AJIVE', 'None', 0, False, 3),\n",
       "       ('bbcsport', 'AJIVE', 'None', 0, False, 4),\n",
       "       ('bbcsport', 'AJIVE', 'None', 0, False, 5),\n",
       "       ('bbcsport', 'AJIVE', 'None', 0, False, 6),\n",
       "       ('bbcsport', 'AJIVE', 'None', 0, False, 7),\n",
       "       ('bbcsport', 'AJIVE', 'None', 0, False, 8),\n",
       "       ('bbcsport', 'AJIVE', 'None', 0, False, 9),\n",
       "       ('bbcsport', 'SNF', 'None', 0, False, 0),\n",
       "       ('bbcsport', 'SNF', 'None', 0, False, 1),\n",
       "       ('bbcsport', 'SNF', 'None', 0, False, 2),\n",
       "       ('bbcsport', 'SNF', 'None', 0, False, 3),\n",
       "       ('bbcsport', 'SNF', 'None', 0, False, 4),\n",
       "       ('bbcsport', 'SNF', 'None', 0, False, 5),\n",
       "       ('bbcsport', 'SNF', 'None', 0, False, 6),\n",
       "       ('bbcsport', 'SNF', 'None', 0, False, 7),\n",
       "       ('bbcsport', 'SNF', 'None', 0, False, 8),\n",
       "       ('bbcsport', 'SNF', 'None', 0, False, 9),\n",
       "       ('bbcsport', 'intNMF', 'None', 0, False, 0),\n",
       "       ('bbcsport', 'intNMF', 'None', 0, False, 1),\n",
       "       ('bbcsport', 'intNMF', 'None', 0, False, 2),\n",
       "       ('bbcsport', 'intNMF', 'None', 0, False, 3),\n",
       "       ('bbcsport', 'intNMF', 'None', 0, False, 4),\n",
       "       ('bbcsport', 'intNMF', 'None', 0, False, 5),\n",
       "       ('bbcsport', 'intNMF', 'None', 0, False, 6),\n",
       "       ('bbcsport', 'intNMF', 'None', 0, False, 7),\n",
       "       ('bbcsport', 'intNMF', 'None', 0, False, 8),\n",
       "       ('bbcsport', 'intNMF', 'None', 0, False, 9),\n",
       "       ('bbcsport', 'jNMF', 'None', 0, False, 0),\n",
       "       ('bbcsport', 'jNMF', 'None', 0, False, 1),\n",
       "       ('bbcsport', 'jNMF', 'None', 0, False, 2),\n",
       "       ('bbcsport', 'jNMF', 'None', 0, False, 3),\n",
       "       ('bbcsport', 'jNMF', 'None', 0, False, 4),\n",
       "       ('bbcsport', 'jNMF', 'None', 0, False, 5),\n",
       "       ('bbcsport', 'jNMF', 'None', 0, False, 6),\n",
       "       ('bbcsport', 'jNMF', 'None', 0, False, 7),\n",
       "       ('bbcsport', 'jNMF', 'None', 0, False, 8),\n",
       "       ('bbcsport', 'jNMF', 'None', 0, False, 9),\n",
       "       ('buaa', 'Concat', 'None', 0, False, 0),\n",
       "       ('buaa', 'Concat', 'None', 0, False, 1),\n",
       "       ('buaa', 'Concat', 'None', 0, False, 2),\n",
       "       ('buaa', 'Concat', 'None', 0, False, 3),\n",
       "       ('buaa', 'Concat', 'None', 0, False, 4),\n",
       "       ('buaa', 'Concat', 'None', 0, False, 5),\n",
       "       ('buaa', 'Concat', 'None', 0, False, 6),\n",
       "       ('buaa', 'Concat', 'None', 0, False, 7),\n",
       "       ('buaa', 'Concat', 'None', 0, False, 8),\n",
       "       ('buaa', 'Concat', 'None', 0, False, 9),\n",
       "       ('buaa', 'NMFC', 'None', 0, False, 0),\n",
       "       ('buaa', 'NMFC', 'None', 0, False, 1),\n",
       "       ('buaa', 'NMFC', 'None', 0, False, 2),\n",
       "       ('buaa', 'NMFC', 'None', 0, False, 3),\n",
       "       ('buaa', 'NMFC', 'None', 0, False, 4),\n",
       "       ('buaa', 'NMFC', 'None', 0, False, 5),\n",
       "       ('buaa', 'NMFC', 'None', 0, False, 6),\n",
       "       ('buaa', 'NMFC', 'None', 0, False, 7),\n",
       "       ('buaa', 'NMFC', 'None', 0, False, 8),\n",
       "       ('buaa', 'NMFC', 'None', 0, False, 9),\n",
       "       ('buaa', 'MVSpectralClustering', 'None', 0, False, 0),\n",
       "       ('buaa', 'MVSpectralClustering', 'None', 0, False, 1),\n",
       "       ('buaa', 'MVSpectralClustering', 'None', 0, False, 2),\n",
       "       ('buaa', 'MVSpectralClustering', 'None', 0, False, 3),\n",
       "       ('buaa', 'MVSpectralClustering', 'None', 0, False, 4),\n",
       "       ('buaa', 'MVSpectralClustering', 'None', 0, False, 5),\n",
       "       ('buaa', 'MVSpectralClustering', 'None', 0, False, 6),\n",
       "       ('buaa', 'MVSpectralClustering', 'None', 0, False, 7),\n",
       "       ('buaa', 'MVSpectralClustering', 'None', 0, False, 8),\n",
       "       ('buaa', 'MVSpectralClustering', 'None', 0, False, 9),\n",
       "       ('buaa', 'MVCoRegSpectralClustering', 'None', 0, False, 0),\n",
       "       ('buaa', 'MVCoRegSpectralClustering', 'None', 0, False, 1),\n",
       "       ('buaa', 'MVCoRegSpectralClustering', 'None', 0, False, 2),\n",
       "       ('buaa', 'MVCoRegSpectralClustering', 'None', 0, False, 3),\n",
       "       ('buaa', 'MVCoRegSpectralClustering', 'None', 0, False, 4),\n",
       "       ('buaa', 'MVCoRegSpectralClustering', 'None', 0, False, 5),\n",
       "       ('buaa', 'MVCoRegSpectralClustering', 'None', 0, False, 6),\n",
       "       ('buaa', 'MVCoRegSpectralClustering', 'None', 0, False, 7),\n",
       "       ('buaa', 'MVCoRegSpectralClustering', 'None', 0, False, 8),\n",
       "       ('buaa', 'MVCoRegSpectralClustering', 'None', 0, False, 9),\n",
       "       ('buaa', 'GroupPCA', 'None', 0, False, 0),\n",
       "       ('buaa', 'GroupPCA', 'None', 0, False, 1),\n",
       "       ('buaa', 'GroupPCA', 'None', 0, False, 2),\n",
       "       ('buaa', 'GroupPCA', 'None', 0, False, 3),\n",
       "       ('buaa', 'GroupPCA', 'None', 0, False, 4),\n",
       "       ('buaa', 'GroupPCA', 'None', 0, False, 5),\n",
       "       ('buaa', 'GroupPCA', 'None', 0, False, 6),\n",
       "       ('buaa', 'GroupPCA', 'None', 0, False, 7),\n",
       "       ('buaa', 'GroupPCA', 'None', 0, False, 8),\n",
       "       ('buaa', 'GroupPCA', 'None', 0, False, 9),\n",
       "       ('buaa', 'AJIVE', 'None', 0, False, 0),\n",
       "       ('buaa', 'AJIVE', 'None', 0, False, 1),\n",
       "       ('buaa', 'AJIVE', 'None', 0, False, 2),\n",
       "       ('buaa', 'AJIVE', 'None', 0, False, 3),\n",
       "       ('buaa', 'AJIVE', 'None', 0, False, 4),\n",
       "       ('buaa', 'AJIVE', 'None', 0, False, 5),\n",
       "       ('buaa', 'AJIVE', 'None', 0, False, 6),\n",
       "       ('buaa', 'AJIVE', 'None', 0, False, 7),\n",
       "       ('buaa', 'AJIVE', 'None', 0, False, 8),\n",
       "       ('buaa', 'AJIVE', 'None', 0, False, 9),\n",
       "       ('buaa', 'SNF', 'None', 0, False, 0),\n",
       "       ('buaa', 'SNF', 'None', 0, False, 1),\n",
       "       ('buaa', 'SNF', 'None', 0, False, 2),\n",
       "       ('buaa', 'SNF', 'None', 0, False, 3),\n",
       "       ('buaa', 'SNF', 'None', 0, False, 4),\n",
       "       ('buaa', 'SNF', 'None', 0, False, 5),\n",
       "       ('buaa', 'SNF', 'None', 0, False, 6),\n",
       "       ('buaa', 'SNF', 'None', 0, False, 7),\n",
       "       ('buaa', 'SNF', 'None', 0, False, 8),\n",
       "       ('buaa', 'SNF', 'None', 0, False, 9),\n",
       "       ('buaa', 'intNMF', 'None', 0, False, 0),\n",
       "       ('buaa', 'intNMF', 'None', 0, False, 1),\n",
       "       ('buaa', 'intNMF', 'None', 0, False, 2),\n",
       "       ('buaa', 'intNMF', 'None', 0, False, 3),\n",
       "       ('buaa', 'intNMF', 'None', 0, False, 4),\n",
       "       ('buaa', 'intNMF', 'None', 0, False, 5),\n",
       "       ('buaa', 'intNMF', 'None', 0, False, 6),\n",
       "       ('buaa', 'intNMF', 'None', 0, False, 7),\n",
       "       ('buaa', 'intNMF', 'None', 0, False, 8),\n",
       "       ('buaa', 'intNMF', 'None', 0, False, 9),\n",
       "       ('buaa', 'jNMF', 'None', 0, False, 0),\n",
       "       ('buaa', 'jNMF', 'None', 0, False, 1),\n",
       "       ('buaa', 'jNMF', 'None', 0, False, 2),\n",
       "       ('buaa', 'jNMF', 'None', 0, False, 3),\n",
       "       ('buaa', 'jNMF', 'None', 0, False, 4),\n",
       "       ('buaa', 'jNMF', 'None', 0, False, 5),\n",
       "       ('buaa', 'jNMF', 'None', 0, False, 6),\n",
       "       ('buaa', 'jNMF', 'None', 0, False, 7),\n",
       "       ('buaa', 'jNMF', 'None', 0, False, 8),\n",
       "       ('buaa', 'jNMF', 'None', 0, False, 9),\n",
       "       ('metabric', 'Concat', 'None', 0, False, 0),\n",
       "       ('metabric', 'Concat', 'None', 0, False, 1),\n",
       "       ('metabric', 'Concat', 'None', 0, False, 2),\n",
       "       ('metabric', 'Concat', 'None', 0, False, 3),\n",
       "       ('metabric', 'Concat', 'None', 0, False, 4),\n",
       "       ('metabric', 'Concat', 'None', 0, False, 5),\n",
       "       ('metabric', 'Concat', 'None', 0, False, 6),\n",
       "       ('metabric', 'Concat', 'None', 0, False, 7),\n",
       "       ('metabric', 'Concat', 'None', 0, False, 8),\n",
       "       ('metabric', 'Concat', 'None', 0, False, 9),\n",
       "       ('metabric', 'NMFC', 'None', 0, False, 0),\n",
       "       ('metabric', 'NMFC', 'None', 0, False, 1),\n",
       "       ('metabric', 'NMFC', 'None', 0, False, 2),\n",
       "       ('metabric', 'NMFC', 'None', 0, False, 3),\n",
       "       ('metabric', 'NMFC', 'None', 0, False, 4),\n",
       "       ('metabric', 'NMFC', 'None', 0, False, 5),\n",
       "       ('metabric', 'NMFC', 'None', 0, False, 6),\n",
       "       ('metabric', 'NMFC', 'None', 0, False, 7),\n",
       "       ('metabric', 'NMFC', 'None', 0, False, 8),\n",
       "       ('metabric', 'NMFC', 'None', 0, False, 9),\n",
       "       ('metabric', 'MVSpectralClustering', 'None', 0, False, 0),\n",
       "       ('metabric', 'MVSpectralClustering', 'None', 0, False, 1),\n",
       "       ('metabric', 'MVSpectralClustering', 'None', 0, False, 2),\n",
       "       ('metabric', 'MVSpectralClustering', 'None', 0, False, 3),\n",
       "       ('metabric', 'MVSpectralClustering', 'None', 0, False, 4),\n",
       "       ('metabric', 'MVSpectralClustering', 'None', 0, False, 5),\n",
       "       ('metabric', 'MVSpectralClustering', 'None', 0, False, 6),\n",
       "       ('metabric', 'MVSpectralClustering', 'None', 0, False, 7),\n",
       "       ('metabric', 'MVSpectralClustering', 'None', 0, False, 8),\n",
       "       ('metabric', 'MVSpectralClustering', 'None', 0, False, 9),\n",
       "       ('metabric', 'MVCoRegSpectralClustering', 'None', 0, False, 0),\n",
       "       ('metabric', 'MVCoRegSpectralClustering', 'None', 0, False, 1),\n",
       "       ('metabric', 'MVCoRegSpectralClustering', 'None', 0, False, 2),\n",
       "       ('metabric', 'MVCoRegSpectralClustering', 'None', 0, False, 3),\n",
       "       ('metabric', 'MVCoRegSpectralClustering', 'None', 0, False, 4),\n",
       "       ('metabric', 'MVCoRegSpectralClustering', 'None', 0, False, 5),\n",
       "       ('metabric', 'MVCoRegSpectralClustering', 'None', 0, False, 6),\n",
       "       ('metabric', 'MVCoRegSpectralClustering', 'None', 0, False, 7),\n",
       "       ('metabric', 'MVCoRegSpectralClustering', 'None', 0, False, 8),\n",
       "       ('metabric', 'MVCoRegSpectralClustering', 'None', 0, False, 9),\n",
       "       ('metabric', 'GroupPCA', 'None', 0, False, 0),\n",
       "       ('metabric', 'GroupPCA', 'None', 0, False, 1),\n",
       "       ('metabric', 'GroupPCA', 'None', 0, False, 2),\n",
       "       ('metabric', 'GroupPCA', 'None', 0, False, 3),\n",
       "       ('metabric', 'GroupPCA', 'None', 0, False, 4),\n",
       "       ('metabric', 'GroupPCA', 'None', 0, False, 5),\n",
       "       ('metabric', 'GroupPCA', 'None', 0, False, 6),\n",
       "       ('metabric', 'GroupPCA', 'None', 0, False, 7),\n",
       "       ('metabric', 'GroupPCA', 'None', 0, False, 8),\n",
       "       ('metabric', 'GroupPCA', 'None', 0, False, 9),\n",
       "       ('metabric', 'AJIVE', 'None', 0, False, 0),\n",
       "       ('metabric', 'AJIVE', 'None', 0, False, 1),\n",
       "       ('metabric', 'AJIVE', 'None', 0, False, 2),\n",
       "       ('metabric', 'AJIVE', 'None', 0, False, 3),\n",
       "       ('metabric', 'AJIVE', 'None', 0, False, 4),\n",
       "       ('metabric', 'AJIVE', 'None', 0, False, 5),\n",
       "       ('metabric', 'AJIVE', 'None', 0, False, 6),\n",
       "       ('metabric', 'AJIVE', 'None', 0, False, 7),\n",
       "       ('metabric', 'AJIVE', 'None', 0, False, 8),\n",
       "       ('metabric', 'AJIVE', 'None', 0, False, 9),\n",
       "       ('metabric', 'SNF', 'None', 0, False, 0),\n",
       "       ('metabric', 'SNF', 'None', 0, False, 1),\n",
       "       ('metabric', 'SNF', 'None', 0, False, 2),\n",
       "       ('metabric', 'SNF', 'None', 0, False, 3),\n",
       "       ('metabric', 'SNF', 'None', 0, False, 4),\n",
       "       ('metabric', 'SNF', 'None', 0, False, 5),\n",
       "       ('metabric', 'SNF', 'None', 0, False, 6),\n",
       "       ('metabric', 'SNF', 'None', 0, False, 7),\n",
       "       ('metabric', 'SNF', 'None', 0, False, 8),\n",
       "       ('metabric', 'SNF', 'None', 0, False, 9),\n",
       "       ('metabric', 'intNMF', 'None', 0, False, 0),\n",
       "       ('metabric', 'intNMF', 'None', 0, False, 1),\n",
       "       ('metabric', 'intNMF', 'None', 0, False, 2),\n",
       "       ('metabric', 'intNMF', 'None', 0, False, 3),\n",
       "       ('metabric', 'intNMF', 'None', 0, False, 4),\n",
       "       ('metabric', 'intNMF', 'None', 0, False, 5),\n",
       "       ('metabric', 'intNMF', 'None', 0, False, 6),\n",
       "       ('metabric', 'intNMF', 'None', 0, False, 7),\n",
       "       ('metabric', 'intNMF', 'None', 0, False, 8),\n",
       "       ('metabric', 'intNMF', 'None', 0, False, 9),\n",
       "       ('metabric', 'jNMF', 'None', 0, False, 0),\n",
       "       ('metabric', 'jNMF', 'None', 0, False, 1),\n",
       "       ('metabric', 'jNMF', 'None', 0, False, 2),\n",
       "       ('metabric', 'jNMF', 'None', 0, False, 3),\n",
       "       ('metabric', 'jNMF', 'None', 0, False, 4),\n",
       "       ('metabric', 'jNMF', 'None', 0, False, 5),\n",
       "       ('metabric', 'jNMF', 'None', 0, False, 6),\n",
       "       ('metabric', 'jNMF', 'None', 0, False, 7),\n",
       "       ('metabric', 'jNMF', 'None', 0, False, 8),\n",
       "       ('metabric', 'jNMF', 'None', 0, False, 9),\n",
       "       ('digits', 'Concat', 'None', 0, False, 0),\n",
       "       ('digits', 'Concat', 'None', 0, False, 1),\n",
       "       ('digits', 'Concat', 'None', 0, False, 2),\n",
       "       ('digits', 'Concat', 'None', 0, False, 3),\n",
       "       ('digits', 'Concat', 'None', 0, False, 4),\n",
       "       ('digits', 'Concat', 'None', 0, False, 5),\n",
       "       ('digits', 'Concat', 'None', 0, False, 6),\n",
       "       ('digits', 'Concat', 'None', 0, False, 7),\n",
       "       ('digits', 'Concat', 'None', 0, False, 8),\n",
       "       ('digits', 'Concat', 'None', 0, False, 9),\n",
       "       ('digits', 'NMFC', 'None', 0, False, 0),\n",
       "       ('digits', 'NMFC', 'None', 0, False, 1),\n",
       "       ('digits', 'NMFC', 'None', 0, False, 2),\n",
       "       ('digits', 'NMFC', 'None', 0, False, 3),\n",
       "       ('digits', 'NMFC', 'None', 0, False, 4),\n",
       "       ('digits', 'NMFC', 'None', 0, False, 5),\n",
       "       ('digits', 'NMFC', 'None', 0, False, 6),\n",
       "       ('digits', 'NMFC', 'None', 0, False, 7),\n",
       "       ('digits', 'NMFC', 'None', 0, False, 8),\n",
       "       ('digits', 'NMFC', 'None', 0, False, 9),\n",
       "       ('digits', 'MVSpectralClustering', 'None', 0, False, 0),\n",
       "       ('digits', 'MVSpectralClustering', 'None', 0, False, 1),\n",
       "       ('digits', 'MVSpectralClustering', 'None', 0, False, 2),\n",
       "       ('digits', 'MVSpectralClustering', 'None', 0, False, 3),\n",
       "       ('digits', 'MVSpectralClustering', 'None', 0, False, 4),\n",
       "       ('digits', 'MVSpectralClustering', 'None', 0, False, 5),\n",
       "       ('digits', 'MVSpectralClustering', 'None', 0, False, 6),\n",
       "       ('digits', 'MVSpectralClustering', 'None', 0, False, 7),\n",
       "       ('digits', 'MVSpectralClustering', 'None', 0, False, 8),\n",
       "       ('digits', 'MVSpectralClustering', 'None', 0, False, 9),\n",
       "       ('digits', 'MVCoRegSpectralClustering', 'None', 0, False, 0),\n",
       "       ('digits', 'MVCoRegSpectralClustering', 'None', 0, False, 1),\n",
       "       ('digits', 'MVCoRegSpectralClustering', 'None', 0, False, 2),\n",
       "       ('digits', 'MVCoRegSpectralClustering', 'None', 0, False, 3),\n",
       "       ('digits', 'MVCoRegSpectralClustering', 'None', 0, False, 4),\n",
       "       ('digits', 'MVCoRegSpectralClustering', 'None', 0, False, 5),\n",
       "       ('digits', 'MVCoRegSpectralClustering', 'None', 0, False, 6),\n",
       "       ('digits', 'MVCoRegSpectralClustering', 'None', 0, False, 7),\n",
       "       ('digits', 'MVCoRegSpectralClustering', 'None', 0, False, 8),\n",
       "       ('digits', 'MVCoRegSpectralClustering', 'None', 0, False, 9),\n",
       "       ('digits', 'GroupPCA', 'None', 0, False, 0),\n",
       "       ('digits', 'GroupPCA', 'None', 0, False, 1),\n",
       "       ('digits', 'GroupPCA', 'None', 0, False, 2),\n",
       "       ('digits', 'GroupPCA', 'None', 0, False, 3),\n",
       "       ('digits', 'GroupPCA', 'None', 0, False, 4),\n",
       "       ('digits', 'GroupPCA', 'None', 0, False, 5),\n",
       "       ('digits', 'GroupPCA', 'None', 0, False, 6),\n",
       "       ('digits', 'GroupPCA', 'None', 0, False, 7),\n",
       "       ('digits', 'GroupPCA', 'None', 0, False, 8),\n",
       "       ('digits', 'GroupPCA', 'None', 0, False, 9),\n",
       "       ('digits', 'AJIVE', 'None', 0, False, 0),\n",
       "       ('digits', 'AJIVE', 'None', 0, False, 1),\n",
       "       ('digits', 'AJIVE', 'None', 0, False, 2),\n",
       "       ('digits', 'AJIVE', 'None', 0, False, 3),\n",
       "       ('digits', 'AJIVE', 'None', 0, False, 4),\n",
       "       ('digits', 'AJIVE', 'None', 0, False, 5),\n",
       "       ('digits', 'AJIVE', 'None', 0, False, 6),\n",
       "       ('digits', 'AJIVE', 'None', 0, False, 7),\n",
       "       ('digits', 'AJIVE', 'None', 0, False, 8),\n",
       "       ('digits', 'AJIVE', 'None', 0, False, 9),\n",
       "       ('digits', 'SNF', 'None', 0, False, 0),\n",
       "       ('digits', 'SNF', 'None', 0, False, 1),\n",
       "       ('digits', 'SNF', 'None', 0, False, 2),\n",
       "       ('digits', 'SNF', 'None', 0, False, 3),\n",
       "       ('digits', 'SNF', 'None', 0, False, 4),\n",
       "       ('digits', 'SNF', 'None', 0, False, 5),\n",
       "       ('digits', 'SNF', 'None', 0, False, 6),\n",
       "       ('digits', 'SNF', 'None', 0, False, 7),\n",
       "       ('digits', 'SNF', 'None', 0, False, 8),\n",
       "       ('digits', 'SNF', 'None', 0, False, 9),\n",
       "       ('digits', 'intNMF', 'None', 0, False, 0),\n",
       "       ('digits', 'intNMF', 'None', 0, False, 1),\n",
       "       ('digits', 'intNMF', 'None', 0, False, 2),\n",
       "       ('digits', 'intNMF', 'None', 0, False, 3),\n",
       "       ('digits', 'intNMF', 'None', 0, False, 4),\n",
       "       ('digits', 'intNMF', 'None', 0, False, 5),\n",
       "       ('digits', 'intNMF', 'None', 0, False, 6),\n",
       "       ('digits', 'intNMF', 'None', 0, False, 7),\n",
       "       ('digits', 'intNMF', 'None', 0, False, 8),\n",
       "       ('digits', 'intNMF', 'None', 0, False, 9),\n",
       "       ('digits', 'jNMF', 'None', 0, False, 0),\n",
       "       ('digits', 'jNMF', 'None', 0, False, 1),\n",
       "       ('digits', 'jNMF', 'None', 0, False, 2),\n",
       "       ('digits', 'jNMF', 'None', 0, False, 3),\n",
       "       ('digits', 'jNMF', 'None', 0, False, 4),\n",
       "       ('digits', 'jNMF', 'None', 0, False, 5),\n",
       "       ('digits', 'jNMF', 'None', 0, False, 6),\n",
       "       ('digits', 'jNMF', 'None', 0, False, 7),\n",
       "       ('digits', 'jNMF', 'None', 0, False, 8),\n",
       "       ('digits', 'jNMF', 'None', 0, False, 9),\n",
       "       ('bdgp', 'Concat', 'None', 0, False, 0),\n",
       "       ('bdgp', 'Concat', 'None', 0, False, 1),\n",
       "       ('bdgp', 'Concat', 'None', 0, False, 2),\n",
       "       ('bdgp', 'Concat', 'None', 0, False, 3),\n",
       "       ('bdgp', 'Concat', 'None', 0, False, 4),\n",
       "       ('bdgp', 'Concat', 'None', 0, False, 5),\n",
       "       ('bdgp', 'Concat', 'None', 0, False, 6),\n",
       "       ('bdgp', 'Concat', 'None', 0, False, 7),\n",
       "       ('bdgp', 'Concat', 'None', 0, False, 8),\n",
       "       ('bdgp', 'Concat', 'None', 0, False, 9),\n",
       "       ('bdgp', 'NMFC', 'None', 0, False, 0),\n",
       "       ('bdgp', 'NMFC', 'None', 0, False, 1),\n",
       "       ('bdgp', 'NMFC', 'None', 0, False, 2),\n",
       "       ('bdgp', 'NMFC', 'None', 0, False, 3),\n",
       "       ('bdgp', 'NMFC', 'None', 0, False, 4),\n",
       "       ('bdgp', 'NMFC', 'None', 0, False, 5),\n",
       "       ('bdgp', 'NMFC', 'None', 0, False, 6),\n",
       "       ('bdgp', 'NMFC', 'None', 0, False, 7),\n",
       "       ('bdgp', 'NMFC', 'None', 0, False, 8),\n",
       "       ('bdgp', 'NMFC', 'None', 0, False, 9),\n",
       "       ('bdgp', 'MVSpectralClustering', 'None', 0, False, 0),\n",
       "       ('bdgp', 'MVSpectralClustering', 'None', 0, False, 1),\n",
       "       ('bdgp', 'MVSpectralClustering', 'None', 0, False, 2),\n",
       "       ('bdgp', 'MVSpectralClustering', 'None', 0, False, 3),\n",
       "       ('bdgp', 'MVSpectralClustering', 'None', 0, False, 4),\n",
       "       ('bdgp', 'MVSpectralClustering', 'None', 0, False, 5),\n",
       "       ('bdgp', 'MVSpectralClustering', 'None', 0, False, 6),\n",
       "       ('bdgp', 'MVSpectralClustering', 'None', 0, False, 7),\n",
       "       ('bdgp', 'MVSpectralClustering', 'None', 0, False, 8),\n",
       "       ('bdgp', 'MVSpectralClustering', 'None', 0, False, 9),\n",
       "       ('bdgp', 'MVCoRegSpectralClustering', 'None', 0, False, 0),\n",
       "       ('bdgp', 'MVCoRegSpectralClustering', 'None', 0, False, 1),\n",
       "       ('bdgp', 'MVCoRegSpectralClustering', 'None', 0, False, 2),\n",
       "       ('bdgp', 'MVCoRegSpectralClustering', 'None', 0, False, 3),\n",
       "       ('bdgp', 'MVCoRegSpectralClustering', 'None', 0, False, 4),\n",
       "       ('bdgp', 'MVCoRegSpectralClustering', 'None', 0, False, 5),\n",
       "       ('bdgp', 'MVCoRegSpectralClustering', 'None', 0, False, 6),\n",
       "       ('bdgp', 'MVCoRegSpectralClustering', 'None', 0, False, 7),\n",
       "       ('bdgp', 'MVCoRegSpectralClustering', 'None', 0, False, 8),\n",
       "       ('bdgp', 'MVCoRegSpectralClustering', 'None', 0, False, 9),\n",
       "       ('bdgp', 'GroupPCA', 'None', 0, False, 0),\n",
       "       ('bdgp', 'GroupPCA', 'None', 0, False, 1),\n",
       "       ('bdgp', 'GroupPCA', 'None', 0, False, 2),\n",
       "       ('bdgp', 'GroupPCA', 'None', 0, False, 3),\n",
       "       ('bdgp', 'GroupPCA', 'None', 0, False, 4),\n",
       "       ('bdgp', 'GroupPCA', 'None', 0, False, 5),\n",
       "       ('bdgp', 'GroupPCA', 'None', 0, False, 6),\n",
       "       ('bdgp', 'GroupPCA', 'None', 0, False, 7),\n",
       "       ('bdgp', 'GroupPCA', 'None', 0, False, 8),\n",
       "       ('bdgp', 'GroupPCA', 'None', 0, False, 9),\n",
       "       ('bdgp', 'AJIVE', 'None', 0, False, 0),\n",
       "       ('bdgp', 'AJIVE', 'None', 0, False, 1),\n",
       "       ('bdgp', 'AJIVE', 'None', 0, False, 2),\n",
       "       ('bdgp', 'AJIVE', 'None', 0, False, 3),\n",
       "       ('bdgp', 'AJIVE', 'None', 0, False, 4),\n",
       "       ('bdgp', 'AJIVE', 'None', 0, False, 5),\n",
       "       ('bdgp', 'AJIVE', 'None', 0, False, 6),\n",
       "       ('bdgp', 'AJIVE', 'None', 0, False, 7),\n",
       "       ('bdgp', 'AJIVE', 'None', 0, False, 8),\n",
       "       ('bdgp', 'AJIVE', 'None', 0, False, 9),\n",
       "       ('bdgp', 'SNF', 'None', 0, False, 0),\n",
       "       ('bdgp', 'SNF', 'None', 0, False, 1),\n",
       "       ('bdgp', 'SNF', 'None', 0, False, 2),\n",
       "       ('bdgp', 'SNF', 'None', 0, False, 3),\n",
       "       ('bdgp', 'SNF', 'None', 0, False, 4),\n",
       "       ('bdgp', 'SNF', 'None', 0, False, 5),\n",
       "       ('bdgp', 'SNF', 'None', 0, False, 6),\n",
       "       ('bdgp', 'SNF', 'None', 0, False, 7),\n",
       "       ('bdgp', 'SNF', 'None', 0, False, 8),\n",
       "       ('bdgp', 'SNF', 'None', 0, False, 9),\n",
       "       ('bdgp', 'intNMF', 'None', 0, False, 0),\n",
       "       ('bdgp', 'intNMF', 'None', 0, False, 1),\n",
       "       ('bdgp', 'intNMF', 'None', 0, False, 2),\n",
       "       ('bdgp', 'intNMF', 'None', 0, False, 3),\n",
       "       ('bdgp', 'intNMF', 'None', 0, False, 4),\n",
       "       ('bdgp', 'intNMF', 'None', 0, False, 5),\n",
       "       ('bdgp', 'intNMF', 'None', 0, False, 6),\n",
       "       ('bdgp', 'intNMF', 'None', 0, False, 7),\n",
       "       ('bdgp', 'intNMF', 'None', 0, False, 8),\n",
       "       ('bdgp', 'intNMF', 'None', 0, False, 9),\n",
       "       ('bdgp', 'jNMF', 'None', 0, False, 0),\n",
       "       ('bdgp', 'jNMF', 'None', 0, False, 1),\n",
       "       ('bdgp', 'jNMF', 'None', 0, False, 2),\n",
       "       ('bdgp', 'jNMF', 'None', 0, False, 3),\n",
       "       ('bdgp', 'jNMF', 'None', 0, False, 4),\n",
       "       ('bdgp', 'jNMF', 'None', 0, False, 5),\n",
       "       ('bdgp', 'jNMF', 'None', 0, False, 6),\n",
       "       ('bdgp', 'jNMF', 'None', 0, False, 7),\n",
       "       ('bdgp', 'jNMF', 'None', 0, False, 8),\n",
       "       ('bdgp', 'jNMF', 'None', 0, False, 9),\n",
       "       ('tcga', 'Concat', 'None', 0, False, 0),\n",
       "       ('tcga', 'Concat', 'None', 0, False, 1),\n",
       "       ('tcga', 'Concat', 'None', 0, False, 2),\n",
       "       ('tcga', 'Concat', 'None', 0, False, 3),\n",
       "       ('tcga', 'Concat', 'None', 0, False, 4),\n",
       "       ('tcga', 'Concat', 'None', 0, False, 5),\n",
       "       ('tcga', 'Concat', 'None', 0, False, 6),\n",
       "       ('tcga', 'Concat', 'None', 0, False, 7),\n",
       "       ('tcga', 'Concat', 'None', 0, False, 8),\n",
       "       ('tcga', 'Concat', 'None', 0, False, 9),\n",
       "       ('tcga', 'NMFC', 'None', 0, False, 0),\n",
       "       ('tcga', 'NMFC', 'None', 0, False, 1),\n",
       "       ('tcga', 'NMFC', 'None', 0, False, 2),\n",
       "       ('tcga', 'NMFC', 'None', 0, False, 3),\n",
       "       ('tcga', 'NMFC', 'None', 0, False, 4),\n",
       "       ('tcga', 'NMFC', 'None', 0, False, 5),\n",
       "       ('tcga', 'NMFC', 'None', 0, False, 6),\n",
       "       ('tcga', 'NMFC', 'None', 0, False, 7),\n",
       "       ('tcga', 'NMFC', 'None', 0, False, 8),\n",
       "       ('tcga', 'NMFC', 'None', 0, False, 9),\n",
       "       ('tcga', 'MVSpectralClustering', 'None', 0, False, 0),\n",
       "       ('tcga', 'MVSpectralClustering', 'None', 0, False, 1),\n",
       "       ('tcga', 'MVSpectralClustering', 'None', 0, False, 2),\n",
       "       ('tcga', 'MVSpectralClustering', 'None', 0, False, 3),\n",
       "       ('tcga', 'MVSpectralClustering', 'None', 0, False, 4),\n",
       "       ('tcga', 'MVSpectralClustering', 'None', 0, False, 5),\n",
       "       ('tcga', 'MVSpectralClustering', 'None', 0, False, 6),\n",
       "       ('tcga', 'MVSpectralClustering', 'None', 0, False, 7),\n",
       "       ('tcga', 'MVSpectralClustering', 'None', 0, False, 8),\n",
       "       ('tcga', 'MVSpectralClustering', 'None', 0, False, 9),\n",
       "       ('tcga', 'MVCoRegSpectralClustering', 'None', 0, False, 0),\n",
       "       ('tcga', 'MVCoRegSpectralClustering', 'None', 0, False, 1),\n",
       "       ('tcga', 'MVCoRegSpectralClustering', 'None', 0, False, 2),\n",
       "       ('tcga', 'MVCoRegSpectralClustering', 'None', 0, False, 3),\n",
       "       ('tcga', 'MVCoRegSpectralClustering', 'None', 0, False, 4),\n",
       "       ('tcga', 'MVCoRegSpectralClustering', 'None', 0, False, 5),\n",
       "       ('tcga', 'MVCoRegSpectralClustering', 'None', 0, False, 6),\n",
       "       ('tcga', 'MVCoRegSpectralClustering', 'None', 0, False, 7),\n",
       "       ('tcga', 'MVCoRegSpectralClustering', 'None', 0, False, 8),\n",
       "       ('tcga', 'MVCoRegSpectralClustering', 'None', 0, False, 9),\n",
       "       ('tcga', 'GroupPCA', 'None', 0, False, 0),\n",
       "       ('tcga', 'GroupPCA', 'None', 0, False, 1),\n",
       "       ('tcga', 'GroupPCA', 'None', 0, False, 2),\n",
       "       ('tcga', 'GroupPCA', 'None', 0, False, 3),\n",
       "       ('tcga', 'GroupPCA', 'None', 0, False, 4),\n",
       "       ('tcga', 'GroupPCA', 'None', 0, False, 5),\n",
       "       ('tcga', 'GroupPCA', 'None', 0, False, 6),\n",
       "       ('tcga', 'GroupPCA', 'None', 0, False, 7),\n",
       "       ('tcga', 'GroupPCA', 'None', 0, False, 8),\n",
       "       ('tcga', 'GroupPCA', 'None', 0, False, 9),\n",
       "       ('tcga', 'AJIVE', 'None', 0, False, 0),\n",
       "       ('tcga', 'AJIVE', 'None', 0, False, 1),\n",
       "       ('tcga', 'AJIVE', 'None', 0, False, 2),\n",
       "       ('tcga', 'AJIVE', 'None', 0, False, 3),\n",
       "       ('tcga', 'AJIVE', 'None', 0, False, 4),\n",
       "       ('tcga', 'AJIVE', 'None', 0, False, 5),\n",
       "       ('tcga', 'AJIVE', 'None', 0, False, 6),\n",
       "       ('tcga', 'AJIVE', 'None', 0, False, 7),\n",
       "       ('tcga', 'AJIVE', 'None', 0, False, 8),\n",
       "       ('tcga', 'AJIVE', 'None', 0, False, 9),\n",
       "       ('tcga', 'SNF', 'None', 0, False, 0),\n",
       "       ('tcga', 'SNF', 'None', 0, False, 1),\n",
       "       ('tcga', 'SNF', 'None', 0, False, 2),\n",
       "       ('tcga', 'SNF', 'None', 0, False, 3),\n",
       "       ('tcga', 'SNF', 'None', 0, False, 4),\n",
       "       ('tcga', 'SNF', 'None', 0, False, 5),\n",
       "       ('tcga', 'SNF', 'None', 0, False, 6),\n",
       "       ('tcga', 'SNF', 'None', 0, False, 7),\n",
       "       ('tcga', 'SNF', 'None', 0, False, 8),\n",
       "       ('tcga', 'SNF', 'None', 0, False, 9),\n",
       "       ('tcga', 'intNMF', 'None', 0, False, 0),\n",
       "       ('tcga', 'intNMF', 'None', 0, False, 1),\n",
       "       ('tcga', 'intNMF', 'None', 0, False, 2),\n",
       "       ('tcga', 'intNMF', 'None', 0, False, 3),\n",
       "       ('tcga', 'intNMF', 'None', 0, False, 4),\n",
       "       ('tcga', 'intNMF', 'None', 0, False, 5),\n",
       "       ('tcga', 'intNMF', 'None', 0, False, 6),\n",
       "       ('tcga', 'intNMF', 'None', 0, False, 7),\n",
       "       ('tcga', 'intNMF', 'None', 0, False, 8),\n",
       "       ('tcga', 'intNMF', 'None', 0, False, 9),\n",
       "       ('tcga', 'jNMF', 'None', 0, False, 0),\n",
       "       ('tcga', 'jNMF', 'None', 0, False, 1),\n",
       "       ('tcga', 'jNMF', 'None', 0, False, 2),\n",
       "       ('tcga', 'jNMF', 'None', 0, False, 3),\n",
       "       ('tcga', 'jNMF', 'None', 0, False, 4),\n",
       "       ('tcga', 'jNMF', 'None', 0, False, 5),\n",
       "       ('tcga', 'jNMF', 'None', 0, False, 6),\n",
       "       ('tcga', 'jNMF', 'None', 0, False, 7),\n",
       "       ('tcga', 'jNMF', 'None', 0, False, 8),\n",
       "       ('tcga', 'jNMF', 'None', 0, False, 9),\n",
       "       ('caltech101', 'Concat', 'None', 0, False, 0),\n",
       "       ('caltech101', 'Concat', 'None', 0, False, 1),\n",
       "       ('caltech101', 'Concat', 'None', 0, False, 2),\n",
       "       ('caltech101', 'Concat', 'None', 0, False, 3),\n",
       "       ('caltech101', 'Concat', 'None', 0, False, 4),\n",
       "       ('caltech101', 'Concat', 'None', 0, False, 5),\n",
       "       ('caltech101', 'Concat', 'None', 0, False, 6),\n",
       "       ('caltech101', 'Concat', 'None', 0, False, 7),\n",
       "       ('caltech101', 'Concat', 'None', 0, False, 8),\n",
       "       ('caltech101', 'Concat', 'None', 0, False, 9),\n",
       "       ('caltech101', 'NMFC', 'None', 0, False, 0),\n",
       "       ('caltech101', 'NMFC', 'None', 0, False, 1),\n",
       "       ('caltech101', 'NMFC', 'None', 0, False, 2),\n",
       "       ('caltech101', 'NMFC', 'None', 0, False, 3),\n",
       "       ('caltech101', 'NMFC', 'None', 0, False, 4),\n",
       "       ('caltech101', 'NMFC', 'None', 0, False, 5),\n",
       "       ('caltech101', 'NMFC', 'None', 0, False, 6),\n",
       "       ('caltech101', 'NMFC', 'None', 0, False, 7),\n",
       "       ('caltech101', 'NMFC', 'None', 0, False, 8),\n",
       "       ('caltech101', 'NMFC', 'None', 0, False, 9),\n",
       "       ('caltech101', 'MVSpectralClustering', 'None', 0, False, 0),\n",
       "       ('caltech101', 'MVSpectralClustering', 'None', 0, False, 1),\n",
       "       ('caltech101', 'MVSpectralClustering', 'None', 0, False, 2),\n",
       "       ('caltech101', 'MVSpectralClustering', 'None', 0, False, 3),\n",
       "       ('caltech101', 'MVSpectralClustering', 'None', 0, False, 4),\n",
       "       ('caltech101', 'MVSpectralClustering', 'None', 0, False, 5),\n",
       "       ('caltech101', 'MVSpectralClustering', 'None', 0, False, 6),\n",
       "       ('caltech101', 'MVSpectralClustering', 'None', 0, False, 7),\n",
       "       ('caltech101', 'MVSpectralClustering', 'None', 0, False, 8),\n",
       "       ('caltech101', 'MVSpectralClustering', 'None', 0, False, 9),\n",
       "       ('caltech101', 'MVCoRegSpectralClustering', 'None', 0, False, 0),\n",
       "       ('caltech101', 'MVCoRegSpectralClustering', 'None', 0, False, 1),\n",
       "       ('caltech101', 'MVCoRegSpectralClustering', 'None', 0, False, 2),\n",
       "       ('caltech101', 'MVCoRegSpectralClustering', 'None', 0, False, 3),\n",
       "       ('caltech101', 'MVCoRegSpectralClustering', 'None', 0, False, 4),\n",
       "       ('caltech101', 'MVCoRegSpectralClustering', 'None', 0, False, 5),\n",
       "       ('caltech101', 'MVCoRegSpectralClustering', 'None', 0, False, 6),\n",
       "       ('caltech101', 'MVCoRegSpectralClustering', 'None', 0, False, 7),\n",
       "       ('caltech101', 'MVCoRegSpectralClustering', 'None', 0, False, 8),\n",
       "       ('caltech101', 'MVCoRegSpectralClustering', 'None', 0, False, 9),\n",
       "       ('caltech101', 'GroupPCA', 'None', 0, False, 0),\n",
       "       ('caltech101', 'GroupPCA', 'None', 0, False, 1),\n",
       "       ('caltech101', 'GroupPCA', 'None', 0, False, 2),\n",
       "       ('caltech101', 'GroupPCA', 'None', 0, False, 3),\n",
       "       ('caltech101', 'GroupPCA', 'None', 0, False, 4),\n",
       "       ('caltech101', 'GroupPCA', 'None', 0, False, 5),\n",
       "       ('caltech101', 'GroupPCA', 'None', 0, False, 6),\n",
       "       ('caltech101', 'GroupPCA', 'None', 0, False, 7),\n",
       "       ('caltech101', 'GroupPCA', 'None', 0, False, 8),\n",
       "       ('caltech101', 'GroupPCA', 'None', 0, False, 9),\n",
       "       ('caltech101', 'AJIVE', 'None', 0, False, 0),\n",
       "       ('caltech101', 'AJIVE', 'None', 0, False, 1),\n",
       "       ('caltech101', 'AJIVE', 'None', 0, False, 2),\n",
       "       ('caltech101', 'AJIVE', 'None', 0, False, 3),\n",
       "       ('caltech101', 'AJIVE', 'None', 0, False, 4),\n",
       "       ('caltech101', 'AJIVE', 'None', 0, False, 5),\n",
       "       ('caltech101', 'AJIVE', 'None', 0, False, 6),\n",
       "       ('caltech101', 'AJIVE', 'None', 0, False, 7),\n",
       "       ('caltech101', 'AJIVE', 'None', 0, False, 8),\n",
       "       ('caltech101', 'AJIVE', 'None', 0, False, 9),\n",
       "       ('caltech101', 'SNF', 'None', 0, False, 0),\n",
       "       ('caltech101', 'SNF', 'None', 0, False, 1),\n",
       "       ('caltech101', 'SNF', 'None', 0, False, 2),\n",
       "       ('caltech101', 'SNF', 'None', 0, False, 3),\n",
       "       ('caltech101', 'SNF', 'None', 0, False, 4),\n",
       "       ('caltech101', 'SNF', 'None', 0, False, 5),\n",
       "       ('caltech101', 'SNF', 'None', 0, False, 6),\n",
       "       ('caltech101', 'SNF', 'None', 0, False, 7),\n",
       "       ('caltech101', 'SNF', 'None', 0, False, 8),\n",
       "       ('caltech101', 'SNF', 'None', 0, False, 9),\n",
       "       ('caltech101', 'intNMF', 'None', 0, False, 0),\n",
       "       ('caltech101', 'intNMF', 'None', 0, False, 1),\n",
       "       ('caltech101', 'intNMF', 'None', 0, False, 2),\n",
       "       ('caltech101', 'intNMF', 'None', 0, False, 3),\n",
       "       ('caltech101', 'intNMF', 'None', 0, False, 4),\n",
       "       ('caltech101', 'intNMF', 'None', 0, False, 5),\n",
       "       ('caltech101', 'intNMF', 'None', 0, False, 6),\n",
       "       ('caltech101', 'intNMF', 'None', 0, False, 7),\n",
       "       ('caltech101', 'intNMF', 'None', 0, False, 8),\n",
       "       ('caltech101', 'intNMF', 'None', 0, False, 9),\n",
       "       ('caltech101', 'jNMF', 'None', 0, False, 0),\n",
       "       ('caltech101', 'jNMF', 'None', 0, False, 1),\n",
       "       ('caltech101', 'jNMF', 'None', 0, False, 2),\n",
       "       ('caltech101', 'jNMF', 'None', 0, False, 3),\n",
       "       ('caltech101', 'jNMF', 'None', 0, False, 4),\n",
       "       ('caltech101', 'jNMF', 'None', 0, False, 5),\n",
       "       ('caltech101', 'jNMF', 'None', 0, False, 6),\n",
       "       ('caltech101', 'jNMF', 'None', 0, False, 7),\n",
       "       ('caltech101', 'jNMF', 'None', 0, False, 8),\n",
       "       ('caltech101', 'jNMF', 'None', 0, False, 9),\n",
       "       ('nuswide', 'Concat', 'None', 0, False, 0),\n",
       "       ('nuswide', 'Concat', 'None', 0, False, 1),\n",
       "       ('nuswide', 'Concat', 'None', 0, False, 2),\n",
       "       ('nuswide', 'Concat', 'None', 0, False, 3),\n",
       "       ('nuswide', 'Concat', 'None', 0, False, 4),\n",
       "       ('nuswide', 'Concat', 'None', 0, False, 5),\n",
       "       ('nuswide', 'Concat', 'None', 0, False, 6),\n",
       "       ('nuswide', 'Concat', 'None', 0, False, 7),\n",
       "       ('nuswide', 'Concat', 'None', 0, False, 8),\n",
       "       ('nuswide', 'Concat', 'None', 0, False, 9),\n",
       "       ('nuswide', 'NMFC', 'None', 0, False, 0),\n",
       "       ('nuswide', 'NMFC', 'None', 0, False, 1),\n",
       "       ('nuswide', 'NMFC', 'None', 0, False, 2),\n",
       "       ('nuswide', 'NMFC', 'None', 0, False, 3),\n",
       "       ('nuswide', 'NMFC', 'None', 0, False, 4),\n",
       "       ('nuswide', 'NMFC', 'None', 0, False, 5),\n",
       "       ('nuswide', 'NMFC', 'None', 0, False, 6),\n",
       "       ('nuswide', 'NMFC', 'None', 0, False, 7),\n",
       "       ('nuswide', 'NMFC', 'None', 0, False, 8),\n",
       "       ('nuswide', 'NMFC', 'None', 0, False, 9),\n",
       "       ('nuswide', 'MVSpectralClustering', 'None', 0, False, 0),\n",
       "       ('nuswide', 'MVSpectralClustering', 'None', 0, False, 1),\n",
       "       ('nuswide', 'MVSpectralClustering', 'None', 0, False, 2),\n",
       "       ('nuswide', 'MVSpectralClustering', 'None', 0, False, 3),\n",
       "       ('nuswide', 'MVSpectralClustering', 'None', 0, False, 4),\n",
       "       ('nuswide', 'MVSpectralClustering', 'None', 0, False, 5),\n",
       "       ('nuswide', 'MVSpectralClustering', 'None', 0, False, 6),\n",
       "       ('nuswide', 'MVSpectralClustering', 'None', 0, False, 7),\n",
       "       ('nuswide', 'MVSpectralClustering', 'None', 0, False, 8),\n",
       "       ('nuswide', 'MVSpectralClustering', 'None', 0, False, 9),\n",
       "       ('nuswide', 'MVCoRegSpectralClustering', 'None', 0, False, 0),\n",
       "       ('nuswide', 'MVCoRegSpectralClustering', 'None', 0, False, 1),\n",
       "       ('nuswide', 'MVCoRegSpectralClustering', 'None', 0, False, 2),\n",
       "       ('nuswide', 'MVCoRegSpectralClustering', 'None', 0, False, 3),\n",
       "       ('nuswide', 'MVCoRegSpectralClustering', 'None', 0, False, 4),\n",
       "       ('nuswide', 'MVCoRegSpectralClustering', 'None', 0, False, 5),\n",
       "       ('nuswide', 'MVCoRegSpectralClustering', 'None', 0, False, 6),\n",
       "       ('nuswide', 'MVCoRegSpectralClustering', 'None', 0, False, 7),\n",
       "       ('nuswide', 'MVCoRegSpectralClustering', 'None', 0, False, 8),\n",
       "       ('nuswide', 'MVCoRegSpectralClustering', 'None', 0, False, 9),\n",
       "       ('nuswide', 'GroupPCA', 'None', 0, False, 0),\n",
       "       ('nuswide', 'GroupPCA', 'None', 0, False, 1),\n",
       "       ('nuswide', 'GroupPCA', 'None', 0, False, 2),\n",
       "       ('nuswide', 'GroupPCA', 'None', 0, False, 3),\n",
       "       ('nuswide', 'GroupPCA', 'None', 0, False, 4),\n",
       "       ('nuswide', 'GroupPCA', 'None', 0, False, 5),\n",
       "       ('nuswide', 'GroupPCA', 'None', 0, False, 6),\n",
       "       ('nuswide', 'GroupPCA', 'None', 0, False, 7),\n",
       "       ('nuswide', 'GroupPCA', 'None', 0, False, 8),\n",
       "       ('nuswide', 'GroupPCA', 'None', 0, False, 9),\n",
       "       ('nuswide', 'AJIVE', 'None', 0, False, 0),\n",
       "       ('nuswide', 'AJIVE', 'None', 0, False, 1),\n",
       "       ('nuswide', 'AJIVE', 'None', 0, False, 2),\n",
       "       ('nuswide', 'AJIVE', 'None', 0, False, 3),\n",
       "       ('nuswide', 'AJIVE', 'None', 0, False, 4),\n",
       "       ('nuswide', 'AJIVE', 'None', 0, False, 5),\n",
       "       ('nuswide', 'AJIVE', 'None', 0, False, 6),\n",
       "       ('nuswide', 'AJIVE', 'None', 0, False, 7),\n",
       "       ('nuswide', 'AJIVE', 'None', 0, False, 8),\n",
       "       ('nuswide', 'AJIVE', 'None', 0, False, 9),\n",
       "       ('nuswide', 'SNF', 'None', 0, False, 0),\n",
       "       ('nuswide', 'SNF', 'None', 0, False, 1),\n",
       "       ('nuswide', 'SNF', 'None', 0, False, 2),\n",
       "       ('nuswide', 'SNF', 'None', 0, False, 3),\n",
       "       ('nuswide', 'SNF', 'None', 0, False, 4),\n",
       "       ('nuswide', 'SNF', 'None', 0, False, 5),\n",
       "       ('nuswide', 'SNF', 'None', 0, False, 6),\n",
       "       ('nuswide', 'SNF', 'None', 0, False, 7),\n",
       "       ('nuswide', 'SNF', 'None', 0, False, 8),\n",
       "       ('nuswide', 'SNF', 'None', 0, False, 9),\n",
       "       ('nuswide', 'intNMF', 'None', 0, False, 0),\n",
       "       ('nuswide', 'intNMF', 'None', 0, False, 1),\n",
       "       ('nuswide', 'intNMF', 'None', 0, False, 2),\n",
       "       ('nuswide', 'intNMF', 'None', 0, False, 3),\n",
       "       ('nuswide', 'intNMF', 'None', 0, False, 4),\n",
       "       ('nuswide', 'intNMF', 'None', 0, False, 5),\n",
       "       ('nuswide', 'intNMF', 'None', 0, False, 6),\n",
       "       ('nuswide', 'intNMF', 'None', 0, False, 7),\n",
       "       ('nuswide', 'intNMF', 'None', 0, False, 8),\n",
       "       ('nuswide', 'intNMF', 'None', 0, False, 9),\n",
       "       ('nuswide', 'jNMF', 'None', 0, False, 0),\n",
       "       ('nuswide', 'jNMF', 'None', 0, False, 1),\n",
       "       ('nuswide', 'jNMF', 'None', 0, False, 2),\n",
       "       ('nuswide', 'jNMF', 'None', 0, False, 3),\n",
       "       ('nuswide', 'jNMF', 'None', 0, False, 4),\n",
       "       ('nuswide', 'jNMF', 'None', 0, False, 5),\n",
       "       ('nuswide', 'jNMF', 'None', 0, False, 6),\n",
       "       ('nuswide', 'jNMF', 'None', 0, False, 7),\n",
       "       ('nuswide', 'jNMF', 'None', 0, False, 8),\n",
       "       ('nuswide', 'jNMF', 'None', 0, False, 9)], dtype=object)"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.MultiIndex.from_frame(results_amputation_mechanism_none_tochange).values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "c14623ef-b0b6-4a66-ac05-2e0203da0aa4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>amputation_mechanism</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset</th>\n",
       "      <th>algorithm</th>\n",
       "      <th>amputation_mechanism</th>\n",
       "      <th>missing_percentage</th>\n",
       "      <th>imputation</th>\n",
       "      <th>run_n</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nutrimouse_genotype</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">Concat</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">ED</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>0</th>\n",
       "      <td>nutrimouse_genotype</td>\n",
       "      <td>Concat</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>nutrimouse_genotype</td>\n",
       "      <td>Concat</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>nutrimouse_genotype</td>\n",
       "      <td>Concat</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>nutrimouse_genotype</td>\n",
       "      <td>Concat</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>nutrimouse_genotype</td>\n",
       "      <td>Concat</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">nuswide</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">jNMF</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">MNAR</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">0</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>5</th>\n",
       "      <td>nuswide</td>\n",
       "      <td>jNMF</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>nuswide</td>\n",
       "      <td>jNMF</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>nuswide</td>\n",
       "      <td>jNMF</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>nuswide</td>\n",
       "      <td>jNMF</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>nuswide</td>\n",
       "      <td>jNMF</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2700 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                    dataset   \n",
       "dataset             algorithm amputation_mechanism missing_percentage imputation run_n                        \n",
       "nutrimouse_genotype Concat    ED                   0                  False      0      nutrimouse_genotype  \\\n",
       "                                                                                 1      nutrimouse_genotype   \n",
       "                                                                                 2      nutrimouse_genotype   \n",
       "                                                                                 3      nutrimouse_genotype   \n",
       "                                                                                 4      nutrimouse_genotype   \n",
       "...                                                                                                     ...   \n",
       "nuswide             jNMF      MNAR                 0                  False      5                  nuswide   \n",
       "                                                                                 6                  nuswide   \n",
       "                                                                                 7                  nuswide   \n",
       "                                                                                 8                  nuswide   \n",
       "                                                                                 9                  nuswide   \n",
       "\n",
       "                                                                                       algorithm   \n",
       "dataset             algorithm amputation_mechanism missing_percentage imputation run_n             \n",
       "nutrimouse_genotype Concat    ED                   0                  False      0        Concat  \\\n",
       "                                                                                 1        Concat   \n",
       "                                                                                 2        Concat   \n",
       "                                                                                 3        Concat   \n",
       "                                                                                 4        Concat   \n",
       "...                                                                                          ...   \n",
       "nuswide             jNMF      MNAR                 0                  False      5          jNMF   \n",
       "                                                                                 6          jNMF   \n",
       "                                                                                 7          jNMF   \n",
       "                                                                                 8          jNMF   \n",
       "                                                                                 9          jNMF   \n",
       "\n",
       "                                                                                       amputation_mechanism   \n",
       "dataset             algorithm amputation_mechanism missing_percentage imputation run_n                        \n",
       "nutrimouse_genotype Concat    ED                   0                  False      0                     None  \\\n",
       "                                                                                 1                     None   \n",
       "                                                                                 2                     None   \n",
       "                                                                                 3                     None   \n",
       "                                                                                 4                     None   \n",
       "...                                                                                                     ...   \n",
       "nuswide             jNMF      MNAR                 0                  False      5                     None   \n",
       "                                                                                 6                     None   \n",
       "                                                                                 7                     None   \n",
       "                                                                                 8                     None   \n",
       "                                                                                 9                     None   \n",
       "\n",
       "                                                                                        missing_percentage   \n",
       "dataset             algorithm amputation_mechanism missing_percentage imputation run_n                       \n",
       "nutrimouse_genotype Concat    ED                   0                  False      0                       0  \\\n",
       "                                                                                 1                       0   \n",
       "                                                                                 2                       0   \n",
       "                                                                                 3                       0   \n",
       "                                                                                 4                       0   \n",
       "...                                                                                                    ...   \n",
       "nuswide             jNMF      MNAR                 0                  False      5                       0   \n",
       "                                                                                 6                       0   \n",
       "                                                                                 7                       0   \n",
       "                                                                                 8                       0   \n",
       "                                                                                 9                       0   \n",
       "\n",
       "                                                                                        imputation   \n",
       "dataset             algorithm amputation_mechanism missing_percentage imputation run_n               \n",
       "nutrimouse_genotype Concat    ED                   0                  False      0           False  \\\n",
       "                                                                                 1           False   \n",
       "                                                                                 2           False   \n",
       "                                                                                 3           False   \n",
       "                                                                                 4           False   \n",
       "...                                                                                            ...   \n",
       "nuswide             jNMF      MNAR                 0                  False      5           False   \n",
       "                                                                                 6           False   \n",
       "                                                                                 7           False   \n",
       "                                                                                 8           False   \n",
       "                                                                                 9           False   \n",
       "\n",
       "                                                                                        run_n  \n",
       "dataset             algorithm amputation_mechanism missing_percentage imputation run_n         \n",
       "nutrimouse_genotype Concat    ED                   0                  False      0          0  \n",
       "                                                                                 1          1  \n",
       "                                                                                 2          2  \n",
       "                                                                                 3          3  \n",
       "                                                                                 4          4  \n",
       "...                                                                                       ...  \n",
       "nuswide             jNMF      MNAR                 0                  False      5          5  \n",
       "                                                                                 6          6  \n",
       "                                                                                 7          7  \n",
       "                                                                                 8          8  \n",
       "                                                                                 9          9  \n",
       "\n",
       "[2700 rows x 6 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "5b37dd84-dd02-4031-8d7a-ad797a68b1c6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n",
       "       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n",
       "       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n",
       "       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n",
       "       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n",
       "       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n",
       "       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "from scipy import stats\n",
    "\n",
    "amp.wss_per_pattern = []\n",
    "amp.probs_per_pattern = []\n",
    "\n",
    "X = amp._validate_data(missing_view_panel)\n",
    "num_samples = X.shape[0]\n",
    "\n",
    "# split complete_data in groups\n",
    "# the number of groups is defined by the number of patterns\n",
    "X_incomplete = X.copy()\n",
    "X_indices = np.arange(num_samples)\n",
    "# set seed for choice, if None it will be random.\n",
    "rng = np.random.default_rng(amp.seed)\n",
    "amp.assigned_group_number = rng.choice(\n",
    "    a=amp.num_patterns, size=num_samples, p=amp.freqs\n",
    ")\n",
    "pattern_idx = 0\n",
    "group_indices = X_indices[amp.assigned_group_number == pattern_idx]\n",
    "pattern = np.squeeze(\n",
    "    np.asarray(amp.observed_var_indicator[pattern_idx, :])\n",
    ")\n",
    "data_group = (\n",
    "    X[group_indices] if isinstance(X, np.ndarray) else X.iloc[group_indices]\n",
    ")\n",
    "# data_group = stats.zscore(data_group)\n",
    "wss = np.dot(data_group, amp.weights[pattern_idx, :].T)\n",
    "wss"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "575d3d78-5dc9-4734-aedf-bc8dfedb3578",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "116"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1368,
   "id": "2a2641bd-0bc2-44d1-813d-31e153151c76",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-12-22 16:25:33,878 [WARNING] Failed to load lookup table for a prespecified score to probability function. It is possible data/shift_lookup.csv is missing, in the wrong location, or corrupted. Try rerunning scripts/generate_shift_lookup_table.py to regenerate the lookup table.\n"
     ]
    },
    {
     "ename": "AssertionError",
     "evalue": "Cannot ampute all features under MAR, since all vars will be missing.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[1368], line 9\u001b[0m\n\u001b[1;32m      7\u001b[0m views_to_remove \u001b[38;5;241m=\u001b[39m missing_view_panel\u001b[38;5;241m.\u001b[39mcolumns[views_to_remove]\n\u001b[1;32m      8\u001b[0m amp \u001b[38;5;241m=\u001b[39m MultivariateAmputation(patterns\u001b[38;5;241m=\u001b[39m [{\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mincomplete_vars\u001b[39m\u001b[38;5;124m\"\u001b[39m: views_to_remove}])\n\u001b[0;32m----> 9\u001b[0m \u001b[43mamp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfit_transform\u001b[49m\u001b[43m(\u001b[49m\u001b[43ma\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39mfillna(\u001b[38;5;241m0\u001b[39m)\u001b[38;5;241m.\u001b[39mastype(\u001b[38;5;28mint\u001b[39m)\n",
      "File \u001b[0;32m~/anaconda3/envs/imc/lib/python3.10/site-packages/sklearn/utils/_set_output.py:140\u001b[0m, in \u001b[0;36m_wrap_method_output.<locals>.wrapped\u001b[0;34m(self, X, *args, **kwargs)\u001b[0m\n\u001b[1;32m    138\u001b[0m \u001b[38;5;129m@wraps\u001b[39m(f)\n\u001b[1;32m    139\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrapped\u001b[39m(\u001b[38;5;28mself\u001b[39m, X, \u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[0;32m--> 140\u001b[0m     data_to_wrap \u001b[38;5;241m=\u001b[39m \u001b[43mf\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mX\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    141\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(data_to_wrap, \u001b[38;5;28mtuple\u001b[39m):\n\u001b[1;32m    142\u001b[0m         \u001b[38;5;66;03m# only wrap the first output for cross decomposition\u001b[39;00m\n\u001b[1;32m    143\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m (\n\u001b[1;32m    144\u001b[0m             _wrap_data_with_container(method, data_to_wrap[\u001b[38;5;241m0\u001b[39m], X, \u001b[38;5;28mself\u001b[39m),\n\u001b[1;32m    145\u001b[0m             \u001b[38;5;241m*\u001b[39mdata_to_wrap[\u001b[38;5;241m1\u001b[39m:],\n\u001b[1;32m    146\u001b[0m         )\n",
      "File \u001b[0;32m~/anaconda3/envs/imc/lib/python3.10/site-packages/sklearn/base.py:878\u001b[0m, in \u001b[0;36mTransformerMixin.fit_transform\u001b[0;34m(self, X, y, **fit_params)\u001b[0m\n\u001b[1;32m    874\u001b[0m \u001b[38;5;66;03m# non-optimized default implementation; override when a better\u001b[39;00m\n\u001b[1;32m    875\u001b[0m \u001b[38;5;66;03m# method is possible for a given clustering algorithm\u001b[39;00m\n\u001b[1;32m    876\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m y \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    877\u001b[0m     \u001b[38;5;66;03m# fit method of arity 1 (unsupervised transformation)\u001b[39;00m\n\u001b[0;32m--> 878\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfit\u001b[49m\u001b[43m(\u001b[49m\u001b[43mX\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mfit_params\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39mtransform(X)\n\u001b[1;32m    879\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    880\u001b[0m     \u001b[38;5;66;03m# fit method of arity 2 (supervised transformation)\u001b[39;00m\n\u001b[1;32m    881\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mfit(X, y, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mfit_params)\u001b[38;5;241m.\u001b[39mtransform(X)\n",
      "File \u001b[0;32m~/anaconda3/envs/imc/lib/python3.10/site-packages/pyampute/ampute.py:888\u001b[0m, in \u001b[0;36mMultivariateAmputation.fit\u001b[0;34m(self, X, y)\u001b[0m\n\u001b[1;32m    886\u001b[0m \u001b[38;5;66;03m# defaults for the rest of the args (depends on patterns being initialized)\u001b[39;00m\n\u001b[1;32m    887\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_set_defaults()\n\u001b[0;32m--> 888\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_validate_args\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    890\u001b[0m \u001b[38;5;66;03m# vars involved in amputation have scores computed and need to be\u001b[39;00m\n\u001b[1;32m    891\u001b[0m \u001b[38;5;66;03m#   complete and numeric\u001b[39;00m\n\u001b[1;32m    892\u001b[0m \u001b[38;5;66;03m# A var (column) is involved if for any pattern (row) it has a weight.\u001b[39;00m\n\u001b[1;32m    893\u001b[0m \u001b[38;5;66;03m# We don't care about numeric restraint for MCAR\u001b[39;00m\n\u001b[1;32m    894\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mvars_involved_in_ampute \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m    895\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mweights[\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmechanisms \u001b[38;5;241m!=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mMCAR\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m0\u001b[39m\n\u001b[1;32m    896\u001b[0m )\u001b[38;5;241m.\u001b[39many(axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m0\u001b[39m)\n",
      "File \u001b[0;32m~/anaconda3/envs/imc/lib/python3.10/site-packages/pyampute/ampute.py:657\u001b[0m, in \u001b[0;36mMultivariateAmputation._validate_args\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    653\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m (\n\u001b[1;32m    654\u001b[0m     (\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mobserved_var_indicator \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m1\u001b[39m)\u001b[38;5;241m.\u001b[39mall(axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m)\n\u001b[1;32m    655\u001b[0m ), \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCannot indicate no features to be amputed, will result in no amputation.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    656\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m isin(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmechanisms, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mMAR\u001b[39m\u001b[38;5;124m\"\u001b[39m)\u001b[38;5;241m.\u001b[39many(axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m0\u001b[39m):\n\u001b[0;32m--> 657\u001b[0m     \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m (\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mobserved_var_indicator[\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmechanisms \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mMAR\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0\u001b[39m)\u001b[38;5;241m.\u001b[39mall(\n\u001b[1;32m    658\u001b[0m         axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m    659\u001b[0m     ), \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCannot ampute all features under MAR, since all vars will be missing.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    661\u001b[0m \u001b[38;5;66;03m##################\u001b[39;00m\n\u001b[1;32m    662\u001b[0m \u001b[38;5;66;03m#      PROP      #\u001b[39;00m\n\u001b[1;32m    663\u001b[0m \u001b[38;5;66;03m##################\u001b[39;00m\n\u001b[1;32m    664\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mprop \u001b[38;5;241m>\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;241m0\u001b[39m \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mprop \u001b[38;5;241m<\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;241m100\u001b[39m, (\n\u001b[1;32m    665\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mProportion of missingness should be a value between 0 and 1\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    666\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m (for a proportion) or between 1 and 100 (for a percentage)\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    667\u001b[0m )\n",
      "\u001b[0;31mAssertionError\u001b[0m: Cannot ampute all features under MAR, since all vars will be missing."
     ]
    }
   ],
   "source": [
    "a = pd.DataFrame(np.zeros((10,2))) + 1\n",
    "if missing_view_panel.shape[1] > 2:\n",
    "    n_views_to_remove = round(missing_view_panel.shape[1]*0.5 + pd.Series([0.1,-0.1]).sample(1, random_state=random_state).iloc[0])\n",
    "else:\n",
    "    n_views_to_remove = 1\n",
    "views_to_remove = missing_view_panel.columns.to_series().sample(n= n_views_to_remove, random_state=random_state)\n",
    "views_to_remove = missing_view_panel.columns[views_to_remove]\n",
    "amp = MultivariateAmputation(patterns= [{\"incomplete_vars\": views_to_remove}])\n",
    "amp.fit_transform(a).fillna(0).astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1166,
   "id": "26b72de5-2183-4a49-b01a-b1b0c297107d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[31mERROR: Could not find a version that satisfies the requirement ampute (from versions: none)\u001b[0m\u001b[31m\n",
      "\u001b[0m\u001b[31mERROR: No matching distribution found for ampute\u001b[0m\u001b[31m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!pip install ampute"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1165,
   "id": "904ddef3-94e1-4f27-94bb-6f20c97a7803",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-12-22 14:15:00,770 [WARNING] Failed to load lookup table for a prespecified score to probability function. It is possible data/shift_lookup.csv is missing, in the wrong location, or corrupted. Try rerunning scripts/generate_shift_lookup_table.py to regenerate the lookup table.\n",
      "2023-12-22 14:15:00,789 [WARNING] Candidates for pattern 0 all have almost the same weighted sum scores. It is possible this is due to the use of binary variables in amputation. This creates problems when using the sigmoid function for the score_to_probability_func. Currently our solution is as follows: if there is just one candidate with a sum score 0, we will ampute it. If there is one candidate with a nonzero sum score, or multiple candidates with the same score, we evenly apply as if MCAR.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(62, 10)"
      ]
     },
     "execution_count": 1165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = pd.DataFrame(np.zeros((100,10))) + 1\n",
    "amp = MultivariateAmputation(prop = 30)\n",
    "amp.fit_transform(a).dropna().shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "id": "d09d35ca-f365-447a-934f-cf298b962466",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "def f(arg):\n",
    "    arg[\"asd\"] = 2\n",
    "    arg[\"asdlkñ\"] = 3\n",
    "    return arg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "id": "1cbc7d5b-30e7-4f03-abcd-a4063499743e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "a = pd.concat([pd.read_csv(\"imvc/datasets/data/nutrimouse/original/genotype.csv\"), pd.read_csv(\"imvc/datasets/data/nutrimouse/original/diet.csv\")], axis= 1)\n",
    "le = LabelEncoder()\n",
    "a.iloc[:,0] = le.fit_transform(a.iloc[:,0])\n",
    "le_ = LabelEncoder()\n",
    "a.iloc[:,1] = le_.fit_transform(a.iloc[:,1])\n",
    "a.to_csv(\"imvc/datasets/data/nutrimouse/nutrimouse_y.csv\", index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "id": "55e17408-7168-45e8-a362-cb262c1192aa",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "metadata = {\"modality\": {0: \"gene\", 1: \"lipid\"}, \"labels\": {\"genotype\": pd.Series(le.classes_).to_dict(), \"diet\": pd.Series(le_.classes_).to_dict()}}\n",
    "\n",
    "with open(os.path.join(\"imvc/datasets/data/nutrimouse\", 'metadata.json'), 'w') as fp:\n",
    "    json.dump(metadata, fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "313007c2-1662-4397-9218-f6688383f325",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "probs = np.arange(0., 1., step= 0.1).round(1)\n",
    "runs_per_alg = np.arange(10).tolist()\n",
    "pd.DataFrame(probs).merge(pd.DataFrame(runs_per_alg), how= \"cross\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "24ea88c5-641c-439c-8257-1a427f4d574d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "a = pd.DataFrame([\"alg\", \"pad\"])\n",
    "b = pd.DataFrame([\"alg1\", \"pad2\"])\n",
    "a.merge(b, how= \"cross\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c68327e9-ab81-4dd7-bd48-496655af9430",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "metadata = {}\n",
    "metadata = {\"modality\": {0: \"morphological features\", 1: \"Karhunen-Love coefficients\", 2: \"profile correlations\", 3: \"Zernike moments\", 4: \"Fourier coefficients of the character shapes\", 5: \"pixel averages of the images from 2x3 windows\"}}\n",
    "import json\n",
    "\n",
    "with open(os.path.join(\"imvc/datasets/data/digits\", 'metadata.json'), 'w') as fp:\n",
    "    json.dump(metadata, fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d430d77a-1056-409a-8317-fe0b6a57fa17",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import scipy\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "b6bf57b4-f042-498f-a25d-0d023c9cf9e7",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>cg00003994</th>\n",
       "      <th>cg00024396</th>\n",
       "      <th>cg00035623</th>\n",
       "      <th>cg00047050</th>\n",
       "      <th>cg00079563</th>\n",
       "      <th>cg00095674</th>\n",
       "      <th>cg00103783</th>\n",
       "      <th>cg00112517</th>\n",
       "      <th>cg00117172</th>\n",
       "      <th>...</th>\n",
       "      <th>cg27529628</th>\n",
       "      <th>cg27543230</th>\n",
       "      <th>cg27544190</th>\n",
       "      <th>cg27555365</th>\n",
       "      <th>cg27560922</th>\n",
       "      <th>cg27574244</th>\n",
       "      <th>cg27601582</th>\n",
       "      <th>cg27626299</th>\n",
       "      <th>cg27654142</th>\n",
       "      <th>cg27662877</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-13-0799</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-13-0800</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-13-0801</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-24-1924</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-24-1930</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2432</th>\n",
       "      <td>TCGA-CV-7434</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2433</th>\n",
       "      <td>TCGA-CV-7435</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2434</th>\n",
       "      <td>TCGA-CV-7437</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2435</th>\n",
       "      <td>TCGA-CV-7438</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2436</th>\n",
       "      <td>TCGA-CV-7440</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2437 rows × 2044 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Unnamed: 0  cg00003994  cg00024396  cg00035623  cg00047050   \n",
       "0     TCGA-13-0799         0.0         0.0         0.0         0.0  \\\n",
       "1     TCGA-13-0800         0.0         1.0         0.0         0.0   \n",
       "2     TCGA-13-0801         1.0         0.0         0.0         0.0   \n",
       "3     TCGA-24-1924         0.0         0.0         0.0         0.0   \n",
       "4     TCGA-24-1930         0.0         0.0         0.0         0.0   \n",
       "...            ...         ...         ...         ...         ...   \n",
       "2432  TCGA-CV-7434         1.0         0.0         0.0         0.0   \n",
       "2433  TCGA-CV-7435         0.0         0.0         0.0         0.0   \n",
       "2434  TCGA-CV-7437         1.0         0.0         0.0         0.0   \n",
       "2435  TCGA-CV-7438         1.0         0.0         0.0         0.0   \n",
       "2436  TCGA-CV-7440         0.0         0.0         0.0         0.0   \n",
       "\n",
       "      cg00079563  cg00095674  cg00103783  cg00112517  cg00117172  ...   \n",
       "0            0.0         0.0         0.0         0.0         0.0  ...  \\\n",
       "1            0.0         0.0         0.0         0.0         1.0  ...   \n",
       "2            0.0         0.0         0.0         0.0         0.0  ...   \n",
       "3            0.0         0.0         0.0         0.0         1.0  ...   \n",
       "4            0.0         0.0         NaN         0.0         0.0  ...   \n",
       "...          ...         ...         ...         ...         ...  ...   \n",
       "2432         0.0         0.0         0.0         1.0         0.0  ...   \n",
       "2433         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "2434         0.0         0.0         0.0         1.0         0.0  ...   \n",
       "2435         0.0         0.0         1.0         0.0         0.0  ...   \n",
       "2436         0.0         0.0         0.0         0.0         0.0  ...   \n",
       "\n",
       "      cg27529628  cg27543230  cg27544190  cg27555365  cg27560922  cg27574244   \n",
       "0            0.0         0.0         0.0         0.0         0.0         0.0  \\\n",
       "1            0.0         0.0         1.0         0.0         0.0         1.0   \n",
       "2            0.0         0.0         0.0         0.0         0.0         0.0   \n",
       "3            0.0         0.0         0.0         0.0         0.0         0.0   \n",
       "4            0.0         0.0         0.0         0.0         0.0         NaN   \n",
       "...          ...         ...         ...         ...         ...         ...   \n",
       "2432         0.0         1.0         0.0         0.0         1.0         0.0   \n",
       "2433         0.0         0.0         0.0         0.0         0.0         0.0   \n",
       "2434         0.0         0.0         0.0         0.0         0.0         0.0   \n",
       "2435         0.0         0.0         0.0         0.0         1.0         0.0   \n",
       "2436         0.0         0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "      cg27601582  cg27626299  cg27654142  cg27662877  \n",
       "0            0.0         1.0         0.0         0.0  \n",
       "1            0.0         0.0         0.0         0.0  \n",
       "2            1.0         1.0         0.0         0.0  \n",
       "3            0.0         1.0         0.0         0.0  \n",
       "4            0.0         0.0         0.0         0.0  \n",
       "...          ...         ...         ...         ...  \n",
       "2432         1.0         1.0         1.0         0.0  \n",
       "2433         0.0         1.0         0.0         0.0  \n",
       "2434         0.0         1.0         0.0         0.0  \n",
       "2435         0.0         1.0         0.0         0.0  \n",
       "2436         0.0         1.0         0.0         0.0  \n",
       "\n",
       "[2437 rows x 2044 columns]"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "id": "0319fa7e-5380-44ac-9b61-f4f78943a8a2",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "simulated_netMUG\n",
      "simulated_netMUG_y.csv\n",
      "\n",
      ".ipynb_checkpoints\n",
      "\n",
      "simulated_InterSIM\n",
      "simulated_InterSIM_y.csv\n",
      "\n",
      "buaa\n",
      "buaa_y.csv\n",
      "\n",
      "bdgp\n",
      "bdgp_y.csv\n",
      "\n",
      "tcga\n",
      "tcga_y.csv\n",
      "\n",
      "nuswide\n",
      "nuswide_y.csv\n",
      "\n",
      "metabric\n",
      "metabric_y.csv\n",
      "\n",
      "nutrimouse\n",
      "nutrimouse_y.csv\n",
      "\n",
      "digits\n",
      "digits_y.csv\n",
      "\n",
      "caltech101\n",
      "caltech101_y.csv\n",
      "\n",
      "bbcsport\n",
      "bbcsport_y.csv\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "for i in os.listdir(\"imvc/datasets/data/\"):\n",
    "    print(i)\n",
    "    for j in os.listdir(os.path.join(\"imvc/datasets/data/\", i)):\n",
    "        if \"_y.csv\" in j:\n",
    "            try:\n",
    "                a = pd.read_csv(os.path.join(\"imvc/datasets/data/\", i, j), index_col = 0)\n",
    "                if a.shape[1] < 2:\n",
    "                    print(j)\n",
    "                    a.columns = list(range(a.shape[1]))\n",
    "                    a.to_csv(os.path.join(\"imvc/datasets/data/\", i, j), index= False)\n",
    "                else:\n",
    "                    a.to_csv(os.path.join(\"imvc/datasets/data/\", i, j), index= False)\n",
    "            except:\n",
    "                pass\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "id": "3f24a9d6-28e6-4a60-97e1-c32f5fe7c481",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "'bool' object is not iterable",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[180], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43ma\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mto_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[43mos\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpath\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mjoin\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mimvc/datasets/data/\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mi\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mj\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindex\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcolumns\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/anaconda3/envs/imc/lib/python3.10/site-packages/pandas/core/generic.py:3772\u001b[0m, in \u001b[0;36mNDFrame.to_csv\u001b[0;34m(self, path_or_buf, sep, na_rep, float_format, columns, header, index, index_label, mode, encoding, compression, quoting, quotechar, lineterminator, chunksize, date_format, doublequote, escapechar, decimal, errors, storage_options)\u001b[0m\n\u001b[1;32m   3761\u001b[0m df \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(\u001b[38;5;28mself\u001b[39m, ABCDataFrame) \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mto_frame()\n\u001b[1;32m   3763\u001b[0m formatter \u001b[38;5;241m=\u001b[39m DataFrameFormatter(\n\u001b[1;32m   3764\u001b[0m     frame\u001b[38;5;241m=\u001b[39mdf,\n\u001b[1;32m   3765\u001b[0m     header\u001b[38;5;241m=\u001b[39mheader,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   3769\u001b[0m     decimal\u001b[38;5;241m=\u001b[39mdecimal,\n\u001b[1;32m   3770\u001b[0m )\n\u001b[0;32m-> 3772\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mDataFrameRenderer\u001b[49m\u001b[43m(\u001b[49m\u001b[43mformatter\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mto_csv\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   3773\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpath_or_buf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3774\u001b[0m \u001b[43m    \u001b[49m\u001b[43mlineterminator\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mlineterminator\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3775\u001b[0m \u001b[43m    \u001b[49m\u001b[43msep\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msep\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3776\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3777\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3778\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcompression\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3779\u001b[0m \u001b[43m    \u001b[49m\u001b[43mquoting\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquoting\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3780\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcolumns\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3781\u001b[0m \u001b[43m    \u001b[49m\u001b[43mindex_label\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mindex_label\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3782\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3783\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchunksize\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunksize\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3784\u001b[0m \u001b[43m    \u001b[49m\u001b[43mquotechar\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquotechar\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3785\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdate_format\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdate_format\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3786\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdoublequote\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdoublequote\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3787\u001b[0m \u001b[43m    \u001b[49m\u001b[43mescapechar\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mescapechar\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3788\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstorage_options\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3789\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/anaconda3/envs/imc/lib/python3.10/site-packages/pandas/io/formats/format.py:1167\u001b[0m, in \u001b[0;36mDataFrameRenderer.to_csv\u001b[0;34m(self, path_or_buf, encoding, sep, columns, index_label, mode, compression, quoting, quotechar, lineterminator, chunksize, date_format, doublequote, escapechar, errors, storage_options)\u001b[0m\n\u001b[1;32m   1164\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1165\u001b[0m     created_buffer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[0;32m-> 1167\u001b[0m csv_formatter \u001b[38;5;241m=\u001b[39m \u001b[43mCSVFormatter\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1168\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpath_or_buf\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpath_or_buf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1169\u001b[0m \u001b[43m    \u001b[49m\u001b[43mlineterminator\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mlineterminator\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1170\u001b[0m \u001b[43m    \u001b[49m\u001b[43msep\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msep\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1171\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1172\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1173\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcompression\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1174\u001b[0m \u001b[43m    \u001b[49m\u001b[43mquoting\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquoting\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1175\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcols\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1176\u001b[0m \u001b[43m    \u001b[49m\u001b[43mindex_label\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mindex_label\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1177\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1178\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchunksize\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunksize\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1179\u001b[0m \u001b[43m    \u001b[49m\u001b[43mquotechar\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquotechar\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1180\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdate_format\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdate_format\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1181\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdoublequote\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdoublequote\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1182\u001b[0m \u001b[43m    \u001b[49m\u001b[43mescapechar\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mescapechar\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1183\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstorage_options\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1184\u001b[0m \u001b[43m    \u001b[49m\u001b[43mformatter\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfmt\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1185\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1186\u001b[0m csv_formatter\u001b[38;5;241m.\u001b[39msave()\n\u001b[1;32m   1188\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m created_buffer:\n",
      "File \u001b[0;32m~/anaconda3/envs/imc/lib/python3.10/site-packages/pandas/io/formats/csvs.py:89\u001b[0m, in \u001b[0;36mCSVFormatter.__init__\u001b[0;34m(self, formatter, path_or_buf, sep, cols, index_label, mode, encoding, errors, compression, quoting, lineterminator, chunksize, quotechar, date_format, doublequote, escapechar, storage_options)\u001b[0m\n\u001b[1;32m     87\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mlineterminator \u001b[38;5;241m=\u001b[39m lineterminator \u001b[38;5;129;01mor\u001b[39;00m os\u001b[38;5;241m.\u001b[39mlinesep\n\u001b[1;32m     88\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdate_format \u001b[38;5;241m=\u001b[39m date_format\n\u001b[0;32m---> 89\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcols \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_initialize_columns\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcols\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     90\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mchunksize \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_initialize_chunksize(chunksize)\n",
      "File \u001b[0;32m~/anaconda3/envs/imc/lib/python3.10/site-packages/pandas/io/formats/csvs.py:155\u001b[0m, in \u001b[0;36mCSVFormatter._initialize_columns\u001b[0;34m(self, cols)\u001b[0m\n\u001b[1;32m    153\u001b[0m         cols \u001b[38;5;241m=\u001b[39m cols\u001b[38;5;241m.\u001b[39m_format_native_types(\u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_number_format)\n\u001b[1;32m    154\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 155\u001b[0m         cols \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mlist\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mcols\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    156\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mobj \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mobj\u001b[38;5;241m.\u001b[39mloc[:, cols]\n\u001b[1;32m    158\u001b[0m \u001b[38;5;66;03m# update columns to include possible multiplicity of dupes\u001b[39;00m\n\u001b[1;32m    159\u001b[0m \u001b[38;5;66;03m# and make sure cols is just a list of labels\u001b[39;00m\n",
      "\u001b[0;31mTypeError\u001b[0m: 'bool' object is not iterable"
     ]
    }
   ],
   "source": [
    "a.to_csv(os.path.join(\"imvc/datasets/data/\", i, j), index= False, columns= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "91a1c544-079d-4a55-a4e4-b2864b48872f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"imvc/datasets/data/tcga/tcga_y.csv\", index_col = 0)\n",
    "df.to_csv(\"imvc/datasets/data/tcga/tcga_y.csv\", index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "6f47bb9b-1c1a-4953-98ca-45ba0e8b9205",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "pd.DataFrame(mat[\"CNA_s\"], index = pd.DataFrame(mat[\"gene_s\"]).squeeze().apply(lambda x: x[0].replace('\"', \"\")).to_list()).T.to_csv(\"imvc/datasets/data/metabric/metabric_1.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "2144204d-1703-42ae-be08-818c5e57252d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "le = LabelEncoder()\n",
    "pd.Series(le.fit_transform(df)).to_csv(\"imvc/datasets/data/metabric/metabric_y.csv\", index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "id": "edbc3153-cc45-47b0-8a7a-592a7cbd8808",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "df = pd.read_excel(\"imvc/datasets/data/metabric/original/Metabric_breast.xlsx\")[\"CLAUDIN_SUBTYPE\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "98015e26-faf2-494a-8724-86dc4b6f0131",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "with open(os.path.join(\"imvc/datasets/data/metabric\", 'metadata.json'), 'r') as fp:\n",
    "    metadata = json.load(fp)\n",
    "\n",
    "metadata = {**metadata, \"samples\": pd.read_excel(\"imvc/datasets/data/metabric/original/Metabric_breast.xlsx\")[\"Patient Identifier\"].str.replace(\".\", \"-\").to_dict()}\n",
    "with open(os.path.join(\"imvc/datasets/data/metabric\", 'metadata.json'), 'w') as fp:\n",
    "    json.dump(metadata, fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "1f8cb08e-33d9-4abf-b33f-946c76e202c1",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "metadata = {}\n",
    "metadata = {\"modality\": {0: \"GE\", 1: \"CNA\"}, \"labels\": {i:j for i,j in enumerate(le.classes_)}}\n",
    "import json, os\n",
    "\n",
    "with open(os.path.join(\"imvc/datasets/data/metabric\", 'metadata.json'), 'w') as fp:\n",
    "    json.dump(metadata, fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "93f6efae-d6ad-43b0-a7c8-c3978801119e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient Identifier</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>OS_STATUS</th>\n",
       "      <th>OS_MONTHS</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>CLAUDIN_SUBTYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MB.0362</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>47.033333</td>\n",
       "      <td>3</td>\n",
       "      <td>C</td>\n",
       "      <td>LumA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MB.0346</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>20.433333</td>\n",
       "      <td>4</td>\n",
       "      <td>D</td>\n",
       "      <td>Her2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MB.0386</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>3</td>\n",
       "      <td>C</td>\n",
       "      <td>LumA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MB.0574</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>4</td>\n",
       "      <td>D</td>\n",
       "      <td>LumA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MB.0503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>3</td>\n",
       "      <td>C</td>\n",
       "      <td>LumA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1899</th>\n",
       "      <td>MB.5465</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>18.800000</td>\n",
       "      <td>3</td>\n",
       "      <td>C</td>\n",
       "      <td>Basal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1900</th>\n",
       "      <td>MB.5453</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>57.300000</td>\n",
       "      <td>2</td>\n",
       "      <td>B</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1901</th>\n",
       "      <td>MB.5471</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>4</td>\n",
       "      <td>D</td>\n",
       "      <td>LumA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1902</th>\n",
       "      <td>MB.5127</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>3</td>\n",
       "      <td>C</td>\n",
       "      <td>LumB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1903</th>\n",
       "      <td>MB.4313</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>LumA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1904 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Patient Identifier  Unnamed: 1 OS_STATUS  OS_MONTHS  Unnamed: 4   \n",
       "0               MB.0362         NaN  DECEASED  47.033333           3  \\\n",
       "1               MB.0346         NaN  DECEASED  20.433333           4   \n",
       "2               MB.0386         NaN    LIVING  60.000000           3   \n",
       "3               MB.0574         NaN    LIVING  60.000000           4   \n",
       "4               MB.0503         NaN    LIVING  60.000000           3   \n",
       "...                 ...         ...       ...        ...         ...   \n",
       "1899            MB.5465         NaN  DECEASED  18.800000           3   \n",
       "1900            MB.5453         NaN  DECEASED  57.300000           2   \n",
       "1901            MB.5471         NaN    LIVING  60.000000           4   \n",
       "1902            MB.5127         NaN    LIVING  60.000000           3   \n",
       "1903            MB.4313         NaN    LIVING  60.000000           1   \n",
       "\n",
       "     Unnamed: 5 CLAUDIN_SUBTYPE  \n",
       "0             C            LumA  \n",
       "1             D            Her2  \n",
       "2             C            LumA  \n",
       "3             D            LumA  \n",
       "4             C            LumA  \n",
       "...         ...             ...  \n",
       "1899          C           Basal  \n",
       "1900          B          Normal  \n",
       "1901          D            LumA  \n",
       "1902          C            LumB  \n",
       "1903          A            LumA  \n",
       "\n",
       "[1904 rows x 7 columns]"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_excel(\"imvc/datasets/data/metabric/original/Metabric_breast.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "7f791ab1-e8c7-4a4f-a38e-d32815dfb7e8",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "aa = pd.read_excel(\"imvc/datasets/data/metabric/Metabric_breast.xlsx\")\n",
    "aa[\"Patient Identifier\"] = aa[\"Patient Identifier\"].str.replace(\".\", \"-\")\n",
    "bb = pd.read_csv(\"imvc/datasets/data/metabric/brca_metabric_clinical_data.tsv\", sep= \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "ddb8b4d9-3186-4540-98dd-ed17d2cd7b02",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from sklearn.preprocessing import LabelEncoder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "8f306362-eeb7-4ff1-8b50-86196645bb3c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient Identifier</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>OS_STATUS</th>\n",
       "      <th>OS_MONTHS</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>CLAUDIN_SUBTYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1791</th>\n",
       "      <td>MB.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>B</td>\n",
       "      <td>claudin-low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290</th>\n",
       "      <td>MB.0002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>4</td>\n",
       "      <td>D</td>\n",
       "      <td>LumA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960</th>\n",
       "      <td>MB.0005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>LumB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>969</th>\n",
       "      <td>MB.0006</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>3</td>\n",
       "      <td>C</td>\n",
       "      <td>LumB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>MB.0008</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>41.366667</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>LumB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1700</th>\n",
       "      <td>MB.7295</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>B</td>\n",
       "      <td>LumA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1703</th>\n",
       "      <td>MB.7296</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>44.733333</td>\n",
       "      <td>4</td>\n",
       "      <td>D</td>\n",
       "      <td>LumB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1701</th>\n",
       "      <td>MB.7297</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>3</td>\n",
       "      <td>C</td>\n",
       "      <td>LumB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1625</th>\n",
       "      <td>MB.7298</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>LumB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1626</th>\n",
       "      <td>MB.7299</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>4</td>\n",
       "      <td>D</td>\n",
       "      <td>LumB</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1904 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Patient Identifier  Unnamed: 1 OS_STATUS  OS_MONTHS  Unnamed: 4   \n",
       "1791            MB.0000         NaN    LIVING  60.000000           2  \\\n",
       "290             MB.0002         NaN    LIVING  60.000000           4   \n",
       "960             MB.0005         NaN    LIVING  60.000000           1   \n",
       "969             MB.0006         NaN    LIVING  60.000000           3   \n",
       "215             MB.0008         NaN  DECEASED  41.366667           1   \n",
       "...                 ...         ...       ...        ...         ...   \n",
       "1700            MB.7295         NaN    LIVING  60.000000           2   \n",
       "1703            MB.7296         NaN  DECEASED  44.733333           4   \n",
       "1701            MB.7297         NaN    LIVING  60.000000           3   \n",
       "1625            MB.7298         NaN    LIVING  60.000000           1   \n",
       "1626            MB.7299         NaN    LIVING  60.000000           4   \n",
       "\n",
       "     Unnamed: 5 CLAUDIN_SUBTYPE  \n",
       "1791          B     claudin-low  \n",
       "290           D            LumA  \n",
       "960           A            LumB  \n",
       "969           C            LumB  \n",
       "215           A            LumB  \n",
       "...         ...             ...  \n",
       "1700          B            LumA  \n",
       "1703          D            LumB  \n",
       "1701          C            LumB  \n",
       "1625          A            LumB  \n",
       "1626          D            LumB  \n",
       "\n",
       "[1904 rows x 7 columns]"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aa.sort_values(\"Patient Identifier\")[[\"Patient Identifier\", \"CLAUDIN_SUBTYPE\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "9fdfdcee-4a3d-4b4a-b526-0ae8748e14ce",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient ID</th>\n",
       "      <th>Pam50 + Claudin-low subtype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MB-0000</td>\n",
       "      <td>claudin-low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MB-0002</td>\n",
       "      <td>LumA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MB-0005</td>\n",
       "      <td>LumB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MB-0006</td>\n",
       "      <td>LumB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MB-0008</td>\n",
       "      <td>LumB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2504</th>\n",
       "      <td>MTS-T2428</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2505</th>\n",
       "      <td>MTS-T2429</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2506</th>\n",
       "      <td>MTS-T2430</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2507</th>\n",
       "      <td>MTS-T2431</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2508</th>\n",
       "      <td>MTS-T2432</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2509 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Patient ID Pam50 + Claudin-low subtype\n",
       "0       MB-0000                 claudin-low\n",
       "1       MB-0002                        LumA\n",
       "2       MB-0005                        LumB\n",
       "3       MB-0006                        LumB\n",
       "4       MB-0008                        LumB\n",
       "...         ...                         ...\n",
       "2504  MTS-T2428                         NaN\n",
       "2505  MTS-T2429                         NaN\n",
       "2506  MTS-T2430                         NaN\n",
       "2507  MTS-T2431                         NaN\n",
       "2508  MTS-T2432                         NaN\n",
       "\n",
       "[2509 rows x 2 columns]"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bb[[\"Patient ID\", \"Pam50 + Claudin-low subtype\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "cdf7d8f7-54b8-4b02-81eb-ab9b947b81d0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1      9\n",
       "95     9\n",
       "97     9\n",
       "98     9\n",
       "99     9\n",
       "      ..\n",
       "51     9\n",
       "52     9\n",
       "53     9\n",
       "54     9\n",
       "150    9\n",
       "Name: count, Length: 150, dtype: int64"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.Series(mat[\"classLabel\"][:,0]).to_csv(\"imvc/datasets/data/buaa/buaa_y.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66c66242-c9b9-465f-aef7-d0caf5fc9bae",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "mat = scipy.io.loadmat('imvc/datasets/data/caltech101/Caltech101-all.mat')\n",
    "for i,x in enumerate(mat[\"X\"][0]):\n",
    "    print(x.shape)\n",
    "    pd.DataFrame(x).to_csv(f'imvc/datasets/data/caltech101/caltech101_{i}.csv', index= False)\n",
    "pd.DataFrame(mat[\"Y\"]).to_csv(f'imvc/datasets/data/caltech101/caltech101_y.csv', index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "516b89bc-1500-4143-8c30-0009f7dbc03b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "x = pd.read_csv(\"imvc/datasets/data/tcga/PanCan.miRNAseq.RPM.215-MIMATs-most-variant-25pc.4229-samples.NMF-input.BCGSC.20140603.csv\", index_col= 0)\n",
    "x.index.name= None\n",
    "x.columns = x.columns.to_series().apply(lambda x: x.split(\"_\")[1]).str[:12]\n",
    "x = x.T\n",
    "x = x[~x.index.duplicated(keep='first')]\n",
    "print(x.shape)\n",
    "x.to_csv(os.path.join(path, \"tcga_0.csv\"))\n",
    "x.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88ce4184-4d41-4fa7-9a12-4f6fe8ced380",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "x = pd.read_csv(\"imvc/datasets/data/tcga/PanCan12.3602-corrected-v3.txt\", sep= \"\\t\", index_col= 0, header= [0,1])\n",
    "x.index.name= None\n",
    "x.columns = x.columns.droplevel(0).str[:12]\n",
    "x = x.T\n",
    "x = x[~x.index.duplicated(keep='first')]\n",
    "print(x.shape)\n",
    "x.to_csv(os.path.join(path, \"tcga_1.csv\"))\n",
    "x.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "63ba8199-3c46-4aeb-a515-c4a1ebe85af8",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "metadata = {}\n",
    "metadata = {\"modality\": {0: \"visual\", 1: \"mRNA\", 2: \"text\", 3: \"methyl\"}, \"labels\": classes}\n",
    "import json\n",
    "\n",
    "with open(os.path.join(path, 'metadata.json'), 'w') as fp:\n",
    "    json.dump(metadata, fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "884f7401-b36d-4cd2-b949-0847c54db423",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "x = pd.read_csv(\"imvc/datasets/data/tcga/PanCan11_RBN_RPPA_without_Duplicates_20130325.csv\", index_col= 0)\n",
    "x.index.name= None\n",
    "x = x[x.columns[5:]]\n",
    "x = x[~x.index.duplicated(keep='first')]\n",
    "print(x.shape)\n",
    "x.to_csv(os.path.join(path, \"tcga_2.csv\"))\n",
    "x.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa60c039-bf61-49dc-82c3-f5aaafb6226d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "x = pd.read_csv(\"imvc/datasets/data/tcga/DNAmethylationClusteringMatrix.csv\", index_col= 0)\n",
    "x.columns = x.columns.str[:12]\n",
    "x = x.T\n",
    "x = x[~x.index.duplicated(keep='first')]\n",
    "print(x.shape)\n",
    "x.to_csv(os.path.join(path, \"tcga_3.csv\"))\n",
    "x.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1403c144-5c85-40e4-897c-07da8fddcadf",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "b = pd.read_csv(os.path.join(path, f\"tcga_y.csv\"), index_col= 0)\n",
    "classes = pd.Series(b.iloc[:, 0].unique()).to_dict()\n",
    "b.iloc[:, 0].replace({v:k for k,v in classes.items()}).to_csv(os.path.join(path, \"tcga_y.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "19f7b08a-f8a5-456b-8e22-08992eaf804f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "b"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b3dd080b-f7eb-425b-85aa-8e86695f3c16",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "i = 1\n",
    "a = pd.read_csv(os.path.join(path, f\"tcga_{i}.csv\"), index_col= 0).index\n",
    "print(len(a))\n",
    "for i in range(4):\n",
    "    a = a.intersection(pd.read_csv(os.path.join(path, f\"tcga_{i}.csv\"), index_col= 0).index)\n",
    "    print(len(a))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "da1289e7-6f55-4711-aebe-dacedff0116b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for i in range(4):\n",
    "    b = pd.read_csv(os.path.join(path, f\"tcga_{i}.csv\"), index_col= 0)\n",
    "    aa = a.intersection(b.index)\n",
    "    print(i, \"\\t\", b.shape, \"\\t\", b.drop_duplicates().shape, \"\\t\", b.loc[aa].shape, \"\\t\", b.loc[aa].drop_duplicates().shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9bac7683-9f9b-42fd-9bad-e537e56f6969",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "c = b.loc[\"TCGA-13-0791\"]\n",
    "c[c.columns[c.iloc[0] != c.iloc[1]]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dca7b617-55bd-48ea-abcb-0ab950b14b92",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "b.loc[a].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d1c78aeb-47b2-45f2-a5a3-656ddab94d26",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "b.loc[a].index.difference(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa2a67a2-27bd-4203-a394-5c5c9df8f344",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "b.drop_duplicates().loc[a.drop_duplicates()].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c72c32ac-d783-4eeb-918b-a57b27bb6153",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "b.index.intersection(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "42004a47-56ac-4e52-8f00-574eac035648",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "a.intersection(b.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "54f1c8f4-8c06-4ca2-95bc-469a2751567b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "x = pd.read_csv(\"imvc/datasets/data/tcga/mrna_y.csv\", index_col= 0)\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "92cc643b-8c86-482f-935d-09788a3f58ba",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "x1 = pd.read_csv(\"imvc/datasets/data/tcga/mrna_y.csv\", index_col =0)\n",
    "x2 = pd.read_csv(\"imvc/datasets/data/tcga/mirna_y.csv\", index_col =0)\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21e204fb-eebc-4f07-8bc1-3401822b5b78",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "x1.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1e69c77d-d9dd-404a-aca2-781e43efa19b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "x2.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "351ad9c5-2f85-4424-8f80-676fa09a492f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "x1.index.intersection(x2.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "77c831f4-439e-4510-b179-e28897928ade",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "x.T.to_csv(os.path.join(path, \"mrna.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d10ba525-192b-40c5-88b9-622f6dc0cf10",
   "metadata": {},
   "outputs": [],
   "source": [
    "x.columns.droplevel(1).to_series().to_csv(os.path.join(path, \"mirna_y.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e8b48d9a-10e4-4392-89f0-eb6b67a51a7e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "a = x.columns.to_frame().set_index(1)\n",
    "# a.columns = [''] * len(a.columns)\n",
    "a.index.name= None\n",
    "a.index = a.index.str[:12]\n",
    "a.to_csv(os.path.join(path, \"mrna_y.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c6eba92-57ac-4fcc-92f8-4f5db4a9ebb8",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "pd.read_csv(\"imvc/datasets/data/tcga/PanCan.miRNAseq.RPM.215-MIMATs-most-variant-25pc.4229-samples.NMF-input.BCGSC.20140603.csv\", index_col= 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4164af31-fb27-4a9d-a190-343b4fea1ddb",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "pd.read_csv(\"imvc/datasets/data/tcga/PanCan.miRNAseq.RPM.215-MIMATs-most-variant-25pc.4229-samples.NMF-input.BCGSC.20140603.csv\").columns.to_series().apply(lambda x: x.split(\"_\")[0]).value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e359b8ba-995a-4600-b5f6-9c4098f861e0",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for i,name in enumerate([\"X\", \"Ya\"]):\n",
    "    x = scipy.io.loadmat(f'imvc/datasets/data/bdgp/{name}.mat')[name]\n",
    "    print(x.shape)\n",
    "    pd.DataFrame(x).to_csv(f'imvc/datasets/data/bdgp/bdgp_{i}.csv', index= False)\n",
    "x = scipy.io.loadmat(f'imvc/datasets/data/bdgp/Yc.mat')[\"Yc\"]\n",
    "pd.DataFrame(x.argmax(1)).to_csv(f'imvc/datasets/data/bdgp/bdgp_y.csv', index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "978c82d6-e3b7-4f35-8380-fedca782ef31",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "path = \"imvc/datasets/data/tcga\"\n",
    "files = [f for f in os.listdir(path) if os.path.isfile(os.path.join(path, f))]\n",
    "for i,x in enumerate([\"exp\", \"methy\", \"mirna\"]):\n",
    "    target = []\n",
    "    files_x = [os.path.join(path, file) for file in files if file.startswith(x)]\n",
    "    ds = []\n",
    "    for file_x in files_x:\n",
    "        d_x = pd.read_csv(file_x, index_col= 0).T\n",
    "        print(file_x, d_x.shape)\n",
    "        target.extend([file_x.split(\"_\")[-1]]* d_x.shape[0])\n",
    "        ds.append(d_x)\n",
    "    d = pd.concat(ds)\n",
    "    print(x, d.shape)\n",
    "    d = d.dropna(axis= 1)\n",
    "    print(x, d.shape)\n",
    "    d.to_csv(os.path.join(path, f'tcga_{i}.csv'))\n",
    "pd.Series(target).to_csv(os.path.join(path, 'tcga_y.csv'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39c1fa46-5710-40f3-a6a4-13c7f9c96006",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "path = \"imvc/datasets/data/tcga\"\n",
    "files = [os.path.join(path, f) for f in os.listdir(path) if f.startswith(\"tcga_\")]\n",
    "d = pd.concat([pd.read_csv(file) for file in files], axis= 1)\n",
    "for i,file in enumerate(files):\n",
    "    d_x = pd.read_csv(file)\n",
    "    print(file, d_x.shape, d_x.loc[d.index])\n",
    "    d_x.loc[d.index].to_csv(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86d7d760-b3cb-43f4-afa4-16f588a1a44f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "path = \"imvc/datasets/data/tcga\"\n",
    "for i,x in enumerate([f for f in os.listdir(path) if os.path.isfile(os.path.join(path, f))]):\n",
    "    with ZipFile(os.path.join(path, x)) as zf:\n",
    "        for file in zf.namelist():\n",
    "            with zf.open(file) as f2:\n",
    "                d = pd.read_csv(f2, sep= \" \")\n",
    "                print(file, d.shape)\n",
    "                d.to_csv(f\"{os.path.join(path, file)}_{x.split('.')[0]}.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2af7b2c8-bc2b-4237-8b04-967f054a8cfb",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "path = \"imvc/datasets/data/digits\"\n",
    "for i,x in enumerate([f for f in os.listdir(path) if os.path.isfile(os.path.join(path, f))]):\n",
    "    d = pd.read_csv(os.path.join(path, x))\n",
    "    print(d.shape)\n",
    "    d.iloc[:, :-1].to_csv(os.path.join(path, f\"digits_{i}.csv\"), index= False)\n",
    "d.iloc[:, -1].to_csv(os.path.join(path, f\"digits_y.csv\"), index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ff058a6d-ddb2-4499-8c57-6d6b19b99b4d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "x = scipy.io.loadmat(f'imvc/datasets/data/bdgp/Yc.mat')[\"Yc\"]\n",
    "x.argmax(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f53749fa-f7af-4e14-b192-a326e883c5ea",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "[f for f in os.listdir(path) if os.path.isfile(os.path.join(path, f))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51afef5b-8792-4244-9e96-fa35b00079c6",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "os.listdir(path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "45a6a5d2-ba01-4d6f-a4fb-7cf5d097fce8",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "mat[\"X\"][0][0].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a4cccf4b-89df-4d0c-a95c-be3977097bc5",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "mat[\"X\"][0][1].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f1306beb-4250-49c7-bb64-7986e6c238d4",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pyreadr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57abfe0c-ebd8-43b8-ba13-7fc7f4c21ad6",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "mat = pyreadr.read_r('imvc/datasets/data/metabric/METABRIC_discovery')\n",
    "mat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9bf5aeed-9351-4151-b3f2-e9d344c9f43b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "mat[\"mydatCNV\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "556223fb-cd7d-4a21-b6e0-1bbe3a119356",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "91ea130b-e706-4025-afd0-a5e827aa1afd",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "pd.DataFrame(mat[\"Y\"]).squeeze().value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f9954cb-dc76-45d9-8ae7-219cffe4d972",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "mat = scipy.io.loadmat('imvc/datasets/data/bdgp/X.mat')\n",
    "mat[\"X\"].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90d74d47-3223-4d7e-aaeb-a2de0873505c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "mat = scipy.io.loadmat('imvc/datasets/data/bdgp/Yc.mat')\n",
    "mat[\"Yc\"][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed22dba5-2b9a-46cf-a9c2-37cf78244a60",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "mat[\"Yc\"].shape"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
